{
  "cells": [
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "L6JlnE4yLY2H",
        "outputId": "5807cf1d-da72-4ac2-8b86-321643bf9f17"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\r0% [Working]\r            \rHit:1 http://archive.ubuntu.com/ubuntu jammy InRelease\n",
            "\r0% [Connecting to security.ubuntu.com (185.125.190.81)] [Waiting for headers] [Connected to r2u.stat\r                                                                                                    \rGet:2 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [128 kB]\n",
            "Get:3 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease [3,626 B]\n",
            "Get:4 https://r2u.stat.illinois.edu/ubuntu jammy InRelease [6,555 B]\n",
            "Get:5 http://security.ubuntu.com/ubuntu jammy-security InRelease [129 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [127 kB]\n",
            "Hit:7 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease\n",
            "Hit:8 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease\n",
            "Get:9 https://r2u.stat.illinois.edu/ubuntu jammy/main amd64 Packages [2,646 kB]\n",
            "Hit:10 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\n",
            "Get:11 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [1,519 kB]\n",
            "Hit:12 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n",
            "Get:13 https://r2u.stat.illinois.edu/ubuntu jammy/main all Packages [8,615 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [2,860 kB]\n",
            "Get:15 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [1,228 kB]\n",
            "Get:16 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [2,561 kB]\n",
            "Fetched 19.8 MB in 2s (9,541 kB/s)\n",
            "Reading package lists... Done\n",
            "W: Skipping acquire of configured file 'main/source/Sources' as repository 'https://r2u.stat.illinois.edu/ubuntu jammy InRelease' does not seem to provide it (sources.list entry misspelt?)\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  libpython3.8-minimal libpython3.8-stdlib python3.8-lib2to3 python3.8-minimal\n",
            "Suggested packages:\n",
            "  python3.8-venv binfmt-support\n",
            "The following NEW packages will be installed:\n",
            "  libpython3.8-minimal libpython3.8-stdlib python3.8 python3.8-distutils python3.8-lib2to3\n",
            "  python3.8-minimal\n",
            "0 upgraded, 6 newly installed, 0 to remove and 64 not upgraded.\n",
            "Need to get 5,395 kB of archives.\n",
            "After this operation, 20.1 MB of additional disk space will be used.\n",
            "Get:1 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy/main amd64 libpython3.8-minimal amd64 3.8.20-1+jammy1 [796 kB]\n",
            "Get:2 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy/main amd64 python3.8-minimal amd64 3.8.20-1+jammy1 [2,023 kB]\n",
            "Get:3 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy/main amd64 libpython3.8-stdlib amd64 3.8.20-1+jammy1 [1,817 kB]\n",
            "Get:4 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy/main amd64 python3.8 amd64 3.8.20-1+jammy1 [440 kB]\n",
            "Get:5 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy/main amd64 python3.8-lib2to3 all 3.8.20-1+jammy1 [126 kB]\n",
            "Get:6 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy/main amd64 python3.8-distutils all 3.8.20-1+jammy1 [193 kB]\n",
            "Fetched 5,395 kB in 8s (649 kB/s)\n",
            "Selecting previously unselected package libpython3.8-minimal:amd64.\n",
            "(Reading database ... 124561 files and directories currently installed.)\n",
            "Preparing to unpack .../0-libpython3.8-minimal_3.8.20-1+jammy1_amd64.deb ...\n",
            "Unpacking libpython3.8-minimal:amd64 (3.8.20-1+jammy1) ...\n",
            "Selecting previously unselected package python3.8-minimal.\n",
            "Preparing to unpack .../1-python3.8-minimal_3.8.20-1+jammy1_amd64.deb ...\n",
            "Unpacking python3.8-minimal (3.8.20-1+jammy1) ...\n",
            "Selecting previously unselected package libpython3.8-stdlib:amd64.\n",
            "Preparing to unpack .../2-libpython3.8-stdlib_3.8.20-1+jammy1_amd64.deb ...\n",
            "Unpacking libpython3.8-stdlib:amd64 (3.8.20-1+jammy1) ...\n",
            "Selecting previously unselected package python3.8.\n",
            "Preparing to unpack .../3-python3.8_3.8.20-1+jammy1_amd64.deb ...\n",
            "Unpacking python3.8 (3.8.20-1+jammy1) ...\n",
            "Selecting previously unselected package python3.8-lib2to3.\n",
            "Preparing to unpack .../4-python3.8-lib2to3_3.8.20-1+jammy1_all.deb ...\n",
            "Unpacking python3.8-lib2to3 (3.8.20-1+jammy1) ...\n",
            "Selecting previously unselected package python3.8-distutils.\n",
            "Preparing to unpack .../5-python3.8-distutils_3.8.20-1+jammy1_all.deb ...\n",
            "Unpacking python3.8-distutils (3.8.20-1+jammy1) ...\n",
            "Setting up libpython3.8-minimal:amd64 (3.8.20-1+jammy1) ...\n",
            "Setting up python3.8-lib2to3 (3.8.20-1+jammy1) ...\n",
            "Setting up python3.8-minimal (3.8.20-1+jammy1) ...\n",
            "Setting up python3.8-distutils (3.8.20-1+jammy1) ...\n",
            "Setting up libpython3.8-stdlib:amd64 (3.8.20-1+jammy1) ...\n",
            "Setting up python3.8 (3.8.20-1+jammy1) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Processing triggers for mailcap (3.70+nmu1ubuntu1) ...\n",
            "There are 3 choices for the alternative python3 (providing /usr/bin/python3).\n",
            "\n",
            "  Selection    Path                 Priority   Status\n",
            "------------------------------------------------------------\n",
            "* 0            /usr/bin/python3.11   2         auto mode\n",
            "  1            /usr/bin/python3.10   1         manual mode\n",
            "  2            /usr/bin/python3.11   2         manual mode\n",
            "  3            /usr/bin/python3.8    1         manual mode\n",
            "\n",
            "Press <enter> to keep the current choice[*], or type selection number: 3\n",
            "update-alternatives: using /usr/bin/python3.8 to provide /usr/bin/python3 (python3) in manual mode\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  python3-setuptools python3-wheel\n",
            "Suggested packages:\n",
            "  python-setuptools-doc\n",
            "The following NEW packages will be installed:\n",
            "  python3-pip python3-setuptools python3-wheel\n",
            "0 upgraded, 3 newly installed, 0 to remove and 64 not upgraded.\n",
            "Need to get 1,677 kB of archives.\n",
            "After this operation, 8,968 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 python3-setuptools all 59.6.0-1.2ubuntu0.22.04.2 [340 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 python3-wheel all 0.37.1-2ubuntu0.22.04.1 [32.0 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 python3-pip all 22.0.2+dfsg-1ubuntu0.5 [1,306 kB]\n",
            "Fetched 1,677 kB in 1s (1,394 kB/s)\n",
            "Selecting previously unselected package python3-setuptools.\n",
            "(Reading database ... 125326 files and directories currently installed.)\n",
            "Preparing to unpack .../python3-setuptools_59.6.0-1.2ubuntu0.22.04.2_all.deb ...\n",
            "Unpacking python3-setuptools (59.6.0-1.2ubuntu0.22.04.2) ...\n",
            "Selecting previously unselected package python3-wheel.\n",
            "Preparing to unpack .../python3-wheel_0.37.1-2ubuntu0.22.04.1_all.deb ...\n",
            "Unpacking python3-wheel (0.37.1-2ubuntu0.22.04.1) ...\n",
            "Selecting previously unselected package python3-pip.\n",
            "Preparing to unpack .../python3-pip_22.0.2+dfsg-1ubuntu0.5_all.deb ...\n",
            "Unpacking python3-pip (22.0.2+dfsg-1ubuntu0.5) ...\n",
            "Setting up python3-setuptools (59.6.0-1.2ubuntu0.22.04.2) ...\n",
            "Setting up python3-wheel (0.37.1-2ubuntu0.22.04.1) ...\n",
            "Setting up python3-pip (22.0.2+dfsg-1ubuntu0.5) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Requirement already satisfied: pip in /usr/lib/python3/dist-packages (22.0.2)\n",
            "Collecting pip\n",
            "  Downloading pip-24.3.1-py3-none-any.whl (1.8 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.8/1.8 MB\u001b[0m \u001b[31m8.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: pip\n",
            "\u001b[33m  WARNING: The scripts pip, pip3 and pip3.8 are installed in '/root/.local/bin' which is not on PATH.\n",
            "  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.\u001b[0m\u001b[33m\n",
            "\u001b[0mSuccessfully installed pip-24.3.1\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
            "\u001b[0m"
          ]
        }
      ],
      "source": [
        "!apt-get update -y\n",
        "!apt-get install python3.8 python3.8-distutils\n",
        "!update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.8 1\n",
        "!update-alternatives --config python3\n",
        "!apt-get install python3-pip\n",
        "!python3 -m pip install --upgrade pip --user"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8c7NqCqsHD4T",
        "outputId": "3aed10a5-5027-45d5-e1fe-826be3d93f9c"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Python 3.8.20\n"
          ]
        }
      ],
      "source": [
        "# Check the Python version inside the environment\n",
        "!python --version"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "YfDB2FS007Nw",
        "outputId": "a35dd4f3-73eb-42b0-dbe6-0c7a4d522f4d"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Cloning into 'UIE'...\n",
            "remote: Enumerating objects: 334, done.\u001b[K\n",
            "remote: Counting objects: 100% (5/5), done.\u001b[K\n",
            "remote: Compressing objects: 100% (3/3), done.\u001b[K\n",
            "remote: Total 334 (delta 3), reused 2 (delta 2), pack-reused 329 (from 1)\u001b[K\n",
            "Receiving objects: 100% (334/334), 153.51 KiB | 2.13 MiB/s, done.\n",
            "Resolving deltas: 100% (152/152), done.\n",
            "/content/UIE\n"
          ]
        }
      ],
      "source": [
        "!git clone https://github.com/universal-ie/UIE.git\n",
        "%cd UIE"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "sSXrY00qRLYq",
        "outputId": "8ff346fd-a8db-4f58-cde9-404f39a85c17"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting absl-py==1.0.0 (from -r requirements.txt (line 1))\n",
            "  Downloading absl_py-1.0.0-py3-none-any.whl.metadata (2.3 kB)\n",
            "Collecting altair==4.2.0 (from -r requirements.txt (line 2))\n",
            "  Downloading altair-4.2.0-py3-none-any.whl.metadata (13 kB)\n",
            "Collecting anyio==3.5.0 (from -r requirements.txt (line 3))\n",
            "  Downloading anyio-3.5.0-py3-none-any.whl.metadata (4.7 kB)\n",
            "Collecting anytree==2.8.0 (from -r requirements.txt (line 4))\n",
            "  Downloading anytree-2.8.0-py2.py3-none-any.whl.metadata (7.3 kB)\n",
            "Collecting argon2-cffi==21.3.0 (from -r requirements.txt (line 5))\n",
            "  Downloading argon2_cffi-21.3.0-py3-none-any.whl.metadata (5.4 kB)\n",
            "Collecting argon2-cffi-bindings==21.2.0 (from -r requirements.txt (line 6))\n",
            "  Downloading argon2_cffi_bindings-21.2.0-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.7 kB)\n",
            "Collecting asgiref==3.5.0 (from -r requirements.txt (line 7))\n",
            "  Downloading asgiref-3.5.0-py3-none-any.whl.metadata (9.2 kB)\n",
            "Collecting astor==0.8.1 (from -r requirements.txt (line 8))\n",
            "  Downloading astor-0.8.1-py2.py3-none-any.whl.metadata (4.2 kB)\n",
            "Collecting asttokens==2.0.5 (from -r requirements.txt (line 9))\n",
            "  Downloading asttokens-2.0.5-py2.py3-none-any.whl.metadata (4.6 kB)\n",
            "Collecting attrs==21.4.0 (from -r requirements.txt (line 10))\n",
            "  Downloading attrs-21.4.0-py2.py3-none-any.whl.metadata (9.8 kB)\n",
            "Collecting autopep8==1.6.0 (from -r requirements.txt (line 11))\n",
            "  Downloading autopep8-1.6.0-py2.py3-none-any.whl.metadata (16 kB)\n",
            "Collecting backcall==0.2.0 (from -r requirements.txt (line 12))\n",
            "  Downloading backcall-0.2.0-py2.py3-none-any.whl.metadata (2.0 kB)\n",
            "Collecting backports.zoneinfo==0.2.1 (from -r requirements.txt (line 13))\n",
            "  Downloading backports.zoneinfo-0.2.1-cp38-cp38-manylinux1_x86_64.whl.metadata (4.7 kB)\n",
            "Collecting base58==2.1.1 (from -r requirements.txt (line 14))\n",
            "  Downloading base58-2.1.1-py3-none-any.whl.metadata (3.1 kB)\n",
            "Collecting black==21.12b0 (from -r requirements.txt (line 15))\n",
            "  Downloading black-21.12b0-py3-none-any.whl.metadata (39 kB)\n",
            "Collecting bleach==4.1.0 (from -r requirements.txt (line 16))\n",
            "  Downloading bleach-4.1.0-py2.py3-none-any.whl.metadata (24 kB)\n",
            "Requirement already satisfied: blinker==1.4 in /usr/lib/python3/dist-packages (from -r requirements.txt (line 17)) (1.4)\n",
            "Collecting cachetools==4.2.4 (from -r requirements.txt (line 18))\n",
            "  Downloading cachetools-4.2.4-py3-none-any.whl.metadata (4.8 kB)\n",
            "Collecting certifi==2021.10.8 (from -r requirements.txt (line 19))\n",
            "  Downloading certifi-2021.10.8-py2.py3-none-any.whl.metadata (3.0 kB)\n",
            "Collecting cffi==1.15.0 (from -r requirements.txt (line 20))\n",
            "  Downloading cffi-1.15.0-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl.metadata (1.2 kB)\n",
            "Collecting charset-normalizer==2.0.10 (from -r requirements.txt (line 21))\n",
            "  Downloading charset_normalizer-2.0.10-py3-none-any.whl.metadata (11 kB)\n",
            "Collecting click==8.0.3 (from -r requirements.txt (line 22))\n",
            "  Downloading click-8.0.3-py3-none-any.whl.metadata (3.2 kB)\n",
            "Collecting colorama==0.4.4 (from -r requirements.txt (line 23))\n",
            "  Downloading colorama-0.4.4-py2.py3-none-any.whl.metadata (14 kB)\n",
            "Collecting colorlog==6.6.0 (from -r requirements.txt (line 24))\n",
            "  Downloading colorlog-6.6.0-py2.py3-none-any.whl.metadata (10 kB)\n",
            "Collecting commonmark==0.9.1 (from -r requirements.txt (line 25))\n",
            "  Downloading commonmark-0.9.1-py2.py3-none-any.whl.metadata (5.7 kB)\n",
            "Collecting conllu==4.4.1 (from -r requirements.txt (line 26))\n",
            "  Downloading conllu-4.4.1-py2.py3-none-any.whl.metadata (19 kB)\n",
            "Collecting cycler==0.11.0 (from -r requirements.txt (line 27))\n",
            "  Downloading cycler-0.11.0-py3-none-any.whl.metadata (785 bytes)\n",
            "Collecting dataclasses==0.6 (from -r requirements.txt (line 28))\n",
            "  Downloading dataclasses-0.6-py3-none-any.whl.metadata (3.0 kB)\n",
            "Collecting datasets==1.9.0 (from -r requirements.txt (line 29))\n",
            "  Downloading datasets-1.9.0-py3-none-any.whl.metadata (9.5 kB)\n",
            "Collecting debugpy==1.5.1 (from -r requirements.txt (line 30))\n",
            "  Downloading debugpy-1.5.1-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl.metadata (1.2 kB)\n",
            "Collecting decorator==5.1.1 (from -r requirements.txt (line 31))\n",
            "  Downloading decorator-5.1.1-py3-none-any.whl.metadata (4.0 kB)\n",
            "Collecting defusedxml==0.7.1 (from -r requirements.txt (line 32))\n",
            "  Downloading defusedxml-0.7.1-py2.py3-none-any.whl.metadata (32 kB)\n",
            "Collecting dill==0.3.4 (from -r requirements.txt (line 33))\n",
            "  Downloading dill-0.3.4-py2.py3-none-any.whl.metadata (9.6 kB)\n",
            "Collecting elasticsearch==7.16.3 (from -r requirements.txt (line 34))\n",
            "  Downloading elasticsearch-7.16.3-py2.py3-none-any.whl.metadata (5.7 kB)\n",
            "Collecting entrypoints==0.3 (from -r requirements.txt (line 35))\n",
            "  Downloading entrypoints-0.3-py2.py3-none-any.whl.metadata (1.4 kB)\n",
            "Collecting executing==0.8.2 (from -r requirements.txt (line 36))\n",
            "  Downloading executing-0.8.2-py2.py3-none-any.whl.metadata (8.4 kB)\n",
            "Collecting faiss-cpu==1.7.2 (from -r requirements.txt (line 37))\n",
            "  Downloading faiss_cpu-1.7.2-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (1.4 kB)\n",
            "Collecting fastapi==0.74.1 (from -r requirements.txt (line 38))\n",
            "  Downloading fastapi-0.74.1-py3-none-any.whl.metadata (24 kB)\n",
            "Collecting filelock==3.0.12 (from -r requirements.txt (line 39))\n",
            "  Downloading filelock-3.0.12-py3-none-any.whl.metadata (4.3 kB)\n",
            "Collecting fire==0.4.0 (from -r requirements.txt (line 40))\n",
            "  Downloading fire-0.4.0.tar.gz (87 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting fonttools==4.28.5 (from -r requirements.txt (line 41))\n",
            "  Downloading fonttools-4.28.5-py3-none-any.whl.metadata (118 kB)\n",
            "Collecting fsspec==2022.1.0 (from -r requirements.txt (line 42))\n",
            "  Downloading fsspec-2022.1.0-py3-none-any.whl.metadata (5.4 kB)\n",
            "Collecting future==0.18.2 (from -r requirements.txt (line 43))\n",
            "  Downloading future-0.18.2.tar.gz (829 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m829.2/829.2 kB\u001b[0m \u001b[31m23.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting git-python==1.0.3 (from -r requirements.txt (line 44))\n",
            "  Downloading git_python-1.0.3-py2.py3-none-any.whl.metadata (331 bytes)\n",
            "Collecting gitdb==4.0.9 (from -r requirements.txt (line 45))\n",
            "  Downloading gitdb-4.0.9-py3-none-any.whl.metadata (998 bytes)\n",
            "Collecting GitPython==3.1.26 (from -r requirements.txt (line 46))\n",
            "  Downloading GitPython-3.1.26-py3-none-any.whl.metadata (1.2 kB)\n",
            "Collecting google-auth==2.3.3 (from -r requirements.txt (line 47))\n",
            "  Downloading google_auth-2.3.3-py2.py3-none-any.whl.metadata (3.7 kB)\n",
            "Collecting google-auth-oauthlib==0.4.6 (from -r requirements.txt (line 48))\n",
            "  Downloading google_auth_oauthlib-0.4.6-py2.py3-none-any.whl.metadata (2.7 kB)\n",
            "Collecting googleapis-common-protos==1.54.0 (from -r requirements.txt (line 49))\n",
            "  Downloading googleapis_common_protos-1.54.0-py2.py3-none-any.whl.metadata (1.3 kB)\n",
            "Collecting grpcio==1.43.0 (from -r requirements.txt (line 50))\n",
            "  Downloading grpcio-1.43.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.0 kB)\n",
            "Collecting h11==0.13.0 (from -r requirements.txt (line 51))\n",
            "  Downloading h11-0.13.0-py3-none-any.whl.metadata (8.2 kB)\n",
            "Collecting h5py==3.6.0 (from -r requirements.txt (line 52))\n",
            "  Downloading h5py-3.6.0-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl.metadata (1.9 kB)\n",
            "Collecting huggingface-hub==0.0.8 (from -r requirements.txt (line 53))\n",
            "  Downloading huggingface_hub-0.0.8-py3-none-any.whl.metadata (8.7 kB)\n",
            "Collecting idna==3.3 (from -r requirements.txt (line 54))\n",
            "  Downloading idna-3.3-py3-none-any.whl.metadata (9.8 kB)\n",
            "Collecting importlib-metadata==4.10.1 (from -r requirements.txt (line 55))\n",
            "  Downloading importlib_metadata-4.10.1-py3-none-any.whl.metadata (4.0 kB)\n",
            "Collecting importlib-resources==5.4.0 (from -r requirements.txt (line 56))\n",
            "  Downloading importlib_resources-5.4.0-py3-none-any.whl.metadata (3.1 kB)\n",
            "Collecting iniconfig==1.1.1 (from -r requirements.txt (line 57))\n",
            "  Downloading iniconfig-1.1.1-py2.py3-none-any.whl.metadata (2.4 kB)\n",
            "Collecting ipykernel==6.7.0 (from -r requirements.txt (line 58))\n",
            "  Downloading ipykernel-6.7.0-py3-none-any.whl.metadata (2.0 kB)\n",
            "Collecting ipython (from -r requirements.txt (line 59))\n",
            "  Downloading ipython-8.12.3-py3-none-any.whl.metadata (5.7 kB)\n",
            "Collecting ipython-genutils==0.2.0 (from -r requirements.txt (line 60))\n",
            "  Downloading ipython_genutils-0.2.0-py2.py3-none-any.whl.metadata (755 bytes)\n",
            "Collecting ipywidgets==7.6.5 (from -r requirements.txt (line 61))\n",
            "  Downloading ipywidgets-7.6.5-py2.py3-none-any.whl.metadata (1.9 kB)\n",
            "Collecting jedi==0.18.1 (from -r requirements.txt (line 62))\n",
            "  Downloading jedi-0.18.1-py2.py3-none-any.whl.metadata (20 kB)\n",
            "Collecting jieba==0.42.1 (from -r requirements.txt (line 63))\n",
            "  Downloading jieba-0.42.1.tar.gz (19.2 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m19.2/19.2 MB\u001b[0m \u001b[31m162.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting Jinja2==3.0.3 (from -r requirements.txt (line 64))\n",
            "  Downloading Jinja2-3.0.3-py3-none-any.whl.metadata (3.5 kB)\n",
            "Collecting joblib==1.1.0 (from -r requirements.txt (line 65))\n",
            "  Downloading joblib-1.1.0-py2.py3-none-any.whl.metadata (5.2 kB)\n",
            "Collecting jsonschema==4.4.0 (from -r requirements.txt (line 66))\n",
            "  Downloading jsonschema-4.4.0-py3-none-any.whl.metadata (7.5 kB)\n",
            "Collecting jupyter-client==7.1.1 (from -r requirements.txt (line 67))\n",
            "  Downloading jupyter_client-7.1.1-py3-none-any.whl.metadata (5.4 kB)\n",
            "Collecting jupyter-core==4.9.1 (from -r requirements.txt (line 68))\n",
            "  Downloading jupyter_core-4.9.1-py3-none-any.whl.metadata (870 bytes)\n",
            "Collecting jupyterlab-pygments==0.1.2 (from -r requirements.txt (line 69))\n",
            "  Downloading jupyterlab_pygments-0.1.2-py2.py3-none-any.whl.metadata (329 bytes)\n",
            "Collecting jupyterlab-widgets==1.0.2 (from -r requirements.txt (line 70))\n",
            "  Downloading jupyterlab_widgets-1.0.2-py3-none-any.whl.metadata (3.7 kB)\n",
            "Collecting kiwisolver==1.3.2 (from -r requirements.txt (line 71))\n",
            "  Downloading kiwisolver-1.3.2-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.whl.metadata (2.1 kB)\n",
            "Collecting Markdown==3.3.6 (from -r requirements.txt (line 72))\n",
            "  Downloading Markdown-3.3.6-py3-none-any.whl.metadata (4.6 kB)\n",
            "Collecting MarkupSafe==2.0.1 (from -r requirements.txt (line 73))\n",
            "  Downloading MarkupSafe-2.0.1-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl.metadata (3.2 kB)\n",
            "Collecting matplotlib==3.5.1 (from -r requirements.txt (line 74))\n",
            "  Downloading matplotlib-3.5.1-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.whl.metadata (6.7 kB)\n",
            "Collecting matplotlib-inline==0.1.3 (from -r requirements.txt (line 75))\n",
            "  Downloading matplotlib_inline-0.1.3-py3-none-any.whl.metadata (397 bytes)\n",
            "Collecting mistune==0.8.4 (from -r requirements.txt (line 76))\n",
            "  Downloading mistune-0.8.4-py2.py3-none-any.whl.metadata (8.5 kB)\n",
            "Collecting multiprocess==0.70.12.2 (from -r requirements.txt (line 77))\n",
            "  Downloading multiprocess-0.70.12.2-py38-none-any.whl.metadata (6.9 kB)\n",
            "Collecting mypy-extensions==0.4.3 (from -r requirements.txt (line 78))\n",
            "  Downloading mypy_extensions-0.4.3-py2.py3-none-any.whl.metadata (1.2 kB)\n",
            "Collecting nbclient==0.5.10 (from -r requirements.txt (line 79))\n",
            "  Downloading nbclient-0.5.10-py3-none-any.whl.metadata (4.9 kB)\n",
            "Collecting nbconvert==6.4.0 (from -r requirements.txt (line 80))\n",
            "  Downloading nbconvert-6.4.0-py3-none-any.whl.metadata (5.8 kB)\n",
            "Collecting nbformat==5.1.3 (from -r requirements.txt (line 81))\n",
            "  Downloading nbformat-5.1.3-py3-none-any.whl.metadata (1.4 kB)\n",
            "Collecting nest-asyncio==1.5.4 (from -r requirements.txt (line 82))\n",
            "  Downloading nest_asyncio-1.5.4-py3-none-any.whl.metadata (2.7 kB)\n",
            "Collecting nltk==3.5 (from -r requirements.txt (line 83))\n",
            "  Downloading nltk-3.5.zip (1.4 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.4/1.4 MB\u001b[0m \u001b[31m77.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting notebook==6.4.7 (from -r requirements.txt (line 84))\n",
            "  Downloading notebook-6.4.7-py3-none-any.whl.metadata (2.4 kB)\n",
            "Collecting numpy==1.19.5 (from -r requirements.txt (line 85))\n",
            "  Downloading numpy-1.19.5-cp38-cp38-manylinux2010_x86_64.whl.metadata (2.0 kB)\n",
            "Collecting oauthlib==3.1.1 (from -r requirements.txt (line 86))\n",
            "  Downloading oauthlib-3.1.1-py2.py3-none-any.whl.metadata (7.4 kB)\n",
            "Collecting packaging==21.3 (from -r requirements.txt (line 87))\n",
            "  Downloading packaging-21.3-py3-none-any.whl.metadata (15 kB)\n",
            "Collecting pandas==1.3.5 (from -r requirements.txt (line 88))\n",
            "  Downloading pandas-1.3.5-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (12 kB)\n",
            "Collecting pandocfilters==1.5.0 (from -r requirements.txt (line 89))\n",
            "  Downloading pandocfilters-1.5.0-py2.py3-none-any.whl.metadata (9.0 kB)\n",
            "Collecting parso==0.8.3 (from -r requirements.txt (line 90))\n",
            "  Downloading parso-0.8.3-py2.py3-none-any.whl.metadata (7.5 kB)\n",
            "Collecting pathspec==0.9.0 (from -r requirements.txt (line 91))\n",
            "  Downloading pathspec-0.9.0-py2.py3-none-any.whl.metadata (12 kB)\n",
            "Collecting pexpect==4.8.0 (from -r requirements.txt (line 92))\n",
            "  Downloading pexpect-4.8.0-py2.py3-none-any.whl.metadata (2.2 kB)\n",
            "Collecting pickleshare==0.7.5 (from -r requirements.txt (line 93))\n",
            "  Downloading pickleshare-0.7.5-py2.py3-none-any.whl.metadata (1.5 kB)\n",
            "Collecting Pillow==9.0.0 (from -r requirements.txt (line 94))\n",
            "  Downloading Pillow-9.0.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (7.6 kB)\n",
            "Collecting platformdirs==2.4.1 (from -r requirements.txt (line 95))\n",
            "  Downloading platformdirs-2.4.1-py3-none-any.whl.metadata (9.1 kB)\n",
            "Collecting pluggy==1.0.0 (from -r requirements.txt (line 96))\n",
            "  Downloading pluggy-1.0.0-py2.py3-none-any.whl.metadata (4.3 kB)\n",
            "Collecting portalocker==2.3.2 (from -r requirements.txt (line 97))\n",
            "  Downloading portalocker-2.3.2-py2.py3-none-any.whl.metadata (7.6 kB)\n",
            "Collecting prometheus-client==0.12.0 (from -r requirements.txt (line 98))\n",
            "  Downloading prometheus_client-0.12.0-py2.py3-none-any.whl.metadata (20 kB)\n",
            "Collecting promise==2.3 (from -r requirements.txt (line 99))\n",
            "  Downloading promise-2.3.tar.gz (19 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting prompt-toolkit==3.0.24 (from -r requirements.txt (line 100))\n",
            "  Downloading prompt_toolkit-3.0.24-py3-none-any.whl.metadata (7.0 kB)\n",
            "Collecting protobuf==3.19.3 (from -r requirements.txt (line 101))\n",
            "  Downloading protobuf-3.19.3-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (806 bytes)\n",
            "Collecting psutil==5.9.0 (from -r requirements.txt (line 102))\n",
            "  Downloading psutil-5.9.0-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (21 kB)\n",
            "Collecting ptyprocess==0.7.0 (from -r requirements.txt (line 103))\n",
            "  Downloading ptyprocess-0.7.0-py2.py3-none-any.whl.metadata (1.3 kB)\n",
            "Collecting pure-eval==0.2.1 (from -r requirements.txt (line 104))\n",
            "  Downloading pure_eval-0.2.1-py3-none-any.whl.metadata (6.2 kB)\n",
            "Collecting py==1.11.0 (from -r requirements.txt (line 105))\n",
            "  Downloading py-1.11.0-py2.py3-none-any.whl.metadata (2.8 kB)\n",
            "Collecting pyarrow==4.0.1 (from -r requirements.txt (line 106))\n",
            "  Downloading pyarrow-4.0.1-cp38-cp38-manylinux2014_x86_64.whl.metadata (2.8 kB)\n",
            "Collecting pyasn1==0.4.8 (from -r requirements.txt (line 107))\n",
            "  Downloading pyasn1-0.4.8-py2.py3-none-any.whl.metadata (1.5 kB)\n",
            "Collecting pyasn1-modules==0.2.8 (from -r requirements.txt (line 108))\n",
            "  Downloading pyasn1_modules-0.2.8-py2.py3-none-any.whl.metadata (1.9 kB)\n",
            "Collecting pycodestyle==2.8.0 (from -r requirements.txt (line 109))\n",
            "  Downloading pycodestyle-2.8.0-py2.py3-none-any.whl.metadata (31 kB)\n",
            "Collecting pycparser==2.21 (from -r requirements.txt (line 110))\n",
            "  Downloading pycparser-2.21-py2.py3-none-any.whl.metadata (1.1 kB)\n",
            "Collecting pydantic==1.9.0 (from -r requirements.txt (line 111))\n",
            "  Downloading pydantic-1.9.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (121 kB)\n",
            "Collecting pydeck==0.7.1 (from -r requirements.txt (line 112))\n",
            "  Downloading pydeck-0.7.1-py2.py3-none-any.whl.metadata (3.9 kB)\n",
            "Collecting Pygments==2.11.2 (from -r requirements.txt (line 113))\n",
            "  Downloading Pygments-2.11.2-py3-none-any.whl.metadata (1.5 kB)\n",
            "Collecting Pympler==1.0.1 (from -r requirements.txt (line 114))\n",
            "  Downloading Pympler-1.0.1-py3-none-any.whl.metadata (3.4 kB)\n",
            "Collecting pyparsing==3.0.6 (from -r requirements.txt (line 115))\n",
            "  Downloading pyparsing-3.0.6-py3-none-any.whl.metadata (4.2 kB)\n",
            "Collecting pyrsistent==0.18.1 (from -r requirements.txt (line 116))\n",
            "  Downloading pyrsistent-0.18.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (26 kB)\n",
            "Collecting pytest==6.2.5 (from -r requirements.txt (line 117))\n",
            "  Downloading pytest-6.2.5-py3-none-any.whl.metadata (7.5 kB)\n",
            "Collecting python-dateutil==2.8.2 (from -r requirements.txt (line 118))\n",
            "  Downloading python_dateutil-2.8.2-py2.py3-none-any.whl.metadata (8.2 kB)\n",
            "Collecting pytz==2021.3 (from -r requirements.txt (line 119))\n",
            "  Downloading pytz-2021.3-py2.py3-none-any.whl.metadata (21 kB)\n",
            "Collecting pytz-deprecation-shim==0.1.0.post0 (from -r requirements.txt (line 120))\n",
            "  Downloading pytz_deprecation_shim-0.1.0.post0-py2.py3-none-any.whl.metadata (5.2 kB)\n",
            "Collecting PyYAML==6.0 (from -r requirements.txt (line 121))\n",
            "  Downloading PyYAML-6.0-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl.metadata (2.0 kB)\n",
            "Collecting pyzmq==22.3.0 (from -r requirements.txt (line 122))\n",
            "  Downloading pyzmq-22.3.0-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl.metadata (5.0 kB)\n",
            "Collecting regex==2022.1.18 (from -r requirements.txt (line 123))\n",
            "  Downloading regex-2022.1.18-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (39 kB)\n",
            "Collecting requests==2.27.1 (from -r requirements.txt (line 124))\n",
            "  Downloading requests-2.27.1-py2.py3-none-any.whl.metadata (5.0 kB)\n",
            "Collecting requests-oauthlib==1.3.0 (from -r requirements.txt (line 125))\n",
            "  Downloading requests_oauthlib-1.3.0-py2.py3-none-any.whl.metadata (9.6 kB)\n",
            "Collecting rich==9.8.2 (from -r requirements.txt (line 126))\n",
            "  Downloading rich-9.8.2-py3-none-any.whl.metadata (16 kB)\n",
            "Collecting rouge-score==0.0.4 (from -r requirements.txt (line 127))\n",
            "  Downloading rouge_score-0.0.4-py2.py3-none-any.whl.metadata (3.8 kB)\n",
            "Collecting rsa==4.8 (from -r requirements.txt (line 128))\n",
            "  Downloading rsa-4.8-py3-none-any.whl.metadata (3.1 kB)\n",
            "Collecting sacrebleu==1.4.14 (from -r requirements.txt (line 129))\n",
            "  Downloading sacrebleu-1.4.14-py3-none-any.whl.metadata (1.3 kB)\n",
            "Collecting sacremoses==0.0.47 (from -r requirements.txt (line 130))\n",
            "  Downloading sacremoses-0.0.47-py2.py3-none-any.whl.metadata (476 bytes)\n",
            "Collecting scikit-learn==1.0.2 (from -r requirements.txt (line 131))\n",
            "  Downloading scikit_learn-1.0.2-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (10 kB)\n",
            "Collecting scipy==1.7.3 (from -r requirements.txt (line 132))\n",
            "  Downloading scipy-1.7.3-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (2.2 kB)\n",
            "Collecting Send2Trash==1.8.0 (from -r requirements.txt (line 133))\n",
            "  Downloading Send2Trash-1.8.0-py3-none-any.whl.metadata (6.8 kB)\n",
            "Collecting sentencepiece==0.1.96 (from -r requirements.txt (line 134))\n",
            "  Downloading sentencepiece-0.1.96-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (10 kB)\n",
            "Collecting seqeval==1.2.2 (from -r requirements.txt (line 135))\n",
            "  Downloading seqeval-1.2.2.tar.gz (43 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: six==1.16.0 in /usr/lib/python3/dist-packages (from -r requirements.txt (line 136)) (1.16.0)\n",
            "Collecting smmap==5.0.0 (from -r requirements.txt (line 137))\n",
            "  Downloading smmap-5.0.0-py3-none-any.whl.metadata (4.2 kB)\n",
            "Collecting sniffio==1.2.0 (from -r requirements.txt (line 138))\n",
            "  Downloading sniffio-1.2.0-py3-none-any.whl.metadata (3.6 kB)\n",
            "Collecting stack-data==0.1.4 (from -r requirements.txt (line 139))\n",
            "  Downloading stack_data-0.1.4-py3-none-any.whl.metadata (18 kB)\n",
            "Collecting starlette==0.17.1 (from -r requirements.txt (line 140))\n",
            "  Downloading starlette-0.17.1-py3-none-any.whl.metadata (6.3 kB)\n",
            "Collecting streamlit==1.4.0 (from -r requirements.txt (line 141))\n",
            "  Downloading streamlit-1.4.0-py2.py3-none-any.whl.metadata (1.2 kB)\n",
            "Collecting tabulate==0.8.9 (from -r requirements.txt (line 142))\n",
            "  Downloading tabulate-0.8.9-py3-none-any.whl.metadata (23 kB)\n",
            "Collecting tensorboard==2.7.0 (from -r requirements.txt (line 143))\n",
            "  Downloading tensorboard-2.7.0-py3-none-any.whl.metadata (1.9 kB)\n",
            "Collecting tensorboard-data-server==0.6.1 (from -r requirements.txt (line 144))\n",
            "  Downloading tensorboard_data_server-0.6.1-py3-none-manylinux2010_x86_64.whl.metadata (1.1 kB)\n",
            "Collecting tensorboard-plugin-wit==1.8.1 (from -r requirements.txt (line 145))\n",
            "  Downloading tensorboard_plugin_wit-1.8.1-py3-none-any.whl.metadata (873 bytes)\n",
            "Collecting tensorflow-datasets==4.4.0 (from -r requirements.txt (line 146))\n",
            "  Downloading tensorflow_datasets-4.4.0-py3-none-any.whl.metadata (5.8 kB)\n",
            "Collecting tensorflow-metadata==1.6.0 (from -r requirements.txt (line 147))\n",
            "  Downloading tensorflow_metadata-1.6.0-py3-none-any.whl.metadata (2.2 kB)\n",
            "Collecting termcolor==1.1.0 (from -r requirements.txt (line 148))\n",
            "  Downloading termcolor-1.1.0.tar.gz (3.9 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting terminado==0.12.1 (from -r requirements.txt (line 149))\n",
            "  Downloading terminado-0.12.1-py3-none-any.whl.metadata (3.7 kB)\n",
            "Collecting testpath==0.5.0 (from -r requirements.txt (line 150))\n",
            "  Downloading testpath-0.5.0-py3-none-any.whl.metadata (1.2 kB)\n",
            "Collecting threadpoolctl==3.0.0 (from -r requirements.txt (line 151))\n",
            "  Downloading threadpoolctl-3.0.0-py3-none-any.whl.metadata (8.8 kB)\n",
            "Collecting tokenizers==0.10.3 (from -r requirements.txt (line 152))\n",
            "  Downloading tokenizers-0.10.3-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl.metadata (5.8 kB)\n",
            "Collecting toml==0.10.2 (from -r requirements.txt (line 153))\n",
            "  Downloading toml-0.10.2-py2.py3-none-any.whl.metadata (7.1 kB)\n",
            "Collecting tomli==1.2.3 (from -r requirements.txt (line 154))\n",
            "  Downloading tomli-1.2.3-py3-none-any.whl.metadata (9.1 kB)\n",
            "Collecting toolz==0.11.2 (from -r requirements.txt (line 155))\n",
            "  Downloading toolz-0.11.2-py3-none-any.whl.metadata (5.1 kB)\n",
            "Collecting tornado==6.1 (from -r requirements.txt (line 156))\n",
            "  Downloading tornado-6.1-cp38-cp38-manylinux2010_x86_64.whl.metadata (2.4 kB)\n",
            "Collecting tqdm==4.62.3 (from -r requirements.txt (line 157))\n",
            "  Downloading tqdm-4.62.3-py2.py3-none-any.whl.metadata (56 kB)\n",
            "Collecting traitlets==5.1.1 (from -r requirements.txt (line 158))\n",
            "  Downloading traitlets-5.1.1-py3-none-any.whl.metadata (6.5 kB)\n",
            "Collecting transformers==4.6.1 (from -r requirements.txt (line 159))\n",
            "  Downloading transformers-4.6.1-py3-none-any.whl.metadata (45 kB)\n",
            "Collecting typing-extensions==3.10.0.2 (from -r requirements.txt (line 160))\n",
            "  Downloading typing_extensions-3.10.0.2-py3-none-any.whl.metadata (2.1 kB)\n",
            "Collecting tzdata==2021.5 (from -r requirements.txt (line 161))\n",
            "  Downloading tzdata-2021.5-py2.py3-none-any.whl.metadata (1.4 kB)\n",
            "Collecting tzlocal==4.1 (from -r requirements.txt (line 162))\n",
            "  Downloading tzlocal-4.1-py3-none-any.whl.metadata (13 kB)\n",
            "Collecting urllib3==1.26.8 (from -r requirements.txt (line 163))\n",
            "  Downloading urllib3-1.26.8-py2.py3-none-any.whl.metadata (45 kB)\n",
            "Collecting uvicorn==0.17.5 (from -r requirements.txt (line 164))\n",
            "  Downloading uvicorn-0.17.5-py3-none-any.whl.metadata (6.3 kB)\n",
            "Collecting validators==0.18.2 (from -r requirements.txt (line 165))\n",
            "  Downloading validators-0.18.2-py3-none-any.whl.metadata (1.3 kB)\n",
            "Collecting watchdog==2.1.6 (from -r requirements.txt (line 166))\n",
            "  Downloading watchdog-2.1.6-py3-none-manylinux2014_x86_64.whl.metadata (30 kB)\n",
            "Collecting wcwidth==0.2.5 (from -r requirements.txt (line 167))\n",
            "  Downloading wcwidth-0.2.5-py2.py3-none-any.whl.metadata (11 kB)\n",
            "Collecting webencodings==0.5.1 (from -r requirements.txt (line 168))\n",
            "  Downloading webencodings-0.5.1-py2.py3-none-any.whl.metadata (2.1 kB)\n",
            "Collecting Werkzeug==2.0.2 (from -r requirements.txt (line 169))\n",
            "  Downloading Werkzeug-2.0.2-py3-none-any.whl.metadata (4.5 kB)\n",
            "Collecting widgetsnbextension==3.5.2 (from -r requirements.txt (line 170))\n",
            "  Downloading widgetsnbextension-3.5.2-py2.py3-none-any.whl.metadata (1.3 kB)\n",
            "Collecting xxhash==2.0.2 (from -r requirements.txt (line 171))\n",
            "  Downloading xxhash-2.0.2-cp38-cp38-manylinux2010_x86_64.whl.metadata (11 kB)\n",
            "Collecting zipp==3.7.0 (from -r requirements.txt (line 172))\n",
            "  Downloading zipp-3.7.0-py3-none-any.whl.metadata (2.3 kB)\n",
            "Requirement already satisfied: setuptools>=40.3.0 in /usr/lib/python3/dist-packages (from google-auth==2.3.3->-r requirements.txt (line 47)) (59.6.0)\n",
            "Requirement already satisfied: wheel>=0.26 in /usr/lib/python3/dist-packages (from tensorboard==2.7.0->-r requirements.txt (line 143)) (0.37.1)\n",
            "INFO: pip is looking at multiple versions of ipython to determine which version is compatible with other requirements. This could take a while.\n",
            "Collecting ipython (from -r requirements.txt (line 59))\n",
            "  Downloading ipython-8.12.2-py3-none-any.whl.metadata (5.8 kB)\n",
            "  Downloading ipython-8.12.1-py3-none-any.whl.metadata (5.8 kB)\n",
            "  Downloading ipython-8.12.0-py3-none-any.whl.metadata (5.8 kB)\n",
            "  Downloading ipython-8.11.0-py3-none-any.whl.metadata (5.7 kB)\n",
            "  Downloading ipython-8.10.0-py3-none-any.whl.metadata (5.7 kB)\n",
            "  Downloading ipython-8.9.0-py3-none-any.whl.metadata (5.7 kB)\n",
            "  Downloading ipython-8.8.0-py3-none-any.whl.metadata (5.7 kB)\n",
            "Downloading absl_py-1.0.0-py3-none-any.whl (126 kB)\n",
            "Downloading altair-4.2.0-py3-none-any.whl (812 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m812.8/812.8 kB\u001b[0m \u001b[31m39.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading anyio-3.5.0-py3-none-any.whl (79 kB)\n",
            "Downloading anytree-2.8.0-py2.py3-none-any.whl (41 kB)\n",
            "Downloading argon2_cffi-21.3.0-py3-none-any.whl (14 kB)\n",
            "Downloading argon2_cffi_bindings-21.2.0-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (86 kB)\n",
            "Downloading asgiref-3.5.0-py3-none-any.whl (22 kB)\n",
            "Downloading astor-0.8.1-py2.py3-none-any.whl (27 kB)\n",
            "Downloading asttokens-2.0.5-py2.py3-none-any.whl (20 kB)\n",
            "Downloading attrs-21.4.0-py2.py3-none-any.whl (60 kB)\n",
            "Downloading autopep8-1.6.0-py2.py3-none-any.whl (45 kB)\n",
            "Downloading backcall-0.2.0-py2.py3-none-any.whl (11 kB)\n",
            "Downloading backports.zoneinfo-0.2.1-cp38-cp38-manylinux1_x86_64.whl (74 kB)\n",
            "Downloading base58-2.1.1-py3-none-any.whl (5.6 kB)\n",
            "Downloading black-21.12b0-py3-none-any.whl (156 kB)\n",
            "Downloading bleach-4.1.0-py2.py3-none-any.whl (157 kB)\n",
            "Downloading cachetools-4.2.4-py3-none-any.whl (10 kB)\n",
            "Downloading certifi-2021.10.8-py2.py3-none-any.whl (149 kB)\n",
            "Downloading cffi-1.15.0-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (446 kB)\n",
            "Downloading charset_normalizer-2.0.10-py3-none-any.whl (39 kB)\n",
            "Downloading click-8.0.3-py3-none-any.whl (97 kB)\n",
            "Downloading colorama-0.4.4-py2.py3-none-any.whl (16 kB)\n",
            "Downloading colorlog-6.6.0-py2.py3-none-any.whl (11 kB)\n",
            "Downloading commonmark-0.9.1-py2.py3-none-any.whl (51 kB)\n",
            "Downloading conllu-4.4.1-py2.py3-none-any.whl (15 kB)\n",
            "Downloading cycler-0.11.0-py3-none-any.whl (6.4 kB)\n",
            "Downloading dataclasses-0.6-py3-none-any.whl (14 kB)\n",
            "Downloading datasets-1.9.0-py3-none-any.whl (262 kB)\n",
            "Downloading debugpy-1.5.1-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (1.9 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.9/1.9 MB\u001b[0m \u001b[31m85.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading decorator-5.1.1-py3-none-any.whl (9.1 kB)\n",
            "Downloading defusedxml-0.7.1-py2.py3-none-any.whl (25 kB)\n",
            "Downloading dill-0.3.4-py2.py3-none-any.whl (86 kB)\n",
            "Downloading elasticsearch-7.16.3-py2.py3-none-any.whl (385 kB)\n",
            "Downloading entrypoints-0.3-py2.py3-none-any.whl (11 kB)\n",
            "Downloading executing-0.8.2-py2.py3-none-any.whl (16 kB)\n",
            "Downloading faiss_cpu-1.7.2-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (8.6 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m8.6/8.6 MB\u001b[0m \u001b[31m139.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading fastapi-0.74.1-py3-none-any.whl (53 kB)\n",
            "Downloading filelock-3.0.12-py3-none-any.whl (7.6 kB)\n",
            "Downloading fonttools-4.28.5-py3-none-any.whl (890 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m890.4/890.4 kB\u001b[0m \u001b[31m46.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading fsspec-2022.1.0-py3-none-any.whl (133 kB)\n",
            "Downloading git_python-1.0.3-py2.py3-none-any.whl (1.9 kB)\n",
            "Downloading gitdb-4.0.9-py3-none-any.whl (63 kB)\n",
            "Downloading GitPython-3.1.26-py3-none-any.whl (180 kB)\n",
            "Downloading google_auth-2.3.3-py2.py3-none-any.whl (155 kB)\n",
            "Downloading google_auth_oauthlib-0.4.6-py2.py3-none-any.whl (18 kB)\n",
            "Downloading googleapis_common_protos-1.54.0-py2.py3-none-any.whl (207 kB)\n",
            "Downloading grpcio-1.43.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.1 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m4.1/4.1 MB\u001b[0m \u001b[31m120.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading h11-0.13.0-py3-none-any.whl (58 kB)\n",
            "Downloading h5py-3.6.0-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (4.5 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m4.5/4.5 MB\u001b[0m \u001b[31m73.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading huggingface_hub-0.0.8-py3-none-any.whl (34 kB)\n",
            "Downloading idna-3.3-py3-none-any.whl (61 kB)\n",
            "Downloading importlib_metadata-4.10.1-py3-none-any.whl (17 kB)\n",
            "Downloading importlib_resources-5.4.0-py3-none-any.whl (28 kB)\n",
            "Downloading iniconfig-1.1.1-py2.py3-none-any.whl (5.0 kB)\n",
            "Downloading ipykernel-6.7.0-py3-none-any.whl (127 kB)\n",
            "Downloading ipython_genutils-0.2.0-py2.py3-none-any.whl (26 kB)\n",
            "Downloading ipywidgets-7.6.5-py2.py3-none-any.whl (121 kB)\n",
            "Downloading jedi-0.18.1-py2.py3-none-any.whl (1.6 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.6/1.6 MB\u001b[0m \u001b[31m66.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading Jinja2-3.0.3-py3-none-any.whl (133 kB)\n",
            "Downloading joblib-1.1.0-py2.py3-none-any.whl (306 kB)\n",
            "Downloading jsonschema-4.4.0-py3-none-any.whl (72 kB)\n",
            "Downloading jupyter_client-7.1.1-py3-none-any.whl (130 kB)\n",
            "Downloading jupyter_core-4.9.1-py3-none-any.whl (86 kB)\n",
            "Downloading jupyterlab_pygments-0.1.2-py2.py3-none-any.whl (4.6 kB)\n",
            "Downloading jupyterlab_widgets-1.0.2-py3-none-any.whl (243 kB)\n",
            "Downloading kiwisolver-1.3.2-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.whl (1.2 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.2/1.2 MB\u001b[0m \u001b[31m61.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading Markdown-3.3.6-py3-none-any.whl (97 kB)\n",
            "Downloading MarkupSafe-2.0.1-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (30 kB)\n",
            "Downloading matplotlib-3.5.1-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.whl (11.3 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m11.3/11.3 MB\u001b[0m \u001b[31m160.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading matplotlib_inline-0.1.3-py3-none-any.whl (8.2 kB)\n",
            "Downloading mistune-0.8.4-py2.py3-none-any.whl (16 kB)\n",
            "Downloading multiprocess-0.70.12.2-py38-none-any.whl (128 kB)\n",
            "Downloading mypy_extensions-0.4.3-py2.py3-none-any.whl (4.5 kB)\n",
            "Downloading nbclient-0.5.10-py3-none-any.whl (69 kB)\n",
            "Downloading nbconvert-6.4.0-py3-none-any.whl (557 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m557.4/557.4 kB\u001b[0m \u001b[31m22.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading nbformat-5.1.3-py3-none-any.whl (178 kB)\n",
            "Downloading nest_asyncio-1.5.4-py3-none-any.whl (5.1 kB)\n",
            "Downloading notebook-6.4.7-py3-none-any.whl (9.9 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m9.9/9.9 MB\u001b[0m \u001b[31m129.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading numpy-1.19.5-cp38-cp38-manylinux2010_x86_64.whl (14.9 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m14.9/14.9 MB\u001b[0m \u001b[31m162.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading oauthlib-3.1.1-py2.py3-none-any.whl (146 kB)\n",
            "Downloading packaging-21.3-py3-none-any.whl (40 kB)\n",
            "Downloading pandas-1.3.5-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (11.5 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m11.5/11.5 MB\u001b[0m \u001b[31m158.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pandocfilters-1.5.0-py2.py3-none-any.whl (8.7 kB)\n",
            "Downloading parso-0.8.3-py2.py3-none-any.whl (100 kB)\n",
            "Downloading pathspec-0.9.0-py2.py3-none-any.whl (31 kB)\n",
            "Downloading pexpect-4.8.0-py2.py3-none-any.whl (59 kB)\n",
            "Downloading pickleshare-0.7.5-py2.py3-none-any.whl (6.9 kB)\n",
            "Downloading Pillow-9.0.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.3 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m4.3/4.3 MB\u001b[0m \u001b[31m124.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading platformdirs-2.4.1-py3-none-any.whl (14 kB)\n",
            "Downloading pluggy-1.0.0-py2.py3-none-any.whl (13 kB)\n",
            "Downloading portalocker-2.3.2-py2.py3-none-any.whl (15 kB)\n",
            "Downloading prometheus_client-0.12.0-py2.py3-none-any.whl (57 kB)\n",
            "Downloading prompt_toolkit-3.0.24-py3-none-any.whl (374 kB)\n",
            "Downloading protobuf-3.19.3-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.1 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.1/1.1 MB\u001b[0m \u001b[31m49.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading psutil-5.9.0-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (283 kB)\n",
            "Downloading ptyprocess-0.7.0-py2.py3-none-any.whl (13 kB)\n",
            "Downloading pure_eval-0.2.1-py3-none-any.whl (11 kB)\n",
            "Downloading py-1.11.0-py2.py3-none-any.whl (98 kB)\n",
            "Downloading pyarrow-4.0.1-cp38-cp38-manylinux2014_x86_64.whl (21.9 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m21.9/21.9 MB\u001b[0m \u001b[31m166.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pyasn1-0.4.8-py2.py3-none-any.whl (77 kB)\n",
            "Downloading pyasn1_modules-0.2.8-py2.py3-none-any.whl (155 kB)\n",
            "Downloading pycodestyle-2.8.0-py2.py3-none-any.whl (42 kB)\n",
            "Downloading pycparser-2.21-py2.py3-none-any.whl (118 kB)\n",
            "Downloading pydantic-1.9.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (12.6 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m12.6/12.6 MB\u001b[0m \u001b[31m164.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pydeck-0.7.1-py2.py3-none-any.whl (4.3 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m4.3/4.3 MB\u001b[0m \u001b[31m119.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading Pygments-2.11.2-py3-none-any.whl (1.1 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.1/1.1 MB\u001b[0m \u001b[31m52.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading Pympler-1.0.1-py3-none-any.whl (164 kB)\n",
            "Downloading pyparsing-3.0.6-py3-none-any.whl (97 kB)\n",
            "Downloading pyrsistent-0.18.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (119 kB)\n",
            "Downloading pytest-6.2.5-py3-none-any.whl (280 kB)\n",
            "Downloading python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)\n",
            "Downloading pytz-2021.3-py2.py3-none-any.whl (503 kB)\n",
            "Downloading pytz_deprecation_shim-0.1.0.post0-py2.py3-none-any.whl (15 kB)\n",
            "Downloading PyYAML-6.0-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (701 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m701.2/701.2 kB\u001b[0m \u001b[31m30.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pyzmq-22.3.0-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (1.1 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.1/1.1 MB\u001b[0m \u001b[31m56.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading regex-2022.1.18-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (764 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m764.6/764.6 kB\u001b[0m \u001b[31m36.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading requests-2.27.1-py2.py3-none-any.whl (63 kB)\n",
            "Downloading requests_oauthlib-1.3.0-py2.py3-none-any.whl (23 kB)\n",
            "Downloading rich-9.8.2-py3-none-any.whl (186 kB)\n",
            "Downloading rouge_score-0.0.4-py2.py3-none-any.whl (22 kB)\n",
            "Downloading rsa-4.8-py3-none-any.whl (39 kB)\n",
            "Downloading sacrebleu-1.4.14-py3-none-any.whl (64 kB)\n",
            "Downloading sacremoses-0.0.47-py2.py3-none-any.whl (895 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m895.2/895.2 kB\u001b[0m \u001b[31m44.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading scikit_learn-1.0.2-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (26.7 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m26.7/26.7 MB\u001b[0m \u001b[31m181.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading scipy-1.7.3-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (39.3 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m39.3/39.3 MB\u001b[0m \u001b[31m166.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading Send2Trash-1.8.0-py3-none-any.whl (18 kB)\n",
            "Downloading sentencepiece-0.1.96-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.2 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.2/1.2 MB\u001b[0m \u001b[31m62.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading smmap-5.0.0-py3-none-any.whl (24 kB)\n",
            "Downloading sniffio-1.2.0-py3-none-any.whl (10 kB)\n",
            "Downloading stack_data-0.1.4-py3-none-any.whl (20 kB)\n",
            "Downloading starlette-0.17.1-py3-none-any.whl (58 kB)\n",
            "Downloading streamlit-1.4.0-py2.py3-none-any.whl (9.3 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m9.3/9.3 MB\u001b[0m \u001b[31m152.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading tabulate-0.8.9-py3-none-any.whl (25 kB)\n",
            "Downloading tensorboard-2.7.0-py3-none-any.whl (5.8 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m5.8/5.8 MB\u001b[0m \u001b[31m127.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading tensorboard_data_server-0.6.1-py3-none-manylinux2010_x86_64.whl (4.9 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m4.9/4.9 MB\u001b[0m \u001b[31m133.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading tensorboard_plugin_wit-1.8.1-py3-none-any.whl (781 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m781.3/781.3 kB\u001b[0m \u001b[31m38.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading tensorflow_datasets-4.4.0-py3-none-any.whl (4.0 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m4.0/4.0 MB\u001b[0m \u001b[31m120.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading tensorflow_metadata-1.6.0-py3-none-any.whl (48 kB)\n",
            "Downloading terminado-0.12.1-py3-none-any.whl (15 kB)\n",
            "Downloading testpath-0.5.0-py3-none-any.whl (84 kB)\n",
            "Downloading threadpoolctl-3.0.0-py3-none-any.whl (14 kB)\n",
            "Downloading tokenizers-0.10.3-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (3.3 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.3/3.3 MB\u001b[0m \u001b[31m119.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading toml-0.10.2-py2.py3-none-any.whl (16 kB)\n",
            "Downloading tomli-1.2.3-py3-none-any.whl (12 kB)\n",
            "Downloading toolz-0.11.2-py3-none-any.whl (55 kB)\n",
            "Downloading tornado-6.1-cp38-cp38-manylinux2010_x86_64.whl (427 kB)\n",
            "Downloading tqdm-4.62.3-py2.py3-none-any.whl (76 kB)\n",
            "Downloading traitlets-5.1.1-py3-none-any.whl (102 kB)\n",
            "Downloading transformers-4.6.1-py3-none-any.whl (2.2 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m2.2/2.2 MB\u001b[0m \u001b[31m85.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading typing_extensions-3.10.0.2-py3-none-any.whl (26 kB)\n",
            "Downloading tzdata-2021.5-py2.py3-none-any.whl (339 kB)\n",
            "Downloading tzlocal-4.1-py3-none-any.whl (19 kB)\n",
            "Downloading urllib3-1.26.8-py2.py3-none-any.whl (138 kB)\n",
            "Downloading uvicorn-0.17.5-py3-none-any.whl (53 kB)\n",
            "Downloading validators-0.18.2-py3-none-any.whl (19 kB)\n",
            "Downloading watchdog-2.1.6-py3-none-manylinux2014_x86_64.whl (76 kB)\n",
            "Downloading wcwidth-0.2.5-py2.py3-none-any.whl (30 kB)\n",
            "Downloading webencodings-0.5.1-py2.py3-none-any.whl (11 kB)\n",
            "Downloading Werkzeug-2.0.2-py3-none-any.whl (288 kB)\n",
            "Downloading widgetsnbextension-3.5.2-py2.py3-none-any.whl (1.6 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.6/1.6 MB\u001b[0m \u001b[31m71.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading xxhash-2.0.2-cp38-cp38-manylinux2010_x86_64.whl (243 kB)\n",
            "Downloading zipp-3.7.0-py3-none-any.whl (5.3 kB)\n",
            "Downloading ipython-8.8.0-py3-none-any.whl (775 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m775.8/775.8 kB\u001b[0m \u001b[31m33.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hBuilding wheels for collected packages: fire, future, jieba, nltk, promise, seqeval, termcolor\n",
            "  Building wheel for fire (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for fire: filename=fire-0.4.0-py2.py3-none-any.whl size=115942 sha256=fc5135aff75931d4f18492c3fafca5ef17561301d19c037ed611111f7d889781\n",
            "  Stored in directory: /root/.cache/pip/wheels/1f/10/06/2a990ee4d73a8479fe2922445e8a876d38cfbfed052284c6a1\n",
            "  Building wheel for future (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for future: filename=future-0.18.2-py3-none-any.whl size=491070 sha256=cfdce42af53e3872bfe2ad62edab4f8b99529f80ed048b707e1247fb796f4da4\n",
            "  Stored in directory: /root/.cache/pip/wheels/8e/70/28/3d6ccd6e315f65f245da085482a2e1c7d14b90b30f239e2cf4\n",
            "  Building wheel for jieba (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for jieba: filename=jieba-0.42.1-py3-none-any.whl size=19314478 sha256=cbb49c753e5155857d1c01b705a86f3dd916da281a23e8da3a15a913510bd1ac\n",
            "  Stored in directory: /root/.cache/pip/wheels/ca/38/d8/dfdfe73bec1d12026b30cb7ce8da650f3f0ea2cf155ea018ae\n",
            "  Building wheel for nltk (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for nltk: filename=nltk-3.5-py3-none-any.whl size=1434690 sha256=aafd204f1b8786bcae9799a97ca05ff3bdbe31b44784e85885ad8a68c40aaec8\n",
            "  Stored in directory: /root/.cache/pip/wheels/ff/d5/7b/f1fb4e1e1603b2f01c2424dd60fbcc50c12ef918bafc44b155\n",
            "  Building wheel for promise (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for promise: filename=promise-2.3-py3-none-any.whl size=21502 sha256=bf6860cf91c93d542b9ab4a8ece3447b4f7cf5b4e9a32fc89d46cbd2d125b1ae\n",
            "  Stored in directory: /root/.cache/pip/wheels/54/aa/01/724885182f93150035a2a91bce34a12877e8067a97baaf5dc8\n",
            "  Building wheel for seqeval (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for seqeval: filename=seqeval-1.2.2-py3-none-any.whl size=16180 sha256=06a1d1f22b13953fad8f139e59295ac65cee96819b997b4696295a61815668d8\n",
            "  Stored in directory: /root/.cache/pip/wheels/ad/5c/ba/05fa33fa5855777b7d686e843ec07452f22a66a138e290e732\n",
            "  Building wheel for termcolor (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for termcolor: filename=termcolor-1.1.0-py3-none-any.whl size=4848 sha256=35cc8a6c51cb5a595bbe349fa482380d309f7fdf60f99c3a30caf45a09402b84\n",
            "  Stored in directory: /root/.cache/pip/wheels/a0/16/9c/5473df82468f958445479c59e784896fa24f4a5fc024b0f501\n",
            "Successfully built fire future jieba nltk promise seqeval termcolor\n",
            "Installing collected packages: webencodings, wcwidth, typing-extensions, tokenizers, termcolor, tensorboard-plugin-wit, tabulate, sentencepiece, Send2Trash, regex, pytz, pyasn1, pure-eval, ptyprocess, pickleshare, mypy-extensions, mistune, jieba, ipython-genutils, iniconfig, filelock, faiss-cpu, executing, dataclasses, commonmark, certifi, backcall, zipp, xxhash, Werkzeug, watchdog, urllib3, tzdata, traitlets, tqdm, tornado, toolz, tomli, toml, threadpoolctl, testpath, tensorboard-data-server, sniffio, smmap, rsa, pyzmq, PyYAML, python-dateutil, pyrsistent, pyparsing, Pympler, Pygments, pydantic, pycparser, pycodestyle, pyasn1-modules, py, psutil, protobuf, prompt-toolkit, promise, prometheus-client, portalocker, pluggy, platformdirs, Pillow, pexpect, pathspec, parso, pandocfilters, oauthlib, numpy, nest-asyncio, MarkupSafe, kiwisolver, jupyterlab-widgets, joblib, idna, h11, grpcio, future, fsspec, fonttools, fire, entrypoints, dill, defusedxml, decorator, debugpy, cycler, conllu, colorlog, colorama, click, charset-normalizer, cachetools, base58, backports.zoneinfo, attrs, asttokens, astor, asgiref, anytree, absl-py, validators, uvicorn, terminado, stack-data, scipy, sacremoses, sacrebleu, rich, requests, pytz-deprecation-shim, pyarrow, pandas, packaging, nltk, multiprocess, matplotlib-inline, jupyterlab-pygments, jupyter-core, Jinja2, jedi, importlib-resources, importlib-metadata, h5py, googleapis-common-protos, google-auth, gitdb, elasticsearch, cffi, black, autopep8, anyio, tzlocal, tensorflow-metadata, starlette, scikit-learn, rouge-score, requests-oauthlib, pytest, matplotlib, Markdown, jupyter-client, jsonschema, ipython, huggingface-hub, GitPython, bleach, argon2-cffi-bindings, transformers, tensorflow-datasets, seqeval, nbformat, ipykernel, google-auth-oauthlib, git-python, fastapi, datasets, argon2-cffi, altair, tensorboard, nbclient, nbconvert, notebook, widgetsnbextension, ipywidgets, pydeck, streamlit\n",
            "  Attempting uninstall: zipp\n",
            "    Found existing installation: zipp 1.0.0\n",
            "    Uninstalling zipp-1.0.0:\n",
            "      Successfully uninstalled zipp-1.0.0\n",
            "  Attempting uninstall: pyparsing\n",
            "    Found existing installation: pyparsing 2.4.7\n",
            "    Uninstalling pyparsing-2.4.7:\n",
            "      Successfully uninstalled pyparsing-2.4.7\n",
            "  Attempting uninstall: oauthlib\n",
            "    Found existing installation: oauthlib 3.2.0\n",
            "    Uninstalling oauthlib-3.2.0:\n",
            "      Successfully uninstalled oauthlib-3.2.0\n",
            "  Attempting uninstall: importlib-metadata\n",
            "    Found existing installation: importlib-metadata 4.6.4\n",
            "    Uninstalling importlib-metadata-4.6.4:\n",
            "      Successfully uninstalled importlib-metadata-4.6.4\n",
            "Successfully installed GitPython-3.1.26 Jinja2-3.0.3 Markdown-3.3.6 MarkupSafe-2.0.1 Pillow-9.0.0 PyYAML-6.0 Pygments-2.11.2 Pympler-1.0.1 Send2Trash-1.8.0 Werkzeug-2.0.2 absl-py-1.0.0 altair-4.2.0 anyio-3.5.0 anytree-2.8.0 argon2-cffi-21.3.0 argon2-cffi-bindings-21.2.0 asgiref-3.5.0 astor-0.8.1 asttokens-2.0.5 attrs-21.4.0 autopep8-1.6.0 backcall-0.2.0 backports.zoneinfo-0.2.1 base58-2.1.1 black-21.12b0 bleach-4.1.0 cachetools-4.2.4 certifi-2021.10.8 cffi-1.15.0 charset-normalizer-2.0.10 click-8.0.3 colorama-0.4.4 colorlog-6.6.0 commonmark-0.9.1 conllu-4.4.1 cycler-0.11.0 dataclasses-0.6 datasets-1.9.0 debugpy-1.5.1 decorator-5.1.1 defusedxml-0.7.1 dill-0.3.4 elasticsearch-7.16.3 entrypoints-0.3 executing-0.8.2 faiss-cpu-1.7.2 fastapi-0.74.1 filelock-3.0.12 fire-0.4.0 fonttools-4.28.5 fsspec-2022.1.0 future-0.18.2 git-python-1.0.3 gitdb-4.0.9 google-auth-2.3.3 google-auth-oauthlib-0.4.6 googleapis-common-protos-1.54.0 grpcio-1.43.0 h11-0.13.0 h5py-3.6.0 huggingface-hub-0.0.8 idna-3.3 importlib-metadata-4.10.1 importlib-resources-5.4.0 iniconfig-1.1.1 ipykernel-6.7.0 ipython-8.8.0 ipython-genutils-0.2.0 ipywidgets-7.6.5 jedi-0.18.1 jieba-0.42.1 joblib-1.1.0 jsonschema-4.4.0 jupyter-client-7.1.1 jupyter-core-4.9.1 jupyterlab-pygments-0.1.2 jupyterlab-widgets-1.0.2 kiwisolver-1.3.2 matplotlib-3.5.1 matplotlib-inline-0.1.3 mistune-0.8.4 multiprocess-0.70.12.2 mypy-extensions-0.4.3 nbclient-0.5.10 nbconvert-6.4.0 nbformat-5.1.3 nest-asyncio-1.5.4 nltk-3.5 notebook-6.4.7 numpy-1.19.5 oauthlib-3.1.1 packaging-21.3 pandas-1.3.5 pandocfilters-1.5.0 parso-0.8.3 pathspec-0.9.0 pexpect-4.8.0 pickleshare-0.7.5 platformdirs-2.4.1 pluggy-1.0.0 portalocker-2.3.2 prometheus-client-0.12.0 promise-2.3 prompt-toolkit-3.0.24 protobuf-3.19.3 psutil-5.9.0 ptyprocess-0.7.0 pure-eval-0.2.1 py-1.11.0 pyarrow-4.0.1 pyasn1-0.4.8 pyasn1-modules-0.2.8 pycodestyle-2.8.0 pycparser-2.21 pydantic-1.9.0 pydeck-0.7.1 pyparsing-3.0.6 pyrsistent-0.18.1 pytest-6.2.5 python-dateutil-2.8.2 pytz-2021.3 pytz-deprecation-shim-0.1.0.post0 pyzmq-22.3.0 regex-2022.1.18 requests-2.27.1 requests-oauthlib-1.3.0 rich-9.8.2 rouge-score-0.0.4 rsa-4.8 sacrebleu-1.4.14 sacremoses-0.0.47 scikit-learn-1.0.2 scipy-1.7.3 sentencepiece-0.1.96 seqeval-1.2.2 smmap-5.0.0 sniffio-1.2.0 stack-data-0.1.4 starlette-0.17.1 streamlit-1.4.0 tabulate-0.8.9 tensorboard-2.7.0 tensorboard-data-server-0.6.1 tensorboard-plugin-wit-1.8.1 tensorflow-datasets-4.4.0 tensorflow-metadata-1.6.0 termcolor-1.1.0 terminado-0.12.1 testpath-0.5.0 threadpoolctl-3.0.0 tokenizers-0.10.3 toml-0.10.2 tomli-1.2.3 toolz-0.11.2 tornado-6.1 tqdm-4.62.3 traitlets-5.1.1 transformers-4.6.1 typing-extensions-3.10.0.2 tzdata-2021.5 tzlocal-4.1 urllib3-1.26.8 uvicorn-0.17.5 validators-0.18.2 watchdog-2.1.6 wcwidth-0.2.5 webencodings-0.5.1 widgetsnbextension-3.5.2 xxhash-2.0.2 zipp-3.7.0\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "application/vnd.colab-display-data+json": {
              "pip_warning": {
                "packages": [
                  "dateutil",
                  "defusedxml",
                  "ipython_genutils",
                  "pickleshare"
                ]
              },
              "id": "24e49adefe744d05b237fd307fef6aa2"
            }
          },
          "metadata": {}
        }
      ],
      "source": [
        "!pip install -r requirements.txt"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 568
        },
        "id": "Azsp-5QZ1yu-",
        "outputId": "41fad6a7-9027-4765-db91-6b4ca2d40996"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Collecting gdown\n",
            "  Downloading gdown-5.2.0-py3-none-any.whl.metadata (5.8 kB)\n",
            "Requirement already satisfied: nltk==3.5 in /usr/local/lib/python3.8/dist-packages (3.5)\n",
            "Requirement already satisfied: click in /usr/local/lib/python3.8/dist-packages (from nltk==3.5) (8.0.3)\n",
            "Requirement already satisfied: joblib in /usr/local/lib/python3.8/dist-packages (from nltk==3.5) (1.1.0)\n",
            "Requirement already satisfied: regex in /usr/local/lib/python3.8/dist-packages (from nltk==3.5) (2022.1.18)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.8/dist-packages (from nltk==3.5) (4.62.3)\n",
            "Collecting beautifulsoup4 (from gdown)\n",
            "  Downloading beautifulsoup4-4.12.3-py3-none-any.whl.metadata (3.8 kB)\n",
            "Requirement already satisfied: filelock in /usr/local/lib/python3.8/dist-packages (from gdown) (3.0.12)\n",
            "Requirement already satisfied: requests[socks] in /usr/local/lib/python3.8/dist-packages (from gdown) (2.27.1)\n",
            "Collecting soupsieve>1.2 (from beautifulsoup4->gdown)\n",
            "  Downloading soupsieve-2.6-py3-none-any.whl.metadata (4.6 kB)\n",
            "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/local/lib/python3.8/dist-packages (from requests[socks]->gdown) (1.26.8)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.8/dist-packages (from requests[socks]->gdown) (2021.10.8)\n",
            "Requirement already satisfied: charset-normalizer~=2.0.0 in /usr/local/lib/python3.8/dist-packages (from requests[socks]->gdown) (2.0.10)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.8/dist-packages (from requests[socks]->gdown) (3.3)\n",
            "Collecting PySocks!=1.5.7,>=1.5.6 (from requests[socks]->gdown)\n",
            "  Downloading PySocks-1.7.1-py3-none-any.whl.metadata (13 kB)\n",
            "Downloading gdown-5.2.0-py3-none-any.whl (18 kB)\n",
            "Downloading beautifulsoup4-4.12.3-py3-none-any.whl (147 kB)\n",
            "Downloading PySocks-1.7.1-py3-none-any.whl (16 kB)\n",
            "Downloading soupsieve-2.6-py3-none-any.whl (36 kB)\n",
            "Installing collected packages: soupsieve, PySocks, beautifulsoup4, gdown\n",
            "Successfully installed PySocks-1.7.1 beautifulsoup4-4.12.3 gdown-5.2.0 soupsieve-2.6\n"
          ]
        },
        {
          "data": {
            "application/vnd.colab-display-data+json": {
              "id": "2e14fc2a8a1743f9ba98991996a4174b",
              "pip_warning": {
                "packages": [
                  "socks"
                ]
              }
            }
          },
          "metadata": {},
          "output_type": "display_data"
        }
      ],
      "source": [
        "pip install gdown nltk==3.5"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "CE13nmSEQK6O",
        "outputId": "77f52f54-5491-44e4-943d-6b42f4746776"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Downloading...\n",
            "From (original): https://drive.google.com/uc?id=15OFkWw8kJA1k2g_zehZ0pxcjTABY2iF1\n",
            "From (redirected): https://drive.google.com/uc?id=15OFkWw8kJA1k2g_zehZ0pxcjTABY2iF1&confirm=t&uuid=a7c9d83c-bca3-4d25-84df-8eb02d6dc532\n",
            "To: /content/UIE/uie-large-en.zip\n",
            "100% 2.90G/2.90G [00:59<00:00, 49.0MB/s]\n",
            "Archive:  uie-large-en.zip\n",
            "   creating: uie-large-en/\n",
            "  inflating: uie-large-en/config.json  \n",
            "  inflating: uie-large-en/special_tokens_map.json  \n",
            "  inflating: uie-large-en/added_tokens.json  \n",
            "  inflating: uie-large-en/spiece.model  \n",
            "  inflating: uie-large-en/pytorch_model.bin  \n",
            "  inflating: uie-large-en/tokenizer.json  \n",
            "  inflating: uie-large-en/tokenizer_config.json  \n"
          ]
        }
      ],
      "source": [
        "!gdown 15OFkWw8kJA1k2g_zehZ0pxcjTABY2iF1\n",
        "\n",
        "!unzip uie-large-en.zip"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "2Fh85XTsN0RB"
      },
      "outputs": [],
      "source": [
        "!mkdir hf_models\n",
        "!mv uie-large-en hf_models/"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "NKMlvM9uzMAe",
        "outputId": "2ac70dde-1167-46ce-b235-315f112c3d76"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/content/UIE/dataset_processing\n"
          ]
        }
      ],
      "source": [
        "cd dataset_processing"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "J789H4puS3gA",
        "outputId": "d79e0a1a-fa84-4e4e-fb2d-ab042e5b14b9"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--2025-01-14 19:25:01--  https://raw.githubusercontent.com/yubowen-ph/JointER/master/dataset/NYT-multi/data/train.json\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.110.133, 185.199.108.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 47296062 (45M) [text/plain]\n",
            "Saving to: ‘data/NYT-multi/train.json’\n",
            "\n",
            "train.json          100%[===================>]  45.10M  --.-KB/s    in 0.1s    \n",
            "\n",
            "2025-01-14 19:25:05 (400 MB/s) - ‘data/NYT-multi/train.json’ saved [47296062/47296062]\n",
            "\n",
            "--2025-01-14 19:25:05--  https://raw.githubusercontent.com/yubowen-ph/JointER/master/dataset/NYT-multi/data/dev.json\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 4219415 (4.0M) [text/plain]\n",
            "Saving to: ‘data/NYT-multi/dev.json’\n",
            "\n",
            "dev.json            100%[===================>]   4.02M  --.-KB/s    in 0.01s   \n",
            "\n",
            "2025-01-14 19:25:07 (374 MB/s) - ‘data/NYT-multi/dev.json’ saved [4219415/4219415]\n",
            "\n",
            "--2025-01-14 19:25:07--  https://raw.githubusercontent.com/yubowen-ph/JointER/master/dataset/NYT-multi/data/test.json\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 4242192 (4.0M) [text/plain]\n",
            "Saving to: ‘data/NYT-multi/test.json’\n",
            "\n",
            "test.json           100%[===================>]   4.04M  --.-KB/s    in 0.01s   \n",
            "\n",
            "2025-01-14 19:25:08 (311 MB/s) - ‘data/NYT-multi/test.json’ saved [4242192/4242192]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!mkdir data/NYT-multi\n",
        "!wget -P data/NYT-multi https://raw.githubusercontent.com/yubowen-ph/JointER/master/dataset/NYT-multi/data/train.json\n",
        "!wget -P data/NYT-multi https://raw.githubusercontent.com/yubowen-ph/JointER/master/dataset/NYT-multi/data/dev.json\n",
        "!wget -P data/NYT-multi https://raw.githubusercontent.com/yubowen-ph/JointER/master/dataset/NYT-multi/data/test.json"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "C7Sj7PS63I2F",
        "outputId": "dc48beed-354c-4a42-ac6b-a22aa3dbe028"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--2025-01-08 17:15:24--  https://raw.githubusercontent.com/synalp/NER/master/corpus/CoNLL-2003/eng.train\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 3281528 (3.1M) [text/plain]\n",
            "Saving to: ‘data/conll03/eng.train’\n",
            "\n",
            "eng.train           100%[===================>]   3.13M  --.-KB/s    in 0.01s   \n",
            "\n",
            "2025-01-08 17:15:24 (299 MB/s) - ‘data/conll03/eng.train’ saved [3281528/3281528]\n",
            "\n",
            "--2025-01-08 17:15:24--  https://raw.githubusercontent.com/synalp/NER/master/corpus/CoNLL-2003/eng.testa\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.111.133, 185.199.109.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 827012 (808K) [text/plain]\n",
            "Saving to: ‘data/conll03/eng.testa’\n",
            "\n",
            "eng.testa           100%[===================>] 807.63K  --.-KB/s    in 0.01s   \n",
            "\n",
            "2025-01-08 17:15:25 (82.5 MB/s) - ‘data/conll03/eng.testa’ saved [827012/827012]\n",
            "\n",
            "--2025-01-08 17:15:25--  https://raw.githubusercontent.com/synalp/NER/master/corpus/CoNLL-2003/eng.testb\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.111.133, 185.199.109.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 748096 (731K) [text/plain]\n",
            "Saving to: ‘data/conll03/eng.testb’\n",
            "\n",
            "eng.testb           100%[===================>] 730.56K  --.-KB/s    in 0.006s  \n",
            "\n",
            "2025-01-08 17:15:25 (111 MB/s) - ‘data/conll03/eng.testb’ saved [748096/748096]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://raw.githubusercontent.com/synalp/NER/master/corpus/CoNLL-2003/eng.train -P data/conll03\n",
        "!wget https://raw.githubusercontent.com/synalp/NER/master/corpus/CoNLL-2003/eng.testa -P data/conll03\n",
        "!wget https://raw.githubusercontent.com/synalp/NER/master/corpus/CoNLL-2003/eng.testb -P data/conll03\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "m6WuYcDyzZcM",
        "outputId": "3eb12c55-9cdd-41c9-824c-8547d9cf4400"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "data/NYT-multi/train.json: 56196\n",
            "data/NYT-multi/dev.json: 5000\n",
            "data/NYT-multi/test.json: 5000\n",
            "{'ORGANIZATION': 'organization', 'LOCATION': 'location', 'PERSON': 'person', '/location/location/contains': 'contains', '/people/person/place_of_birth': 'place of birth', '/business/person/company': 'company', '/people/person/place_lived': 'place lived', '/location/administrative_division/country': 'country', '/location/country/administrative_divisions': 'administrative divisions', '/people/person/religion': 'religion', '/people/person/nationality': 'nationality', '/people/person/children': 'children', '/location/country/capital': 'capital', '/business/company/place_founded': 'place founded', '/people/deceased_person/place_of_death': 'place of death', '/business/company/founders': 'founders', '/location/neighborhood/neighborhood_of': 'neighborhood of', '/business/company/advisors': 'advisors', '/people/ethnicity/geographic_distribution': 'geographic distribution', '/sports/sports_team_location/teams': 'teams', '/sports/sports_team/location': 'location', '/business/company_shareholder/major_shareholder_of': 'major shareholder of', '/business/company/major_shareholders': 'major shareholders', '/people/person/ethnicity': 'ethnicity', '/people/ethnicity/people': 'people', '/people/person/profession': 'profession', '/business/company/industry': 'industry'}\n",
            "100% 56196/56196 [00:04<00:00, 12785.14it/s]\n",
            "100% 5000/5000 [00:00<00:00, 12178.75it/s]\n",
            "100% 5000/5000 [00:00<00:00, 12025.64it/s]\n",
            "Counter({'train sent': 56196, 'val sent': 5000, 'test sent': 5000})\n",
            "converted_data/text2spotasoc/relation//NYT\n",
            "==========================\n"
          ]
        }
      ],
      "source": [
        "!python uie_convert.py -format spotasoc -config data_config/relation/NYT-multi.yaml -output relation/\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 52
        },
        "id": "eE2klK3A5eWz",
        "outputId": "8e741275-29a5-4bcf-8006-7c6d173622ef"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/content/UIE\n"
          ]
        },
        {
          "data": {
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            },
            "text/plain": [
              "'/content/UIE'"
            ]
          },
          "execution_count": 8,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "%cd /content/UIE\n",
        "\n",
        "%pwd\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "E0uqdC9a__qb"
      },
      "outputs": [],
      "source": [
        "!ln -s dataset_processing/converted_data/ data\n",
        "\n",
        "!mkdir output"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "YKPekK3vSchE",
        "outputId": "abe45458-2dd0-464f-cd29-315e0b99802a"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: datasets in /usr/local/lib/python3.8/dist-packages (1.9.0)\n",
            "Requirement already satisfied: numpy>=1.17 in /usr/local/lib/python3.8/dist-packages (from datasets) (1.19.5)\n",
            "Requirement already satisfied: pyarrow!=4.0.0,>=1.0.0 in /usr/local/lib/python3.8/dist-packages (from datasets) (4.0.1)\n",
            "Requirement already satisfied: dill in /usr/local/lib/python3.8/dist-packages (from datasets) (0.3.4)\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.8/dist-packages (from datasets) (1.3.5)\n",
            "Requirement already satisfied: requests>=2.19.0 in /usr/local/lib/python3.8/dist-packages (from datasets) (2.27.1)\n",
            "Requirement already satisfied: tqdm>=4.27 in /usr/local/lib/python3.8/dist-packages (from datasets) (4.62.3)\n",
            "Requirement already satisfied: xxhash in /usr/local/lib/python3.8/dist-packages (from datasets) (2.0.2)\n",
            "Requirement already satisfied: multiprocess in /usr/local/lib/python3.8/dist-packages (from datasets) (0.70.12.2)\n",
            "Requirement already satisfied: fsspec>=2021.05.0 in /usr/local/lib/python3.8/dist-packages (from datasets) (2022.1.0)\n",
            "Requirement already satisfied: huggingface-hub<0.1.0 in /usr/local/lib/python3.8/dist-packages (from datasets) (0.0.8)\n",
            "Requirement already satisfied: packaging in /usr/local/lib/python3.8/dist-packages (from datasets) (21.3)\n",
            "Requirement already satisfied: filelock in /usr/local/lib/python3.8/dist-packages (from huggingface-hub<0.1.0->datasets) (3.0.12)\n",
            "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /usr/local/lib/python3.8/dist-packages (from requests>=2.19.0->datasets) (1.26.8)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.8/dist-packages (from requests>=2.19.0->datasets) (2021.10.8)\n",
            "Requirement already satisfied: charset-normalizer~=2.0.0 in /usr/local/lib/python3.8/dist-packages (from requests>=2.19.0->datasets) (2.0.10)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.8/dist-packages (from requests>=2.19.0->datasets) (3.3)\n",
            "Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /usr/local/lib/python3.8/dist-packages (from packaging->datasets) (3.0.6)\n",
            "Requirement already satisfied: python-dateutil>=2.7.3 in /usr/local/lib/python3.8/dist-packages (from pandas->datasets) (2.8.2)\n",
            "Requirement already satisfied: pytz>=2017.3 in /usr/local/lib/python3.8/dist-packages (from pandas->datasets) (2021.3)\n",
            "Requirement already satisfied: six>=1.5 in /usr/lib/python3/dist-packages (from python-dateutil>=2.7.3->pandas->datasets) (1.16.0)\n"
          ]
        }
      ],
      "source": [
        "!pip install datasets"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install wandb\n",
        "\n",
        "!wandb login"
      ],
      "metadata": {
        "id": "kM-lAY5tp31D"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "ovk2ghDJSBh_"
      },
      "outputs": [],
      "source": [
        "!python --version"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "id": "f_J6lsTSo8Wu",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "e2d0a99e-b7ca-400a-e2b8-4e16cf5ae812"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Looking in links: https://download.pytorch.org/whl/torch_stable.html\n",
            "Collecting torch==1.8.0+cu111\n",
            "  Downloading https://download.pytorch.org/whl/cu111/torch-1.8.0%2Bcu111-cp38-cp38-linux_x86_64.whl (1982.2 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m2.0/2.0 GB\u001b[0m \u001b[31m18.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting torchvision==0.9.0+cu111\n",
            "  Downloading https://download.pytorch.org/whl/cu111/torchvision-0.9.0%2Bcu111-cp38-cp38-linux_x86_64.whl (17.6 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m17.6/17.6 MB\u001b[0m \u001b[31m89.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting torchaudio==0.8.0\n",
            "  Downloading torchaudio-0.8.0-cp38-cp38-manylinux1_x86_64.whl.metadata (1.1 kB)\n",
            "Requirement already satisfied: typing-extensions in /usr/local/lib/python3.8/dist-packages (from torch==1.8.0+cu111) (3.10.0.2)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.8/dist-packages (from torch==1.8.0+cu111) (1.19.5)\n",
            "Requirement already satisfied: pillow>=4.1.1 in /usr/local/lib/python3.8/dist-packages (from torchvision==0.9.0+cu111) (9.0.0)\n",
            "Downloading torchaudio-0.8.0-cp38-cp38-manylinux1_x86_64.whl (1.9 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.9/1.9 MB\u001b[0m \u001b[31m33.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: torch, torchvision, torchaudio\n",
            "Successfully installed torch-1.8.0+cu111 torchaudio-0.8.0 torchvision-0.9.0+cu111\n"
          ]
        }
      ],
      "source": [
        "!pip install torch==1.8.0+cu111 torchvision==0.9.0+cu111 torchaudio==0.8.0 -f https://download.pytorch.org/whl/torch_stable.html\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "collapsed": true,
        "id": "vEqBVRVUP_d6",
        "outputId": "8a6f8d62-b7d8-430b-f379-cbef31ad3763"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/content/UIE\n",
            "Map Config config/offset_map/closest_offset_en.yaml\n",
            "01/18/2025 17:30:14 - INFO - __main__ -   Options:\n",
            "01/18/2025 17:30:14 - INFO - __main__ -   ModelArguments(model_name_or_path='hf_models/uie-large-en', config_name=None, tokenizer_name=None, cache_dir=None, use_fast_tokenizer=True, model_revision='main', use_auth_token=False, from_checkpoint=False)\n",
            "01/18/2025 17:30:14 - INFO - __main__ -   DataTrainingArguments(task='meta', dataset_name=None, dataset_config_name=None, text_column='text', record_column='record', train_file='data/text2spotasoc/relation/GWAS/train.json', validation_file='data/text2spotasoc/relation/GWAS/val.json', test_file='data/text2spotasoc/relation/GWAS/test.json', overwrite_cache=False, preprocessing_num_workers=4, preprocess=True, preprocessed_folder=None, max_source_length=512, max_target_length=512, max_prefix_length=-1, val_max_target_length=512, pad_to_max_length=False, max_train_samples=None, max_val_samples=None, max_test_samples=None, num_beams=None, ignore_pad_token_for_loss=True, source_prefix='meta: ', meta_negative=-1, ordered_prompt=False, decoding_format='spotasoc', record_schema='data/text2spotasoc/relation/GWAS/record.schema', spot_noise=0.1, asoc_noise=0.1, meta_positive_rate=1.0)\n",
            "01/18/2025 17:30:14 - INFO - __main__ -   ConstraintSeq2SeqTrainingArguments(output_dir='output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1', overwrite_output_dir=False, do_train=True, do_eval=True, do_predict=True, evaluation_strategy=<IntervalStrategy.STEPS: 'steps'>, prediction_loss_only=False, per_device_train_batch_size=4, per_device_eval_batch_size=16, per_gpu_train_batch_size=None, per_gpu_eval_batch_size=None, gradient_accumulation_steps=1, eval_accumulation_steps=None, learning_rate=0.0005, weight_decay=0.0, adam_beta1=0.9, adam_beta2=0.999, adam_epsilon=1e-08, max_grad_norm=1.0, num_train_epochs=400.0, max_steps=-1, lr_scheduler_type=<SchedulerType.LINEAR: 'linear'>, warmup_ratio=0.06, warmup_steps=0, logging_dir='output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1_log', logging_strategy=<IntervalStrategy.STEPS: 'steps'>, logging_first_step=False, logging_steps=500, save_strategy=<IntervalStrategy.STEPS: 'steps'>, save_steps=500, save_total_limit=1, no_cuda=False, seed=421, fp16=False, fp16_opt_level='O1', fp16_backend='auto', fp16_full_eval=False, local_rank=-1, tpu_num_cores=None, tpu_metrics_debug=False, debug=[], dataloader_drop_last=False, eval_steps=5000, dataloader_num_workers=0, past_index=-1, run_name='output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1', disable_tqdm=False, remove_unused_columns=False, label_names=None, load_best_model_at_end=True, metric_for_best_model='eval_overall-F1', greater_is_better=True, ignore_data_skip=False, sharded_ddp=[], deepspeed=None, label_smoothing_factor=0.0, adafactor=False, group_by_length=False, length_column_name='length', report_to=['tensorboard', 'wandb'], ddp_find_unused_parameters=False, dataloader_pin_memory=True, skip_memory_metrics=True, use_legacy_prediction_loop=False, push_to_hub=False, resume_from_checkpoint=None, mp_parameters='', sortish_sampler=False, predict_with_generate=True, constraint_decoding=False, save_better_checkpoint=False, start_eval_step=0)\n",
            "01/18/2025 17:30:14 - WARNING - __main__ -   Process rank: -1, device: cuda:0, n_gpu: 1distributed training: False, 16-bits training: False\n",
            "01/18/2025 17:30:14 - INFO - __main__ -   Training/evaluation parameters ConstraintSeq2SeqTrainingArguments(output_dir='output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1', overwrite_output_dir=False, do_train=True, do_eval=True, do_predict=True, evaluation_strategy=<IntervalStrategy.STEPS: 'steps'>, prediction_loss_only=False, per_device_train_batch_size=4, per_device_eval_batch_size=16, per_gpu_train_batch_size=None, per_gpu_eval_batch_size=None, gradient_accumulation_steps=1, eval_accumulation_steps=None, learning_rate=0.0005, weight_decay=0.0, adam_beta1=0.9, adam_beta2=0.999, adam_epsilon=1e-08, max_grad_norm=1.0, num_train_epochs=400.0, max_steps=-1, lr_scheduler_type=<SchedulerType.LINEAR: 'linear'>, warmup_ratio=0.06, warmup_steps=0, logging_dir='output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1_log', logging_strategy=<IntervalStrategy.STEPS: 'steps'>, logging_first_step=False, logging_steps=500, save_strategy=<IntervalStrategy.STEPS: 'steps'>, save_steps=500, save_total_limit=1, no_cuda=False, seed=421, fp16=False, fp16_opt_level='O1', fp16_backend='auto', fp16_full_eval=False, local_rank=-1, tpu_num_cores=None, tpu_metrics_debug=False, debug=[], dataloader_drop_last=False, eval_steps=5000, dataloader_num_workers=0, past_index=-1, run_name='output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1', disable_tqdm=False, remove_unused_columns=False, label_names=None, load_best_model_at_end=True, metric_for_best_model='eval_overall-F1', greater_is_better=True, ignore_data_skip=False, sharded_ddp=[], deepspeed=None, label_smoothing_factor=0.0, adafactor=False, group_by_length=False, length_column_name='length', report_to=['tensorboard', 'wandb'], ddp_find_unused_parameters=False, dataloader_pin_memory=True, skip_memory_metrics=True, use_legacy_prediction_loop=False, push_to_hub=False, resume_from_checkpoint=None, mp_parameters='', sortish_sampler=False, predict_with_generate=True, constraint_decoding=False, save_better_checkpoint=False, start_eval_step=0)\n",
            "01/18/2025 17:30:14 - INFO - __main__ -   {'train': 'data/text2spotasoc/relation/GWAS/train.json', 'validation': 'data/text2spotasoc/relation/GWAS/val.json', 'test': 'data/text2spotasoc/relation/GWAS/test.json'}\n",
            "01/18/2025 17:30:14 - WARNING - datasets.builder -   Using custom data configuration default-15814f03dfeab079\n",
            "01/18/2025 17:30:14 - WARNING - datasets.builder -   Reusing dataset json (/root/.cache/huggingface/datasets/json/default-15814f03dfeab079/0.0.0/958c48074b7307f8fc8e71dd3c233cccd95331f830807be63e197c10611ad12a)\n",
            "01/18/2025 17:30:14 - INFO - __main__ -   DatasetDict({\n",
            "    train: Dataset({\n",
            "        features: ['text', 'tokens', 'record', 'entity', 'relation', 'event', 'spot', 'asoc', 'spot_asoc', 'task'],\n",
            "        num_rows: 538\n",
            "    })\n",
            "    validation: Dataset({\n",
            "        features: ['text', 'tokens', 'record', 'entity', 'relation', 'event', 'spot', 'asoc', 'spot_asoc', 'task'],\n",
            "        num_rows: 35\n",
            "    })\n",
            "    test: Dataset({\n",
            "        features: ['text', 'tokens', 'record', 'entity', 'relation', 'event', 'spot', 'asoc', 'spot_asoc', 'task'],\n",
            "        num_rows: 35\n",
            "    })\n",
            "})\n",
            "01/18/2025 17:30:14 - INFO - __main__ -   hf_models/uie-large-en\n",
            "loading configuration file hf_models/uie-large-en/config.json\n",
            "Model config T5Config {\n",
            "  \"_name_or_path\": \"hf_models/t5-v1_1-large\",\n",
            "  \"architectures\": [\n",
            "    \"T5ForConditionalGeneration\"\n",
            "  ],\n",
            "  \"d_ff\": 2816,\n",
            "  \"d_kv\": 64,\n",
            "  \"d_model\": 1024,\n",
            "  \"decoder_start_token_id\": 0,\n",
            "  \"dropout_rate\": 0.1,\n",
            "  \"eos_token_id\": 1,\n",
            "  \"feed_forward_proj\": \"gated-gelu\",\n",
            "  \"gradient_checkpointing\": false,\n",
            "  \"initializer_factor\": 1.0,\n",
            "  \"is_encoder_decoder\": true,\n",
            "  \"layer_norm_epsilon\": 1e-06,\n",
            "  \"max_length\": 128,\n",
            "  \"model_type\": \"t5\",\n",
            "  \"num_decoder_layers\": 24,\n",
            "  \"num_heads\": 16,\n",
            "  \"num_layers\": 24,\n",
            "  \"output_past\": true,\n",
            "  \"pad_token_id\": 0,\n",
            "  \"relative_attention_num_buckets\": 32,\n",
            "  \"tie_word_embeddings\": false,\n",
            "  \"transformers_version\": \"4.6.1\",\n",
            "  \"use_cache\": true,\n",
            "  \"vocab_size\": 32102\n",
            "}\n",
            "\n",
            "loading configuration file hf_models/uie-large-en/config.json\n",
            "Model config T5Config {\n",
            "  \"_name_or_path\": \"hf_models/t5-v1_1-large\",\n",
            "  \"architectures\": [\n",
            "    \"T5ForConditionalGeneration\"\n",
            "  ],\n",
            "  \"d_ff\": 2816,\n",
            "  \"d_kv\": 64,\n",
            "  \"d_model\": 1024,\n",
            "  \"decoder_start_token_id\": 0,\n",
            "  \"dropout_rate\": 0.1,\n",
            "  \"eos_token_id\": 1,\n",
            "  \"feed_forward_proj\": \"gated-gelu\",\n",
            "  \"gradient_checkpointing\": false,\n",
            "  \"initializer_factor\": 1.0,\n",
            "  \"is_encoder_decoder\": true,\n",
            "  \"layer_norm_epsilon\": 1e-06,\n",
            "  \"max_length\": 128,\n",
            "  \"model_type\": \"t5\",\n",
            "  \"num_decoder_layers\": 24,\n",
            "  \"num_heads\": 16,\n",
            "  \"num_layers\": 24,\n",
            "  \"output_past\": true,\n",
            "  \"pad_token_id\": 0,\n",
            "  \"relative_attention_num_buckets\": 32,\n",
            "  \"tie_word_embeddings\": false,\n",
            "  \"transformers_version\": \"4.6.1\",\n",
            "  \"use_cache\": true,\n",
            "  \"vocab_size\": 32102\n",
            "}\n",
            "\n",
            "loading file hf_models/uie-large-en/spiece.model\n",
            "loading file hf_models/uie-large-en/tokenizer.json\n",
            "loading file hf_models/uie-large-en/added_tokens.json\n",
            "loading file hf_models/uie-large-en/special_tokens_map.json\n",
            "loading file hf_models/uie-large-en/tokenizer_config.json\n",
            "Using bos_token, but it is not set yet.\n",
            "loading weights file hf_models/uie-large-en/pytorch_model.bin\n",
            "All model checkpoint weights were used when initializing T5ForConditionalGeneration.\n",
            "\n",
            "All the weights of T5ForConditionalGeneration were initialized from the model checkpoint at hf_models/uie-large-en.\n",
            "If your task is similar to the task the model of the checkpoint was trained on, you can already use T5ForConditionalGeneration for predictions without further training.\n",
            "Assigning ['<spot>', '<asoc>'] to the additional_special_tokens key of the tokenizer\n",
            "01/18/2025 17:30:23 - INFO - __main__ -   PreTrainedTokenizerFast(name_or_path='hf_models/uie-large-en', vocab_size=32100, model_max_len=512, is_fast=True, padding_side='right', special_tokens={'eos_token': '</s>', 'unk_token': '<unk>', 'pad_token': '<pad>', 'additional_special_tokens': ['<spot>', '<asoc>']})\n",
            "01/18/2025 17:30:23 - INFO - __main__ -   Prefix: \n",
            "01/18/2025 17:30:23 - INFO - __main__ -   Prefix Length: 0\n",
            "01/18/2025 17:30:23 - INFO - __main__ -   Using src: text and tgt: record\n",
            "01/18/2025 17:30:23 - INFO - __main__ -   Start Data Preprocessing ...\n",
            "01/18/2025 17:30:23 - WARNING - datasets.arrow_dataset -   Loading cached processed dataset at /root/.cache/huggingface/datasets/json/default-15814f03dfeab079/0.0.0/958c48074b7307f8fc8e71dd3c233cccd95331f830807be63e197c10611ad12a/cache-2d04497df1d56d54.arrow\n",
            "01/18/2025 17:30:23 - WARNING - datasets.arrow_dataset -   Loading cached processed dataset at /root/.cache/huggingface/datasets/json/default-15814f03dfeab079/0.0.0/958c48074b7307f8fc8e71dd3c233cccd95331f830807be63e197c10611ad12a/cache-f8c919ebe36379b0.arrow\n",
            "01/18/2025 17:30:23 - WARNING - datasets.arrow_dataset -   Loading cached processed dataset at /root/.cache/huggingface/datasets/json/default-15814f03dfeab079/0.0.0/958c48074b7307f8fc8e71dd3c233cccd95331f830807be63e197c10611ad12a/cache-e4a9b70407962748.arrow\n",
            "01/18/2025 17:30:23 - WARNING - datasets.arrow_dataset -   Loading cached processed dataset at /root/.cache/huggingface/datasets/json/default-15814f03dfeab079/0.0.0/958c48074b7307f8fc8e71dd3c233cccd95331f830807be63e197c10611ad12a/cache-274ec89775ae0698.arrow\n",
            "01/18/2025 17:30:24 - WARNING - datasets.arrow_dataset -   Loading cached processed dataset at /root/.cache/huggingface/datasets/json/default-15814f03dfeab079/0.0.0/958c48074b7307f8fc8e71dd3c233cccd95331f830807be63e197c10611ad12a/cache-23616f072ba8d33b.arrow\n",
            "01/18/2025 17:30:24 - WARNING - datasets.arrow_dataset -   Loading cached processed dataset at /root/.cache/huggingface/datasets/json/default-15814f03dfeab079/0.0.0/958c48074b7307f8fc8e71dd3c233cccd95331f830807be63e197c10611ad12a/cache-95a88f860470dac4.arrow\n",
            "01/18/2025 17:30:24 - WARNING - datasets.arrow_dataset -   Loading cached processed dataset at /root/.cache/huggingface/datasets/json/default-15814f03dfeab079/0.0.0/958c48074b7307f8fc8e71dd3c233cccd95331f830807be63e197c10611ad12a/cache-c54c4363a8c354c6.arrow\n",
            "01/18/2025 17:30:24 - WARNING - datasets.arrow_dataset -   Loading cached processed dataset at /root/.cache/huggingface/datasets/json/default-15814f03dfeab079/0.0.0/958c48074b7307f8fc8e71dd3c233cccd95331f830807be63e197c10611ad12a/cache-764f669d01bef009.arrow\n",
            "01/18/2025 17:30:24 - WARNING - datasets.arrow_dataset -   Loading cached processed dataset at /root/.cache/huggingface/datasets/json/default-15814f03dfeab079/0.0.0/958c48074b7307f8fc8e71dd3c233cccd95331f830807be63e197c10611ad12a/cache-982fc0ab6249b29d.arrow\n",
            "01/18/2025 17:30:24 - WARNING - datasets.arrow_dataset -   Loading cached processed dataset at /root/.cache/huggingface/datasets/json/default-15814f03dfeab079/0.0.0/958c48074b7307f8fc8e71dd3c233cccd95331f830807be63e197c10611ad12a/cache-be954fbd78504112.arrow\n",
            "01/18/2025 17:30:24 - WARNING - datasets.arrow_dataset -   Loading cached processed dataset at /root/.cache/huggingface/datasets/json/default-15814f03dfeab079/0.0.0/958c48074b7307f8fc8e71dd3c233cccd95331f830807be63e197c10611ad12a/cache-6ea1b4471dc42a16.arrow\n",
            "01/18/2025 17:30:24 - WARNING - datasets.arrow_dataset -   Loading cached processed dataset at /root/.cache/huggingface/datasets/json/default-15814f03dfeab079/0.0.0/958c48074b7307f8fc8e71dd3c233cccd95331f830807be63e197c10611ad12a/cache-1cfe3580499ebbac.arrow\n",
            "01/18/2025 17:30:24 - INFO - __main__ -   End Data Preprocessing ...\n",
            "01/18/2025 17:30:24 - INFO - __main__ -   Meta Sample, Negative: -1, Ordered Prompt: False\n",
            "***** Running training *****\n",
            "  Num examples = 538\n",
            "  Num Epochs = 400\n",
            "  Instantaneous batch size per device = 4\n",
            "  Total train batch size (w. parallel, distributed & accumulation) = 4\n",
            "  Gradient Accumulation steps = 1\n",
            "  Total optimization steps = 54000\n",
            "Automatic Weights & Biases logging enabled, to disable set os.environ[\"WANDB_DISABLED\"] = \"true\"\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Currently logged in as: \u001b[33mjb810\u001b[0m. Use \u001b[1m`wandb login --relogin`\u001b[0m to force relogin\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Using wandb-core as the SDK backend.  Please refer to https://wandb.me/wandb-core for more information.\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Tracking run with wandb version 0.19.4\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Run data is saved locally in \u001b[35m\u001b[1m/content/UIE/wandb/run-20250118_173031-hpbdzeix\u001b[0m\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Run \u001b[1m`wandb offline`\u001b[0m to turn off syncing.\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Syncing run \u001b[33moutput/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1\u001b[0m\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: ⭐️ View project at \u001b[34m\u001b[4mhttps://wandb.ai/jb810/huggingface\u001b[0m\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: 🚀 View run at \u001b[34m\u001b[4mhttps://wandb.ai/jb810/huggingface/runs/hpbdzeix\u001b[0m\n",
            "{'loss': 0.9075, 'learning_rate': 7.716049382716049e-05, 'epoch': 3.7}\n",
            "{'loss': 0.207, 'learning_rate': 0.00015432098765432098, 'epoch': 7.41}\n",
            "{'loss': 0.1422, 'learning_rate': 0.0002314814814814815, 'epoch': 11.11}\n",
            "{'loss': 0.1123, 'learning_rate': 0.00030864197530864197, 'epoch': 14.81}\n",
            "{'loss': 0.1032, 'learning_rate': 0.00038580246913580245, 'epoch': 18.52}\n",
            "{'loss': 0.1126, 'learning_rate': 0.000462962962962963, 'epoch': 22.22}\n",
            "{'loss': 0.1121, 'learning_rate': 0.0004974389282899922, 'epoch': 25.93}\n",
            "{'loss': 0.0866, 'learning_rate': 0.0004925137903861308, 'epoch': 29.63}\n",
            "{'loss': 0.0804, 'learning_rate': 0.0004875886524822695, 'epoch': 33.33}\n",
            "{'loss': 0.0741, 'learning_rate': 0.0004826635145784082, 'epoch': 37.04}\n",
            "  9% 5000/54000 [26:51<6:38:54,  2.05it/s]***** Running Evaluation *****\n",
            "  Num examples = 35\n",
            "  Batch size = 16\n",
            "\n",
            "  0% 0/3 [00:00<?, ?it/s]\u001b[A\n",
            " 67% 2/3 [00:12<00:06,  6.24s/it]\u001b[A\n",
            "                                          \n",
            "\u001b[A{'eval_loss': 0.31040000915527344, 'eval_overall-F1': 79.0216, 'eval_spot-tp': 87.0, 'eval_spot-gold': 162.0, 'eval_spot-pred': 154.0, 'eval_spot-P': 56.4935, 'eval_spot-R': 53.7037, 'eval_spot-F1': 55.0633, 'eval_asoc-tp': 23.0, 'eval_asoc-gold': 97.0, 'eval_asoc-pred': 95.0, 'eval_asoc-P': 24.2105, 'eval_asoc-R': 23.7113, 'eval_asoc-F1': 23.9583, 'eval_record-tp': 46.0, 'eval_record-gold': 162.0, 'eval_record-pred': 154.0, 'eval_record-P': 29.8701, 'eval_record-R': 28.3951, 'eval_record-F1': 29.1139, 'eval_ordered-record-tp': 46.0, 'eval_ordered-record-gold': 162.0, 'eval_ordered-record-pred': 154.0, 'eval_ordered-record-P': 29.8701, 'eval_ordered-record-R': 28.3951, 'eval_ordered-record-F1': 29.1139, 'eval_gold_tree': 162, 'eval_pred_tree': 156, 'eval_well-formed': 35, 'eval_gen_len': 106.4286, 'eval_runtime': 30.011, 'eval_samples_per_second': 1.166, 'epoch': 37.04}\n",
            "  9% 5000/54000 [27:21<6:38:54,  2.05it/s]\n",
            "100% 3/3 [00:18<00:00,  6.00s/it]\u001b[A\n",
            "                                 \u001b[ASaving model checkpoint to output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-5000\n",
            "Configuration saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-5000/config.json\n",
            "Model weights saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-5000/pytorch_model.bin\n",
            "tokenizer config file saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-5000/tokenizer_config.json\n",
            "Special tokens file saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-5000/special_tokens_map.json\n",
            "Copy vocab file to output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-5000/spiece.model\n",
            "{'loss': 0.0774, 'learning_rate': 0.00047773837667454687, 'epoch': 40.74}\n",
            "{'loss': 0.0749, 'learning_rate': 0.0004728132387706856, 'epoch': 44.44}\n",
            "{'loss': 0.0737, 'learning_rate': 0.00046788810086682424, 'epoch': 48.15}\n",
            "{'loss': 0.0698, 'learning_rate': 0.000462962962962963, 'epoch': 51.85}\n",
            "{'loss': 0.0688, 'learning_rate': 0.00045803782505910167, 'epoch': 55.56}\n",
            "{'loss': 0.0668, 'learning_rate': 0.00045311268715524035, 'epoch': 59.26}\n",
            "{'loss': 0.066, 'learning_rate': 0.00044818754925137903, 'epoch': 62.96}\n",
            "{'loss': 0.0688, 'learning_rate': 0.0004432624113475177, 'epoch': 66.67}\n",
            "{'loss': 0.0708, 'learning_rate': 0.0004383372734436564, 'epoch': 70.37}\n",
            "{'loss': 0.0655, 'learning_rate': 0.00043341213553979514, 'epoch': 74.07}\n",
            " 19% 10000/54000 [54:40<5:56:36,  2.06it/s]***** Running Evaluation *****\n",
            "  Num examples = 35\n",
            "  Batch size = 16\n",
            "\n",
            "  0% 0/3 [00:00<?, ?it/s]\u001b[A\n",
            " 67% 2/3 [00:09<00:04,  4.92s/it]\u001b[A\n",
            "                                           \n",
            "\u001b[A{'eval_loss': 0.3688962161540985, 'eval_overall-F1': 72.1101, 'eval_spot-tp': 77.0, 'eval_spot-gold': 162.0, 'eval_spot-pred': 157.0, 'eval_spot-P': 49.0446, 'eval_spot-R': 47.5309, 'eval_spot-F1': 48.2759, 'eval_asoc-tp': 23.0, 'eval_asoc-gold': 97.0, 'eval_asoc-pred': 96.0, 'eval_asoc-P': 23.9583, 'eval_asoc-R': 23.7113, 'eval_asoc-F1': 23.8342, 'eval_record-tp': 41.0, 'eval_record-gold': 162.0, 'eval_record-pred': 157.0, 'eval_record-P': 26.1146, 'eval_record-R': 25.3086, 'eval_record-F1': 25.7053, 'eval_ordered-record-tp': 41.0, 'eval_ordered-record-gold': 162.0, 'eval_ordered-record-pred': 157.0, 'eval_ordered-record-P': 26.1146, 'eval_ordered-record-R': 25.3086, 'eval_ordered-record-F1': 25.7053, 'eval_gold_tree': 162, 'eval_pred_tree': 165, 'eval_well-formed': 35, 'eval_gen_len': 105.2571, 'eval_runtime': 25.4033, 'eval_samples_per_second': 1.378, 'epoch': 74.07}\n",
            " 19% 10000/54000 [55:05<5:56:36,  2.06it/s]\n",
            "100% 3/3 [00:16<00:00,  5.72s/it]\u001b[A\n",
            "                                 \u001b[ASaving model checkpoint to output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-10000\n",
            "Configuration saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-10000/config.json\n",
            "Model weights saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-10000/pytorch_model.bin\n",
            "tokenizer config file saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-10000/tokenizer_config.json\n",
            "Special tokens file saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-10000/special_tokens_map.json\n",
            "Copy vocab file to output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-10000/spiece.model\n",
            "{'loss': 0.0614, 'learning_rate': 0.0004284869976359338, 'epoch': 77.78}\n",
            "{'loss': 0.0604, 'learning_rate': 0.0004235618597320725, 'epoch': 81.48}\n",
            "{'loss': 0.0633, 'learning_rate': 0.0004186367218282112, 'epoch': 85.19}\n",
            "{'loss': 0.0619, 'learning_rate': 0.0004137115839243499, 'epoch': 88.89}\n",
            "{'loss': 0.064, 'learning_rate': 0.00040878644602048857, 'epoch': 92.59}\n",
            "{'loss': 0.0615, 'learning_rate': 0.00040386130811662725, 'epoch': 96.3}\n",
            "{'loss': 0.0596, 'learning_rate': 0.00039893617021276594, 'epoch': 100.0}\n",
            "{'loss': 0.0604, 'learning_rate': 0.0003940110323089047, 'epoch': 103.7}\n",
            "{'loss': 0.0599, 'learning_rate': 0.0003890858944050433, 'epoch': 107.41}\n",
            "{'loss': 0.0592, 'learning_rate': 0.00038416075650118205, 'epoch': 111.11}\n",
            " 28% 15000/54000 [1:22:23<5:18:35,  2.04it/s]***** Running Evaluation *****\n",
            "  Num examples = 35\n",
            "  Batch size = 16\n",
            "\n",
            "  0% 0/3 [00:00<?, ?it/s]\u001b[A\n",
            " 67% 2/3 [00:13<00:06,  6.84s/it]\u001b[A\n",
            "                                             \n",
            "\u001b[A{'eval_loss': 0.46435803174972534, 'eval_overall-F1': 63.269, 'eval_spot-tp': 69.0, 'eval_spot-gold': 162.0, 'eval_spot-pred': 153.0, 'eval_spot-P': 45.098, 'eval_spot-R': 42.5926, 'eval_spot-F1': 43.8095, 'eval_asoc-tp': 18.0, 'eval_asoc-gold': 97.0, 'eval_asoc-pred': 88.0, 'eval_asoc-P': 20.4545, 'eval_asoc-R': 18.5567, 'eval_asoc-F1': 19.4595, 'eval_record-tp': 32.0, 'eval_record-gold': 162.0, 'eval_record-pred': 153.0, 'eval_record-P': 20.915, 'eval_record-R': 19.7531, 'eval_record-F1': 20.3175, 'eval_ordered-record-tp': 32.0, 'eval_ordered-record-gold': 162.0, 'eval_ordered-record-pred': 153.0, 'eval_ordered-record-P': 20.915, 'eval_ordered-record-R': 19.7531, 'eval_ordered-record-F1': 20.3175, 'eval_gold_tree': 162, 'eval_pred_tree': 156, 'eval_well-formed': 35, 'eval_gen_len': 101.4857, 'eval_runtime': 30.5263, 'eval_samples_per_second': 1.147, 'epoch': 111.11}\n",
            " 28% 15000/54000 [1:22:53<5:18:35,  2.04it/s]\n",
            "100% 3/3 [00:19<00:00,  6.60s/it]\u001b[A\n",
            "                                 \u001b[ASaving model checkpoint to output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-15000\n",
            "Configuration saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-15000/config.json\n",
            "Model weights saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-15000/pytorch_model.bin\n",
            "tokenizer config file saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-15000/tokenizer_config.json\n",
            "Special tokens file saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-15000/special_tokens_map.json\n",
            "Copy vocab file to output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-15000/spiece.model\n",
            "Deleting older checkpoint [output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-10000] due to args.save_total_limit\n",
            "{'loss': 0.0612, 'learning_rate': 0.00037923561859732073, 'epoch': 114.81}\n",
            "{'loss': 0.0595, 'learning_rate': 0.0003743104806934594, 'epoch': 118.52}\n",
            "{'loss': 0.0587, 'learning_rate': 0.0003693853427895981, 'epoch': 122.22}\n",
            "{'loss': 0.059, 'learning_rate': 0.00036446020488573684, 'epoch': 125.93}\n",
            "{'loss': 0.058, 'learning_rate': 0.00035953506698187547, 'epoch': 129.63}\n",
            "{'loss': 0.0555, 'learning_rate': 0.0003546099290780142, 'epoch': 133.33}\n",
            "{'loss': 0.0564, 'learning_rate': 0.00034968479117415284, 'epoch': 137.04}\n",
            "{'loss': 0.0587, 'learning_rate': 0.0003447596532702916, 'epoch': 140.74}\n",
            "{'loss': 0.0586, 'learning_rate': 0.00033983451536643026, 'epoch': 144.44}\n",
            "{'loss': 0.0534, 'learning_rate': 0.00033490937746256895, 'epoch': 148.15}\n",
            " 37% 20000/54000 [1:50:23<4:42:42,  2.00it/s]***** Running Evaluation *****\n",
            "  Num examples = 35\n",
            "  Batch size = 16\n",
            "\n",
            "  0% 0/3 [00:00<?, ?it/s]\u001b[A\n",
            " 67% 2/3 [00:23<00:11, 11.94s/it]\u001b[A\n",
            "                                             \n",
            "\u001b[A{'eval_loss': 0.5344852209091187, 'eval_overall-F1': 57.363, 'eval_spot-tp': 71.0, 'eval_spot-gold': 162.0, 'eval_spot-pred': 166.0, 'eval_spot-P': 42.7711, 'eval_spot-R': 43.8272, 'eval_spot-F1': 43.2927, 'eval_asoc-tp': 14.0, 'eval_asoc-gold': 97.0, 'eval_asoc-pred': 102.0, 'eval_asoc-P': 13.7255, 'eval_asoc-R': 14.433, 'eval_asoc-F1': 14.0704, 'eval_record-tp': 34.0, 'eval_record-gold': 162.0, 'eval_record-pred': 166.0, 'eval_record-P': 20.4819, 'eval_record-R': 20.9877, 'eval_record-F1': 20.7317, 'eval_ordered-record-tp': 34.0, 'eval_ordered-record-gold': 162.0, 'eval_ordered-record-pred': 166.0, 'eval_ordered-record-P': 20.4819, 'eval_ordered-record-R': 20.9877, 'eval_ordered-record-F1': 20.7317, 'eval_gold_tree': 162, 'eval_pred_tree': 172, 'eval_well-formed': 35, 'eval_gen_len': 125.7714, 'eval_runtime': 41.7245, 'eval_samples_per_second': 0.839, 'epoch': 148.15}\n",
            " 37% 20000/54000 [1:51:05<4:42:42,  2.00it/s]\n",
            "100% 3/3 [00:30<00:00,  9.85s/it]\u001b[A\n",
            "                                 \u001b[ASaving model checkpoint to output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-20000\n",
            "Configuration saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-20000/config.json\n",
            "Model weights saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-20000/pytorch_model.bin\n",
            "tokenizer config file saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-20000/tokenizer_config.json\n",
            "Special tokens file saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-20000/special_tokens_map.json\n",
            "Copy vocab file to output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-20000/spiece.model\n",
            "Deleting older checkpoint [output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-15000] due to args.save_total_limit\n",
            "{'loss': 0.0571, 'learning_rate': 0.00032998423955870763, 'epoch': 151.85}\n",
            "{'loss': 0.056, 'learning_rate': 0.00032505910165484637, 'epoch': 155.56}\n",
            "{'loss': 0.0541, 'learning_rate': 0.000320133963750985, 'epoch': 159.26}\n",
            "{'loss': 0.0552, 'learning_rate': 0.00031520882584712374, 'epoch': 162.96}\n",
            "{'loss': 0.0545, 'learning_rate': 0.00031028368794326237, 'epoch': 166.67}\n",
            "{'loss': 0.0547, 'learning_rate': 0.0003053585500394011, 'epoch': 170.37}\n",
            "{'loss': 0.0584, 'learning_rate': 0.0003004334121355398, 'epoch': 174.07}\n",
            "{'loss': 0.0557, 'learning_rate': 0.0002955082742316785, 'epoch': 177.78}\n",
            "{'loss': 0.0548, 'learning_rate': 0.00029058313632781716, 'epoch': 181.48}\n",
            "{'loss': 0.0548, 'learning_rate': 0.0002856579984239559, 'epoch': 185.19}\n",
            " 46% 25000/54000 [2:18:24<3:59:47,  2.02it/s]***** Running Evaluation *****\n",
            "  Num examples = 35\n",
            "  Batch size = 16\n",
            "\n",
            "  0% 0/3 [00:00<?, ?it/s]\u001b[A\n",
            " 67% 2/3 [00:11<00:05,  5.72s/it]\u001b[A\n",
            "                                             \n",
            "\u001b[A{'eval_loss': 0.5314256548881531, 'eval_overall-F1': 64.1941, 'eval_spot-tp': 75.0, 'eval_spot-gold': 162.0, 'eval_spot-pred': 164.0, 'eval_spot-P': 45.7317, 'eval_spot-R': 46.2963, 'eval_spot-F1': 46.0123, 'eval_asoc-tp': 17.0, 'eval_asoc-gold': 97.0, 'eval_asoc-pred': 90.0, 'eval_asoc-P': 18.8889, 'eval_asoc-R': 17.5258, 'eval_asoc-F1': 18.1818, 'eval_record-tp': 44.0, 'eval_record-gold': 162.0, 'eval_record-pred': 164.0, 'eval_record-P': 26.8293, 'eval_record-R': 27.1605, 'eval_record-F1': 26.9939, 'eval_ordered-record-tp': 44.0, 'eval_ordered-record-gold': 162.0, 'eval_ordered-record-pred': 164.0, 'eval_ordered-record-P': 26.8293, 'eval_ordered-record-R': 27.1605, 'eval_ordered-record-F1': 26.9939, 'eval_gold_tree': 162, 'eval_pred_tree': 170, 'eval_well-formed': 35, 'eval_gen_len': 108.9429, 'eval_runtime': 31.2229, 'eval_samples_per_second': 1.121, 'epoch': 185.19}\n",
            " 46% 25000/54000 [2:18:56<3:59:47,  2.02it/s]\n",
            "100% 3/3 [00:19<00:00,  6.62s/it]\u001b[A\n",
            "                                 \u001b[ASaving model checkpoint to output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-25000\n",
            "Configuration saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-25000/config.json\n",
            "Model weights saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-25000/pytorch_model.bin\n",
            "tokenizer config file saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-25000/tokenizer_config.json\n",
            "Special tokens file saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-25000/special_tokens_map.json\n",
            "Copy vocab file to output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-25000/spiece.model\n",
            "Deleting older checkpoint [output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-20000] due to args.save_total_limit\n",
            "{'loss': 0.053, 'learning_rate': 0.00028073286052009453, 'epoch': 188.89}\n",
            "{'loss': 0.0553, 'learning_rate': 0.0002758077226162333, 'epoch': 192.59}\n",
            "{'loss': 0.0544, 'learning_rate': 0.0002708825847123719, 'epoch': 196.3}\n",
            "{'loss': 0.0538, 'learning_rate': 0.00026595744680851064, 'epoch': 200.0}\n",
            "{'loss': 0.0525, 'learning_rate': 0.00026103230890464933, 'epoch': 203.7}\n",
            "{'loss': 0.055, 'learning_rate': 0.000256107171000788, 'epoch': 207.41}\n",
            "{'loss': 0.0546, 'learning_rate': 0.0002511820330969267, 'epoch': 211.11}\n",
            "{'loss': 0.0521, 'learning_rate': 0.0002462568951930654, 'epoch': 214.81}\n",
            "{'loss': 0.0514, 'learning_rate': 0.0002413317572892041, 'epoch': 218.52}\n",
            "{'loss': 0.0509, 'learning_rate': 0.0002364066193853428, 'epoch': 222.22}\n",
            " 56% 30000/54000 [2:46:20<3:16:29,  2.04it/s]***** Running Evaluation *****\n",
            "  Num examples = 35\n",
            "  Batch size = 16\n",
            "\n",
            "  0% 0/3 [00:00<?, ?it/s]\u001b[A\n",
            " 67% 2/3 [00:12<00:06,  6.31s/it]\u001b[A\n",
            "                                             \n",
            "\u001b[A{'eval_loss': 0.5197909474372864, 'eval_overall-F1': 64.4274, 'eval_spot-tp': 74.0, 'eval_spot-gold': 162.0, 'eval_spot-pred': 163.0, 'eval_spot-P': 45.3988, 'eval_spot-R': 45.679, 'eval_spot-F1': 45.5385, 'eval_asoc-tp': 17.0, 'eval_asoc-gold': 97.0, 'eval_asoc-pred': 83.0, 'eval_asoc-P': 20.4819, 'eval_asoc-R': 17.5258, 'eval_asoc-F1': 18.8889, 'eval_record-tp': 40.0, 'eval_record-gold': 162.0, 'eval_record-pred': 163.0, 'eval_record-P': 24.5399, 'eval_record-R': 24.6914, 'eval_record-F1': 24.6154, 'eval_ordered-record-tp': 40.0, 'eval_ordered-record-gold': 162.0, 'eval_ordered-record-pred': 163.0, 'eval_ordered-record-P': 24.5399, 'eval_ordered-record-R': 24.6914, 'eval_ordered-record-F1': 24.6154, 'eval_gold_tree': 162, 'eval_pred_tree': 168, 'eval_well-formed': 35, 'eval_gen_len': 101.6571, 'eval_runtime': 28.2171, 'eval_samples_per_second': 1.24, 'epoch': 222.22}\n",
            " 56% 30000/54000 [2:46:48<3:16:29,  2.04it/s]\n",
            "100% 3/3 [00:19<00:00,  6.37s/it]\u001b[A\n",
            "                                 \u001b[ASaving model checkpoint to output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-30000\n",
            "Configuration saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-30000/config.json\n",
            "Model weights saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-30000/pytorch_model.bin\n",
            "tokenizer config file saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-30000/tokenizer_config.json\n",
            "Special tokens file saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-30000/special_tokens_map.json\n",
            "Copy vocab file to output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-30000/spiece.model\n",
            "Deleting older checkpoint [output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-25000] due to args.save_total_limit\n",
            "{'loss': 0.0534, 'learning_rate': 0.0002314814814814815, 'epoch': 225.93}\n",
            "{'loss': 0.0526, 'learning_rate': 0.00022655634357762018, 'epoch': 229.63}\n",
            "{'loss': 0.055, 'learning_rate': 0.00022163120567375886, 'epoch': 233.33}\n",
            "{'loss': 0.0527, 'learning_rate': 0.00021670606776989757, 'epoch': 237.04}\n",
            "{'loss': 0.0535, 'learning_rate': 0.00021178092986603626, 'epoch': 240.74}\n",
            "{'loss': 0.0501, 'learning_rate': 0.00020685579196217494, 'epoch': 244.44}\n",
            "{'loss': 0.0511, 'learning_rate': 0.00020193065405831363, 'epoch': 248.15}\n",
            "{'loss': 0.0529, 'learning_rate': 0.00019700551615445234, 'epoch': 251.85}\n",
            "{'loss': 0.0536, 'learning_rate': 0.00019208037825059102, 'epoch': 255.56}\n",
            "{'loss': 0.0533, 'learning_rate': 0.0001871552403467297, 'epoch': 259.26}\n",
            " 65% 35000/54000 [3:14:12<2:35:45,  2.03it/s]***** Running Evaluation *****\n",
            "  Num examples = 35\n",
            "  Batch size = 16\n",
            "\n",
            "  0% 0/3 [00:00<?, ?it/s]\u001b[A\n",
            " 67% 2/3 [00:13<00:06,  6.97s/it]\u001b[A\n",
            "                                             \n",
            "\u001b[A{'eval_loss': 0.6034294962882996, 'eval_overall-F1': 70.3491, 'eval_spot-tp': 73.0, 'eval_spot-gold': 162.0, 'eval_spot-pred': 162.0, 'eval_spot-P': 45.0617, 'eval_spot-R': 45.0617, 'eval_spot-F1': 45.0617, 'eval_asoc-tp': 22.0, 'eval_asoc-gold': 97.0, 'eval_asoc-pred': 77.0, 'eval_asoc-P': 28.5714, 'eval_asoc-R': 22.6804, 'eval_asoc-F1': 25.2874, 'eval_record-tp': 36.0, 'eval_record-gold': 162.0, 'eval_record-pred': 162.0, 'eval_record-P': 22.2222, 'eval_record-R': 22.2222, 'eval_record-F1': 22.2222, 'eval_ordered-record-tp': 36.0, 'eval_ordered-record-gold': 162.0, 'eval_ordered-record-pred': 162.0, 'eval_ordered-record-P': 22.2222, 'eval_ordered-record-R': 22.2222, 'eval_ordered-record-F1': 22.2222, 'eval_gold_tree': 162, 'eval_pred_tree': 166, 'eval_well-formed': 35, 'eval_gen_len': 101.5143, 'eval_runtime': 28.852, 'eval_samples_per_second': 1.213, 'epoch': 259.26}\n",
            " 65% 35000/54000 [3:14:41<2:35:45,  2.03it/s]\n",
            "100% 3/3 [00:20<00:00,  6.62s/it]\u001b[A\n",
            "                                 \u001b[ASaving model checkpoint to output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-35000\n",
            "Configuration saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-35000/config.json\n",
            "Model weights saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-35000/pytorch_model.bin\n",
            "tokenizer config file saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-35000/tokenizer_config.json\n",
            "Special tokens file saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-35000/special_tokens_map.json\n",
            "Copy vocab file to output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-35000/spiece.model\n",
            "Deleting older checkpoint [output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-30000] due to args.save_total_limit\n",
            "{'loss': 0.052, 'learning_rate': 0.00018223010244286842, 'epoch': 262.96}\n",
            "{'loss': 0.0525, 'learning_rate': 0.0001773049645390071, 'epoch': 266.67}\n",
            "{'loss': 0.0525, 'learning_rate': 0.0001723798266351458, 'epoch': 270.37}\n",
            "{'loss': 0.0526, 'learning_rate': 0.00016745468873128447, 'epoch': 274.07}\n",
            "{'loss': 0.0521, 'learning_rate': 0.00016252955082742319, 'epoch': 277.78}\n",
            "{'loss': 0.0523, 'learning_rate': 0.00015760441292356187, 'epoch': 281.48}\n",
            "{'loss': 0.0516, 'learning_rate': 0.00015267927501970056, 'epoch': 285.19}\n",
            "{'loss': 0.0506, 'learning_rate': 0.00014775413711583924, 'epoch': 288.89}\n",
            "{'loss': 0.0508, 'learning_rate': 0.00014282899921197795, 'epoch': 292.59}\n",
            "{'loss': 0.0522, 'learning_rate': 0.00013790386130811664, 'epoch': 296.3}\n",
            " 74% 40000/54000 [3:42:03<1:54:38,  2.04it/s]***** Running Evaluation *****\n",
            "  Num examples = 35\n",
            "  Batch size = 16\n",
            "\n",
            "  0% 0/3 [00:00<?, ?it/s]\u001b[A\n",
            " 67% 2/3 [00:13<00:06,  6.63s/it]\u001b[A\n",
            "                                             \n",
            "\u001b[A{'eval_loss': 0.643721878528595, 'eval_overall-F1': 65.9108, 'eval_spot-tp': 75.0, 'eval_spot-gold': 162.0, 'eval_spot-pred': 157.0, 'eval_spot-P': 47.7707, 'eval_spot-R': 46.2963, 'eval_spot-F1': 47.0219, 'eval_asoc-tp': 17.0, 'eval_asoc-gold': 97.0, 'eval_asoc-pred': 83.0, 'eval_asoc-P': 20.4819, 'eval_asoc-R': 17.5258, 'eval_asoc-F1': 18.8889, 'eval_record-tp': 41.0, 'eval_record-gold': 162.0, 'eval_record-pred': 157.0, 'eval_record-P': 26.1146, 'eval_record-R': 25.3086, 'eval_record-F1': 25.7053, 'eval_ordered-record-tp': 41.0, 'eval_ordered-record-gold': 162.0, 'eval_ordered-record-pred': 157.0, 'eval_ordered-record-P': 26.1146, 'eval_ordered-record-R': 25.3086, 'eval_ordered-record-F1': 25.7053, 'eval_gold_tree': 162, 'eval_pred_tree': 166, 'eval_well-formed': 35, 'eval_gen_len': 100.1714, 'eval_runtime': 30.9976, 'eval_samples_per_second': 1.129, 'epoch': 296.3}\n",
            " 74% 40000/54000 [3:42:34<1:54:38,  2.04it/s]\n",
            "100% 3/3 [00:20<00:00,  6.94s/it]\u001b[A\n",
            "                                 \u001b[ASaving model checkpoint to output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-40000\n",
            "Configuration saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-40000/config.json\n",
            "Model weights saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-40000/pytorch_model.bin\n",
            "tokenizer config file saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-40000/tokenizer_config.json\n",
            "Special tokens file saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-40000/special_tokens_map.json\n",
            "Copy vocab file to output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-40000/spiece.model\n",
            "Deleting older checkpoint [output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-35000] due to args.save_total_limit\n",
            "{'loss': 0.0509, 'learning_rate': 0.00013297872340425532, 'epoch': 300.0}\n",
            "{'loss': 0.0486, 'learning_rate': 0.000128053585500394, 'epoch': 303.7}\n",
            "{'loss': 0.0509, 'learning_rate': 0.0001231284475965327, 'epoch': 307.41}\n",
            "{'loss': 0.0499, 'learning_rate': 0.0001182033096926714, 'epoch': 311.11}\n",
            "{'loss': 0.0506, 'learning_rate': 0.00011327817178881009, 'epoch': 314.81}\n",
            "{'loss': 0.0504, 'learning_rate': 0.00010835303388494879, 'epoch': 318.52}\n",
            "{'loss': 0.0487, 'learning_rate': 0.00010342789598108747, 'epoch': 322.22}\n",
            "{'loss': 0.0508, 'learning_rate': 9.850275807722617e-05, 'epoch': 325.93}\n",
            "{'loss': 0.0512, 'learning_rate': 9.357762017336485e-05, 'epoch': 329.63}\n",
            "{'loss': 0.0488, 'learning_rate': 8.865248226950355e-05, 'epoch': 333.33}\n",
            " 83% 45000/54000 [4:09:53<1:13:56,  2.03it/s]***** Running Evaluation *****\n",
            "  Num examples = 35\n",
            "  Batch size = 16\n",
            "\n",
            "  0% 0/3 [00:00<?, ?it/s]\u001b[A\n",
            " 67% 2/3 [00:14<00:07,  7.48s/it]\u001b[A\n",
            "                                             \n",
            "\u001b[A{'eval_loss': 0.6380999088287354, 'eval_overall-F1': 65.759, 'eval_spot-tp': 76.0, 'eval_spot-gold': 162.0, 'eval_spot-pred': 147.0, 'eval_spot-P': 51.7007, 'eval_spot-R': 46.9136, 'eval_spot-F1': 49.1909, 'eval_asoc-tp': 14.0, 'eval_asoc-gold': 97.0, 'eval_asoc-pred': 72.0, 'eval_asoc-P': 19.4444, 'eval_asoc-R': 14.433, 'eval_asoc-F1': 16.568, 'eval_record-tp': 40.0, 'eval_record-gold': 162.0, 'eval_record-pred': 147.0, 'eval_record-P': 27.2109, 'eval_record-R': 24.6914, 'eval_record-F1': 25.89, 'eval_ordered-record-tp': 40.0, 'eval_ordered-record-gold': 162.0, 'eval_ordered-record-pred': 147.0, 'eval_ordered-record-P': 27.2109, 'eval_ordered-record-R': 24.6914, 'eval_ordered-record-F1': 25.89, 'eval_gold_tree': 162, 'eval_pred_tree': 149, 'eval_well-formed': 35, 'eval_gen_len': 96.3429, 'eval_runtime': 29.3344, 'eval_samples_per_second': 1.193, 'epoch': 333.33}\n",
            " 83% 45000/54000 [4:10:23<1:13:56,  2.03it/s]\n",
            "100% 3/3 [00:21<00:00,  6.97s/it]\u001b[A\n",
            "                                 \u001b[ASaving model checkpoint to output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-45000\n",
            "Configuration saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-45000/config.json\n",
            "Model weights saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-45000/pytorch_model.bin\n",
            "tokenizer config file saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-45000/tokenizer_config.json\n",
            "Special tokens file saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-45000/special_tokens_map.json\n",
            "Copy vocab file to output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-45000/spiece.model\n",
            "Deleting older checkpoint [output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-40000] due to args.save_total_limit\n",
            "{'loss': 0.05, 'learning_rate': 8.372734436564224e-05, 'epoch': 337.04}\n",
            "{'loss': 0.0508, 'learning_rate': 7.880220646178094e-05, 'epoch': 340.74}\n",
            "{'loss': 0.0494, 'learning_rate': 7.387706855791962e-05, 'epoch': 344.44}\n",
            "{'loss': 0.0493, 'learning_rate': 6.895193065405832e-05, 'epoch': 348.15}\n",
            "{'loss': 0.0492, 'learning_rate': 6.4026792750197e-05, 'epoch': 351.85}\n",
            "{'loss': 0.0496, 'learning_rate': 5.91016548463357e-05, 'epoch': 355.56}\n",
            "{'loss': 0.0518, 'learning_rate': 5.417651694247439e-05, 'epoch': 359.26}\n",
            "{'loss': 0.0497, 'learning_rate': 4.9251379038613084e-05, 'epoch': 362.96}\n",
            "{'loss': 0.0494, 'learning_rate': 4.4326241134751776e-05, 'epoch': 366.67}\n",
            "{'loss': 0.0503, 'learning_rate': 3.940110323089047e-05, 'epoch': 370.37}\n",
            " 93% 50000/54000 [4:37:47<33:09,  2.01it/s]***** Running Evaluation *****\n",
            "  Num examples = 35\n",
            "  Batch size = 16\n",
            "\n",
            "  0% 0/3 [00:00<?, ?it/s]\u001b[A\n",
            " 67% 2/3 [00:14<00:07,  7.50s/it]\u001b[A\n",
            "                                           \n",
            "\u001b[A{'eval_loss': 0.6845572590827942, 'eval_overall-F1': 68.6484, 'eval_spot-tp': 75.0, 'eval_spot-gold': 162.0, 'eval_spot-pred': 150.0, 'eval_spot-P': 50.0, 'eval_spot-R': 46.2963, 'eval_spot-F1': 48.0769, 'eval_asoc-tp': 18.0, 'eval_asoc-gold': 97.0, 'eval_asoc-pred': 78.0, 'eval_asoc-P': 23.0769, 'eval_asoc-R': 18.5567, 'eval_asoc-F1': 20.5714, 'eval_record-tp': 41.0, 'eval_record-gold': 162.0, 'eval_record-pred': 150.0, 'eval_record-P': 27.3333, 'eval_record-R': 25.3086, 'eval_record-F1': 26.2821, 'eval_ordered-record-tp': 41.0, 'eval_ordered-record-gold': 162.0, 'eval_ordered-record-pred': 150.0, 'eval_ordered-record-P': 27.3333, 'eval_ordered-record-R': 25.3086, 'eval_ordered-record-F1': 26.2821, 'eval_gold_tree': 162, 'eval_pred_tree': 165, 'eval_well-formed': 35, 'eval_gen_len': 95.7429, 'eval_runtime': 32.6173, 'eval_samples_per_second': 1.073, 'epoch': 370.37}\n",
            " 93% 50000/54000 [4:38:20<33:09,  2.01it/s]\n",
            "100% 3/3 [00:21<00:00,  7.00s/it]\u001b[A\n",
            "                                 \u001b[ASaving model checkpoint to output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-50000\n",
            "Configuration saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-50000/config.json\n",
            "Model weights saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-50000/pytorch_model.bin\n",
            "tokenizer config file saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-50000/tokenizer_config.json\n",
            "Special tokens file saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-50000/special_tokens_map.json\n",
            "Copy vocab file to output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-50000/spiece.model\n",
            "Deleting older checkpoint [output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-45000] due to args.save_total_limit\n",
            "{'loss': 0.0491, 'learning_rate': 3.447596532702916e-05, 'epoch': 374.07}\n",
            "{'loss': 0.051, 'learning_rate': 2.955082742316785e-05, 'epoch': 377.78}\n",
            "{'loss': 0.0504, 'learning_rate': 2.4625689519306542e-05, 'epoch': 381.48}\n",
            "{'loss': 0.0503, 'learning_rate': 1.9700551615445234e-05, 'epoch': 385.19}\n",
            "{'loss': 0.049, 'learning_rate': 1.4775413711583925e-05, 'epoch': 388.89}\n",
            "{'loss': 0.0499, 'learning_rate': 9.850275807722617e-06, 'epoch': 392.59}\n",
            "{'loss': 0.0489, 'learning_rate': 4.9251379038613084e-06, 'epoch': 396.3}\n",
            "{'loss': 0.0499, 'learning_rate': 0.0, 'epoch': 400.0}\n",
            "100% 54000/54000 [5:00:22<00:00,  2.00it/s]\n",
            "\n",
            "Training completed. Do not forget to share your model on huggingface.co/models =)\n",
            "\n",
            "\n",
            "Loading best model from output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/checkpoint-5000 (score: 79.0216).\n",
            "{'train_runtime': 18026.6228, 'train_samples_per_second': 2.996, 'epoch': 400.0}\n",
            "100% 54000/54000 [5:00:25<00:00,  3.00it/s]\n",
            "Saving model checkpoint to output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1\n",
            "Configuration saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/config.json\n",
            "Model weights saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/pytorch_model.bin\n",
            "tokenizer config file saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/tokenizer_config.json\n",
            "Special tokens file saved in output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/special_tokens_map.json\n",
            "Copy vocab file to output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/spiece.model\n",
            "01/18/2025 22:31:06 - INFO - __main__ -   ***** Train results *****\n",
            "01/18/2025 22:31:06 - INFO - __main__ -     epoch = 400.0\n",
            "01/18/2025 22:31:06 - INFO - __main__ -     train_runtime = 18026.6228\n",
            "01/18/2025 22:31:06 - INFO - __main__ -     train_samples_per_second = 2.996\n",
            "01/18/2025 22:31:06 - INFO - __main__ -   *** Evaluate ***\n",
            "***** Running Evaluation *****\n",
            "  Num examples = 35\n",
            "  Batch size = 16\n",
            "100% 3/3 [00:18<00:00,  6.12s/it]\n",
            "***** Running Prediction *****\n",
            "  Num examples = 35\n",
            "  Batch size = 16\n",
            "100% 3/3 [00:18<00:00,  6.04s/it]Traceback (most recent call last):\n",
            "  File \"run_uie_finetune.py\", line 520, in <module>\n",
            "    main()\n",
            "  File \"run_uie_finetune.py\", line 467, in main\n",
            "    with open(output_eval_file, \"w\") as writer:\n",
            "FileNotFoundError: [Errno 2] No such file or directory: 'output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/eval_results_seq2seq.txt'\n",
            "\u001b[1;34mwandb\u001b[0m: \n",
            "\u001b[1;34mwandb\u001b[0m: 🚀 View run \u001b[33moutput/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1\u001b[0m at: \u001b[34mhttps://wandb.ai/jb810/huggingface/runs/hpbdzeix\u001b[0m\n",
            "\u001b[1;34mwandb\u001b[0m: Find logs at: \u001b[1;35mwandb/run-20250118_173031-hpbdzeix/logs\u001b[0m\n",
            "Map Config config/offset_map/closest_offset_en.yaml\n",
            "output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/eval_preds_seq2seq.txt not found.\n",
            "\n",
            "output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/test_preds_seq2seq.txt not found.\n",
            "\n",
            "output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/eval_preds_record.txt not found.\n",
            "output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/test_preds_record.txt not found.\n",
            "===========> AVG <===========\n",
            "find: ‘output/meta_2025-01-18-17-30-28278_hf_models_uie-large-en_spotasoc_relation_GWAS_e400_linear_lr5e-4_ls0_b4_wu0.06_n-1_RP_sn0.1_an0.1_run1/’: No such file or directory\n"
          ]
        }
      ],
      "source": [
        "# Make sure you're in the top-level UIE folder\n",
        "%cd /content/UIE\n",
        "\n",
        "!bash run_uie_finetune.bash -v -d 0 \\\n",
        "  -b 4 \\\n",
        "  -k 1 \\\n",
        "  --lr 5e-4 \\\n",
        "  --warmup_ratio 0.06 \\\n",
        "  -i relation/GWAS \\\n",
        "  --epoch 400 \\\n",
        "  --spot_noise 0.1 \\\n",
        "  --asoc_noise 0.1 \\\n",
        "  -f spotasoc \\\n",
        "  --map_config config/offset_map/closest_offset_en.yaml \\\n",
        "  -m hf_models/uie-large-en \\\n",
        "  --random_prompt \\\n",
        "  --eval_steps 5000 \\\n",
        "  --lr_scheduler linear \\\n",
        "  --max_source_length 512 \\\n",
        "  --max_target_length 512 \\"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "metadata": {
        "id": "hXD_jn6jYMTr",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "f248f8d2-c03e-47a7-9b46-308b64836d34"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Mounted at /content/drive\n"
          ]
        }
      ],
      "source": [
        "from google.colab import drive\n",
        "drive.mount('/content/drive')\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "BtsiNTS9ViIp"
      },
      "outputs": [],
      "source": [
        "!cp -r /content/UIE/output/Model_V4 /content/drive/MyDrive"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!cp -r \"/content/drive/MyDrive/Model_V4\" \"/content/UIE/output\""
      ],
      "metadata": {
        "id": "ryJAXLSXp9Pu"
      },
      "execution_count": 7,
      "outputs": []
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "o9RX3HRvYfJM"
      },
      "outputs": [],
      "source": [
        "!mv \"/content/drive/MyDrive/UIE/pytorch_model.bin\" \"/content/UIE/output/Model_V3\"\n"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!python evaluate_uie_manual.py \\\n",
        "  --data_dir data/text2spotasoc/relation/GWAS \\\n",
        "  --model_path output/Model_V4 \\\n",
        "  --schema_file data/text2spotasoc/relation/GWAS/record.schema \\\n",
        "  --batch_size 32"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "nyztCuO9ymCA",
        "outputId": "21c9e280-5cb1-468a-c3b2-2c05747db24e"
      },
      "execution_count": 21,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "--- Evaluating on val.json from: data/text2spotasoc/relation/GWAS/val.json ---\n",
            "Inference on val: 100% 2/2 [00:19<00:00,  9.75s/it]\n",
            "[Cannot reconstruct]: linked DNA polymorphisms genome-wide ['By', 'coupling', 'these', 'events', 'with', 'linked', 'DNA', 'polymorphisms', 'genome', '-', 'wide', ',', 'the', 'phenotypic', 'effect', 'of', 'multiple', 'alleles', 'across', 'the', 'genome', 'can', 'be', 'tested', '.']\n",
            "\n",
            "[Cannot reconstruct]: [9] ['Because', 'of', 'this', 'serious', 'and', 'widespread', 'food', 'safety', 'concern', 'research', 'into', 'understanding', 'the', 'mechanisms', 'driving', 'arsenic', 'accumulation', 'in', 'plants', 'has', 'become', 'a', 'priority', '[', '9', ']', '.']\n",
            "\n",
            "[Cannot reconstruct]: M.T. et al., unpublished observations ['This', 'association', 'has', 'also', 'recently', 'been', 'experimentally', 'confirmed', '(', 'M.T.', 'et', 'al', '.', ',', 'unpublished', 'observations', ')', '.']\n",
            "\n",
            "[Cannot reconstruct]: sites with elevated soil Pb content (and acidic pH) ['We', 'observed', 'that', 'plants', 'coming', 'from', 'sites', 'with', 'elevated', 'soil', 'Pb', 'content', '(', 'and', 'acidic', 'pH', ')', 'had', 'better', 'capacity', 'to', 'tolerate', 'this', 'stress', '(', 'Figure', '2B', ')', '.']\n",
            "\n",
            "[Cannot reconstruct]: the (often much larger set of) strong associations ['This', 'is', 'an', 'upper', 'bound', 'because', 'it', 'seems', 'almost', 'certain', 'that', 'many', 'of', 'the', '(', 'often', 'much', 'larger', 'set', 'of', ')', 'strong', 'associations', 'that', 'are', 'not', 'close', 'to', 'a', '-', 'priori', 'candidates', 'will', 'also', 'turn', 'out', 'to', 'be', 'real', '.']\n",
            "\n",
            "[Cannot reconstruct]: a bi-parental cross ['However', ',', 'unlike', 'in', 'synthetic', 'RILs', 'made', 'from', 'a', 'bi', '-', 'parental', 'cross', ',', 'the', 'low', 'frequency', 'of', 'rare', 'alleles', 'in', 'a', 'natural', 'population', 'makes', 'it', 'difficult', 'to', 'detect', 'their', 'phenotypic', 'effect', 'using', 'GWA', 'mapping', '.']\n",
            "\n",
            "[Cannot reconstruct]: [28] ['This', 'natural', 'variation', 'has', 'been', 'used', 'to', 'identify', 'specific', 'genes', 'involved', 'in', 'controlling', 'variation', 'in', 'many', 'traits', '[', '28', ']', '.']\n",
            "\n",
            "[Cannot reconstruct]: ACCELERATED CELL DEATH 6 (ACD6) ['Finally', ',', 'ACCELERATED', 'CELL', 'DEATH', '6', '(', 'ACD6', ')', ',', 'which', 'has', 'been', 'experimentally', 'shown', 'to', 'be', 'directly', 'involved', 'in', 'lesioning', '[', '22', ']', ',', 'is', 'detected', 'here', 'as', 'associated', 'with', 'several', 'lesioning', 'and', 'chlorosis', 'phenotypes', '.']\n",
            "\n",
            "[Cannot reconstruct]: single, strong peaks of association ['Regardless', 'of', 'method', 'used', ',', 'six', 'phenotypes', 'yield', 'single', ',', 'strong', 'peaks', 'of', 'association', 'that', 'can', 'be', 'seen', 'by', 'inspection', '.']\n",
            "\n",
            "[Cannot reconstruct]: hac1-1 ['On', 'the', 'other', 'hand', ',', 'arsenate', 'uptake', ',', 'measured', 'as', 'depletion', 'of', 'arsenate', 'from', 'the', 'nutrient', 'solution', ',', 'was', 'no', 'different', 'in', 'hac1', '-', '1', ',', 'hac1', '-', '2', ',', 'or', 'Kr-0', 'compared', 'to', 'wild', '-', 'type', 'Col-0', 'at', 'both', 'time', 'points', 'tested', '(', 'Figure', '8B', ')', '.']\n",
            "\n",
            "[Cannot reconstruct]: reducing arsenate (As[V]) to arsenite (As[III] ['This', 'suggested', 'the', 'possibility', 'that', 'HAC1', 'functions', 'as', 'an', 'arsenate', 'reductase', ',', 'reducing', 'arsenate', '(', 'As[V', ']', ')', 'to', 'arsenite', '(', 'As[III', ']', ')', '.']\n",
            "\n",
            "[Cannot reconstruct]: Nd-1) ['Here', ',', 'we', 'expand', 'on', 'this', 'observation', 'by', 'establishing', 'the', 'strength', 'of', 'the', 'AtHKT1;1', 'alleles', 'in', 'four', 'further', 'high', 'Na[+', ']', 'accumulating', 'accessions', '(', 'Bur-1', ',', 'Duk', ',', 'PHW-20', 'and', 'UKNW06', '-', '386', ')', 'that', 'all', 'contain', 'a', 'T', 'at', 'Chr4:6392276', ',', 'along', 'with', 'a', 'low', 'leaf', 'Na[+', ']', 'accession', '(', 'Nd-1', ')', 'with', 'a', 'C', 'at', 'Chr4:6392276', '.']\n",
            "\n",
            "[Cannot reconstruct]: TRAM/LAG1/CLN8 lipid-sensing domain ['TRAM', '/', 'LAG1', '/', 'CLN8', 'lipid', '-', 'sensing', 'domain', 'has', 'been', 'well', 'described', 'in', 'humans', 'and', 'yeast', ',', 'where', 'it', 'is', 'associated', 'with', 'ER', '-', 'Golgi', 'trafficking', '(', 'Winter', 'and', 'Ponting', ',', '2002', ')', '.']\n",
            "\n",
            "[Cannot reconstruct]: this food-chain contaminant ['This', 'knowledge', 'creates', 'new', 'opportunities', 'to', 'limit', 'arsenic', 'accumulation', 'in', 'food', 'crops', ',', 'thereby', 'helping', 'to', 'reduce', 'the', 'cancer', 'risk', 'from', 'this', 'food', '-', 'chain', 'contaminant', '.']\n",
            "\n",
            "[Cannot reconstruct]: Pb-tolerant accessions ['In', 'particular', ',', 'our', 'goals', 'are', '(', '1', ')', 'to', 'elucidate', 'the', 'mechanisms', 'developed', 'by', 'Pb', '-', 'tolerant', 'accessions', 'that', 'differ', 'in', 'their', 'Pb', 'sequestration', 'strategies', ',', 'and', '(', '2', ')', 'to', 'stablish', 'the', 'role', 'of', 'the', 'novel', 'candidate', 'genes', 'identified', 'by', 'GWAs', 'in', 'Pb', 'tolerance', '.']\n",
            "\n",
            "[Cannot reconstruct]: hac1-1 ['However', ',', 'hac1', '-', '1', 'and', 'hac1', '-', '2', 'were', 'both', 'significantly', 'more', 'sensitive', 'to', 'arsenate', 'compared', 'to', 'the', 'Col-0', 'wild', 'type', 'at', 'relatively', 'high', 'concentrations', 'of', 'arsenate', 'when', 'growth', 'was', 'measured', 'as', 'root', 'length', 'or', 'overall', 'plant', 'fresh', 'weight', '(', 'Figure', '10', ')', '.']\n",
            "\n",
            "[Cannot reconstruct]: T-HLL ['On', 'the', 'other', 'hand', ',', 'both', 'T', '-', 'HLL', 'and', 'Pbsensitive', '(', 'S', ')', 'accessions', 'translocated', 'similar', 'Pb', 'concentrations', '(', 'Figure', '3F', ')', ',', 'suggesting', 'that', '(', 'S', ')', 'plants', 'are', 'not', 'able', 'to', 'maintain', 'the', 'Pb', 'out', 'of', 'the', 'cytosol', 'and', 'it', 'becomes', 'toxic', 'for', 'them', '.']\n",
            "\n",
            "\n",
            "=== Validation-Set Metrics ===\n",
            "Entities: TP=86, FP=55, FN=76 => Precision=0.6099, Recall=0.5309, F1=0.5677\n",
            "Relations: TP=12, FP=71, FN=85 => Precision=0.1446, Recall=0.1237, F1=0.1333\n",
            "\n",
            "[INFO] CSVs written to uie_manual_eval/\n",
            "\n",
            "--- Consistency Checks ---\n",
            "Entity FN in code: 76\n",
            "Rows in entity_fn.csv: 76\n",
            "[OK] The number of missed entities matches the number of rows in entity_fn.csv\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "df = pd.read_csv(\"uie_manual_eval/entity_fn.csv\")\n",
        "\n",
        "tp_count = 0\n",
        "fn_count = 0\n",
        "\n",
        "for index, row in df.iterrows():\n",
        "    # Extract columns of interest from the row\n",
        "    text = row.get(\"text\", \"\")\n",
        "    gold_entity_type = row.get(\"gold_entity_type\", \"\")\n",
        "    gold_entity_text = row.get(\"gold_entity_text\", \"\")\n",
        "    all_pred_entities = row.get(\"all_pred_entities\", \"\")\n",
        "\n",
        "    # Display\n",
        "    print(\"\\n---------------------------------------------------------\")\n",
        "    print(f\"Row index: {index}\")\n",
        "    print(\"Text: \", text)\n",
        "    print(\"Gold Entity Type: \", gold_entity_type)\n",
        "    print(\"Gold Entity Text: \", gold_entity_text)\n",
        "    print(\"All Predicted Entities: \", all_pred_entities)\n",
        "\n",
        "    # Ask user input: 'y' or 'n'\n",
        "    choice = input(\n",
        "        \"Is this gold entity sufficiently covered by one of the predicted entities? (y/n) \"\n",
        "    ).strip().lower()\n",
        "\n",
        "    if choice == 'y':\n",
        "        tp_count += 1\n",
        "    else:\n",
        "        fn_count += 1\n",
        "\n",
        "# Final summary\n",
        "print(\"\\n=== Manual Review Complete ===\")\n",
        "print(f\"TP Count (covered)   = {tp_count}\")\n",
        "print(f\"FN Count (not covered) = {fn_count}\")"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "YAH9qZYdFejR",
        "outputId": "7c522185-40ae-4e0b-ca50-a329e85e00cf"
      },
      "execution_count": 22,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 0\n",
            "Text:   Because slightly different sets were used in different phenotyping experiments, the total number of lines used was 199 (Supplementary Table 1).\n",
            "Gold Entity Type:  Study / Analysis\n",
            "Gold Entity Text:  phenotyping experiments\n",
            "All Predicted Entities:  [('Study / Analysis', 'different phenotyping experiments'), ('Population / Group', 'the total number of lines used'), ('Concept / Abstract Entity', 'Supplementary Table 1')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 1\n",
            "Text:   Because slightly different sets were used in different phenotyping experiments, the total number of lines used was 199 (Supplementary Table 1).\n",
            "Gold Entity Type:  Population / Group\n",
            "Gold Entity Text:  lines\n",
            "All Predicted Entities:  [('Study / Analysis', 'different phenotyping experiments'), ('Population / Group', 'the total number of lines used'), ('Concept / Abstract Entity', 'Supplementary Table 1')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 2\n",
            "Text:  By coupling these events with linked DNA polymorphisms genome-wide, the phenotypic effect of multiple alleles across the genome can be tested.\n",
            "Gold Entity Type:  Concept / Abstract Entity\n",
            "Gold Entity Text:  these events\n",
            "All Predicted Entities:  [('Biological Process', 'these events'), ('Allele / Variant', 'linked DNA polymorphisms genome-wide'), ('Trait / Phenotype', 'the phenotypic effect'), ('Allele / Variant', 'multiple alleles across the genome')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 3\n",
            "Text:  By coupling these events with linked DNA polymorphisms genome-wide, the phenotypic effect of multiple alleles across the genome can be tested.\n",
            "Gold Entity Type:  Allele / Variant\n",
            "Gold Entity Text:  linked DNA polymorphisms\n",
            "All Predicted Entities:  [('Biological Process', 'these events'), ('Allele / Variant', 'linked DNA polymorphisms genome-wide'), ('Trait / Phenotype', 'the phenotypic effect'), ('Allele / Variant', 'multiple alleles across the genome')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 4\n",
            "Text:  By coupling these events with linked DNA polymorphisms genome-wide, the phenotypic effect of multiple alleles across the genome can be tested.\n",
            "Gold Entity Type:  Allele / Variant\n",
            "Gold Entity Text:  multiple alleles\n",
            "All Predicted Entities:  [('Biological Process', 'these events'), ('Allele / Variant', 'linked DNA polymorphisms genome-wide'), ('Trait / Phenotype', 'the phenotypic effect'), ('Allele / Variant', 'multiple alleles across the genome')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 5\n",
            "Text:  By coupling these events with linked DNA polymorphisms genome-wide, the phenotypic effect of multiple alleles across the genome can be tested.\n",
            "Gold Entity Type:  Genomic Region\n",
            "Gold Entity Text:  the genome\n",
            "All Predicted Entities:  [('Biological Process', 'these events'), ('Allele / Variant', 'linked DNA polymorphisms genome-wide'), ('Trait / Phenotype', 'the phenotypic effect'), ('Allele / Variant', 'multiple alleles across the genome')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 6\n",
            "Text:  Because of this serious and widespread food safety concern research into understanding the mechanisms driving arsenic accumulation in plants has become a priority [9].\n",
            "Gold Entity Type:  Concept / Abstract Entity\n",
            "Gold Entity Text:  serious and widespread food safety concern\n",
            "All Predicted Entities:  [('Biological Process', 'the mechanisms'), ('Trait / Phenotype', 'arsenic accumulation in plants'), ('Study / Analysis', '[9]')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 7\n",
            "Text:  Because of this serious and widespread food safety concern research into understanding the mechanisms driving arsenic accumulation in plants has become a priority [9].\n",
            "Gold Entity Type:  Study / Analysis\n",
            "Gold Entity Text:  research into understanding the mechanisms driving arsenic accumulation in plants\n",
            "All Predicted Entities:  [('Biological Process', 'the mechanisms'), ('Trait / Phenotype', 'arsenic accumulation in plants'), ('Study / Analysis', '[9]')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 8\n",
            "Text:  This association has also recently been experimentally confirmed (M.T. et al., unpublished observations).\n",
            "Gold Entity Type:  Concept / Abstract Entity\n",
            "Gold Entity Text:  association\n",
            "All Predicted Entities:  [('Concept / Abstract Entity', 'This association'), ('Study / Analysis', 'M.T. et al., unpublished observations')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 9\n",
            "Text:  We observed that plants coming from sites with elevated soil Pb content (and acidic pH) had better capacity to tolerate this stress (Figure 2B).\n",
            "Gold Entity Type:  Environmental Factor\n",
            "Gold Entity Text:  elevated soil Pb content\n",
            "All Predicted Entities:  [('Population / Group', 'plants'), ('Environmental Factor', 'sites with elevated soil Pb content (and acidic pH)')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 10\n",
            "Text:  We observed that plants coming from sites with elevated soil Pb content (and acidic pH) had better capacity to tolerate this stress (Figure 2B).\n",
            "Gold Entity Type:  Environmental Factor\n",
            "Gold Entity Text:  acidic pH\n",
            "All Predicted Entities:  [('Population / Group', 'plants'), ('Environmental Factor', 'sites with elevated soil Pb content (and acidic pH)')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 11\n",
            "Text:  We observed that plants coming from sites with elevated soil Pb content (and acidic pH) had better capacity to tolerate this stress (Figure 2B).\n",
            "Gold Entity Type:  Trait / Phenotype\n",
            "Gold Entity Text:  better capacity to tolerate this stress\n",
            "All Predicted Entities:  [('Population / Group', 'plants'), ('Environmental Factor', 'sites with elevated soil Pb content (and acidic pH)')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 12\n",
            "Text:  This is an upper bound because it seems almost certain that many of the (often much larger set of) strong associations that are not close to a-priori candidates will also turn out to be real. \n",
            "Gold Entity Type:  Concept / Abstract Entity\n",
            "Gold Entity Text:  upper bound\n",
            "All Predicted Entities:  [('Concept / Abstract Entity', 'an upper bound'), ('Concept / Abstract Entity', 'the (often much larger set of) strong associations'), ('Concept / Abstract Entity', 'a-priori candidates')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 13\n",
            "Text:  This is an upper bound because it seems almost certain that many of the (often much larger set of) strong associations that are not close to a-priori candidates will also turn out to be real. \n",
            "Gold Entity Type:  Concept / Abstract Entity\n",
            "Gold Entity Text:  strong associations\n",
            "All Predicted Entities:  [('Concept / Abstract Entity', 'an upper bound'), ('Concept / Abstract Entity', 'the (often much larger set of) strong associations'), ('Concept / Abstract Entity', 'a-priori candidates')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 14\n",
            "Text:  At2g21045 was named HAC1.\n",
            "Gold Entity Type:  Gene\n",
            "Gold Entity Text:  HAC1\n",
            "All Predicted Entities:  [('Gene', 'At2g21045')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 15\n",
            "Text:  However, unlike in synthetic RILs made from a bi-parental cross, the low frequency of rare alleles in a natural population makes it difficult to detect their phenotypic effect using GWA mapping.\n",
            "Gold Entity Type:  Tool / Method\n",
            "Gold Entity Text:  bi-parental cross\n",
            "All Predicted Entities:  [('Population / Group', 'synthetic RILs'), ('Population / Group', 'a bi-parental cross'), ('Allele / Variant', 'rare alleles'), ('Population / Group', 'a natural population'), ('Trait / Phenotype', 'their phenotypic effect'), ('Tool / Method', 'GWA mapping')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 16\n",
            "Text:  However, unlike in synthetic RILs made from a bi-parental cross, the low frequency of rare alleles in a natural population makes it difficult to detect their phenotypic effect using GWA mapping.\n",
            "Gold Entity Type:  Trait / Phenotype\n",
            "Gold Entity Text:  phenotypic effect\n",
            "All Predicted Entities:  [('Population / Group', 'synthetic RILs'), ('Population / Group', 'a bi-parental cross'), ('Allele / Variant', 'rare alleles'), ('Population / Group', 'a natural population'), ('Trait / Phenotype', 'their phenotypic effect'), ('Tool / Method', 'GWA mapping')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 17\n",
            "Text:  However, unlike in synthetic RILs made from a bi-parental cross, the low frequency of rare alleles in a natural population makes it difficult to detect their phenotypic effect using GWA mapping.\n",
            "Gold Entity Type:  Study / Analysis\n",
            "Gold Entity Text:  GWA mapping\n",
            "All Predicted Entities:  [('Population / Group', 'synthetic RILs'), ('Population / Group', 'a bi-parental cross'), ('Allele / Variant', 'rare alleles'), ('Population / Group', 'a natural population'), ('Trait / Phenotype', 'their phenotypic effect'), ('Tool / Method', 'GWA mapping')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 18\n",
            "Text:  However, enzymes forming phosphorylated products can also promote arsenate reduction by incorrectly using arsenate in place of phosphate to generate arsenylated products in which the arsenate is more easily reduced by thiols such as glutathione [67].\n",
            "Gold Entity Type:  Protein / Polypeptide\n",
            "Gold Entity Text:  enzymes forming phosphorylated products\n",
            "All Predicted Entities:  [('Protein / Polypeptide', 'enzymes'), ('Chemical Substance', 'arsenate'), ('Chemical Substance', 'phosphate'), ('Biological Process', 'arsenate reduction'), ('Chemical Substance', 'arsenylated products'), ('Chemical Substance', 'glutathione')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 19\n",
            "Text:  However, enzymes forming phosphorylated products can also promote arsenate reduction by incorrectly using arsenate in place of phosphate to generate arsenylated products in which the arsenate is more easily reduced by thiols such as glutathione [67].\n",
            "Gold Entity Type:  Chemical Substance\n",
            "Gold Entity Text:  thiols\n",
            "All Predicted Entities:  [('Protein / Polypeptide', 'enzymes'), ('Chemical Substance', 'arsenate'), ('Chemical Substance', 'phosphate'), ('Biological Process', 'arsenate reduction'), ('Chemical Substance', 'arsenylated products'), ('Chemical Substance', 'glutathione')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 20\n",
            "Text:  Finally, ACCELERATED CELL DEATH 6 (ACD6), which has been experimentally shown to be directly involved in lesioning [22], is detected here as associated with several lesioning and chlorosis phenotypes. \n",
            "Gold Entity Type:  Trait / Phenotype\n",
            "Gold Entity Text:  lesioning and chlorosis phenotypes\n",
            "All Predicted Entities:  [('Gene', 'ACCELERATED CELL DEATH 6 (ACD6)'), ('Trait / Phenotype', 'lesioning'), ('Trait / Phenotype', 'several lesioning and chlorosis phenotypes')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 21\n",
            "Text:  Finally, ACCELERATED CELL DEATH 6 (ACD6), which has been experimentally shown to be directly involved in lesioning [22], is detected here as associated with several lesioning and chlorosis phenotypes. \n",
            "Gold Entity Type:  Trait / Phenotype\n",
            "Gold Entity Text:  chlorosis phenotypes\n",
            "All Predicted Entities:  [('Gene', 'ACCELERATED CELL DEATH 6 (ACD6)'), ('Trait / Phenotype', 'lesioning'), ('Trait / Phenotype', 'several lesioning and chlorosis phenotypes')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 22\n",
            "Text:  Arsenic concentrations were measured in leaves from 315 F2 plants (Figure 1C) from the Kr06Col-0 cross and approximately one-quarter of these plants (75 of 315, X[2] = 0.24,X[2]0.05(1) [= 3.84) had high leaf arsenic, suggest-] ing that high leaf arsenic in Kr-0 is controlled by a single major locus.\n",
            "Gold Entity Type:  Population / Group\n",
            "Gold Entity Text:  the Kr06Col-0 cross\n",
            "All Predicted Entities:  [('Trait / Phenotype', 'Arsenic concentrations'), ('Tissue / Plant Structure', 'leaves'), ('Population / Group', '315 F2 plants'), ('Population / Group', 'Kr06Col-0 cross'), ('Measurement / Statistical Result', 'X[2] = 0.24'), ('Trait / Phenotype', 'high leaf arsenic'), ('Genomic Region', 'a single major locus')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 23\n",
            "Text:  Arsenic concentrations were measured in leaves from 315 F2 plants (Figure 1C) from the Kr06Col-0 cross and approximately one-quarter of these plants (75 of 315, X[2] = 0.24,X[2]0.05(1) [= 3.84) had high leaf arsenic, suggest-] ing that high leaf arsenic in Kr-0 is controlled by a single major locus.\n",
            "Gold Entity Type:  Accession / Line\n",
            "Gold Entity Text:  Kr-0\n",
            "All Predicted Entities:  [('Trait / Phenotype', 'Arsenic concentrations'), ('Tissue / Plant Structure', 'leaves'), ('Population / Group', '315 F2 plants'), ('Population / Group', 'Kr06Col-0 cross'), ('Measurement / Statistical Result', 'X[2] = 0.24'), ('Trait / Phenotype', 'high leaf arsenic'), ('Genomic Region', 'a single major locus')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 24\n",
            "Text:  Arsenic concentrations were measured in leaves from 315 F2 plants (Figure 1C) from the Kr06Col-0 cross and approximately one-quarter of these plants (75 of 315, X[2] = 0.24,X[2]0.05(1) [= 3.84) had high leaf arsenic, suggest-] ing that high leaf arsenic in Kr-0 is controlled by a single major locus.\n",
            "Gold Entity Type:  Accession / Line\n",
            "Gold Entity Text:  Kr06Col-0\n",
            "All Predicted Entities:  [('Trait / Phenotype', 'Arsenic concentrations'), ('Tissue / Plant Structure', 'leaves'), ('Population / Group', '315 F2 plants'), ('Population / Group', 'Kr06Col-0 cross'), ('Measurement / Statistical Result', 'X[2] = 0.24'), ('Trait / Phenotype', 'high leaf arsenic'), ('Genomic Region', 'a single major locus')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 25\n",
            "Text:  Arsenic concentrations were measured in leaves from 315 F2 plants (Figure 1C) from the Kr06Col-0 cross and approximately one-quarter of these plants (75 of 315, X[2] = 0.24,X[2]0.05(1) [= 3.84) had high leaf arsenic, suggest-] ing that high leaf arsenic in Kr-0 is controlled by a single major locus.\n",
            "Gold Entity Type:  Measurement / Statistical Result\n",
            "Gold Entity Text:  75 of 315, X[2] = 0.24,X[2]0.05(1) [= 3.84\n",
            "All Predicted Entities:  [('Trait / Phenotype', 'Arsenic concentrations'), ('Tissue / Plant Structure', 'leaves'), ('Population / Group', '315 F2 plants'), ('Population / Group', 'Kr06Col-0 cross'), ('Measurement / Statistical Result', 'X[2] = 0.24'), ('Trait / Phenotype', 'high leaf arsenic'), ('Genomic Region', 'a single major locus')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 26\n",
            "Text:  Such imaging revealed that in the absence of a functional HAC1 gene, and with the loss of arsenite efflux, arsenic appears to accumulate mainly in the root stele to concentrations 120%–180% higher than in the Col-0 wild type (Figure 9).\n",
            "Gold Entity Type:  Measurement / Statistical Result\n",
            "Gold Entity Text:  concentrations 120%–180% higher than in the Col-0 wild type\n",
            "All Predicted Entities:  [('Gene', 'HAC1 gene'), ('Biological Process', 'arsenite efflux'), ('Chemical Substance', 'arsenic'), ('Tissue / Plant Structure', 'root stele'), ('Accession / Line', 'Col-0 wild type')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 27\n",
            "Text:  Regardless of method used, six phenotypes yield single, strong peaks of association that can be seen by inspection.\n",
            "Gold Entity Type:  Measurement / Statistical Result\n",
            "Gold Entity Text:  single, strong peaks of association\n",
            "All Predicted Entities:  [('Trait / Phenotype', 'six phenotypes'), ('Concept / Abstract Entity', 'single, strong peaks of association'), ('Tool / Method', 'method')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 28\n",
            "Text:  On the other hand, arsenate uptake, measured as depletion of arsenate from the nutrient solution, was no different in hac1-1, hac1-2, or Kr-0 compared to wild-type Col-0 at both time points tested (Figure 8B).\n",
            "Gold Entity Type:  Measurement / Statistical Result\n",
            "Gold Entity Text:  depletion of arsenate from the nutrient solution\n",
            "All Predicted Entities:  [('Trait / Phenotype', 'arsenate uptake'), ('Trait / Phenotype', 'depletion of arsenate from the nutrient solution'), ('Accession / Line', 'hac1-1'), ('Accession / Line', 'hac1-2'), ('Accession / Line', 'Kr-0'), ('Accession / Line', 'wild-type Col-0')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 29\n",
            "Text:  On the other hand, arsenate uptake, measured as depletion of arsenate from the nutrient solution, was no different in hac1-1, hac1-2, or Kr-0 compared to wild-type Col-0 at both time points tested (Figure 8B).\n",
            "Gold Entity Type:  Allele / Variant\n",
            "Gold Entity Text:  hac1-1\n",
            "All Predicted Entities:  [('Trait / Phenotype', 'arsenate uptake'), ('Trait / Phenotype', 'depletion of arsenate from the nutrient solution'), ('Accession / Line', 'hac1-1'), ('Accession / Line', 'hac1-2'), ('Accession / Line', 'Kr-0'), ('Accession / Line', 'wild-type Col-0')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 30\n",
            "Text:  On the other hand, arsenate uptake, measured as depletion of arsenate from the nutrient solution, was no different in hac1-1, hac1-2, or Kr-0 compared to wild-type Col-0 at both time points tested (Figure 8B).\n",
            "Gold Entity Type:  Allele / Variant\n",
            "Gold Entity Text:  hac1-2\n",
            "All Predicted Entities:  [('Trait / Phenotype', 'arsenate uptake'), ('Trait / Phenotype', 'depletion of arsenate from the nutrient solution'), ('Accession / Line', 'hac1-1'), ('Accession / Line', 'hac1-2'), ('Accession / Line', 'Kr-0'), ('Accession / Line', 'wild-type Col-0')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 31\n",
            "Text:  On the other hand, arsenate uptake, measured as depletion of arsenate from the nutrient solution, was no different in hac1-1, hac1-2, or Kr-0 compared to wild-type Col-0 at both time points tested (Figure 8B).\n",
            "Gold Entity Type:  Chemical Substance\n",
            "Gold Entity Text:  arsenate\n",
            "All Predicted Entities:  [('Trait / Phenotype', 'arsenate uptake'), ('Trait / Phenotype', 'depletion of arsenate from the nutrient solution'), ('Accession / Line', 'hac1-1'), ('Accession / Line', 'hac1-2'), ('Accession / Line', 'Kr-0'), ('Accession / Line', 'wild-type Col-0')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 32\n",
            "Text:  On the other hand, arsenate uptake, measured as depletion of arsenate from the nutrient solution, was no different in hac1-1, hac1-2, or Kr-0 compared to wild-type Col-0 at both time points tested (Figure 8B).\n",
            "Gold Entity Type:  Environmental Factor\n",
            "Gold Entity Text:  nutrient solution\n",
            "All Predicted Entities:  [('Trait / Phenotype', 'arsenate uptake'), ('Trait / Phenotype', 'depletion of arsenate from the nutrient solution'), ('Accession / Line', 'hac1-1'), ('Accession / Line', 'hac1-2'), ('Accession / Line', 'Kr-0'), ('Accession / Line', 'wild-type Col-0')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 33\n",
            "Text:  Both factors are likely to be important in the present study: for example, it is easy to imagine that plants from northern Sweden will tend to share cold-adaptive alleles at many causal loci as a result of selection, and marker alleles genome-wide as a result of demographic history.\n",
            "Gold Entity Type:  Allele / Variant\n",
            "Gold Entity Text:  cold-adaptive alleles\n",
            "All Predicted Entities:  [('Study / Analysis', 'the present study'), ('Population / Group', 'plants from northern Sweden'), ('Biological Process', 'selection'), ('Allele / Variant', 'marker alleles genome-wide')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 34\n",
            "Text:  Both factors are likely to be important in the present study: for example, it is easy to imagine that plants from northern Sweden will tend to share cold-adaptive alleles at many causal loci as a result of selection, and marker alleles genome-wide as a result of demographic history.\n",
            "Gold Entity Type:  Genomic Region\n",
            "Gold Entity Text:  many causal loci\n",
            "All Predicted Entities:  [('Study / Analysis', 'the present study'), ('Population / Group', 'plants from northern Sweden'), ('Biological Process', 'selection'), ('Allele / Variant', 'marker alleles genome-wide')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 35\n",
            "Text:  Both factors are likely to be important in the present study: for example, it is easy to imagine that plants from northern Sweden will tend to share cold-adaptive alleles at many causal loci as a result of selection, and marker alleles genome-wide as a result of demographic history.\n",
            "Gold Entity Type:  Concept / Abstract Entity\n",
            "Gold Entity Text:  demographic history\n",
            "All Predicted Entities:  [('Study / Analysis', 'the present study'), ('Population / Group', 'plants from northern Sweden'), ('Biological Process', 'selection'), ('Allele / Variant', 'marker alleles genome-wide')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 36\n",
            "Text:  In summary, A. thaliana has evolved two main strategies to cope with high substrate Pb: (1) exclusion from the shoot by sequestration in the roots and (2) translocation to the shoots and storage in inactive organelles or intracellular spaces.\n",
            "Gold Entity Type:  Environmental Factor\n",
            "Gold Entity Text:  high substrate Pb\n",
            "All Predicted Entities:  [('Organism', 'A. thaliana'), ('Chemical Substance', 'high substrate Pb'), ('Biological Process', 'exclusion from the shoot'), ('Biological Process', 'sequestration'), ('Tissue / Plant Structure', 'roots'), ('Biological Process', 'translocation to the shoots')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 37\n",
            "Text:  In summary, A. thaliana has evolved two main strategies to cope with high substrate Pb: (1) exclusion from the shoot by sequestration in the roots and (2) translocation to the shoots and storage in inactive organelles or intracellular spaces.\n",
            "Gold Entity Type:  Biological Process\n",
            "Gold Entity Text:  exclusion from the shoot by sequestration in the roots\n",
            "All Predicted Entities:  [('Organism', 'A. thaliana'), ('Chemical Substance', 'high substrate Pb'), ('Biological Process', 'exclusion from the shoot'), ('Biological Process', 'sequestration'), ('Tissue / Plant Structure', 'roots'), ('Biological Process', 'translocation to the shoots')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 38\n",
            "Text:  In summary, A. thaliana has evolved two main strategies to cope with high substrate Pb: (1) exclusion from the shoot by sequestration in the roots and (2) translocation to the shoots and storage in inactive organelles or intracellular spaces.\n",
            "Gold Entity Type:  Biological Process\n",
            "Gold Entity Text:  translocation to the shoots and storage in inactive organelles or intracellular spaces\n",
            "All Predicted Entities:  [('Organism', 'A. thaliana'), ('Chemical Substance', 'high substrate Pb'), ('Biological Process', 'exclusion from the shoot'), ('Biological Process', 'sequestration'), ('Tissue / Plant Structure', 'roots'), ('Biological Process', 'translocation to the shoots')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 39\n",
            "Text:  This suggested the possibility that HAC1 functions as an arsenate reductase, reducing arsenate (As[V]) to arsenite (As[III]).\n",
            "Gold Entity Type:  Chemical Substance\n",
            "Gold Entity Text:  arsenite (As[III])\n",
            "All Predicted Entities:  [('Gene', 'HAC1'), ('Protein / Polypeptide', 'arsenate reductase'), ('Chemical Substance', 'arsenate (As[V])'), ('Chemical Substance', 'arsenite (As[III]')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 40\n",
            "Text:  This suggested the possibility that HAC1 functions as an arsenate reductase, reducing arsenate (As[V]) to arsenite (As[III]).\n",
            "Gold Entity Type:  Biological Process\n",
            "Gold Entity Text:  reducing arsenate (As[V]) to arsenite (As[III])\n",
            "All Predicted Entities:  [('Gene', 'HAC1'), ('Protein / Polypeptide', 'arsenate reductase'), ('Chemical Substance', 'arsenate (As[V])'), ('Chemical Substance', 'arsenite (As[III]')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 41\n",
            "Text:  However, foliar uptake is an important second pathway in areas with high fallout of airborne particles (Uzu et al., 2010; Luo et al., 2019).\n",
            "Gold Entity Type:  Environmental Factor\n",
            "Gold Entity Text:  areas with high fallout of airborne particles\n",
            "All Predicted Entities:  [('Biological Process', 'foliar uptake'), ('Environmental Factor', 'high fallout of airborne particles')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 42\n",
            "Text:  However, foliar uptake is an important second pathway in areas with high fallout of airborne particles (Uzu et al., 2010; Luo et al., 2019).\n",
            "Gold Entity Type:  Study / Analysis\n",
            "Gold Entity Text:  Uzu et al., 2010\n",
            "All Predicted Entities:  [('Biological Process', 'foliar uptake'), ('Environmental Factor', 'high fallout of airborne particles')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 43\n",
            "Text:  However, foliar uptake is an important second pathway in areas with high fallout of airborne particles (Uzu et al., 2010; Luo et al., 2019).\n",
            "Gold Entity Type:  Study / Analysis\n",
            "Gold Entity Text:  Luo et al., 2019\n",
            "All Predicted Entities:  [('Biological Process', 'foliar uptake'), ('Environmental Factor', 'high fallout of airborne particles')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 44\n",
            "Text:  It is also interesting to note that since the majority of the A. thaliana accessions tested have the HAC1 allele associated with higher leaf arsenic, they are likely to have weak alleles of HAC1.\n",
            "Gold Entity Type:  Allele / Variant\n",
            "Gold Entity Text:  the HAC1 allele\n",
            "All Predicted Entities:  [('Population / Group', 'the majority of the A. thaliana accessions tested'), ('Allele / Variant', 'HAC1 allele'), ('Trait / Phenotype', 'higher leaf arsenic'), ('Allele / Variant', 'weak alleles of HAC1'), ('Gene', 'HAC1')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 45\n",
            "Text:  Here, we expand on this observation by establishing the strength of the AtHKT1;1 alleles in four further high Na[+] accumulating accessions (Bur-1, Duk, PHW-20 and UKNW06-386) that all contain a T at Chr4:6392276, along with a low leaf Na[+] accession (Nd-1) with a C at Chr4:6392276.\n",
            "Gold Entity Type:  Gene\n",
            "Gold Entity Text:  AtHKT1;1\n",
            "All Predicted Entities:  [('Allele / Variant', 'AtHKT1;1 alleles'), ('Accession / Line', 'Bur-1'), ('Accession / Line', 'PHW-20'), ('Accession / Line', 'Duk'), ('Accession / Line', 'Bur-1'), ('Accession / Line', 'Nd-1)'), ('Allele / Variant', 'T at Chr4:6392276'), ('Allele / Variant', 'C at Chr4:6392276')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 46\n",
            "Text:  Here, we expand on this observation by establishing the strength of the AtHKT1;1 alleles in four further high Na[+] accumulating accessions (Bur-1, Duk, PHW-20 and UKNW06-386) that all contain a T at Chr4:6392276, along with a low leaf Na[+] accession (Nd-1) with a C at Chr4:6392276.\n",
            "Gold Entity Type:  Accession / Line\n",
            "Gold Entity Text:  UKNW06-386\n",
            "All Predicted Entities:  [('Allele / Variant', 'AtHKT1;1 alleles'), ('Accession / Line', 'Bur-1'), ('Accession / Line', 'PHW-20'), ('Accession / Line', 'Duk'), ('Accession / Line', 'Bur-1'), ('Accession / Line', 'Nd-1)'), ('Allele / Variant', 'T at Chr4:6392276'), ('Allele / Variant', 'C at Chr4:6392276')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 47\n",
            "Text:  Here, we expand on this observation by establishing the strength of the AtHKT1;1 alleles in four further high Na[+] accumulating accessions (Bur-1, Duk, PHW-20 and UKNW06-386) that all contain a T at Chr4:6392276, along with a low leaf Na[+] accession (Nd-1) with a C at Chr4:6392276.\n",
            "Gold Entity Type:  Accession / Line\n",
            "Gold Entity Text:  Nd-1\n",
            "All Predicted Entities:  [('Allele / Variant', 'AtHKT1;1 alleles'), ('Accession / Line', 'Bur-1'), ('Accession / Line', 'PHW-20'), ('Accession / Line', 'Duk'), ('Accession / Line', 'Bur-1'), ('Accession / Line', 'Nd-1)'), ('Allele / Variant', 'T at Chr4:6392276'), ('Allele / Variant', 'C at Chr4:6392276')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 48\n",
            "Text:  Here, we expand on this observation by establishing the strength of the AtHKT1;1 alleles in four further high Na[+] accumulating accessions (Bur-1, Duk, PHW-20 and UKNW06-386) that all contain a T at Chr4:6392276, along with a low leaf Na[+] accession (Nd-1) with a C at Chr4:6392276.\n",
            "Gold Entity Type:  Trait / Phenotype\n",
            "Gold Entity Text:  high Na[+] accumulating\n",
            "All Predicted Entities:  [('Allele / Variant', 'AtHKT1;1 alleles'), ('Accession / Line', 'Bur-1'), ('Accession / Line', 'PHW-20'), ('Accession / Line', 'Duk'), ('Accession / Line', 'Bur-1'), ('Accession / Line', 'Nd-1)'), ('Allele / Variant', 'T at Chr4:6392276'), ('Allele / Variant', 'C at Chr4:6392276')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 49\n",
            "Text:  Here, we expand on this observation by establishing the strength of the AtHKT1;1 alleles in four further high Na[+] accumulating accessions (Bur-1, Duk, PHW-20 and UKNW06-386) that all contain a T at Chr4:6392276, along with a low leaf Na[+] accession (Nd-1) with a C at Chr4:6392276.\n",
            "Gold Entity Type:  Trait / Phenotype\n",
            "Gold Entity Text:  low leaf Na[+]\n",
            "All Predicted Entities:  [('Allele / Variant', 'AtHKT1;1 alleles'), ('Accession / Line', 'Bur-1'), ('Accession / Line', 'PHW-20'), ('Accession / Line', 'Duk'), ('Accession / Line', 'Bur-1'), ('Accession / Line', 'Nd-1)'), ('Allele / Variant', 'T at Chr4:6392276'), ('Allele / Variant', 'C at Chr4:6392276')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 50\n",
            "Text:  In the hac1 A. thaliana mutant that lacks the arsenate reductase activity required for arsenite efflux, we observe an over-accumulation of arsenic within the central stele of the root. \n",
            "Gold Entity Type:  Biological Process\n",
            "Gold Entity Text:  arsenate reductase activity\n",
            "All Predicted Entities:  [('Accession / Line', 'hac1 A. thaliana mutant'), ('Protein / Polypeptide', 'arsenate reductase activity'), ('Biological Process', 'arsenite efflux'), ('Chemical Substance', 'arsenic'), ('Tissue / Plant Structure', 'central stele of the root')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 51\n",
            "Text:  In the hac1 A. thaliana mutant that lacks the arsenate reductase activity required for arsenite efflux, we observe an over-accumulation of arsenic within the central stele of the root. \n",
            "Gold Entity Type:  Trait / Phenotype\n",
            "Gold Entity Text:  over-accumulation of arsenic\n",
            "All Predicted Entities:  [('Accession / Line', 'hac1 A. thaliana mutant'), ('Protein / Polypeptide', 'arsenate reductase activity'), ('Biological Process', 'arsenite efflux'), ('Chemical Substance', 'arsenic'), ('Tissue / Plant Structure', 'central stele of the root')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 52\n",
            "Text:  In the hac1 A. thaliana mutant that lacks the arsenate reductase activity required for arsenite efflux, we observe an over-accumulation of arsenic within the central stele of the root. \n",
            "Gold Entity Type:  Tissue / Plant Structure\n",
            "Gold Entity Text:  central stele\n",
            "All Predicted Entities:  [('Accession / Line', 'hac1 A. thaliana mutant'), ('Protein / Polypeptide', 'arsenate reductase activity'), ('Biological Process', 'arsenite efflux'), ('Chemical Substance', 'arsenic'), ('Tissue / Plant Structure', 'central stele of the root')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 53\n",
            "Text:  In the hac1 A. thaliana mutant that lacks the arsenate reductase activity required for arsenite efflux, we observe an over-accumulation of arsenic within the central stele of the root. \n",
            "Gold Entity Type:  Tissue / Plant Structure\n",
            "Gold Entity Text:  root\n",
            "All Predicted Entities:  [('Accession / Line', 'hac1 A. thaliana mutant'), ('Protein / Polypeptide', 'arsenate reductase activity'), ('Biological Process', 'arsenite efflux'), ('Chemical Substance', 'arsenic'), ('Tissue / Plant Structure', 'central stele of the root')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 54\n",
            "Text:  Pb entering the symplast is removed by the activity of efflux pumps present in the plasma membrane.\n",
            "Gold Entity Type:  Protein / Polypeptide\n",
            "Gold Entity Text:  efflux pumps\n",
            "All Predicted Entities:  [('Chemical Substance', 'Pb'), ('Tissue / Plant Structure', 'symplast'), ('Tissue / Plant Structure', 'plasma membrane'), ('Biological Process', 'efflux pumps')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 55\n",
            "Text:  In addition, Na[+]  compartmentalization facilitates vacuolar osmotic adjustment that is necessary to compensate for the osmotic effects of salinity by maintaining turgor pressure for cell expansion and growth.\n",
            "Gold Entity Type:  Biological Process\n",
            "Gold Entity Text:  Na[+]  compartmentalization\n",
            "All Predicted Entities:  [('Biological Process', 'vacuolar osmotic adjustment'), ('Chemical Substance', 'salinity'), ('Trait / Phenotype', 'turgor pressure'), ('Biological Process', 'growth')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 56\n",
            "Text:  In addition, Na[+]  compartmentalization facilitates vacuolar osmotic adjustment that is necessary to compensate for the osmotic effects of salinity by maintaining turgor pressure for cell expansion and growth.\n",
            "Gold Entity Type:  Environmental Factor\n",
            "Gold Entity Text:  salinity\n",
            "All Predicted Entities:  [('Biological Process', 'vacuolar osmotic adjustment'), ('Chemical Substance', 'salinity'), ('Trait / Phenotype', 'turgor pressure'), ('Biological Process', 'growth')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 57\n",
            "Text:  In addition, Na[+]  compartmentalization facilitates vacuolar osmotic adjustment that is necessary to compensate for the osmotic effects of salinity by maintaining turgor pressure for cell expansion and growth.\n",
            "Gold Entity Type:  Biological Process\n",
            "Gold Entity Text:  cell expansion and growth\n",
            "All Predicted Entities:  [('Biological Process', 'vacuolar osmotic adjustment'), ('Chemical Substance', 'salinity'), ('Trait / Phenotype', 'turgor pressure'), ('Biological Process', 'growth')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 58\n",
            "Text:  Whether the wide variety of natural phenotypic and genetic variation observed in A. thaliana [12] contributes to its local adaptation is an important unresolved question that is currently attracting a significant amount of attention [13].\n",
            "Gold Entity Type:  Trait / Phenotype\n",
            "Gold Entity Text:  natural phenotypic and\n",
            "All Predicted Entities:  [('Concept / Abstract Entity', 'the wide variety of natural phenotypic and genetic variation'), ('Organism', 'A. thaliana'), ('Biological Process', 'local adaptation')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 59\n",
            "Text:  Whether the wide variety of natural phenotypic and genetic variation observed in A. thaliana [12] contributes to its local adaptation is an important unresolved question that is currently attracting a significant amount of attention [13].\n",
            "Gold Entity Type:  Concept / Abstract Entity\n",
            "Gold Entity Text:  genetic variation\n",
            "All Predicted Entities:  [('Concept / Abstract Entity', 'the wide variety of natural phenotypic and genetic variation'), ('Organism', 'A. thaliana'), ('Biological Process', 'local adaptation')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 60\n",
            "Text:  This increase in HAC1 mRNA levels reaches a maximum at 100 mM arsenate in the growth media (Figure 7A).\n",
            "Gold Entity Type:  Trait / Phenotype\n",
            "Gold Entity Text:  HAC1 mRNA levels\n",
            "All Predicted Entities:  [('Trait / Phenotype', 'increase in HAC1 mRNA levels'), ('Chemical Substance', '100 mM arsenate'), ('Environmental Factor', 'growth media')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 61\n",
            "Text:  To validate the existence of this leaf arsenic QTL identified by GWA analysis we created a synthetic F2 recombinant population in which alternative alleles of the dialleic SNP Chr2:9008060 were segregating.\n",
            "Gold Entity Type:  Allele / Variant\n",
            "Gold Entity Text:  leaf arsenic QTL\n",
            "All Predicted Entities:  [('Trait / Phenotype', 'leaf arsenic QTL'), ('Study / Analysis', 'GWA analysis'), ('Population / Group', 'a synthetic F2 recombinant population'), ('Allele / Variant', 'alternative alleles of the dialleic SNP Chr2:9008060'), ('Allele / Variant', 'the dialleic SNP Chr2:9008060')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 62\n",
            "Text:  To validate the existence of this leaf arsenic QTL identified by GWA analysis we created a synthetic F2 recombinant population in which alternative alleles of the dialleic SNP Chr2:9008060 were segregating.\n",
            "Gold Entity Type:  Population / Group\n",
            "Gold Entity Text:  synthetic F2 recombinant population\n",
            "All Predicted Entities:  [('Trait / Phenotype', 'leaf arsenic QTL'), ('Study / Analysis', 'GWA analysis'), ('Population / Group', 'a synthetic F2 recombinant population'), ('Allele / Variant', 'alternative alleles of the dialleic SNP Chr2:9008060'), ('Allele / Variant', 'the dialleic SNP Chr2:9008060')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 63\n",
            "Text:  This knowledge creates new opportunities to limit arsenic accumulation in food crops, thereby helping to reduce the cancer risk from this food-chain contaminant.\n",
            "Gold Entity Type:  Trait / Phenotype\n",
            "Gold Entity Text:  arsenic accumulation\n",
            "All Predicted Entities:  [('Concept / Abstract Entity', 'This knowledge'), ('Trait / Phenotype', 'arsenic accumulation in food crops'), ('Chemical Substance', 'this food-chain contaminant')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 64\n",
            "Text:  This knowledge creates new opportunities to limit arsenic accumulation in food crops, thereby helping to reduce the cancer risk from this food-chain contaminant.\n",
            "Gold Entity Type:  Population / Group\n",
            "Gold Entity Text:  food crops\n",
            "All Predicted Entities:  [('Concept / Abstract Entity', 'This knowledge'), ('Trait / Phenotype', 'arsenic accumulation in food crops'), ('Chemical Substance', 'this food-chain contaminant')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 65\n",
            "Text:  This knowledge creates new opportunities to limit arsenic accumulation in food crops, thereby helping to reduce the cancer risk from this food-chain contaminant.\n",
            "Gold Entity Type:  Concept / Abstract Entity\n",
            "Gold Entity Text:  cancer risk\n",
            "All Predicted Entities:  [('Concept / Abstract Entity', 'This knowledge'), ('Trait / Phenotype', 'arsenic accumulation in food crops'), ('Chemical Substance', 'this food-chain contaminant')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 66\n",
            "Text:  This knowledge creates new opportunities to limit arsenic accumulation in food crops, thereby helping to reduce the cancer risk from this food-chain contaminant.\n",
            "Gold Entity Type:  Chemical Substance\n",
            "Gold Entity Text:  food-chain contaminant\n",
            "All Predicted Entities:  [('Concept / Abstract Entity', 'This knowledge'), ('Trait / Phenotype', 'arsenic accumulation in food crops'), ('Chemical Substance', 'this food-chain contaminant')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 67\n",
            "Text:  If polymorphism at FRI is taken into account, CRP is no longer significantly associated, but this does not necessarily mean that the association is spurious.\n",
            "Gold Entity Type:  Protein / Polypeptide\n",
            "Gold Entity Text:  CRP\n",
            "All Predicted Entities:  [('Allele / Variant', 'polymorphism at FRI'), ('Gene', 'CRP'), ('Concept / Abstract Entity', 'the association')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 68\n",
            "Text:  However, hac1-1 and hac1-2 were both significantly more sensitive to arsenate compared to the Col-0 wild type at relatively high concentrations of arsenate when growth was measured as root length or overall plant fresh weight (Figure 10).\n",
            "Gold Entity Type:  Trait / Phenotype\n",
            "Gold Entity Text:  more sensitive to arsenate\n",
            "All Predicted Entities:  [('Allele / Variant', 'hac1-1'), ('Allele / Variant', 'hac1-2'), ('Accession / Line', 'Col-0 wild type'), ('Chemical Substance', 'arsenate'), ('Trait / Phenotype', 'significantly more sensitive')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 69\n",
            "Text:  However, hac1-1 and hac1-2 were both significantly more sensitive to arsenate compared to the Col-0 wild type at relatively high concentrations of arsenate when growth was measured as root length or overall plant fresh weight (Figure 10).\n",
            "Gold Entity Type:  Environmental Factor\n",
            "Gold Entity Text:  relatively high concentrations of arsenate\n",
            "All Predicted Entities:  [('Allele / Variant', 'hac1-1'), ('Allele / Variant', 'hac1-2'), ('Accession / Line', 'Col-0 wild type'), ('Chemical Substance', 'arsenate'), ('Trait / Phenotype', 'significantly more sensitive')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 70\n",
            "Text:  However, hac1-1 and hac1-2 were both significantly more sensitive to arsenate compared to the Col-0 wild type at relatively high concentrations of arsenate when growth was measured as root length or overall plant fresh weight (Figure 10).\n",
            "Gold Entity Type:  Trait / Phenotype\n",
            "Gold Entity Text:  root length\n",
            "All Predicted Entities:  [('Allele / Variant', 'hac1-1'), ('Allele / Variant', 'hac1-2'), ('Accession / Line', 'Col-0 wild type'), ('Chemical Substance', 'arsenate'), ('Trait / Phenotype', 'significantly more sensitive')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 71\n",
            "Text:  However, hac1-1 and hac1-2 were both significantly more sensitive to arsenate compared to the Col-0 wild type at relatively high concentrations of arsenate when growth was measured as root length or overall plant fresh weight (Figure 10).\n",
            "Gold Entity Type:  Trait / Phenotype\n",
            "Gold Entity Text:  overall plant fresh weight\n",
            "All Predicted Entities:  [('Allele / Variant', 'hac1-1'), ('Allele / Variant', 'hac1-2'), ('Accession / Line', 'Col-0 wild type'), ('Chemical Substance', 'arsenate'), ('Trait / Phenotype', 'significantly more sensitive')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 72\n",
            "Text:  On the other hand, both T-HLL and Pbsensitive (S) accessions translocated similar Pb concentrations (Figure 3F), suggesting that (S) plants are not able to maintain the Pb out of the cytosol and it becomes toxic for them.\n",
            "Gold Entity Type:  Trait / Phenotype\n",
            "Gold Entity Text:  Pb concentrations\n",
            "All Predicted Entities:  [('Accession / Line', 'T-HLL'), ('Accession / Line', 'Pbsensitive (S) accessions'), ('Chemical Substance', 'Pb'), ('Tissue / Plant Structure', 'cytosol')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 73\n",
            "Text:  On the other hand, both T-HLL and Pbsensitive (S) accessions translocated similar Pb concentrations (Figure 3F), suggesting that (S) plants are not able to maintain the Pb out of the cytosol and it becomes toxic for them.\n",
            "Gold Entity Type:  Trait / Phenotype\n",
            "Gold Entity Text:  toxic\n",
            "All Predicted Entities:  [('Accession / Line', 'T-HLL'), ('Accession / Line', 'Pbsensitive (S) accessions'), ('Chemical Substance', 'Pb'), ('Tissue / Plant Structure', 'cytosol')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 74\n",
            "Text:  To test this hypothesis, we evaluated the oxidation state of arsenic in plants lacking a functional HAC1 gene.\n",
            "Gold Entity Type:  Trait / Phenotype\n",
            "Gold Entity Text:  the oxidation state of arsenic\n",
            "All Predicted Entities:  [('Trait / Phenotype', 'oxidation state of arsenic'), ('Chemical Substance', 'arsenic'), ('Population / Group', 'plants lacking a functional HAC1 gene')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 75\n",
            "Text:  To test this hypothesis, we evaluated the oxidation state of arsenic in plants lacking a functional HAC1 gene.\n",
            "Gold Entity Type:  Gene\n",
            "Gold Entity Text:  HAC1 gene\n",
            "All Predicted Entities:  [('Trait / Phenotype', 'oxidation state of arsenic'), ('Chemical Substance', 'arsenic'), ('Population / Group', 'plants lacking a functional HAC1 gene')]\n",
            "Is this gold entity sufficiently covered by one of the predicted entities? (y/n) y\n",
            "\n",
            "=== Manual Review Complete ===\n",
            "TP Count (covered)   = 39\n",
            "FN Count (not covered) = 37\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "\"\"\"\n",
        "Reads 'entity_fp.csv' and for each row:\n",
        "  - Prints out relevant columns.\n",
        "  - Asks user if the predicted entity is valid.\n",
        "      * 'y' => increment tp_extra_count\n",
        "      * 'n' => increment fp_count\n",
        "      * 'c' => increment covered_count\n",
        "Finally, prints the totals.\n",
        "\"\"\"\n",
        "df = pd.read_csv(\"uie_manual_eval/entity_fp.csv\")\n",
        "\n",
        "tp_extra_count = 0  # how many FPs were actually correct (missing from gold)\n",
        "fp_count = 0        # how many FPs are truly invalid\n",
        "covered_count = 0   # how many FPs are partially covered by gold in some other way\n",
        "\n",
        "for index, row in df.iterrows():\n",
        "    # Extract columns of interest from the row\n",
        "    text = row.get(\"text\", \"\")\n",
        "    pred_entity_type = row.get(\"pred_entity_type\", \"\")\n",
        "    pred_entity_text = row.get(\"pred_entity_text\", \"\")\n",
        "    all_gold_entities = row.get(\"all_gold_entities\", \"\")\n",
        "\n",
        "    # Display\n",
        "    print(\"\\n---------------------------------------------------------\")\n",
        "    print(f\"Row index: {index}\")\n",
        "    print(\"Text: \", text)\n",
        "    print(\"Predicted Entity Type: \", pred_entity_type)\n",
        "    print(\"Predicted Entity Text: \", pred_entity_text)\n",
        "    print(\"All Gold Entities: \", all_gold_entities)\n",
        "\n",
        "    # Prompt user\n",
        "    choice = input(\n",
        "        \"[y] if the prediction is valid and should have been in gold\\n\"\n",
        "        \"[n] if the prediction is truly wrong\\n\"\n",
        "        \"[c] if it is covered by an existing gold entity already\\n\"\n",
        "        \"=> Your choice (y/n/c)? \"\n",
        "    ).strip().lower()\n",
        "\n",
        "    if choice == 'y':\n",
        "        tp_extra_count += 1\n",
        "    elif choice == 'c':\n",
        "        covered_count += 1\n",
        "    else:  # default to 'n'\n",
        "        fp_count += 1\n",
        "\n",
        "# Final summary\n",
        "print(\"\\n=== Manual Review Complete ===\")\n",
        "print(f\"TP-extra Count (valid but missing in gold) = {tp_extra_count}\")\n",
        "print(f\"Covered Count                              = {covered_count}\")\n",
        "print(f\"FP Count (truly invalid)                   = {fp_count}\")"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "aTPN5DmWOq-3",
        "outputId": "a88649d4-19e2-4926-bc8d-4dcfb189912f"
      },
      "execution_count": 23,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 0\n",
            "Text:   Because slightly different sets were used in different phenotyping experiments, the total number of lines used was 199 (Supplementary Table 1).\n",
            "Predicted Entity Type:  Study / Analysis\n",
            "Predicted Entity Text:  different phenotyping experiments\n",
            "All Gold Entities:  [('Study / Analysis', 'phenotyping experiments'), ('Population / Group', 'lines'), ('Concept / Abstract Entity', 'Supplementary Table 1')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 1\n",
            "Text:   Because slightly different sets were used in different phenotyping experiments, the total number of lines used was 199 (Supplementary Table 1).\n",
            "Predicted Entity Type:  Population / Group\n",
            "Predicted Entity Text:  the total number of lines used\n",
            "All Gold Entities:  [('Study / Analysis', 'phenotyping experiments'), ('Population / Group', 'lines'), ('Concept / Abstract Entity', 'Supplementary Table 1')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 2\n",
            "Text:  By coupling these events with linked DNA polymorphisms genome-wide, the phenotypic effect of multiple alleles across the genome can be tested.\n",
            "Predicted Entity Type:  Biological Process\n",
            "Predicted Entity Text:  these events\n",
            "All Gold Entities:  [('Concept / Abstract Entity', 'these events'), ('Allele / Variant', 'linked DNA polymorphisms'), ('Allele / Variant', 'multiple alleles'), ('Genomic Region', 'the genome'), ('Trait / Phenotype', 'the phenotypic effect')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 3\n",
            "Text:  By coupling these events with linked DNA polymorphisms genome-wide, the phenotypic effect of multiple alleles across the genome can be tested.\n",
            "Predicted Entity Type:  Allele / Variant\n",
            "Predicted Entity Text:  linked DNA polymorphisms genome-wide\n",
            "All Gold Entities:  [('Concept / Abstract Entity', 'these events'), ('Allele / Variant', 'linked DNA polymorphisms'), ('Allele / Variant', 'multiple alleles'), ('Genomic Region', 'the genome'), ('Trait / Phenotype', 'the phenotypic effect')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 4\n",
            "Text:  By coupling these events with linked DNA polymorphisms genome-wide, the phenotypic effect of multiple alleles across the genome can be tested.\n",
            "Predicted Entity Type:  Allele / Variant\n",
            "Predicted Entity Text:  multiple alleles across the genome\n",
            "All Gold Entities:  [('Concept / Abstract Entity', 'these events'), ('Allele / Variant', 'linked DNA polymorphisms'), ('Allele / Variant', 'multiple alleles'), ('Genomic Region', 'the genome'), ('Trait / Phenotype', 'the phenotypic effect')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 5\n",
            "Text:  Because of this serious and widespread food safety concern research into understanding the mechanisms driving arsenic accumulation in plants has become a priority [9].\n",
            "Predicted Entity Type:  Biological Process\n",
            "Predicted Entity Text:  the mechanisms\n",
            "All Gold Entities:  [('Concept / Abstract Entity', 'serious and widespread food safety concern'), ('Trait / Phenotype', 'arsenic accumulation in plants'), ('Study / Analysis', 'research into understanding the mechanisms driving arsenic accumulation in plants')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 6\n",
            "Text:  Because of this serious and widespread food safety concern research into understanding the mechanisms driving arsenic accumulation in plants has become a priority [9].\n",
            "Predicted Entity Type:  Study / Analysis\n",
            "Predicted Entity Text:  [9]\n",
            "All Gold Entities:  [('Concept / Abstract Entity', 'serious and widespread food safety concern'), ('Trait / Phenotype', 'arsenic accumulation in plants'), ('Study / Analysis', 'research into understanding the mechanisms driving arsenic accumulation in plants')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 7\n",
            "Text:  This association has also recently been experimentally confirmed (M.T. et al., unpublished observations).\n",
            "Predicted Entity Type:  Concept / Abstract Entity\n",
            "Predicted Entity Text:  This association\n",
            "All Gold Entities:  [('Concept / Abstract Entity', 'association'), ('Study / Analysis', 'M.T. et al., unpublished observations')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 8\n",
            "Text:  We observed that plants coming from sites with elevated soil Pb content (and acidic pH) had better capacity to tolerate this stress (Figure 2B).\n",
            "Predicted Entity Type:  Environmental Factor\n",
            "Predicted Entity Text:  sites with elevated soil Pb content (and acidic pH)\n",
            "All Gold Entities:  [('Population / Group', 'plants'), ('Environmental Factor', 'elevated soil Pb content'), ('Environmental Factor', 'acidic pH'), ('Trait / Phenotype', 'better capacity to tolerate this stress')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 9\n",
            "Text:  This is an upper bound because it seems almost certain that many of the (often much larger set of) strong associations that are not close to a-priori candidates will also turn out to be real. \n",
            "Predicted Entity Type:  Concept / Abstract Entity\n",
            "Predicted Entity Text:  an upper bound\n",
            "All Gold Entities:  [('Concept / Abstract Entity', 'upper bound'), ('Concept / Abstract Entity', 'strong associations'), ('Concept / Abstract Entity', 'a-priori candidates')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 10\n",
            "Text:  This is an upper bound because it seems almost certain that many of the (often much larger set of) strong associations that are not close to a-priori candidates will also turn out to be real. \n",
            "Predicted Entity Type:  Concept / Abstract Entity\n",
            "Predicted Entity Text:  the (often much larger set of) strong associations\n",
            "All Gold Entities:  [('Concept / Abstract Entity', 'upper bound'), ('Concept / Abstract Entity', 'strong associations'), ('Concept / Abstract Entity', 'a-priori candidates')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 11\n",
            "Text:  However, unlike in synthetic RILs made from a bi-parental cross, the low frequency of rare alleles in a natural population makes it difficult to detect their phenotypic effect using GWA mapping.\n",
            "Predicted Entity Type:  Population / Group\n",
            "Predicted Entity Text:  a bi-parental cross\n",
            "All Gold Entities:  [('Population / Group', 'synthetic RILs'), ('Tool / Method', 'bi-parental cross'), ('Allele / Variant', 'rare alleles'), ('Population / Group', 'a natural population'), ('Trait / Phenotype', 'phenotypic effect'), ('Study / Analysis', 'GWA mapping')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 12\n",
            "Text:  However, unlike in synthetic RILs made from a bi-parental cross, the low frequency of rare alleles in a natural population makes it difficult to detect their phenotypic effect using GWA mapping.\n",
            "Predicted Entity Type:  Trait / Phenotype\n",
            "Predicted Entity Text:  their phenotypic effect\n",
            "All Gold Entities:  [('Population / Group', 'synthetic RILs'), ('Tool / Method', 'bi-parental cross'), ('Allele / Variant', 'rare alleles'), ('Population / Group', 'a natural population'), ('Trait / Phenotype', 'phenotypic effect'), ('Study / Analysis', 'GWA mapping')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 13\n",
            "Text:  However, unlike in synthetic RILs made from a bi-parental cross, the low frequency of rare alleles in a natural population makes it difficult to detect their phenotypic effect using GWA mapping.\n",
            "Predicted Entity Type:  Tool / Method\n",
            "Predicted Entity Text:  GWA mapping\n",
            "All Gold Entities:  [('Population / Group', 'synthetic RILs'), ('Tool / Method', 'bi-parental cross'), ('Allele / Variant', 'rare alleles'), ('Population / Group', 'a natural population'), ('Trait / Phenotype', 'phenotypic effect'), ('Study / Analysis', 'GWA mapping')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 14\n",
            "Text:  This natural variation has been used to identify specific genes involved in controlling variation in many traits [28].\n",
            "Predicted Entity Type:  Study / Analysis\n",
            "Predicted Entity Text:  [28]\n",
            "All Gold Entities:  [('Concept / Abstract Entity', 'natural variation'), ('Gene', 'specific genes'), ('Trait / Phenotype', 'many traits')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 15\n",
            "Text:  However, enzymes forming phosphorylated products can also promote arsenate reduction by incorrectly using arsenate in place of phosphate to generate arsenylated products in which the arsenate is more easily reduced by thiols such as glutathione [67].\n",
            "Predicted Entity Type:  Protein / Polypeptide\n",
            "Predicted Entity Text:  enzymes\n",
            "All Gold Entities:  [('Protein / Polypeptide', 'enzymes forming phosphorylated products'), ('Biological Process', 'arsenate reduction'), ('Chemical Substance', 'arsenate'), ('Chemical Substance', 'phosphate'), ('Chemical Substance', 'arsenylated products'), ('Chemical Substance', 'thiols'), ('Chemical Substance', 'glutathione')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 16\n",
            "Text:  Finally, ACCELERATED CELL DEATH 6 (ACD6), which has been experimentally shown to be directly involved in lesioning [22], is detected here as associated with several lesioning and chlorosis phenotypes. \n",
            "Predicted Entity Type:  Trait / Phenotype\n",
            "Predicted Entity Text:  several lesioning and chlorosis phenotypes\n",
            "All Gold Entities:  [('Gene', 'ACCELERATED CELL DEATH 6 (ACD6)'), ('Trait / Phenotype', 'lesioning'), ('Trait / Phenotype', 'lesioning and chlorosis phenotypes'), ('Trait / Phenotype', 'chlorosis phenotypes')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 17\n",
            "Text:  Arsenic concentrations were measured in leaves from 315 F2 plants (Figure 1C) from the Kr06Col-0 cross and approximately one-quarter of these plants (75 of 315, X[2] = 0.24,X[2]0.05(1) [= 3.84) had high leaf arsenic, suggest-] ing that high leaf arsenic in Kr-0 is controlled by a single major locus.\n",
            "Predicted Entity Type:  Population / Group\n",
            "Predicted Entity Text:  Kr06Col-0 cross\n",
            "All Gold Entities:  [('Trait / Phenotype', 'Arsenic concentrations'), ('Tissue / Plant Structure', 'leaves'), ('Population / Group', '315 F2 plants'), ('Population / Group', 'the Kr06Col-0 cross'), ('Accession / Line', 'Kr-0'), ('Accession / Line', 'Kr06Col-0'), ('Measurement / Statistical Result', '75 of 315, X[2] = 0.24,X[2]0.05(1) [= 3.84'), ('Trait / Phenotype', 'high leaf arsenic'), ('Genomic Region', 'a single major locus')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 18\n",
            "Text:  Arsenic concentrations were measured in leaves from 315 F2 plants (Figure 1C) from the Kr06Col-0 cross and approximately one-quarter of these plants (75 of 315, X[2] = 0.24,X[2]0.05(1) [= 3.84) had high leaf arsenic, suggest-] ing that high leaf arsenic in Kr-0 is controlled by a single major locus.\n",
            "Predicted Entity Type:  Measurement / Statistical Result\n",
            "Predicted Entity Text:  X[2] = 0.24\n",
            "All Gold Entities:  [('Trait / Phenotype', 'Arsenic concentrations'), ('Tissue / Plant Structure', 'leaves'), ('Population / Group', '315 F2 plants'), ('Population / Group', 'the Kr06Col-0 cross'), ('Accession / Line', 'Kr-0'), ('Accession / Line', 'Kr06Col-0'), ('Measurement / Statistical Result', '75 of 315, X[2] = 0.24,X[2]0.05(1) [= 3.84'), ('Trait / Phenotype', 'high leaf arsenic'), ('Genomic Region', 'a single major locus')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 19\n",
            "Text:  Such imaging revealed that in the absence of a functional HAC1 gene, and with the loss of arsenite efflux, arsenic appears to accumulate mainly in the root stele to concentrations 120%–180% higher than in the Col-0 wild type (Figure 9).\n",
            "Predicted Entity Type:  Accession / Line\n",
            "Predicted Entity Text:  Col-0 wild type\n",
            "All Gold Entities:  [('Gene', 'HAC1 gene'), ('Biological Process', 'arsenite efflux'), ('Chemical Substance', 'arsenic'), ('Tissue / Plant Structure', 'root stele'), ('Measurement / Statistical Result', 'concentrations 120%–180% higher than in the Col-0 wild type')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 20\n",
            "Text:  Regardless of method used, six phenotypes yield single, strong peaks of association that can be seen by inspection.\n",
            "Predicted Entity Type:  Concept / Abstract Entity\n",
            "Predicted Entity Text:  single, strong peaks of association\n",
            "All Gold Entities:  [('Trait / Phenotype', 'six phenotypes'), ('Measurement / Statistical Result', 'single, strong peaks of association')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 21\n",
            "Text:  Regardless of method used, six phenotypes yield single, strong peaks of association that can be seen by inspection.\n",
            "Predicted Entity Type:  Tool / Method\n",
            "Predicted Entity Text:  method\n",
            "All Gold Entities:  [('Trait / Phenotype', 'six phenotypes'), ('Measurement / Statistical Result', 'single, strong peaks of association')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 22\n",
            "Text:  On the other hand, arsenate uptake, measured as depletion of arsenate from the nutrient solution, was no different in hac1-1, hac1-2, or Kr-0 compared to wild-type Col-0 at both time points tested (Figure 8B).\n",
            "Predicted Entity Type:  Trait / Phenotype\n",
            "Predicted Entity Text:  depletion of arsenate from the nutrient solution\n",
            "All Gold Entities:  [('Trait / Phenotype', 'arsenate uptake'), ('Measurement / Statistical Result', 'depletion of arsenate from the nutrient solution'), ('Allele / Variant', 'hac1-1'), ('Allele / Variant', 'hac1-2'), ('Accession / Line', 'Kr-0'), ('Accession / Line', 'wild-type Col-0'), ('Chemical Substance', 'arsenate'), ('Environmental Factor', 'nutrient solution')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 23\n",
            "Text:  On the other hand, arsenate uptake, measured as depletion of arsenate from the nutrient solution, was no different in hac1-1, hac1-2, or Kr-0 compared to wild-type Col-0 at both time points tested (Figure 8B).\n",
            "Predicted Entity Type:  Accession / Line\n",
            "Predicted Entity Text:  hac1-1\n",
            "All Gold Entities:  [('Trait / Phenotype', 'arsenate uptake'), ('Measurement / Statistical Result', 'depletion of arsenate from the nutrient solution'), ('Allele / Variant', 'hac1-1'), ('Allele / Variant', 'hac1-2'), ('Accession / Line', 'Kr-0'), ('Accession / Line', 'wild-type Col-0'), ('Chemical Substance', 'arsenate'), ('Environmental Factor', 'nutrient solution')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 24\n",
            "Text:  On the other hand, arsenate uptake, measured as depletion of arsenate from the nutrient solution, was no different in hac1-1, hac1-2, or Kr-0 compared to wild-type Col-0 at both time points tested (Figure 8B).\n",
            "Predicted Entity Type:  Accession / Line\n",
            "Predicted Entity Text:  hac1-2\n",
            "All Gold Entities:  [('Trait / Phenotype', 'arsenate uptake'), ('Measurement / Statistical Result', 'depletion of arsenate from the nutrient solution'), ('Allele / Variant', 'hac1-1'), ('Allele / Variant', 'hac1-2'), ('Accession / Line', 'Kr-0'), ('Accession / Line', 'wild-type Col-0'), ('Chemical Substance', 'arsenate'), ('Environmental Factor', 'nutrient solution')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 25\n",
            "Text:  In summary, A. thaliana has evolved two main strategies to cope with high substrate Pb: (1) exclusion from the shoot by sequestration in the roots and (2) translocation to the shoots and storage in inactive organelles or intracellular spaces.\n",
            "Predicted Entity Type:  Chemical Substance\n",
            "Predicted Entity Text:  high substrate Pb\n",
            "All Gold Entities:  [('Organism', 'A. thaliana'), ('Environmental Factor', 'high substrate Pb'), ('Biological Process', 'exclusion from the shoot by sequestration in the roots'), ('Biological Process', 'translocation to the shoots and storage in inactive organelles or intracellular spaces')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 26\n",
            "Text:  In summary, A. thaliana has evolved two main strategies to cope with high substrate Pb: (1) exclusion from the shoot by sequestration in the roots and (2) translocation to the shoots and storage in inactive organelles or intracellular spaces.\n",
            "Predicted Entity Type:  Biological Process\n",
            "Predicted Entity Text:  exclusion from the shoot\n",
            "All Gold Entities:  [('Organism', 'A. thaliana'), ('Environmental Factor', 'high substrate Pb'), ('Biological Process', 'exclusion from the shoot by sequestration in the roots'), ('Biological Process', 'translocation to the shoots and storage in inactive organelles or intracellular spaces')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 27\n",
            "Text:  In summary, A. thaliana has evolved two main strategies to cope with high substrate Pb: (1) exclusion from the shoot by sequestration in the roots and (2) translocation to the shoots and storage in inactive organelles or intracellular spaces.\n",
            "Predicted Entity Type:  Biological Process\n",
            "Predicted Entity Text:  sequestration\n",
            "All Gold Entities:  [('Organism', 'A. thaliana'), ('Environmental Factor', 'high substrate Pb'), ('Biological Process', 'exclusion from the shoot by sequestration in the roots'), ('Biological Process', 'translocation to the shoots and storage in inactive organelles or intracellular spaces')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 28\n",
            "Text:  In summary, A. thaliana has evolved two main strategies to cope with high substrate Pb: (1) exclusion from the shoot by sequestration in the roots and (2) translocation to the shoots and storage in inactive organelles or intracellular spaces.\n",
            "Predicted Entity Type:  Tissue / Plant Structure\n",
            "Predicted Entity Text:  roots\n",
            "All Gold Entities:  [('Organism', 'A. thaliana'), ('Environmental Factor', 'high substrate Pb'), ('Biological Process', 'exclusion from the shoot by sequestration in the roots'), ('Biological Process', 'translocation to the shoots and storage in inactive organelles or intracellular spaces')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 29\n",
            "Text:  In summary, A. thaliana has evolved two main strategies to cope with high substrate Pb: (1) exclusion from the shoot by sequestration in the roots and (2) translocation to the shoots and storage in inactive organelles or intracellular spaces.\n",
            "Predicted Entity Type:  Biological Process\n",
            "Predicted Entity Text:  translocation to the shoots\n",
            "All Gold Entities:  [('Organism', 'A. thaliana'), ('Environmental Factor', 'high substrate Pb'), ('Biological Process', 'exclusion from the shoot by sequestration in the roots'), ('Biological Process', 'translocation to the shoots and storage in inactive organelles or intracellular spaces')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 30\n",
            "Text:  This suggested the possibility that HAC1 functions as an arsenate reductase, reducing arsenate (As[V]) to arsenite (As[III]).\n",
            "Predicted Entity Type:  Chemical Substance\n",
            "Predicted Entity Text:  arsenite (As[III]\n",
            "All Gold Entities:  [('Gene', 'HAC1'), ('Protein / Polypeptide', 'arsenate reductase'), ('Chemical Substance', 'arsenate (As[V])'), ('Chemical Substance', 'arsenite (As[III])'), ('Biological Process', 'reducing arsenate (As[V]) to arsenite (As[III])')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 31\n",
            "Text:  However, foliar uptake is an important second pathway in areas with high fallout of airborne particles (Uzu et al., 2010; Luo et al., 2019).\n",
            "Predicted Entity Type:  Environmental Factor\n",
            "Predicted Entity Text:  high fallout of airborne particles\n",
            "All Gold Entities:  [('Biological Process', 'foliar uptake'), ('Environmental Factor', 'areas with high fallout of airborne particles'), ('Study / Analysis', 'Uzu et al., 2010'), ('Study / Analysis', 'Luo et al., 2019')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 32\n",
            "Text:  It is also interesting to note that since the majority of the A. thaliana accessions tested have the HAC1 allele associated with higher leaf arsenic, they are likely to have weak alleles of HAC1.\n",
            "Predicted Entity Type:  Allele / Variant\n",
            "Predicted Entity Text:  HAC1 allele\n",
            "All Gold Entities:  [('Population / Group', 'the majority of the A. thaliana accessions tested'), ('Allele / Variant', 'the HAC1 allele'), ('Trait / Phenotype', 'higher leaf arsenic'), ('Allele / Variant', 'weak alleles of HAC1'), ('Gene', 'HAC1')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 33\n",
            "Text:  Here, we expand on this observation by establishing the strength of the AtHKT1;1 alleles in four further high Na[+] accumulating accessions (Bur-1, Duk, PHW-20 and UKNW06-386) that all contain a T at Chr4:6392276, along with a low leaf Na[+] accession (Nd-1) with a C at Chr4:6392276.\n",
            "Predicted Entity Type:  Allele / Variant\n",
            "Predicted Entity Text:  AtHKT1;1 alleles\n",
            "All Gold Entities:  [('Gene', 'AtHKT1;1'), ('Accession / Line', 'Bur-1'), ('Accession / Line', 'Duk'), ('Accession / Line', 'PHW-20'), ('Accession / Line', 'UKNW06-386'), ('Allele / Variant', 'T at Chr4:6392276'), ('Accession / Line', 'Nd-1'), ('Allele / Variant', 'C at Chr4:6392276'), ('Trait / Phenotype', 'high Na[+] accumulating'), ('Trait / Phenotype', 'low leaf Na[+]')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 34\n",
            "Text:  Here, we expand on this observation by establishing the strength of the AtHKT1;1 alleles in four further high Na[+] accumulating accessions (Bur-1, Duk, PHW-20 and UKNW06-386) that all contain a T at Chr4:6392276, along with a low leaf Na[+] accession (Nd-1) with a C at Chr4:6392276.\n",
            "Predicted Entity Type:  Accession / Line\n",
            "Predicted Entity Text:  Bur-1\n",
            "All Gold Entities:  [('Gene', 'AtHKT1;1'), ('Accession / Line', 'Bur-1'), ('Accession / Line', 'Duk'), ('Accession / Line', 'PHW-20'), ('Accession / Line', 'UKNW06-386'), ('Allele / Variant', 'T at Chr4:6392276'), ('Accession / Line', 'Nd-1'), ('Allele / Variant', 'C at Chr4:6392276'), ('Trait / Phenotype', 'high Na[+] accumulating'), ('Trait / Phenotype', 'low leaf Na[+]')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 35\n",
            "Text:  Here, we expand on this observation by establishing the strength of the AtHKT1;1 alleles in four further high Na[+] accumulating accessions (Bur-1, Duk, PHW-20 and UKNW06-386) that all contain a T at Chr4:6392276, along with a low leaf Na[+] accession (Nd-1) with a C at Chr4:6392276.\n",
            "Predicted Entity Type:  Accession / Line\n",
            "Predicted Entity Text:  Nd-1)\n",
            "All Gold Entities:  [('Gene', 'AtHKT1;1'), ('Accession / Line', 'Bur-1'), ('Accession / Line', 'Duk'), ('Accession / Line', 'PHW-20'), ('Accession / Line', 'UKNW06-386'), ('Allele / Variant', 'T at Chr4:6392276'), ('Accession / Line', 'Nd-1'), ('Allele / Variant', 'C at Chr4:6392276'), ('Trait / Phenotype', 'high Na[+] accumulating'), ('Trait / Phenotype', 'low leaf Na[+]')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 36\n",
            "Text:  In the hac1 A. thaliana mutant that lacks the arsenate reductase activity required for arsenite efflux, we observe an over-accumulation of arsenic within the central stele of the root. \n",
            "Predicted Entity Type:  Protein / Polypeptide\n",
            "Predicted Entity Text:  arsenate reductase activity\n",
            "All Gold Entities:  [('Accession / Line', 'hac1 A. thaliana mutant'), ('Biological Process', 'arsenate reductase activity'), ('Biological Process', 'arsenite efflux'), ('Trait / Phenotype', 'over-accumulation of arsenic'), ('Tissue / Plant Structure', 'central stele'), ('Tissue / Plant Structure', 'root')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 37\n",
            "Text:  In the hac1 A. thaliana mutant that lacks the arsenate reductase activity required for arsenite efflux, we observe an over-accumulation of arsenic within the central stele of the root. \n",
            "Predicted Entity Type:  Chemical Substance\n",
            "Predicted Entity Text:  arsenic\n",
            "All Gold Entities:  [('Accession / Line', 'hac1 A. thaliana mutant'), ('Biological Process', 'arsenate reductase activity'), ('Biological Process', 'arsenite efflux'), ('Trait / Phenotype', 'over-accumulation of arsenic'), ('Tissue / Plant Structure', 'central stele'), ('Tissue / Plant Structure', 'root')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 38\n",
            "Text:  In the hac1 A. thaliana mutant that lacks the arsenate reductase activity required for arsenite efflux, we observe an over-accumulation of arsenic within the central stele of the root. \n",
            "Predicted Entity Type:  Tissue / Plant Structure\n",
            "Predicted Entity Text:  central stele of the root\n",
            "All Gold Entities:  [('Accession / Line', 'hac1 A. thaliana mutant'), ('Biological Process', 'arsenate reductase activity'), ('Biological Process', 'arsenite efflux'), ('Trait / Phenotype', 'over-accumulation of arsenic'), ('Tissue / Plant Structure', 'central stele'), ('Tissue / Plant Structure', 'root')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 39\n",
            "Text:  Pb entering the symplast is removed by the activity of efflux pumps present in the plasma membrane.\n",
            "Predicted Entity Type:  Biological Process\n",
            "Predicted Entity Text:  efflux pumps\n",
            "All Gold Entities:  [('Chemical Substance', 'Pb'), ('Tissue / Plant Structure', 'symplast'), ('Protein / Polypeptide', 'efflux pumps'), ('Tissue / Plant Structure', 'plasma membrane')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 40\n",
            "Text:  In addition, Na[+]  compartmentalization facilitates vacuolar osmotic adjustment that is necessary to compensate for the osmotic effects of salinity by maintaining turgor pressure for cell expansion and growth.\n",
            "Predicted Entity Type:  Chemical Substance\n",
            "Predicted Entity Text:  salinity\n",
            "All Gold Entities:  [('Biological Process', 'Na[+]  compartmentalization'), ('Biological Process', 'vacuolar osmotic adjustment'), ('Environmental Factor', 'salinity'), ('Trait / Phenotype', 'turgor pressure'), ('Biological Process', 'cell expansion and growth')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 41\n",
            "Text:  In addition, Na[+]  compartmentalization facilitates vacuolar osmotic adjustment that is necessary to compensate for the osmotic effects of salinity by maintaining turgor pressure for cell expansion and growth.\n",
            "Predicted Entity Type:  Biological Process\n",
            "Predicted Entity Text:  growth\n",
            "All Gold Entities:  [('Biological Process', 'Na[+]  compartmentalization'), ('Biological Process', 'vacuolar osmotic adjustment'), ('Environmental Factor', 'salinity'), ('Trait / Phenotype', 'turgor pressure'), ('Biological Process', 'cell expansion and growth')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 42\n",
            "Text:  Whether the wide variety of natural phenotypic and genetic variation observed in A. thaliana [12] contributes to its local adaptation is an important unresolved question that is currently attracting a significant amount of attention [13].\n",
            "Predicted Entity Type:  Concept / Abstract Entity\n",
            "Predicted Entity Text:  the wide variety of natural phenotypic and genetic variation\n",
            "All Gold Entities:  [('Organism', 'A. thaliana'), ('Trait / Phenotype', 'natural phenotypic and'), ('Concept / Abstract Entity', 'genetic variation'), ('Biological Process', 'local adaptation')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 43\n",
            "Text:  This increase in HAC1 mRNA levels reaches a maximum at 100 mM arsenate in the growth media (Figure 7A).\n",
            "Predicted Entity Type:  Trait / Phenotype\n",
            "Predicted Entity Text:  increase in HAC1 mRNA levels\n",
            "All Gold Entities:  [('Trait / Phenotype', 'HAC1 mRNA levels'), ('Chemical Substance', '100 mM arsenate'), ('Environmental Factor', 'growth media')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 44\n",
            "Text:  To validate the existence of this leaf arsenic QTL identified by GWA analysis we created a synthetic F2 recombinant population in which alternative alleles of the dialleic SNP Chr2:9008060 were segregating.\n",
            "Predicted Entity Type:  Trait / Phenotype\n",
            "Predicted Entity Text:  leaf arsenic QTL\n",
            "All Gold Entities:  [('Allele / Variant', 'leaf arsenic QTL'), ('Study / Analysis', 'GWA analysis'), ('Population / Group', 'synthetic F2 recombinant population'), ('Allele / Variant', 'alternative alleles of the dialleic SNP Chr2:9008060')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 45\n",
            "Text:  To validate the existence of this leaf arsenic QTL identified by GWA analysis we created a synthetic F2 recombinant population in which alternative alleles of the dialleic SNP Chr2:9008060 were segregating.\n",
            "Predicted Entity Type:  Population / Group\n",
            "Predicted Entity Text:  a synthetic F2 recombinant population\n",
            "All Gold Entities:  [('Allele / Variant', 'leaf arsenic QTL'), ('Study / Analysis', 'GWA analysis'), ('Population / Group', 'synthetic F2 recombinant population'), ('Allele / Variant', 'alternative alleles of the dialleic SNP Chr2:9008060')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 46\n",
            "Text:  To validate the existence of this leaf arsenic QTL identified by GWA analysis we created a synthetic F2 recombinant population in which alternative alleles of the dialleic SNP Chr2:9008060 were segregating.\n",
            "Predicted Entity Type:  Allele / Variant\n",
            "Predicted Entity Text:  the dialleic SNP Chr2:9008060\n",
            "All Gold Entities:  [('Allele / Variant', 'leaf arsenic QTL'), ('Study / Analysis', 'GWA analysis'), ('Population / Group', 'synthetic F2 recombinant population'), ('Allele / Variant', 'alternative alleles of the dialleic SNP Chr2:9008060')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 47\n",
            "Text:  This knowledge creates new opportunities to limit arsenic accumulation in food crops, thereby helping to reduce the cancer risk from this food-chain contaminant.\n",
            "Predicted Entity Type:  Trait / Phenotype\n",
            "Predicted Entity Text:  arsenic accumulation in food crops\n",
            "All Gold Entities:  [('Concept / Abstract Entity', 'This knowledge'), ('Trait / Phenotype', 'arsenic accumulation'), ('Population / Group', 'food crops'), ('Concept / Abstract Entity', 'cancer risk'), ('Chemical Substance', 'food-chain contaminant')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 48\n",
            "Text:  This knowledge creates new opportunities to limit arsenic accumulation in food crops, thereby helping to reduce the cancer risk from this food-chain contaminant.\n",
            "Predicted Entity Type:  Chemical Substance\n",
            "Predicted Entity Text:  this food-chain contaminant\n",
            "All Gold Entities:  [('Concept / Abstract Entity', 'This knowledge'), ('Trait / Phenotype', 'arsenic accumulation'), ('Population / Group', 'food crops'), ('Concept / Abstract Entity', 'cancer risk'), ('Chemical Substance', 'food-chain contaminant')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 49\n",
            "Text:  If polymorphism at FRI is taken into account, CRP is no longer significantly associated, but this does not necessarily mean that the association is spurious.\n",
            "Predicted Entity Type:  Gene\n",
            "Predicted Entity Text:  CRP\n",
            "All Gold Entities:  [('Allele / Variant', 'polymorphism at FRI'), ('Protein / Polypeptide', 'CRP')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 50\n",
            "Text:  If polymorphism at FRI is taken into account, CRP is no longer significantly associated, but this does not necessarily mean that the association is spurious.\n",
            "Predicted Entity Type:  Concept / Abstract Entity\n",
            "Predicted Entity Text:  the association\n",
            "All Gold Entities:  [('Allele / Variant', 'polymorphism at FRI'), ('Protein / Polypeptide', 'CRP')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 51\n",
            "Text:  However, hac1-1 and hac1-2 were both significantly more sensitive to arsenate compared to the Col-0 wild type at relatively high concentrations of arsenate when growth was measured as root length or overall plant fresh weight (Figure 10).\n",
            "Predicted Entity Type:  Chemical Substance\n",
            "Predicted Entity Text:  arsenate\n",
            "All Gold Entities:  [('Allele / Variant', 'hac1-1'), ('Allele / Variant', 'hac1-2'), ('Accession / Line', 'Col-0 wild type'), ('Trait / Phenotype', 'more sensitive to arsenate'), ('Environmental Factor', 'relatively high concentrations of arsenate'), ('Trait / Phenotype', 'root length'), ('Trait / Phenotype', 'overall plant fresh weight')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 52\n",
            "Text:  However, hac1-1 and hac1-2 were both significantly more sensitive to arsenate compared to the Col-0 wild type at relatively high concentrations of arsenate when growth was measured as root length or overall plant fresh weight (Figure 10).\n",
            "Predicted Entity Type:  Trait / Phenotype\n",
            "Predicted Entity Text:  significantly more sensitive\n",
            "All Gold Entities:  [('Allele / Variant', 'hac1-1'), ('Allele / Variant', 'hac1-2'), ('Accession / Line', 'Col-0 wild type'), ('Trait / Phenotype', 'more sensitive to arsenate'), ('Environmental Factor', 'relatively high concentrations of arsenate'), ('Trait / Phenotype', 'root length'), ('Trait / Phenotype', 'overall plant fresh weight')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 53\n",
            "Text:  To test this hypothesis, we evaluated the oxidation state of arsenic in plants lacking a functional HAC1 gene.\n",
            "Predicted Entity Type:  Trait / Phenotype\n",
            "Predicted Entity Text:  oxidation state of arsenic\n",
            "All Gold Entities:  [('Trait / Phenotype', 'the oxidation state of arsenic'), ('Population / Group', 'plants lacking a functional HAC1 gene'), ('Gene', 'HAC1 gene')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 54\n",
            "Text:  To test this hypothesis, we evaluated the oxidation state of arsenic in plants lacking a functional HAC1 gene.\n",
            "Predicted Entity Type:  Chemical Substance\n",
            "Predicted Entity Text:  arsenic\n",
            "All Gold Entities:  [('Trait / Phenotype', 'the oxidation state of arsenic'), ('Population / Group', 'plants lacking a functional HAC1 gene'), ('Gene', 'HAC1 gene')]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is covered by an existing gold entity already\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "=== Manual Review Complete ===\n",
            "TP-extra Count (valid but missing in gold) = 15\n",
            "Covered Count                              = 40\n",
            "FP Count (truly invalid)                   = 0\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "\n",
        "# Load the relation_fn.csv\n",
        "df = pd.read_csv(\"uie_manual_eval/relation_fn.csv\")\n",
        "\n",
        "tp_count = 0\n",
        "fn_count = 0\n",
        "\n",
        "for index, row in df.iterrows():\n",
        "    text = row.get(\"text\", \"\")\n",
        "    gold_relation_type = row.get(\"gold_relation_type\", \"\")\n",
        "    gold_arg1_type = row.get(\"gold_arg1_type\", \"\")\n",
        "    gold_arg1_text = row.get(\"gold_arg1_text\", \"\")\n",
        "    gold_arg2_type = row.get(\"gold_arg2_type\", \"\")\n",
        "    gold_arg2_text = row.get(\"gold_arg2_text\", \"\")\n",
        "    all_pred_relations = row.get(\"all_pred_relations\", \"\")\n",
        "\n",
        "    print(\"\\n---------------------------------------------------------\")\n",
        "    print(f\"Row index: {index}\")\n",
        "    print(\"Text: \", text)\n",
        "    print(\"Gold Relation Type: \", gold_relation_type)\n",
        "    print(\"Gold Arg1 (Type/Text): \", gold_arg1_type, \" // \", gold_arg1_text)\n",
        "    print(\"Gold Arg2 (Type/Text): \", gold_arg2_type, \" // \", gold_arg2_text)\n",
        "    print(\"All Predicted Relations: \", all_pred_relations)\n",
        "\n",
        "    choice = input(\n",
        "        \"Is this gold relation sufficiently covered by the predicted relations? (y/n) \"\n",
        "    ).strip().lower()\n",
        "\n",
        "    if choice == 'y':\n",
        "        tp_count += 1\n",
        "    else:\n",
        "        fn_count += 1\n",
        "\n",
        "print(\"\\n=== Manual Review Complete ===\")\n",
        "print(f\"TP Count (covered)   = {tp_count}\")\n",
        "print(f\"FN Count (not covered) = {fn_count}\")\n",
        "\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Mh6ro3HjSk6w",
        "outputId": "1b33fd73-fce1-4ef1-a53c-a4cdc9cee619"
      },
      "execution_count": 24,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 0\n",
            "Text:   Because slightly different sets were used in different phenotyping experiments, the total number of lines used was 199 (Supplementary Table 1).\n",
            "Gold Relation Type:  HAS_MEASUREMENT\n",
            "Gold Arg1 (Type/Text):  Study / Analysis  //  phenotyping experiments\n",
            "Gold Arg2 (Type/Text):  Population / Group  //  lines\n",
            "All Predicted Relations:  [('IDENTIFIED_IN', ('Population / Group', 'the total number of lines used'), ('Concept / Abstract Entity', 'Supplementary Table 1'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 1\n",
            "Text:  By coupling these events with linked DNA polymorphisms genome-wide, the phenotypic effect of multiple alleles across the genome can be tested.\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Concept / Abstract Entity  //  these events\n",
            "Gold Arg2 (Type/Text):  Allele / Variant  //  linked DNA polymorphisms\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Biological Process', 'these events'), ('Allele / Variant', 'linked DNA polymorphisms genome-wide')), ('CONTRIBUTES_TO', ('Biological Process', 'these events'), ('Trait / Phenotype', 'the phenotypic effect')), ('PART_OF', ('Allele / Variant', 'multiple alleles across the genome'), ('Allele / Variant', 'the genome'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 2\n",
            "Text:  By coupling these events with linked DNA polymorphisms genome-wide, the phenotypic effect of multiple alleles across the genome can be tested.\n",
            "Gold Relation Type:  LOCATED_IN\n",
            "Gold Arg1 (Type/Text):  Allele / Variant  //  multiple alleles\n",
            "Gold Arg2 (Type/Text):  Genomic Region  //  the genome\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Biological Process', 'these events'), ('Allele / Variant', 'linked DNA polymorphisms genome-wide')), ('CONTRIBUTES_TO', ('Biological Process', 'these events'), ('Trait / Phenotype', 'the phenotypic effect')), ('PART_OF', ('Allele / Variant', 'multiple alleles across the genome'), ('Allele / Variant', 'the genome'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 3\n",
            "Text:  By coupling these events with linked DNA polymorphisms genome-wide, the phenotypic effect of multiple alleles across the genome can be tested.\n",
            "Gold Relation Type:  CONTRIBUTES_TO\n",
            "Gold Arg1 (Type/Text):  Allele / Variant  //  multiple alleles\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  the phenotypic effect\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Biological Process', 'these events'), ('Allele / Variant', 'linked DNA polymorphisms genome-wide')), ('CONTRIBUTES_TO', ('Biological Process', 'these events'), ('Trait / Phenotype', 'the phenotypic effect')), ('PART_OF', ('Allele / Variant', 'multiple alleles across the genome'), ('Allele / Variant', 'the genome'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 4\n",
            "Text:  Because of this serious and widespread food safety concern research into understanding the mechanisms driving arsenic accumulation in plants has become a priority [9].\n",
            "Gold Relation Type:  CONTRIBUTES_TO\n",
            "Gold Arg1 (Type/Text):  Concept / Abstract Entity  //  serious and widespread food safety concern\n",
            "Gold Arg2 (Type/Text):  Study / Analysis  //  research into understanding the mechanisms driving arsenic accumulation in plants\n",
            "All Predicted Relations:  [('CAUSES', ('Biological Process', 'the mechanisms'), ('Trait / Phenotype', 'arsenic accumulation in plants')), ('IDENTIFIED_IN', ('Biological Process', 'the mechanisms'), ('Study / Analysis', '[9]'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 5\n",
            "Text:  Because of this serious and widespread food safety concern research into understanding the mechanisms driving arsenic accumulation in plants has become a priority [9].\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Study / Analysis  //  research into understanding the mechanisms driving arsenic accumulation in plants\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  arsenic accumulation in plants\n",
            "All Predicted Relations:  [('CAUSES', ('Biological Process', 'the mechanisms'), ('Trait / Phenotype', 'arsenic accumulation in plants')), ('IDENTIFIED_IN', ('Biological Process', 'the mechanisms'), ('Study / Analysis', '[9]'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 6\n",
            "Text:  This association has also recently been experimentally confirmed (M.T. et al., unpublished observations).\n",
            "Gold Relation Type:  IDENTIFIED_IN\n",
            "Gold Arg1 (Type/Text):  Concept / Abstract Entity  //  association\n",
            "Gold Arg2 (Type/Text):  Study / Analysis  //  M.T. et al., unpublished observations\n",
            "All Predicted Relations:  [('IDENTIFIED_IN', ('Concept / Abstract Entity', 'This association'), ('Study / Analysis', 'M.T. et al., unpublished observations'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 7\n",
            "Text:  We observed that plants coming from sites with elevated soil Pb content (and acidic pH) had better capacity to tolerate this stress (Figure 2B).\n",
            "Gold Relation Type:  LOCATED_IN\n",
            "Gold Arg1 (Type/Text):  Population / Group  //  plants\n",
            "Gold Arg2 (Type/Text):  Environmental Factor  //  elevated soil Pb content\n",
            "All Predicted Relations:  [('LOCATED_IN', ('Population / Group', 'plants'), ('Environmental Factor', 'sites with elevated soil Pb content (and acidic pH)')), ('ASSOCIATED_WITH', ('Environmental Factor', 'sites with elevated soil Pb content (and acidic pH)'), ('Environmental Factor', 'better capacity to tolerate this stress'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 8\n",
            "Text:  We observed that plants coming from sites with elevated soil Pb content (and acidic pH) had better capacity to tolerate this stress (Figure 2B).\n",
            "Gold Relation Type:  LOCATED_IN\n",
            "Gold Arg1 (Type/Text):  Population / Group  //  plants\n",
            "Gold Arg2 (Type/Text):  Environmental Factor  //  acidic pH\n",
            "All Predicted Relations:  [('LOCATED_IN', ('Population / Group', 'plants'), ('Environmental Factor', 'sites with elevated soil Pb content (and acidic pH)')), ('ASSOCIATED_WITH', ('Environmental Factor', 'sites with elevated soil Pb content (and acidic pH)'), ('Environmental Factor', 'better capacity to tolerate this stress'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 9\n",
            "Text:  We observed that plants coming from sites with elevated soil Pb content (and acidic pH) had better capacity to tolerate this stress (Figure 2B).\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Population / Group  //  plants\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  better capacity to tolerate this stress\n",
            "All Predicted Relations:  [('LOCATED_IN', ('Population / Group', 'plants'), ('Environmental Factor', 'sites with elevated soil Pb content (and acidic pH)')), ('ASSOCIATED_WITH', ('Environmental Factor', 'sites with elevated soil Pb content (and acidic pH)'), ('Environmental Factor', 'better capacity to tolerate this stress'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 10\n",
            "Text:  At2g21045 was named HAC1.\n",
            "Gold Relation Type:  IS_A\n",
            "Gold Arg1 (Type/Text):  Gene  //  At2g21045\n",
            "Gold Arg2 (Type/Text):  Gene  //  HAC1\n",
            "All Predicted Relations:  [('IS_A', ('Gene', 'At2g21045'), ('Gene', 'At2g21045'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 11\n",
            "Text:  However, unlike in synthetic RILs made from a bi-parental cross, the low frequency of rare alleles in a natural population makes it difficult to detect their phenotypic effect using GWA mapping.\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Population / Group  //  synthetic RILs\n",
            "Gold Arg2 (Type/Text):  Tool / Method  //  bi-parental cross\n",
            "All Predicted Relations:  [('PART_OF', ('Population / Group', 'synthetic RILs'), ('Population / Group', 'a bi-parental cross')), ('PART_OF', ('Allele / Variant', 'rare alleles'), ('Population / Group', 'a natural population')), ('CAUSES', ('Allele / Variant', 'rare alleles'), ('Allele / Variant', 'difficult to detect their phenotypic effect'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 12\n",
            "Text:  However, unlike in synthetic RILs made from a bi-parental cross, the low frequency of rare alleles in a natural population makes it difficult to detect their phenotypic effect using GWA mapping.\n",
            "Gold Relation Type:  LOCATED_IN\n",
            "Gold Arg1 (Type/Text):  Allele / Variant  //  rare alleles\n",
            "Gold Arg2 (Type/Text):  Population / Group  //  a natural population\n",
            "All Predicted Relations:  [('PART_OF', ('Population / Group', 'synthetic RILs'), ('Population / Group', 'a bi-parental cross')), ('PART_OF', ('Allele / Variant', 'rare alleles'), ('Population / Group', 'a natural population')), ('CAUSES', ('Allele / Variant', 'rare alleles'), ('Allele / Variant', 'difficult to detect their phenotypic effect'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 13\n",
            "Text:  However, unlike in synthetic RILs made from a bi-parental cross, the low frequency of rare alleles in a natural population makes it difficult to detect their phenotypic effect using GWA mapping.\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Allele / Variant  //  rare alleles\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  phenotypic effect\n",
            "All Predicted Relations:  [('PART_OF', ('Population / Group', 'synthetic RILs'), ('Population / Group', 'a bi-parental cross')), ('PART_OF', ('Allele / Variant', 'rare alleles'), ('Population / Group', 'a natural population')), ('CAUSES', ('Allele / Variant', 'rare alleles'), ('Allele / Variant', 'difficult to detect their phenotypic effect'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 14\n",
            "Text:  However, unlike in synthetic RILs made from a bi-parental cross, the low frequency of rare alleles in a natural population makes it difficult to detect their phenotypic effect using GWA mapping.\n",
            "Gold Relation Type:  IDENTIFIED_IN\n",
            "Gold Arg1 (Type/Text):  Trait / Phenotype  //  phenotypic effect\n",
            "Gold Arg2 (Type/Text):  Study / Analysis  //  GWA mapping\n",
            "All Predicted Relations:  [('PART_OF', ('Population / Group', 'synthetic RILs'), ('Population / Group', 'a bi-parental cross')), ('PART_OF', ('Allele / Variant', 'rare alleles'), ('Population / Group', 'a natural population')), ('CAUSES', ('Allele / Variant', 'rare alleles'), ('Allele / Variant', 'difficult to detect their phenotypic effect'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 15\n",
            "Text:  This natural variation has been used to identify specific genes involved in controlling variation in many traits [28].\n",
            "Gold Relation Type:  IDENTIFIED_IN\n",
            "Gold Arg1 (Type/Text):  Gene  //  specific genes\n",
            "Gold Arg2 (Type/Text):  Concept / Abstract Entity  //  natural variation\n",
            "All Predicted Relations:  [('CONTRIBUTES_TO', ('Gene', 'specific genes'), ('Gene', 'controlling variation in many traits')), ('IDENTIFIED_IN', ('Gene', 'specific genes'), ('Study / Analysis', '[28]'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 16\n",
            "Text:  This natural variation has been used to identify specific genes involved in controlling variation in many traits [28].\n",
            "Gold Relation Type:  REGULATES\n",
            "Gold Arg1 (Type/Text):  Gene  //  specific genes\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  many traits\n",
            "All Predicted Relations:  [('CONTRIBUTES_TO', ('Gene', 'specific genes'), ('Gene', 'controlling variation in many traits')), ('IDENTIFIED_IN', ('Gene', 'specific genes'), ('Study / Analysis', '[28]'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 17\n",
            "Text:  This redundancy has also been observed in E. coli [66] and S. cerevisiae [26].\n",
            "Gold Relation Type:  IDENTIFIED_IN\n",
            "Gold Arg1 (Type/Text):  Concept / Abstract Entity  //  This redundancy\n",
            "Gold Arg2 (Type/Text):  Organism  //  E. coli\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Concept / Abstract Entity', 'This redundancy'), ('Organism', 'E. coli')), ('ASSOCIATED_WITH', ('Concept / Abstract Entity', 'This redundancy'), ('Organism', 'S. cerevisiae'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 18\n",
            "Text:  This redundancy has also been observed in E. coli [66] and S. cerevisiae [26].\n",
            "Gold Relation Type:  IDENTIFIED_IN\n",
            "Gold Arg1 (Type/Text):  Concept / Abstract Entity  //  This redundancy\n",
            "Gold Arg2 (Type/Text):  Organism  //  S. cerevisiae\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Concept / Abstract Entity', 'This redundancy'), ('Organism', 'E. coli')), ('ASSOCIATED_WITH', ('Concept / Abstract Entity', 'This redundancy'), ('Organism', 'S. cerevisiae'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 19\n",
            "Text:  However, enzymes forming phosphorylated products can also promote arsenate reduction by incorrectly using arsenate in place of phosphate to generate arsenylated products in which the arsenate is more easily reduced by thiols such as glutathione [67].\n",
            "Gold Relation Type:  CONTRIBUTES_TO\n",
            "Gold Arg1 (Type/Text):  Protein / Polypeptide  //  enzymes forming phosphorylated products\n",
            "Gold Arg2 (Type/Text):  Biological Process  //  arsenate reduction\n",
            "All Predicted Relations:  [('CAUSES', ('Protein / Polypeptide', 'enzymes'), ('Biological Process', 'arsenate reduction')), ('IS_A', ('Chemical Substance', 'arsenate'), ('Chemical Substance', 'phosphate')), ('ASSOCIATED_WITH', ('Chemical Substance', 'arsenylated products'), ('Chemical Substance', 'thiols')), ('CAUSES', ('Chemical Substance', 'glutathione'), ('Chemical Substance', 'arsenylated products'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 20\n",
            "Text:  However, enzymes forming phosphorylated products can also promote arsenate reduction by incorrectly using arsenate in place of phosphate to generate arsenylated products in which the arsenate is more easily reduced by thiols such as glutathione [67].\n",
            "Gold Relation Type:  CONTRIBUTES_TO\n",
            "Gold Arg1 (Type/Text):  Chemical Substance  //  thiols\n",
            "Gold Arg2 (Type/Text):  Biological Process  //  arsenate reduction\n",
            "All Predicted Relations:  [('CAUSES', ('Protein / Polypeptide', 'enzymes'), ('Biological Process', 'arsenate reduction')), ('IS_A', ('Chemical Substance', 'arsenate'), ('Chemical Substance', 'phosphate')), ('ASSOCIATED_WITH', ('Chemical Substance', 'arsenylated products'), ('Chemical Substance', 'thiols')), ('CAUSES', ('Chemical Substance', 'glutathione'), ('Chemical Substance', 'arsenylated products'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 21\n",
            "Text:  However, enzymes forming phosphorylated products can also promote arsenate reduction by incorrectly using arsenate in place of phosphate to generate arsenylated products in which the arsenate is more easily reduced by thiols such as glutathione [67].\n",
            "Gold Relation Type:  IS_A\n",
            "Gold Arg1 (Type/Text):  Chemical Substance  //  glutathione\n",
            "Gold Arg2 (Type/Text):  Chemical Substance  //  thiols\n",
            "All Predicted Relations:  [('CAUSES', ('Protein / Polypeptide', 'enzymes'), ('Biological Process', 'arsenate reduction')), ('IS_A', ('Chemical Substance', 'arsenate'), ('Chemical Substance', 'phosphate')), ('ASSOCIATED_WITH', ('Chemical Substance', 'arsenylated products'), ('Chemical Substance', 'thiols')), ('CAUSES', ('Chemical Substance', 'glutathione'), ('Chemical Substance', 'arsenylated products'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 22\n",
            "Text:  Finally, ACCELERATED CELL DEATH 6 (ACD6), which has been experimentally shown to be directly involved in lesioning [22], is detected here as associated with several lesioning and chlorosis phenotypes. \n",
            "Gold Relation Type:  CAUSES\n",
            "Gold Arg1 (Type/Text):  Gene  //  ACCELERATED CELL DEATH 6 (ACD6)\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  lesioning\n",
            "All Predicted Relations:  [('CONTRIBUTES_TO', ('Gene', 'ACCELERATED CELL DEATH 6 (ACD6)'), ('Trait / Phenotype', 'lesioning')), ('ASSOCIATED_WITH', ('Gene', 'ACCELERATED CELL DEATH 6 (ACD6)'), ('Trait / Phenotype', 'several lesioning and chlorosis phenotypes'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 23\n",
            "Text:  Finally, ACCELERATED CELL DEATH 6 (ACD6), which has been experimentally shown to be directly involved in lesioning [22], is detected here as associated with several lesioning and chlorosis phenotypes. \n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Gene  //  ACCELERATED CELL DEATH 6 (ACD6)\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  lesioning and chlorosis phenotypes\n",
            "All Predicted Relations:  [('CONTRIBUTES_TO', ('Gene', 'ACCELERATED CELL DEATH 6 (ACD6)'), ('Trait / Phenotype', 'lesioning')), ('ASSOCIATED_WITH', ('Gene', 'ACCELERATED CELL DEATH 6 (ACD6)'), ('Trait / Phenotype', 'several lesioning and chlorosis phenotypes'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 24\n",
            "Text:  Finally, ACCELERATED CELL DEATH 6 (ACD6), which has been experimentally shown to be directly involved in lesioning [22], is detected here as associated with several lesioning and chlorosis phenotypes. \n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Gene  //  ACCELERATED CELL DEATH 6 (ACD6)\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  chlorosis phenotypes\n",
            "All Predicted Relations:  [('CONTRIBUTES_TO', ('Gene', 'ACCELERATED CELL DEATH 6 (ACD6)'), ('Trait / Phenotype', 'lesioning')), ('ASSOCIATED_WITH', ('Gene', 'ACCELERATED CELL DEATH 6 (ACD6)'), ('Trait / Phenotype', 'several lesioning and chlorosis phenotypes'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 25\n",
            "Text:  Arsenic concentrations were measured in leaves from 315 F2 plants (Figure 1C) from the Kr06Col-0 cross and approximately one-quarter of these plants (75 of 315, X[2] = 0.24,X[2]0.05(1) [= 3.84) had high leaf arsenic, suggest-] ing that high leaf arsenic in Kr-0 is controlled by a single major locus.\n",
            "Gold Relation Type:  PART_OF\n",
            "Gold Arg1 (Type/Text):  Population / Group  //  315 F2 plants\n",
            "Gold Arg2 (Type/Text):  Population / Group  //  the Kr06Col-0 cross\n",
            "All Predicted Relations:  [('LOCATED_IN', ('Trait / Phenotype', 'Arsenic concentrations'), ('Tissue / Plant Structure', 'leaves')), ('PART_OF', ('Population / Group', '315 F2 plants'), ('Population / Group', 'Kr06Col-0 cross')), ('ASSOCIATED_WITH', ('Trait / Phenotype', 'high leaf arsenic'), ('Trait / Phenotype', 'Kr-0')), ('REGULATES', ('Genomic Region', 'a single major locus'), ('Trait / Phenotype', 'high leaf arsenic'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 26\n",
            "Text:  Arsenic concentrations were measured in leaves from 315 F2 plants (Figure 1C) from the Kr06Col-0 cross and approximately one-quarter of these plants (75 of 315, X[2] = 0.24,X[2]0.05(1) [= 3.84) had high leaf arsenic, suggest-] ing that high leaf arsenic in Kr-0 is controlled by a single major locus.\n",
            "Gold Relation Type:  HAS_MEASUREMENT\n",
            "Gold Arg1 (Type/Text):  Population / Group  //  315 F2 plants\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  Arsenic concentrations\n",
            "All Predicted Relations:  [('LOCATED_IN', ('Trait / Phenotype', 'Arsenic concentrations'), ('Tissue / Plant Structure', 'leaves')), ('PART_OF', ('Population / Group', '315 F2 plants'), ('Population / Group', 'Kr06Col-0 cross')), ('ASSOCIATED_WITH', ('Trait / Phenotype', 'high leaf arsenic'), ('Trait / Phenotype', 'Kr-0')), ('REGULATES', ('Genomic Region', 'a single major locus'), ('Trait / Phenotype', 'high leaf arsenic'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 27\n",
            "Text:  Arsenic concentrations were measured in leaves from 315 F2 plants (Figure 1C) from the Kr06Col-0 cross and approximately one-quarter of these plants (75 of 315, X[2] = 0.24,X[2]0.05(1) [= 3.84) had high leaf arsenic, suggest-] ing that high leaf arsenic in Kr-0 is controlled by a single major locus.\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Accession / Line  //  Kr-0\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  high leaf arsenic\n",
            "All Predicted Relations:  [('LOCATED_IN', ('Trait / Phenotype', 'Arsenic concentrations'), ('Tissue / Plant Structure', 'leaves')), ('PART_OF', ('Population / Group', '315 F2 plants'), ('Population / Group', 'Kr06Col-0 cross')), ('ASSOCIATED_WITH', ('Trait / Phenotype', 'high leaf arsenic'), ('Trait / Phenotype', 'Kr-0')), ('REGULATES', ('Genomic Region', 'a single major locus'), ('Trait / Phenotype', 'high leaf arsenic'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 28\n",
            "Text:  Arsenic concentrations were measured in leaves from 315 F2 plants (Figure 1C) from the Kr06Col-0 cross and approximately one-quarter of these plants (75 of 315, X[2] = 0.24,X[2]0.05(1) [= 3.84) had high leaf arsenic, suggest-] ing that high leaf arsenic in Kr-0 is controlled by a single major locus.\n",
            "Gold Relation Type:  CAUSES\n",
            "Gold Arg1 (Type/Text):  Genomic Region  //  a single major locus\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  high leaf arsenic\n",
            "All Predicted Relations:  [('LOCATED_IN', ('Trait / Phenotype', 'Arsenic concentrations'), ('Tissue / Plant Structure', 'leaves')), ('PART_OF', ('Population / Group', '315 F2 plants'), ('Population / Group', 'Kr06Col-0 cross')), ('ASSOCIATED_WITH', ('Trait / Phenotype', 'high leaf arsenic'), ('Trait / Phenotype', 'Kr-0')), ('REGULATES', ('Genomic Region', 'a single major locus'), ('Trait / Phenotype', 'high leaf arsenic'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 29\n",
            "Text:  Such imaging revealed that in the absence of a functional HAC1 gene, and with the loss of arsenite efflux, arsenic appears to accumulate mainly in the root stele to concentrations 120%–180% higher than in the Col-0 wild type (Figure 9).\n",
            "Gold Relation Type:  HAS_MEASUREMENT\n",
            "Gold Arg1 (Type/Text):  Chemical Substance  //  arsenic\n",
            "Gold Arg2 (Type/Text):  Measurement / Statistical Result  //  concentrations 120%–180% higher than in the Col-0 wild type\n",
            "All Predicted Relations:  [('REGULATES', ('Gene', 'HAC1 gene'), ('Biological Process', 'arsenite efflux')), ('LOCATED_IN', ('Chemical Substance', 'arsenic'), ('Tissue / Plant Structure', 'root stele'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 30\n",
            "Text:  Regardless of method used, six phenotypes yield single, strong peaks of association that can be seen by inspection.\n",
            "Gold Relation Type:  HAS_MEASUREMENT\n",
            "Gold Arg1 (Type/Text):  Trait / Phenotype  //  six phenotypes\n",
            "Gold Arg2 (Type/Text):  Measurement / Statistical Result  //  single, strong peaks of association\n",
            "All Predicted Relations:  [('CAUSES', ('Trait / Phenotype', 'six phenotypes'), ('Concept / Abstract Entity', 'single, strong peaks of association')), ('IDENTIFIED_IN', ('Concept / Abstract Entity', 'single, strong peaks of association'), ('Concept / Abstract Entity', 'inspection'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 31\n",
            "Text:  On the other hand, arsenate uptake, measured as depletion of arsenate from the nutrient solution, was no different in hac1-1, hac1-2, or Kr-0 compared to wild-type Col-0 at both time points tested (Figure 8B).\n",
            "Gold Relation Type:  HAS_MEASUREMENT\n",
            "Gold Arg1 (Type/Text):  Trait / Phenotype  //  arsenate uptake\n",
            "Gold Arg2 (Type/Text):  Measurement / Statistical Result  //  depletion of arsenate from the nutrient solution\n",
            "All Predicted Relations:  [('HAS_MEASUREMENT', ('Trait / Phenotype', 'arsenate uptake'), ('Trait / Phenotype', 'depletion of arsenate from the nutrient solution')), ('NOT_ASSOCIATED_WITH', ('Accession / Line', 'hac1-1'), ('Trait / Phenotype', 'arsenate uptake')), ('NOT_ASSOCIATED_WITH', ('Accession / Line', 'hac1-2'), ('Trait / Phenotype', 'arsenate uptake')), ('NOT_ASSOCIATED_WITH', ('Accession / Line', 'Kr-0'), ('Trait / Phenotype', 'arsenate uptake')), ('NOT_ASSOCIATED_WITH', ('Accession / Line', 'wild-type Col-0'), ('Trait / Phenotype', 'arsenate uptake'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 32\n",
            "Text:  On the other hand, arsenate uptake, measured as depletion of arsenate from the nutrient solution, was no different in hac1-1, hac1-2, or Kr-0 compared to wild-type Col-0 at both time points tested (Figure 8B).\n",
            "Gold Relation Type:  NOT_ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Allele / Variant  //  hac1-1\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  arsenate uptake\n",
            "All Predicted Relations:  [('HAS_MEASUREMENT', ('Trait / Phenotype', 'arsenate uptake'), ('Trait / Phenotype', 'depletion of arsenate from the nutrient solution')), ('NOT_ASSOCIATED_WITH', ('Accession / Line', 'hac1-1'), ('Trait / Phenotype', 'arsenate uptake')), ('NOT_ASSOCIATED_WITH', ('Accession / Line', 'hac1-2'), ('Trait / Phenotype', 'arsenate uptake')), ('NOT_ASSOCIATED_WITH', ('Accession / Line', 'Kr-0'), ('Trait / Phenotype', 'arsenate uptake')), ('NOT_ASSOCIATED_WITH', ('Accession / Line', 'wild-type Col-0'), ('Trait / Phenotype', 'arsenate uptake'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 33\n",
            "Text:  On the other hand, arsenate uptake, measured as depletion of arsenate from the nutrient solution, was no different in hac1-1, hac1-2, or Kr-0 compared to wild-type Col-0 at both time points tested (Figure 8B).\n",
            "Gold Relation Type:  NOT_ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Allele / Variant  //  hac1-2\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  arsenate uptake\n",
            "All Predicted Relations:  [('HAS_MEASUREMENT', ('Trait / Phenotype', 'arsenate uptake'), ('Trait / Phenotype', 'depletion of arsenate from the nutrient solution')), ('NOT_ASSOCIATED_WITH', ('Accession / Line', 'hac1-1'), ('Trait / Phenotype', 'arsenate uptake')), ('NOT_ASSOCIATED_WITH', ('Accession / Line', 'hac1-2'), ('Trait / Phenotype', 'arsenate uptake')), ('NOT_ASSOCIATED_WITH', ('Accession / Line', 'Kr-0'), ('Trait / Phenotype', 'arsenate uptake')), ('NOT_ASSOCIATED_WITH', ('Accession / Line', 'wild-type Col-0'), ('Trait / Phenotype', 'arsenate uptake'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 34\n",
            "Text:  Both factors are likely to be important in the present study: for example, it is easy to imagine that plants from northern Sweden will tend to share cold-adaptive alleles at many causal loci as a result of selection, and marker alleles genome-wide as a result of demographic history.\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Population / Group  //  plants from northern Sweden\n",
            "Gold Arg2 (Type/Text):  Allele / Variant  //  cold-adaptive alleles\n",
            "All Predicted Relations:  []\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 35\n",
            "Text:  Both factors are likely to be important in the present study: for example, it is easy to imagine that plants from northern Sweden will tend to share cold-adaptive alleles at many causal loci as a result of selection, and marker alleles genome-wide as a result of demographic history.\n",
            "Gold Relation Type:  LOCATED_IN\n",
            "Gold Arg1 (Type/Text):  Allele / Variant  //  cold-adaptive alleles\n",
            "Gold Arg2 (Type/Text):  Genomic Region  //  many causal loci\n",
            "All Predicted Relations:  []\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 36\n",
            "Text:  Both factors are likely to be important in the present study: for example, it is easy to imagine that plants from northern Sweden will tend to share cold-adaptive alleles at many causal loci as a result of selection, and marker alleles genome-wide as a result of demographic history.\n",
            "Gold Relation Type:  CAUSES\n",
            "Gold Arg1 (Type/Text):  Biological Process  //  selection\n",
            "Gold Arg2 (Type/Text):  Allele / Variant  //  cold-adaptive alleles\n",
            "All Predicted Relations:  []\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 37\n",
            "Text:  Both factors are likely to be important in the present study: for example, it is easy to imagine that plants from northern Sweden will tend to share cold-adaptive alleles at many causal loci as a result of selection, and marker alleles genome-wide as a result of demographic history.\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Population / Group  //  plants from northern Sweden\n",
            "Gold Arg2 (Type/Text):  Allele / Variant  //  marker alleles genome-wide\n",
            "All Predicted Relations:  []\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 38\n",
            "Text:  Both factors are likely to be important in the present study: for example, it is easy to imagine that plants from northern Sweden will tend to share cold-adaptive alleles at many causal loci as a result of selection, and marker alleles genome-wide as a result of demographic history.\n",
            "Gold Relation Type:  CAUSES\n",
            "Gold Arg1 (Type/Text):  Concept / Abstract Entity  //  demographic history\n",
            "Gold Arg2 (Type/Text):  Allele / Variant  //  marker alleles genome-wide\n",
            "All Predicted Relations:  []\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 39\n",
            "Text:  In summary, A. thaliana has evolved two main strategies to cope with high substrate Pb: (1) exclusion from the shoot by sequestration in the roots and (2) translocation to the shoots and storage in inactive organelles or intracellular spaces.\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Organism  //  A. thaliana\n",
            "Gold Arg2 (Type/Text):  Environmental Factor  //  high substrate Pb\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Chemical Substance', 'high substrate Pb')), ('ASSOCIATED_WITH', ('Biological Process', 'exclusion from the shoot'), ('Chemical Substance', 'high substrate Pb')), ('LOCATED_IN', ('Biological Process', 'exclusion from the shoot'), ('Tissue / Plant Structure', 'roots')), ('ASSOCIATED_WITH', ('Biological Process', 'sequestration'), ('Chemical Substance', 'high substrate Pb')), ('LOCATED_IN', ('Biological Process', 'sequestration'), ('Tissue / Plant Structure', 'roots')), ('ASSOCIATED_WITH', ('Biological Process', 'translocation to the shoots'), ('Chemical Substance', 'high substrate Pb')), ('LOCATED_IN', ('Biological Process', 'translocation to the shoots'), ('Biological Process', 'inactive organelles')), ('CONTRIBUTES_TO', ('Biological Process', 'translocation to the shoots'), ('Biological Process', 'storage in inactive organelles or intracellular spaces'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 40\n",
            "Text:  In summary, A. thaliana has evolved two main strategies to cope with high substrate Pb: (1) exclusion from the shoot by sequestration in the roots and (2) translocation to the shoots and storage in inactive organelles or intracellular spaces.\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Organism  //  A. thaliana\n",
            "Gold Arg2 (Type/Text):  Biological Process  //  exclusion from the shoot by sequestration in the roots\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Chemical Substance', 'high substrate Pb')), ('ASSOCIATED_WITH', ('Biological Process', 'exclusion from the shoot'), ('Chemical Substance', 'high substrate Pb')), ('LOCATED_IN', ('Biological Process', 'exclusion from the shoot'), ('Tissue / Plant Structure', 'roots')), ('ASSOCIATED_WITH', ('Biological Process', 'sequestration'), ('Chemical Substance', 'high substrate Pb')), ('LOCATED_IN', ('Biological Process', 'sequestration'), ('Tissue / Plant Structure', 'roots')), ('ASSOCIATED_WITH', ('Biological Process', 'translocation to the shoots'), ('Chemical Substance', 'high substrate Pb')), ('LOCATED_IN', ('Biological Process', 'translocation to the shoots'), ('Biological Process', 'inactive organelles')), ('CONTRIBUTES_TO', ('Biological Process', 'translocation to the shoots'), ('Biological Process', 'storage in inactive organelles or intracellular spaces'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 41\n",
            "Text:  In summary, A. thaliana has evolved two main strategies to cope with high substrate Pb: (1) exclusion from the shoot by sequestration in the roots and (2) translocation to the shoots and storage in inactive organelles or intracellular spaces.\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Organism  //  A. thaliana\n",
            "Gold Arg2 (Type/Text):  Biological Process  //  translocation to the shoots and storage in inactive organelles or intracellular spaces\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Chemical Substance', 'high substrate Pb')), ('ASSOCIATED_WITH', ('Biological Process', 'exclusion from the shoot'), ('Chemical Substance', 'high substrate Pb')), ('LOCATED_IN', ('Biological Process', 'exclusion from the shoot'), ('Tissue / Plant Structure', 'roots')), ('ASSOCIATED_WITH', ('Biological Process', 'sequestration'), ('Chemical Substance', 'high substrate Pb')), ('LOCATED_IN', ('Biological Process', 'sequestration'), ('Tissue / Plant Structure', 'roots')), ('ASSOCIATED_WITH', ('Biological Process', 'translocation to the shoots'), ('Chemical Substance', 'high substrate Pb')), ('LOCATED_IN', ('Biological Process', 'translocation to the shoots'), ('Biological Process', 'inactive organelles')), ('CONTRIBUTES_TO', ('Biological Process', 'translocation to the shoots'), ('Biological Process', 'storage in inactive organelles or intracellular spaces'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 42\n",
            "Text:  This suggested the possibility that HAC1 functions as an arsenate reductase, reducing arsenate (As[V]) to arsenite (As[III]).\n",
            "Gold Relation Type:  CAUSES\n",
            "Gold Arg1 (Type/Text):  Protein / Polypeptide  //  arsenate reductase\n",
            "Gold Arg2 (Type/Text):  Biological Process  //  reducing arsenate (As[V]) to arsenite (As[III])\n",
            "All Predicted Relations:  [('ENCODES', ('Gene', 'HAC1'), ('Protein / Polypeptide', 'arsenate reductase')), ('CONTRIBUTES_TO', ('Protein / Polypeptide', 'arsenate reductase'), ('Protein / Polypeptide', 'reducing arsenate (As[V]) to arsenite (As[III]'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 43\n",
            "Text:  However, foliar uptake is an important second pathway in areas with high fallout of airborne particles (Uzu et al., 2010; Luo et al., 2019).\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Biological Process  //  foliar uptake\n",
            "Gold Arg2 (Type/Text):  Environmental Factor  //  areas with high fallout of airborne particles\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Biological Process', 'foliar uptake'), ('Environmental Factor', 'high fallout of airborne particles'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 44\n",
            "Text:  However, foliar uptake is an important second pathway in areas with high fallout of airborne particles (Uzu et al., 2010; Luo et al., 2019).\n",
            "Gold Relation Type:  IDENTIFIED_IN\n",
            "Gold Arg1 (Type/Text):  Biological Process  //  foliar uptake\n",
            "Gold Arg2 (Type/Text):  Study / Analysis  //  Uzu et al., 2010\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Biological Process', 'foliar uptake'), ('Environmental Factor', 'high fallout of airborne particles'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 45\n",
            "Text:  However, foliar uptake is an important second pathway in areas with high fallout of airborne particles (Uzu et al., 2010; Luo et al., 2019).\n",
            "Gold Relation Type:  IDENTIFIED_IN\n",
            "Gold Arg1 (Type/Text):  Biological Process  //  foliar uptake\n",
            "Gold Arg2 (Type/Text):  Study / Analysis  //  Luo et al., 2019\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Biological Process', 'foliar uptake'), ('Environmental Factor', 'high fallout of airborne particles'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 46\n",
            "Text:  It is also interesting to note that since the majority of the A. thaliana accessions tested have the HAC1 allele associated with higher leaf arsenic, they are likely to have weak alleles of HAC1.\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Allele / Variant  //  the HAC1 allele\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  higher leaf arsenic\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Population / Group', 'the majority of the A. thaliana accessions tested'), ('Allele / Variant', 'HAC1 allele')), ('ASSOCIATED_WITH', ('Allele / Variant', 'HAC1 allele'), ('Trait / Phenotype', 'higher leaf arsenic')), ('PART_OF', ('Allele / Variant', 'HAC1 allele'), ('Gene', 'HAC1')), ('PART_OF', ('Allele / Variant', 'weak alleles of HAC1'), ('Gene', 'HAC1'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 47\n",
            "Text:  It is also interesting to note that since the majority of the A. thaliana accessions tested have the HAC1 allele associated with higher leaf arsenic, they are likely to have weak alleles of HAC1.\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Population / Group  //  the majority of the A. thaliana accessions tested\n",
            "Gold Arg2 (Type/Text):  Allele / Variant  //  the HAC1 allele\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Population / Group', 'the majority of the A. thaliana accessions tested'), ('Allele / Variant', 'HAC1 allele')), ('ASSOCIATED_WITH', ('Allele / Variant', 'HAC1 allele'), ('Trait / Phenotype', 'higher leaf arsenic')), ('PART_OF', ('Allele / Variant', 'HAC1 allele'), ('Gene', 'HAC1')), ('PART_OF', ('Allele / Variant', 'weak alleles of HAC1'), ('Gene', 'HAC1'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 48\n",
            "Text:  It is also interesting to note that since the majority of the A. thaliana accessions tested have the HAC1 allele associated with higher leaf arsenic, they are likely to have weak alleles of HAC1.\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Population / Group  //  the majority of the A. thaliana accessions tested\n",
            "Gold Arg2 (Type/Text):  Allele / Variant  //  weak alleles of HAC1\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Population / Group', 'the majority of the A. thaliana accessions tested'), ('Allele / Variant', 'HAC1 allele')), ('ASSOCIATED_WITH', ('Allele / Variant', 'HAC1 allele'), ('Trait / Phenotype', 'higher leaf arsenic')), ('PART_OF', ('Allele / Variant', 'HAC1 allele'), ('Gene', 'HAC1')), ('PART_OF', ('Allele / Variant', 'weak alleles of HAC1'), ('Gene', 'HAC1'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 49\n",
            "Text:  Here, we expand on this observation by establishing the strength of the AtHKT1;1 alleles in four further high Na[+] accumulating accessions (Bur-1, Duk, PHW-20 and UKNW06-386) that all contain a T at Chr4:6392276, along with a low leaf Na[+] accession (Nd-1) with a C at Chr4:6392276.\n",
            "Gold Relation Type:  IS_A\n",
            "Gold Arg1 (Type/Text):  Accession / Line  //  Bur-1\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  high Na[+] accumulating\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Accession / Line', 'Bur-1'), ('Allele / Variant', 'AtHKT1;1 alleles')), ('ASSOCIATED_WITH', ('Accession / Line', 'PHW-20'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Duk'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Bur-1'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Nd-1)'), ('Allele / Variant', 'C at Chr4:6392276'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 50\n",
            "Text:  Here, we expand on this observation by establishing the strength of the AtHKT1;1 alleles in four further high Na[+] accumulating accessions (Bur-1, Duk, PHW-20 and UKNW06-386) that all contain a T at Chr4:6392276, along with a low leaf Na[+] accession (Nd-1) with a C at Chr4:6392276.\n",
            "Gold Relation Type:  IS_A\n",
            "Gold Arg1 (Type/Text):  Accession / Line  //  Duk\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  high Na[+] accumulating\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Accession / Line', 'Bur-1'), ('Allele / Variant', 'AtHKT1;1 alleles')), ('ASSOCIATED_WITH', ('Accession / Line', 'PHW-20'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Duk'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Bur-1'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Nd-1)'), ('Allele / Variant', 'C at Chr4:6392276'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 51\n",
            "Text:  Here, we expand on this observation by establishing the strength of the AtHKT1;1 alleles in four further high Na[+] accumulating accessions (Bur-1, Duk, PHW-20 and UKNW06-386) that all contain a T at Chr4:6392276, along with a low leaf Na[+] accession (Nd-1) with a C at Chr4:6392276.\n",
            "Gold Relation Type:  IS_A\n",
            "Gold Arg1 (Type/Text):  Accession / Line  //  PHW-20\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  high Na[+] accumulating\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Accession / Line', 'Bur-1'), ('Allele / Variant', 'AtHKT1;1 alleles')), ('ASSOCIATED_WITH', ('Accession / Line', 'PHW-20'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Duk'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Bur-1'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Nd-1)'), ('Allele / Variant', 'C at Chr4:6392276'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 52\n",
            "Text:  Here, we expand on this observation by establishing the strength of the AtHKT1;1 alleles in four further high Na[+] accumulating accessions (Bur-1, Duk, PHW-20 and UKNW06-386) that all contain a T at Chr4:6392276, along with a low leaf Na[+] accession (Nd-1) with a C at Chr4:6392276.\n",
            "Gold Relation Type:  IS_A\n",
            "Gold Arg1 (Type/Text):  Accession / Line  //  UKNW06-386\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  high Na[+] accumulating\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Accession / Line', 'Bur-1'), ('Allele / Variant', 'AtHKT1;1 alleles')), ('ASSOCIATED_WITH', ('Accession / Line', 'PHW-20'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Duk'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Bur-1'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Nd-1)'), ('Allele / Variant', 'C at Chr4:6392276'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 53\n",
            "Text:  Here, we expand on this observation by establishing the strength of the AtHKT1;1 alleles in four further high Na[+] accumulating accessions (Bur-1, Duk, PHW-20 and UKNW06-386) that all contain a T at Chr4:6392276, along with a low leaf Na[+] accession (Nd-1) with a C at Chr4:6392276.\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Accession / Line  //  UKNW06-386\n",
            "Gold Arg2 (Type/Text):  Allele / Variant  //  T at Chr4:6392276\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Accession / Line', 'Bur-1'), ('Allele / Variant', 'AtHKT1;1 alleles')), ('ASSOCIATED_WITH', ('Accession / Line', 'PHW-20'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Duk'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Bur-1'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Nd-1)'), ('Allele / Variant', 'C at Chr4:6392276'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 54\n",
            "Text:  Here, we expand on this observation by establishing the strength of the AtHKT1;1 alleles in four further high Na[+] accumulating accessions (Bur-1, Duk, PHW-20 and UKNW06-386) that all contain a T at Chr4:6392276, along with a low leaf Na[+] accession (Nd-1) with a C at Chr4:6392276.\n",
            "Gold Relation Type:  IS_A\n",
            "Gold Arg1 (Type/Text):  Accession / Line  //  Nd-1\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  low leaf Na[+]\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Accession / Line', 'Bur-1'), ('Allele / Variant', 'AtHKT1;1 alleles')), ('ASSOCIATED_WITH', ('Accession / Line', 'PHW-20'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Duk'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Bur-1'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Nd-1)'), ('Allele / Variant', 'C at Chr4:6392276'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 55\n",
            "Text:  Here, we expand on this observation by establishing the strength of the AtHKT1;1 alleles in four further high Na[+] accumulating accessions (Bur-1, Duk, PHW-20 and UKNW06-386) that all contain a T at Chr4:6392276, along with a low leaf Na[+] accession (Nd-1) with a C at Chr4:6392276.\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Accession / Line  //  Nd-1\n",
            "Gold Arg2 (Type/Text):  Allele / Variant  //  C at Chr4:6392276\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Accession / Line', 'Bur-1'), ('Allele / Variant', 'AtHKT1;1 alleles')), ('ASSOCIATED_WITH', ('Accession / Line', 'PHW-20'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Duk'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Bur-1'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Nd-1)'), ('Allele / Variant', 'C at Chr4:6392276'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 56\n",
            "Text:  In the hac1 A. thaliana mutant that lacks the arsenate reductase activity required for arsenite efflux, we observe an over-accumulation of arsenic within the central stele of the root. \n",
            "Gold Relation Type:  NOT_ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Accession / Line  //  hac1 A. thaliana mutant\n",
            "Gold Arg2 (Type/Text):  Biological Process  //  arsenate reductase activity\n",
            "All Predicted Relations:  [('NOT_ASSOCIATED_WITH', ('Accession / Line', 'hac1 A. thaliana mutant'), ('Protein / Polypeptide', 'arsenate reductase activity')), ('CONTRIBUTES_TO', ('Protein / Polypeptide', 'arsenate reductase activity'), ('Biological Process', 'arsenite efflux')), ('LOCATED_IN', ('Chemical Substance', 'arsenic'), ('Tissue / Plant Structure', 'central stele of the root'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 57\n",
            "Text:  In the hac1 A. thaliana mutant that lacks the arsenate reductase activity required for arsenite efflux, we observe an over-accumulation of arsenic within the central stele of the root. \n",
            "Gold Relation Type:  CONTRIBUTES_TO\n",
            "Gold Arg1 (Type/Text):  Biological Process  //  arsenate reductase activity\n",
            "Gold Arg2 (Type/Text):  Biological Process  //  arsenite efflux\n",
            "All Predicted Relations:  [('NOT_ASSOCIATED_WITH', ('Accession / Line', 'hac1 A. thaliana mutant'), ('Protein / Polypeptide', 'arsenate reductase activity')), ('CONTRIBUTES_TO', ('Protein / Polypeptide', 'arsenate reductase activity'), ('Biological Process', 'arsenite efflux')), ('LOCATED_IN', ('Chemical Substance', 'arsenic'), ('Tissue / Plant Structure', 'central stele of the root'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 58\n",
            "Text:  In the hac1 A. thaliana mutant that lacks the arsenate reductase activity required for arsenite efflux, we observe an over-accumulation of arsenic within the central stele of the root. \n",
            "Gold Relation Type:  LOCATED_IN\n",
            "Gold Arg1 (Type/Text):  Trait / Phenotype  //  over-accumulation of arsenic\n",
            "Gold Arg2 (Type/Text):  Tissue / Plant Structure  //  central stele\n",
            "All Predicted Relations:  [('NOT_ASSOCIATED_WITH', ('Accession / Line', 'hac1 A. thaliana mutant'), ('Protein / Polypeptide', 'arsenate reductase activity')), ('CONTRIBUTES_TO', ('Protein / Polypeptide', 'arsenate reductase activity'), ('Biological Process', 'arsenite efflux')), ('LOCATED_IN', ('Chemical Substance', 'arsenic'), ('Tissue / Plant Structure', 'central stele of the root'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 59\n",
            "Text:  In the hac1 A. thaliana mutant that lacks the arsenate reductase activity required for arsenite efflux, we observe an over-accumulation of arsenic within the central stele of the root. \n",
            "Gold Relation Type:  PART_OF\n",
            "Gold Arg1 (Type/Text):  Tissue / Plant Structure  //  central stele\n",
            "Gold Arg2 (Type/Text):  Tissue / Plant Structure  //  root\n",
            "All Predicted Relations:  [('NOT_ASSOCIATED_WITH', ('Accession / Line', 'hac1 A. thaliana mutant'), ('Protein / Polypeptide', 'arsenate reductase activity')), ('CONTRIBUTES_TO', ('Protein / Polypeptide', 'arsenate reductase activity'), ('Biological Process', 'arsenite efflux')), ('LOCATED_IN', ('Chemical Substance', 'arsenic'), ('Tissue / Plant Structure', 'central stele of the root'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 60\n",
            "Text:  Pb entering the symplast is removed by the activity of efflux pumps present in the plasma membrane.\n",
            "Gold Relation Type:  LOCATED_IN\n",
            "Gold Arg1 (Type/Text):  Protein / Polypeptide  //  efflux pumps\n",
            "Gold Arg2 (Type/Text):  Tissue / Plant Structure  //  plasma membrane\n",
            "All Predicted Relations:  [('REGULATES', ('Biological Process', 'efflux pumps'), ('Biological Process', 'efflux pumps'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 61\n",
            "Text:  Pb entering the symplast is removed by the activity of efflux pumps present in the plasma membrane.\n",
            "Gold Relation Type:  LOCATED_IN\n",
            "Gold Arg1 (Type/Text):  Chemical Substance  //  Pb\n",
            "Gold Arg2 (Type/Text):  Tissue / Plant Structure  //  symplast\n",
            "All Predicted Relations:  [('REGULATES', ('Biological Process', 'efflux pumps'), ('Biological Process', 'efflux pumps'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 62\n",
            "Text:  In addition, Na[+]  compartmentalization facilitates vacuolar osmotic adjustment that is necessary to compensate for the osmotic effects of salinity by maintaining turgor pressure for cell expansion and growth.\n",
            "Gold Relation Type:  CONTRIBUTES_TO\n",
            "Gold Arg1 (Type/Text):  Biological Process  //  Na[+]  compartmentalization\n",
            "Gold Arg2 (Type/Text):  Biological Process  //  vacuolar osmotic adjustment\n",
            "All Predicted Relations:  [('CONTRIBUTES_TO', ('Biological Process', 'vacuolar osmotic adjustment'), ('Trait / Phenotype', 'turgor pressure')), ('CONTRIBUTES_TO', ('Biological Process', 'vacuolar osmotic adjustment'), ('Biological Process', 'growth'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 63\n",
            "Text:  In addition, Na[+]  compartmentalization facilitates vacuolar osmotic adjustment that is necessary to compensate for the osmotic effects of salinity by maintaining turgor pressure for cell expansion and growth.\n",
            "Gold Relation Type:  CONTRIBUTES_TO\n",
            "Gold Arg1 (Type/Text):  Trait / Phenotype  //  turgor pressure\n",
            "Gold Arg2 (Type/Text):  Biological Process  //  cell expansion and growth\n",
            "All Predicted Relations:  [('CONTRIBUTES_TO', ('Biological Process', 'vacuolar osmotic adjustment'), ('Trait / Phenotype', 'turgor pressure')), ('CONTRIBUTES_TO', ('Biological Process', 'vacuolar osmotic adjustment'), ('Biological Process', 'growth'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 64\n",
            "Text:  In addition, Na[+]  compartmentalization facilitates vacuolar osmotic adjustment that is necessary to compensate for the osmotic effects of salinity by maintaining turgor pressure for cell expansion and growth.\n",
            "Gold Relation Type:  CONTRIBUTES_TO\n",
            "Gold Arg1 (Type/Text):  Biological Process  //  vacuolar osmotic adjustment\n",
            "Gold Arg2 (Type/Text):  Environmental Factor  //  salinity\n",
            "All Predicted Relations:  [('CONTRIBUTES_TO', ('Biological Process', 'vacuolar osmotic adjustment'), ('Trait / Phenotype', 'turgor pressure')), ('CONTRIBUTES_TO', ('Biological Process', 'vacuolar osmotic adjustment'), ('Biological Process', 'growth'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 65\n",
            "Text:  Whether the wide variety of natural phenotypic and genetic variation observed in A. thaliana [12] contributes to its local adaptation is an important unresolved question that is currently attracting a significant amount of attention [13].\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Trait / Phenotype  //  natural phenotypic and\n",
            "Gold Arg2 (Type/Text):  Organism  //  A. thaliana\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Concept / Abstract Entity', 'the wide variety of natural phenotypic and genetic variation'), ('Organism', 'A. thaliana')), ('CONTRIBUTES_TO', ('Concept / Abstract Entity', 'the wide variety of natural phenotypic and genetic variation'), ('Biological Process', 'local adaptation'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 66\n",
            "Text:  Whether the wide variety of natural phenotypic and genetic variation observed in A. thaliana [12] contributes to its local adaptation is an important unresolved question that is currently attracting a significant amount of attention [13].\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Concept / Abstract Entity  //  genetic variation\n",
            "Gold Arg2 (Type/Text):  Organism  //  A. thaliana\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Concept / Abstract Entity', 'the wide variety of natural phenotypic and genetic variation'), ('Organism', 'A. thaliana')), ('CONTRIBUTES_TO', ('Concept / Abstract Entity', 'the wide variety of natural phenotypic and genetic variation'), ('Biological Process', 'local adaptation'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 67\n",
            "Text:  Whether the wide variety of natural phenotypic and genetic variation observed in A. thaliana [12] contributes to its local adaptation is an important unresolved question that is currently attracting a significant amount of attention [13].\n",
            "Gold Relation Type:  CONTRIBUTES_TO\n",
            "Gold Arg1 (Type/Text):  Trait / Phenotype  //  natural phenotypic and\n",
            "Gold Arg2 (Type/Text):  Biological Process  //  local adaptation\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Concept / Abstract Entity', 'the wide variety of natural phenotypic and genetic variation'), ('Organism', 'A. thaliana')), ('CONTRIBUTES_TO', ('Concept / Abstract Entity', 'the wide variety of natural phenotypic and genetic variation'), ('Biological Process', 'local adaptation'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 68\n",
            "Text:  Whether the wide variety of natural phenotypic and genetic variation observed in A. thaliana [12] contributes to its local adaptation is an important unresolved question that is currently attracting a significant amount of attention [13].\n",
            "Gold Relation Type:  CONTRIBUTES_TO\n",
            "Gold Arg1 (Type/Text):  Concept / Abstract Entity  //  genetic variation\n",
            "Gold Arg2 (Type/Text):  Biological Process  //  local adaptation\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Concept / Abstract Entity', 'the wide variety of natural phenotypic and genetic variation'), ('Organism', 'A. thaliana')), ('CONTRIBUTES_TO', ('Concept / Abstract Entity', 'the wide variety of natural phenotypic and genetic variation'), ('Biological Process', 'local adaptation'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 69\n",
            "Text:  This increase in HAC1 mRNA levels reaches a maximum at 100 mM arsenate in the growth media (Figure 7A).\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Trait / Phenotype  //  HAC1 mRNA levels\n",
            "Gold Arg2 (Type/Text):  Chemical Substance  //  100 mM arsenate\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Trait / Phenotype', 'increase in HAC1 mRNA levels'), ('Trait / Phenotype', '100 mM arsenate in the growth media')), ('LOCATED_IN', ('Chemical Substance', '100 mM arsenate'), ('Environmental Factor', 'growth media'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 70\n",
            "Text:  To validate the existence of this leaf arsenic QTL identified by GWA analysis we created a synthetic F2 recombinant population in which alternative alleles of the dialleic SNP Chr2:9008060 were segregating.\n",
            "Gold Relation Type:  IDENTIFIED_IN\n",
            "Gold Arg1 (Type/Text):  Allele / Variant  //  leaf arsenic QTL\n",
            "Gold Arg2 (Type/Text):  Study / Analysis  //  GWA analysis\n",
            "All Predicted Relations:  [('IDENTIFIED_IN', ('Trait / Phenotype', 'leaf arsenic QTL'), ('Study / Analysis', 'GWA analysis')), ('PART_OF', ('Allele / Variant', 'alternative alleles of the dialleic SNP Chr2:9008060'), ('Population / Group', 'a synthetic F2 recombinant population')), ('PART_OF', ('Allele / Variant', 'the dialleic SNP Chr2:9008060'), ('Population / Group', 'a synthetic F2 recombinant population'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 71\n",
            "Text:  To validate the existence of this leaf arsenic QTL identified by GWA analysis we created a synthetic F2 recombinant population in which alternative alleles of the dialleic SNP Chr2:9008060 were segregating.\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Population / Group  //  synthetic F2 recombinant population\n",
            "Gold Arg2 (Type/Text):  Allele / Variant  //  alternative alleles of the dialleic SNP Chr2:9008060\n",
            "All Predicted Relations:  [('IDENTIFIED_IN', ('Trait / Phenotype', 'leaf arsenic QTL'), ('Study / Analysis', 'GWA analysis')), ('PART_OF', ('Allele / Variant', 'alternative alleles of the dialleic SNP Chr2:9008060'), ('Population / Group', 'a synthetic F2 recombinant population')), ('PART_OF', ('Allele / Variant', 'the dialleic SNP Chr2:9008060'), ('Population / Group', 'a synthetic F2 recombinant population'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 72\n",
            "Text:  This knowledge creates new opportunities to limit arsenic accumulation in food crops, thereby helping to reduce the cancer risk from this food-chain contaminant.\n",
            "Gold Relation Type:  LOCATED_IN\n",
            "Gold Arg1 (Type/Text):  Trait / Phenotype  //  arsenic accumulation\n",
            "Gold Arg2 (Type/Text):  Population / Group  //  food crops\n",
            "All Predicted Relations:  [('CONTRIBUTES_TO', ('Concept / Abstract Entity', 'This knowledge'), ('Trait / Phenotype', 'arsenic accumulation in food crops')), ('CONTRIBUTES_TO', ('Trait / Phenotype', 'arsenic accumulation in food crops'), ('Chemical Substance', 'this food-chain contaminant'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 73\n",
            "Text:  This knowledge creates new opportunities to limit arsenic accumulation in food crops, thereby helping to reduce the cancer risk from this food-chain contaminant.\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Trait / Phenotype  //  arsenic accumulation\n",
            "Gold Arg2 (Type/Text):  Chemical Substance  //  food-chain contaminant\n",
            "All Predicted Relations:  [('CONTRIBUTES_TO', ('Concept / Abstract Entity', 'This knowledge'), ('Trait / Phenotype', 'arsenic accumulation in food crops')), ('CONTRIBUTES_TO', ('Trait / Phenotype', 'arsenic accumulation in food crops'), ('Chemical Substance', 'this food-chain contaminant'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 74\n",
            "Text:  This knowledge creates new opportunities to limit arsenic accumulation in food crops, thereby helping to reduce the cancer risk from this food-chain contaminant.\n",
            "Gold Relation Type:  CAUSES\n",
            "Gold Arg1 (Type/Text):  Chemical Substance  //  food-chain contaminant\n",
            "Gold Arg2 (Type/Text):  Concept / Abstract Entity  //  cancer risk\n",
            "All Predicted Relations:  [('CONTRIBUTES_TO', ('Concept / Abstract Entity', 'This knowledge'), ('Trait / Phenotype', 'arsenic accumulation in food crops')), ('CONTRIBUTES_TO', ('Trait / Phenotype', 'arsenic accumulation in food crops'), ('Chemical Substance', 'this food-chain contaminant'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 75\n",
            "Text:  In particular, our goals are (1) to elucidate the mechanisms developed by Pb-tolerant accessions that differ in their Pb sequestration strategies, and (2) to stablish the role of the novel candidate genes identified by GWAs in Pb tolerance.\n",
            "Gold Relation Type:  CONTRIBUTES_TO\n",
            "Gold Arg1 (Type/Text):  Biological Process  //  Pb sequestration strategies\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  Pb tolerance\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Accession / Line', 'Pb-tolerant accessions'), ('Biological Process', 'Pb sequestration strategies')), ('IDENTIFIED_IN', ('Gene', 'novel candidate genes'), ('Study / Analysis', 'GWAs')), ('CONTRIBUTES_TO', ('Gene', 'novel candidate genes'), ('Trait / Phenotype', 'Pb tolerance'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 76\n",
            "Text:  If polymorphism at FRI is taken into account, CRP is no longer significantly associated, but this does not necessarily mean that the association is spurious.\n",
            "Gold Relation Type:  NOT_ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Allele / Variant  //  polymorphism at FRI\n",
            "Gold Arg2 (Type/Text):  Protein / Polypeptide  //  CRP\n",
            "All Predicted Relations:  [('NOT_ASSOCIATED_WITH', ('Allele / Variant', 'polymorphism at FRI'), ('Gene', 'CRP'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 77\n",
            "Text:  However, hac1-1 and hac1-2 were both significantly more sensitive to arsenate compared to the Col-0 wild type at relatively high concentrations of arsenate when growth was measured as root length or overall plant fresh weight (Figure 10).\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Allele / Variant  //  hac1-1\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  more sensitive to arsenate\n",
            "All Predicted Relations:  [('CAUSES', ('Allele / Variant', 'hac1-1'), ('Trait / Phenotype', 'significantly more sensitive')), ('CAUSES', ('Allele / Variant', 'hac1-2'), ('Trait / Phenotype', 'significantly more sensitive')), ('ASSOCIATED_WITH', ('Accession / Line', 'Col-0 wild type'), ('Chemical Substance', 'arsenate'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 78\n",
            "Text:  However, hac1-1 and hac1-2 were both significantly more sensitive to arsenate compared to the Col-0 wild type at relatively high concentrations of arsenate when growth was measured as root length or overall plant fresh weight (Figure 10).\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Allele / Variant  //  hac1-2\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  more sensitive to arsenate\n",
            "All Predicted Relations:  [('CAUSES', ('Allele / Variant', 'hac1-1'), ('Trait / Phenotype', 'significantly more sensitive')), ('CAUSES', ('Allele / Variant', 'hac1-2'), ('Trait / Phenotype', 'significantly more sensitive')), ('ASSOCIATED_WITH', ('Accession / Line', 'Col-0 wild type'), ('Chemical Substance', 'arsenate'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 79\n",
            "Text:  However, hac1-1 and hac1-2 were both significantly more sensitive to arsenate compared to the Col-0 wild type at relatively high concentrations of arsenate when growth was measured as root length or overall plant fresh weight (Figure 10).\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Trait / Phenotype  //  more sensitive to arsenate\n",
            "Gold Arg2 (Type/Text):  Environmental Factor  //  relatively high concentrations of arsenate\n",
            "All Predicted Relations:  [('CAUSES', ('Allele / Variant', 'hac1-1'), ('Trait / Phenotype', 'significantly more sensitive')), ('CAUSES', ('Allele / Variant', 'hac1-2'), ('Trait / Phenotype', 'significantly more sensitive')), ('ASSOCIATED_WITH', ('Accession / Line', 'Col-0 wild type'), ('Chemical Substance', 'arsenate'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 80\n",
            "Text:  On the other hand, both T-HLL and Pbsensitive (S) accessions translocated similar Pb concentrations (Figure 3F), suggesting that (S) plants are not able to maintain the Pb out of the cytosol and it becomes toxic for them.\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Accession / Line  //  T-HLL\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  Pb concentrations\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Accession / Line', 'T-HLL'), ('Accession / Line', 'Pbsensitive (S) accessions')), ('NOT_ASSOCIATED_WITH', ('Chemical Substance', 'Pb'), ('Tissue / Plant Structure', 'cytosol'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 81\n",
            "Text:  On the other hand, both T-HLL and Pbsensitive (S) accessions translocated similar Pb concentrations (Figure 3F), suggesting that (S) plants are not able to maintain the Pb out of the cytosol and it becomes toxic for them.\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Accession / Line  //  Pbsensitive (S) accessions\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  Pb concentrations\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Accession / Line', 'T-HLL'), ('Accession / Line', 'Pbsensitive (S) accessions')), ('NOT_ASSOCIATED_WITH', ('Chemical Substance', 'Pb'), ('Tissue / Plant Structure', 'cytosol'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 82\n",
            "Text:  On the other hand, both T-HLL and Pbsensitive (S) accessions translocated similar Pb concentrations (Figure 3F), suggesting that (S) plants are not able to maintain the Pb out of the cytosol and it becomes toxic for them.\n",
            "Gold Relation Type:  LOCATED_IN\n",
            "Gold Arg1 (Type/Text):  Chemical Substance  //  Pb\n",
            "Gold Arg2 (Type/Text):  Tissue / Plant Structure  //  cytosol\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Accession / Line', 'T-HLL'), ('Accession / Line', 'Pbsensitive (S) accessions')), ('NOT_ASSOCIATED_WITH', ('Chemical Substance', 'Pb'), ('Tissue / Plant Structure', 'cytosol'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 83\n",
            "Text:  On the other hand, both T-HLL and Pbsensitive (S) accessions translocated similar Pb concentrations (Figure 3F), suggesting that (S) plants are not able to maintain the Pb out of the cytosol and it becomes toxic for them.\n",
            "Gold Relation Type:  CAUSES\n",
            "Gold Arg1 (Type/Text):  Chemical Substance  //  Pb\n",
            "Gold Arg2 (Type/Text):  Trait / Phenotype  //  toxic\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Accession / Line', 'T-HLL'), ('Accession / Line', 'Pbsensitive (S) accessions')), ('NOT_ASSOCIATED_WITH', ('Chemical Substance', 'Pb'), ('Tissue / Plant Structure', 'cytosol'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 84\n",
            "Text:  To test this hypothesis, we evaluated the oxidation state of arsenic in plants lacking a functional HAC1 gene.\n",
            "Gold Relation Type:  ASSOCIATED_WITH\n",
            "Gold Arg1 (Type/Text):  Trait / Phenotype  //  the oxidation state of arsenic\n",
            "Gold Arg2 (Type/Text):  Population / Group  //  plants lacking a functional HAC1 gene\n",
            "All Predicted Relations:  [('ASSOCIATED_WITH', ('Trait / Phenotype', 'oxidation state of arsenic'), ('Chemical Substance', 'arsenic'))]\n",
            "Is this gold relation sufficiently covered by the predicted relations? (y/n) n\n",
            "\n",
            "=== Manual Review Complete ===\n",
            "TP Count (covered)   = 46\n",
            "FN Count (not covered) = 39\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "\n",
        "# Load the relation_fp.csv\n",
        "df = pd.read_csv(\"uie_manual_eval/relation_fp.csv\")\n",
        "\n",
        "tp_extra_count = 0  # how many predicted relations are actually correct (missing from gold)\n",
        "fp_count = 0        # how many predicted relations are truly invalid\n",
        "covered_count = 0   # how many predicted relations are effectively covered by the existing gold relations\n",
        "\n",
        "for index, row in df.iterrows():\n",
        "    text = row.get(\"text\", \"\")\n",
        "    pred_relation_type = row.get(\"pred_relation_type\", \"\")\n",
        "    pred_arg1_type = row.get(\"pred_arg1_type\", \"\")\n",
        "    pred_arg1_text = row.get(\"pred_arg1_text\", \"\")\n",
        "    pred_arg2_type = row.get(\"pred_arg2_type\", \"\")\n",
        "    pred_arg2_text = row.get(\"pred_arg2_text\", \"\")\n",
        "    all_gold_relations = row.get(\"all_gold_relations\", \"\")\n",
        "\n",
        "    print(\"\\n---------------------------------------------------------\")\n",
        "    print(f\"Row index: {index}\")\n",
        "    print(\"Text:\", text)\n",
        "    print(\"Predicted Relation Type:\", pred_relation_type)\n",
        "    print(\"Pred Arg1 (Type/Text):\", pred_arg1_type, \" // \", pred_arg1_text)\n",
        "    print(\"Pred Arg2 (Type/Text):\", pred_arg2_type, \" // \", pred_arg2_text)\n",
        "    print(\"All Gold Relations:\", all_gold_relations)\n",
        "\n",
        "    # Prompt user\n",
        "    choice = input(\n",
        "        \"[y] if the prediction is valid and should have been in gold\\n\"\n",
        "        \"[n] if the prediction is truly wrong\\n\"\n",
        "        \"[c] if it is basically covered by an existing gold relation\\n\"\n",
        "        \"=> Your choice (y/n/c)? \"\n",
        "    ).strip().lower()\n",
        "\n",
        "    if choice == 'y':\n",
        "        tp_extra_count += 1\n",
        "    elif choice == 'c':\n",
        "        covered_count += 1\n",
        "    else:\n",
        "        fp_count += 1\n",
        "\n",
        "# Final summary\n",
        "print(\"\\n=== Manual Review Complete ===\")\n",
        "print(f\"TP-extra Count (valid but missing in gold) = {tp_extra_count}\")\n",
        "print(f\"Covered Count                              = {covered_count}\")\n",
        "print(f\"FP Count (truly invalid)                   = {fp_count}\")\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ZlJahm4bX-Bs",
        "outputId": "0a3ef346-3c55-4377-f5e8-9d33635b164e"
      },
      "execution_count": 25,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 0\n",
            "Text:  Because slightly different sets were used in different phenotyping experiments, the total number of lines used was 199 (Supplementary Table 1).\n",
            "Predicted Relation Type: IDENTIFIED_IN\n",
            "Pred Arg1 (Type/Text): Population / Group  //  the total number of lines used\n",
            "Pred Arg2 (Type/Text): Concept / Abstract Entity  //  Supplementary Table 1\n",
            "All Gold Relations: [('HAS_MEASUREMENT', ('Study / Analysis', 'phenotyping experiments'), ('Population / Group', 'lines'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 1\n",
            "Text: By coupling these events with linked DNA polymorphisms genome-wide, the phenotypic effect of multiple alleles across the genome can be tested.\n",
            "Predicted Relation Type: ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Biological Process  //  these events\n",
            "Pred Arg2 (Type/Text): Allele / Variant  //  linked DNA polymorphisms genome-wide\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Concept / Abstract Entity', 'these events'), ('Allele / Variant', 'linked DNA polymorphisms')), ('LOCATED_IN', ('Allele / Variant', 'multiple alleles'), ('Genomic Region', 'the genome')), ('CONTRIBUTES_TO', ('Allele / Variant', 'multiple alleles'), ('Trait / Phenotype', 'the phenotypic effect'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 2\n",
            "Text: By coupling these events with linked DNA polymorphisms genome-wide, the phenotypic effect of multiple alleles across the genome can be tested.\n",
            "Predicted Relation Type: CONTRIBUTES_TO\n",
            "Pred Arg1 (Type/Text): Biological Process  //  these events\n",
            "Pred Arg2 (Type/Text): Trait / Phenotype  //  the phenotypic effect\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Concept / Abstract Entity', 'these events'), ('Allele / Variant', 'linked DNA polymorphisms')), ('LOCATED_IN', ('Allele / Variant', 'multiple alleles'), ('Genomic Region', 'the genome')), ('CONTRIBUTES_TO', ('Allele / Variant', 'multiple alleles'), ('Trait / Phenotype', 'the phenotypic effect'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 3\n",
            "Text: By coupling these events with linked DNA polymorphisms genome-wide, the phenotypic effect of multiple alleles across the genome can be tested.\n",
            "Predicted Relation Type: PART_OF\n",
            "Pred Arg1 (Type/Text): Allele / Variant  //  multiple alleles across the genome\n",
            "Pred Arg2 (Type/Text): Allele / Variant  //  the genome\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Concept / Abstract Entity', 'these events'), ('Allele / Variant', 'linked DNA polymorphisms')), ('LOCATED_IN', ('Allele / Variant', 'multiple alleles'), ('Genomic Region', 'the genome')), ('CONTRIBUTES_TO', ('Allele / Variant', 'multiple alleles'), ('Trait / Phenotype', 'the phenotypic effect'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 4\n",
            "Text: Because of this serious and widespread food safety concern research into understanding the mechanisms driving arsenic accumulation in plants has become a priority [9].\n",
            "Predicted Relation Type: CAUSES\n",
            "Pred Arg1 (Type/Text): Biological Process  //  the mechanisms\n",
            "Pred Arg2 (Type/Text): Trait / Phenotype  //  arsenic accumulation in plants\n",
            "All Gold Relations: [('CONTRIBUTES_TO', ('Concept / Abstract Entity', 'serious and widespread food safety concern'), ('Study / Analysis', 'research into understanding the mechanisms driving arsenic accumulation in plants')), ('ASSOCIATED_WITH', ('Study / Analysis', 'research into understanding the mechanisms driving arsenic accumulation in plants'), ('Trait / Phenotype', 'arsenic accumulation in plants'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 5\n",
            "Text: Because of this serious and widespread food safety concern research into understanding the mechanisms driving arsenic accumulation in plants has become a priority [9].\n",
            "Predicted Relation Type: IDENTIFIED_IN\n",
            "Pred Arg1 (Type/Text): Biological Process  //  the mechanisms\n",
            "Pred Arg2 (Type/Text): Study / Analysis  //  [9]\n",
            "All Gold Relations: [('CONTRIBUTES_TO', ('Concept / Abstract Entity', 'serious and widespread food safety concern'), ('Study / Analysis', 'research into understanding the mechanisms driving arsenic accumulation in plants')), ('ASSOCIATED_WITH', ('Study / Analysis', 'research into understanding the mechanisms driving arsenic accumulation in plants'), ('Trait / Phenotype', 'arsenic accumulation in plants'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 6\n",
            "Text: This association has also recently been experimentally confirmed (M.T. et al., unpublished observations).\n",
            "Predicted Relation Type: IDENTIFIED_IN\n",
            "Pred Arg1 (Type/Text): Concept / Abstract Entity  //  This association\n",
            "Pred Arg2 (Type/Text): Study / Analysis  //  M.T. et al., unpublished observations\n",
            "All Gold Relations: [('IDENTIFIED_IN', ('Concept / Abstract Entity', 'association'), ('Study / Analysis', 'M.T. et al., unpublished observations'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 7\n",
            "Text: We observed that plants coming from sites with elevated soil Pb content (and acidic pH) had better capacity to tolerate this stress (Figure 2B).\n",
            "Predicted Relation Type: LOCATED_IN\n",
            "Pred Arg1 (Type/Text): Population / Group  //  plants\n",
            "Pred Arg2 (Type/Text): Environmental Factor  //  sites with elevated soil Pb content (and acidic pH)\n",
            "All Gold Relations: [('LOCATED_IN', ('Population / Group', 'plants'), ('Environmental Factor', 'elevated soil Pb content')), ('LOCATED_IN', ('Population / Group', 'plants'), ('Environmental Factor', 'acidic pH')), ('ASSOCIATED_WITH', ('Population / Group', 'plants'), ('Trait / Phenotype', 'better capacity to tolerate this stress'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 8\n",
            "Text: We observed that plants coming from sites with elevated soil Pb content (and acidic pH) had better capacity to tolerate this stress (Figure 2B).\n",
            "Predicted Relation Type: ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Environmental Factor  //  sites with elevated soil Pb content (and acidic pH)\n",
            "Pred Arg2 (Type/Text): Environmental Factor  //  better capacity to tolerate this stress\n",
            "All Gold Relations: [('LOCATED_IN', ('Population / Group', 'plants'), ('Environmental Factor', 'elevated soil Pb content')), ('LOCATED_IN', ('Population / Group', 'plants'), ('Environmental Factor', 'acidic pH')), ('ASSOCIATED_WITH', ('Population / Group', 'plants'), ('Trait / Phenotype', 'better capacity to tolerate this stress'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 9\n",
            "Text: This is an upper bound because it seems almost certain that many of the (often much larger set of) strong associations that are not close to a-priori candidates will also turn out to be real. \n",
            "Predicted Relation Type: NOT_ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Concept / Abstract Entity  //  the (often much larger set of) strong associations\n",
            "Pred Arg2 (Type/Text): Concept / Abstract Entity  //  a-priori candidates\n",
            "All Gold Relations: []\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 10\n",
            "Text: At2g21045 was named HAC1.\n",
            "Predicted Relation Type: IS_A\n",
            "Pred Arg1 (Type/Text): Gene  //  At2g21045\n",
            "Pred Arg2 (Type/Text): Gene  //  At2g21045\n",
            "All Gold Relations: [('IS_A', ('Gene', 'At2g21045'), ('Gene', 'HAC1'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 11\n",
            "Text: However, unlike in synthetic RILs made from a bi-parental cross, the low frequency of rare alleles in a natural population makes it difficult to detect their phenotypic effect using GWA mapping.\n",
            "Predicted Relation Type: PART_OF\n",
            "Pred Arg1 (Type/Text): Population / Group  //  synthetic RILs\n",
            "Pred Arg2 (Type/Text): Population / Group  //  a bi-parental cross\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Population / Group', 'synthetic RILs'), ('Tool / Method', 'bi-parental cross')), ('LOCATED_IN', ('Allele / Variant', 'rare alleles'), ('Population / Group', 'a natural population')), ('ASSOCIATED_WITH', ('Allele / Variant', 'rare alleles'), ('Trait / Phenotype', 'phenotypic effect')), ('IDENTIFIED_IN', ('Trait / Phenotype', 'phenotypic effect'), ('Study / Analysis', 'GWA mapping'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 12\n",
            "Text: However, unlike in synthetic RILs made from a bi-parental cross, the low frequency of rare alleles in a natural population makes it difficult to detect their phenotypic effect using GWA mapping.\n",
            "Predicted Relation Type: PART_OF\n",
            "Pred Arg1 (Type/Text): Allele / Variant  //  rare alleles\n",
            "Pred Arg2 (Type/Text): Population / Group  //  a natural population\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Population / Group', 'synthetic RILs'), ('Tool / Method', 'bi-parental cross')), ('LOCATED_IN', ('Allele / Variant', 'rare alleles'), ('Population / Group', 'a natural population')), ('ASSOCIATED_WITH', ('Allele / Variant', 'rare alleles'), ('Trait / Phenotype', 'phenotypic effect')), ('IDENTIFIED_IN', ('Trait / Phenotype', 'phenotypic effect'), ('Study / Analysis', 'GWA mapping'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 13\n",
            "Text: However, unlike in synthetic RILs made from a bi-parental cross, the low frequency of rare alleles in a natural population makes it difficult to detect their phenotypic effect using GWA mapping.\n",
            "Predicted Relation Type: CAUSES\n",
            "Pred Arg1 (Type/Text): Allele / Variant  //  rare alleles\n",
            "Pred Arg2 (Type/Text): Allele / Variant  //  difficult to detect their phenotypic effect\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Population / Group', 'synthetic RILs'), ('Tool / Method', 'bi-parental cross')), ('LOCATED_IN', ('Allele / Variant', 'rare alleles'), ('Population / Group', 'a natural population')), ('ASSOCIATED_WITH', ('Allele / Variant', 'rare alleles'), ('Trait / Phenotype', 'phenotypic effect')), ('IDENTIFIED_IN', ('Trait / Phenotype', 'phenotypic effect'), ('Study / Analysis', 'GWA mapping'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 14\n",
            "Text: This natural variation has been used to identify specific genes involved in controlling variation in many traits [28].\n",
            "Predicted Relation Type: CONTRIBUTES_TO\n",
            "Pred Arg1 (Type/Text): Gene  //  specific genes\n",
            "Pred Arg2 (Type/Text): Gene  //  controlling variation in many traits\n",
            "All Gold Relations: [('IDENTIFIED_IN', ('Gene', 'specific genes'), ('Concept / Abstract Entity', 'natural variation')), ('REGULATES', ('Gene', 'specific genes'), ('Trait / Phenotype', 'many traits'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 15\n",
            "Text: This natural variation has been used to identify specific genes involved in controlling variation in many traits [28].\n",
            "Predicted Relation Type: IDENTIFIED_IN\n",
            "Pred Arg1 (Type/Text): Gene  //  specific genes\n",
            "Pred Arg2 (Type/Text): Study / Analysis  //  [28]\n",
            "All Gold Relations: [('IDENTIFIED_IN', ('Gene', 'specific genes'), ('Concept / Abstract Entity', 'natural variation')), ('REGULATES', ('Gene', 'specific genes'), ('Trait / Phenotype', 'many traits'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 16\n",
            "Text: This redundancy has also been observed in E. coli [66] and S. cerevisiae [26].\n",
            "Predicted Relation Type: ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Concept / Abstract Entity  //  This redundancy\n",
            "Pred Arg2 (Type/Text): Organism  //  E. coli\n",
            "All Gold Relations: [('IDENTIFIED_IN', ('Concept / Abstract Entity', 'This redundancy'), ('Organism', 'E. coli')), ('IDENTIFIED_IN', ('Concept / Abstract Entity', 'This redundancy'), ('Organism', 'S. cerevisiae'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 17\n",
            "Text: This redundancy has also been observed in E. coli [66] and S. cerevisiae [26].\n",
            "Predicted Relation Type: ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Concept / Abstract Entity  //  This redundancy\n",
            "Pred Arg2 (Type/Text): Organism  //  S. cerevisiae\n",
            "All Gold Relations: [('IDENTIFIED_IN', ('Concept / Abstract Entity', 'This redundancy'), ('Organism', 'E. coli')), ('IDENTIFIED_IN', ('Concept / Abstract Entity', 'This redundancy'), ('Organism', 'S. cerevisiae'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 18\n",
            "Text: However, enzymes forming phosphorylated products can also promote arsenate reduction by incorrectly using arsenate in place of phosphate to generate arsenylated products in which the arsenate is more easily reduced by thiols such as glutathione [67].\n",
            "Predicted Relation Type: CAUSES\n",
            "Pred Arg1 (Type/Text): Protein / Polypeptide  //  enzymes\n",
            "Pred Arg2 (Type/Text): Biological Process  //  arsenate reduction\n",
            "All Gold Relations: [('CONTRIBUTES_TO', ('Protein / Polypeptide', 'enzymes forming phosphorylated products'), ('Biological Process', 'arsenate reduction')), ('CONTRIBUTES_TO', ('Chemical Substance', 'thiols'), ('Biological Process', 'arsenate reduction')), ('IS_A', ('Chemical Substance', 'glutathione'), ('Chemical Substance', 'thiols'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 19\n",
            "Text: However, enzymes forming phosphorylated products can also promote arsenate reduction by incorrectly using arsenate in place of phosphate to generate arsenylated products in which the arsenate is more easily reduced by thiols such as glutathione [67].\n",
            "Predicted Relation Type: IS_A\n",
            "Pred Arg1 (Type/Text): Chemical Substance  //  arsenate\n",
            "Pred Arg2 (Type/Text): Chemical Substance  //  phosphate\n",
            "All Gold Relations: [('CONTRIBUTES_TO', ('Protein / Polypeptide', 'enzymes forming phosphorylated products'), ('Biological Process', 'arsenate reduction')), ('CONTRIBUTES_TO', ('Chemical Substance', 'thiols'), ('Biological Process', 'arsenate reduction')), ('IS_A', ('Chemical Substance', 'glutathione'), ('Chemical Substance', 'thiols'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 20\n",
            "Text: However, enzymes forming phosphorylated products can also promote arsenate reduction by incorrectly using arsenate in place of phosphate to generate arsenylated products in which the arsenate is more easily reduced by thiols such as glutathione [67].\n",
            "Predicted Relation Type: ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Chemical Substance  //  arsenylated products\n",
            "Pred Arg2 (Type/Text): Chemical Substance  //  thiols\n",
            "All Gold Relations: [('CONTRIBUTES_TO', ('Protein / Polypeptide', 'enzymes forming phosphorylated products'), ('Biological Process', 'arsenate reduction')), ('CONTRIBUTES_TO', ('Chemical Substance', 'thiols'), ('Biological Process', 'arsenate reduction')), ('IS_A', ('Chemical Substance', 'glutathione'), ('Chemical Substance', 'thiols'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 21\n",
            "Text: However, enzymes forming phosphorylated products can also promote arsenate reduction by incorrectly using arsenate in place of phosphate to generate arsenylated products in which the arsenate is more easily reduced by thiols such as glutathione [67].\n",
            "Predicted Relation Type: CAUSES\n",
            "Pred Arg1 (Type/Text): Chemical Substance  //  glutathione\n",
            "Pred Arg2 (Type/Text): Chemical Substance  //  arsenylated products\n",
            "All Gold Relations: [('CONTRIBUTES_TO', ('Protein / Polypeptide', 'enzymes forming phosphorylated products'), ('Biological Process', 'arsenate reduction')), ('CONTRIBUTES_TO', ('Chemical Substance', 'thiols'), ('Biological Process', 'arsenate reduction')), ('IS_A', ('Chemical Substance', 'glutathione'), ('Chemical Substance', 'thiols'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 22\n",
            "Text: Finally, ACCELERATED CELL DEATH 6 (ACD6), which has been experimentally shown to be directly involved in lesioning [22], is detected here as associated with several lesioning and chlorosis phenotypes. \n",
            "Predicted Relation Type: CONTRIBUTES_TO\n",
            "Pred Arg1 (Type/Text): Gene  //  ACCELERATED CELL DEATH 6 (ACD6)\n",
            "Pred Arg2 (Type/Text): Trait / Phenotype  //  lesioning\n",
            "All Gold Relations: [('CAUSES', ('Gene', 'ACCELERATED CELL DEATH 6 (ACD6)'), ('Trait / Phenotype', 'lesioning')), ('ASSOCIATED_WITH', ('Gene', 'ACCELERATED CELL DEATH 6 (ACD6)'), ('Trait / Phenotype', 'lesioning and chlorosis phenotypes')), ('ASSOCIATED_WITH', ('Gene', 'ACCELERATED CELL DEATH 6 (ACD6)'), ('Trait / Phenotype', 'chlorosis phenotypes'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 23\n",
            "Text: Finally, ACCELERATED CELL DEATH 6 (ACD6), which has been experimentally shown to be directly involved in lesioning [22], is detected here as associated with several lesioning and chlorosis phenotypes. \n",
            "Predicted Relation Type: ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Gene  //  ACCELERATED CELL DEATH 6 (ACD6)\n",
            "Pred Arg2 (Type/Text): Trait / Phenotype  //  several lesioning and chlorosis phenotypes\n",
            "All Gold Relations: [('CAUSES', ('Gene', 'ACCELERATED CELL DEATH 6 (ACD6)'), ('Trait / Phenotype', 'lesioning')), ('ASSOCIATED_WITH', ('Gene', 'ACCELERATED CELL DEATH 6 (ACD6)'), ('Trait / Phenotype', 'lesioning and chlorosis phenotypes')), ('ASSOCIATED_WITH', ('Gene', 'ACCELERATED CELL DEATH 6 (ACD6)'), ('Trait / Phenotype', 'chlorosis phenotypes'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 24\n",
            "Text: Arsenic concentrations were measured in leaves from 315 F2 plants (Figure 1C) from the Kr06Col-0 cross and approximately one-quarter of these plants (75 of 315, X[2] = 0.24,X[2]0.05(1) [= 3.84) had high leaf arsenic, suggest-] ing that high leaf arsenic in Kr-0 is controlled by a single major locus.\n",
            "Predicted Relation Type: LOCATED_IN\n",
            "Pred Arg1 (Type/Text): Trait / Phenotype  //  Arsenic concentrations\n",
            "Pred Arg2 (Type/Text): Tissue / Plant Structure  //  leaves\n",
            "All Gold Relations: [('PART_OF', ('Population / Group', '315 F2 plants'), ('Population / Group', 'the Kr06Col-0 cross')), ('HAS_MEASUREMENT', ('Population / Group', '315 F2 plants'), ('Trait / Phenotype', 'Arsenic concentrations')), ('ASSOCIATED_WITH', ('Accession / Line', 'Kr-0'), ('Trait / Phenotype', 'high leaf arsenic')), ('CAUSES', ('Genomic Region', 'a single major locus'), ('Trait / Phenotype', 'high leaf arsenic'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 25\n",
            "Text: Arsenic concentrations were measured in leaves from 315 F2 plants (Figure 1C) from the Kr06Col-0 cross and approximately one-quarter of these plants (75 of 315, X[2] = 0.24,X[2]0.05(1) [= 3.84) had high leaf arsenic, suggest-] ing that high leaf arsenic in Kr-0 is controlled by a single major locus.\n",
            "Predicted Relation Type: PART_OF\n",
            "Pred Arg1 (Type/Text): Population / Group  //  315 F2 plants\n",
            "Pred Arg2 (Type/Text): Population / Group  //  Kr06Col-0 cross\n",
            "All Gold Relations: [('PART_OF', ('Population / Group', '315 F2 plants'), ('Population / Group', 'the Kr06Col-0 cross')), ('HAS_MEASUREMENT', ('Population / Group', '315 F2 plants'), ('Trait / Phenotype', 'Arsenic concentrations')), ('ASSOCIATED_WITH', ('Accession / Line', 'Kr-0'), ('Trait / Phenotype', 'high leaf arsenic')), ('CAUSES', ('Genomic Region', 'a single major locus'), ('Trait / Phenotype', 'high leaf arsenic'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 26\n",
            "Text: Arsenic concentrations were measured in leaves from 315 F2 plants (Figure 1C) from the Kr06Col-0 cross and approximately one-quarter of these plants (75 of 315, X[2] = 0.24,X[2]0.05(1) [= 3.84) had high leaf arsenic, suggest-] ing that high leaf arsenic in Kr-0 is controlled by a single major locus.\n",
            "Predicted Relation Type: ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Trait / Phenotype  //  high leaf arsenic\n",
            "Pred Arg2 (Type/Text): Trait / Phenotype  //  Kr-0\n",
            "All Gold Relations: [('PART_OF', ('Population / Group', '315 F2 plants'), ('Population / Group', 'the Kr06Col-0 cross')), ('HAS_MEASUREMENT', ('Population / Group', '315 F2 plants'), ('Trait / Phenotype', 'Arsenic concentrations')), ('ASSOCIATED_WITH', ('Accession / Line', 'Kr-0'), ('Trait / Phenotype', 'high leaf arsenic')), ('CAUSES', ('Genomic Region', 'a single major locus'), ('Trait / Phenotype', 'high leaf arsenic'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 27\n",
            "Text: Arsenic concentrations were measured in leaves from 315 F2 plants (Figure 1C) from the Kr06Col-0 cross and approximately one-quarter of these plants (75 of 315, X[2] = 0.24,X[2]0.05(1) [= 3.84) had high leaf arsenic, suggest-] ing that high leaf arsenic in Kr-0 is controlled by a single major locus.\n",
            "Predicted Relation Type: REGULATES\n",
            "Pred Arg1 (Type/Text): Genomic Region  //  a single major locus\n",
            "Pred Arg2 (Type/Text): Trait / Phenotype  //  high leaf arsenic\n",
            "All Gold Relations: [('PART_OF', ('Population / Group', '315 F2 plants'), ('Population / Group', 'the Kr06Col-0 cross')), ('HAS_MEASUREMENT', ('Population / Group', '315 F2 plants'), ('Trait / Phenotype', 'Arsenic concentrations')), ('ASSOCIATED_WITH', ('Accession / Line', 'Kr-0'), ('Trait / Phenotype', 'high leaf arsenic')), ('CAUSES', ('Genomic Region', 'a single major locus'), ('Trait / Phenotype', 'high leaf arsenic'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 28\n",
            "Text: Regardless of method used, six phenotypes yield single, strong peaks of association that can be seen by inspection.\n",
            "Predicted Relation Type: CAUSES\n",
            "Pred Arg1 (Type/Text): Trait / Phenotype  //  six phenotypes\n",
            "Pred Arg2 (Type/Text): Concept / Abstract Entity  //  single, strong peaks of association\n",
            "All Gold Relations: [('HAS_MEASUREMENT', ('Trait / Phenotype', 'six phenotypes'), ('Measurement / Statistical Result', 'single, strong peaks of association'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 29\n",
            "Text: Regardless of method used, six phenotypes yield single, strong peaks of association that can be seen by inspection.\n",
            "Predicted Relation Type: IDENTIFIED_IN\n",
            "Pred Arg1 (Type/Text): Concept / Abstract Entity  //  single, strong peaks of association\n",
            "Pred Arg2 (Type/Text): Concept / Abstract Entity  //  inspection\n",
            "All Gold Relations: [('HAS_MEASUREMENT', ('Trait / Phenotype', 'six phenotypes'), ('Measurement / Statistical Result', 'single, strong peaks of association'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 30\n",
            "Text: On the other hand, arsenate uptake, measured as depletion of arsenate from the nutrient solution, was no different in hac1-1, hac1-2, or Kr-0 compared to wild-type Col-0 at both time points tested (Figure 8B).\n",
            "Predicted Relation Type: HAS_MEASUREMENT\n",
            "Pred Arg1 (Type/Text): Trait / Phenotype  //  arsenate uptake\n",
            "Pred Arg2 (Type/Text): Trait / Phenotype  //  depletion of arsenate from the nutrient solution\n",
            "All Gold Relations: [('HAS_MEASUREMENT', ('Trait / Phenotype', 'arsenate uptake'), ('Measurement / Statistical Result', 'depletion of arsenate from the nutrient solution')), ('NOT_ASSOCIATED_WITH', ('Allele / Variant', 'hac1-1'), ('Trait / Phenotype', 'arsenate uptake')), ('NOT_ASSOCIATED_WITH', ('Allele / Variant', 'hac1-2'), ('Trait / Phenotype', 'arsenate uptake')), ('NOT_ASSOCIATED_WITH', ('Accession / Line', 'Kr-0'), ('Trait / Phenotype', 'arsenate uptake'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 31\n",
            "Text: On the other hand, arsenate uptake, measured as depletion of arsenate from the nutrient solution, was no different in hac1-1, hac1-2, or Kr-0 compared to wild-type Col-0 at both time points tested (Figure 8B).\n",
            "Predicted Relation Type: NOT_ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Accession / Line  //  hac1-1\n",
            "Pred Arg2 (Type/Text): Trait / Phenotype  //  arsenate uptake\n",
            "All Gold Relations: [('HAS_MEASUREMENT', ('Trait / Phenotype', 'arsenate uptake'), ('Measurement / Statistical Result', 'depletion of arsenate from the nutrient solution')), ('NOT_ASSOCIATED_WITH', ('Allele / Variant', 'hac1-1'), ('Trait / Phenotype', 'arsenate uptake')), ('NOT_ASSOCIATED_WITH', ('Allele / Variant', 'hac1-2'), ('Trait / Phenotype', 'arsenate uptake')), ('NOT_ASSOCIATED_WITH', ('Accession / Line', 'Kr-0'), ('Trait / Phenotype', 'arsenate uptake'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 32\n",
            "Text: On the other hand, arsenate uptake, measured as depletion of arsenate from the nutrient solution, was no different in hac1-1, hac1-2, or Kr-0 compared to wild-type Col-0 at both time points tested (Figure 8B).\n",
            "Predicted Relation Type: NOT_ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Accession / Line  //  hac1-2\n",
            "Pred Arg2 (Type/Text): Trait / Phenotype  //  arsenate uptake\n",
            "All Gold Relations: [('HAS_MEASUREMENT', ('Trait / Phenotype', 'arsenate uptake'), ('Measurement / Statistical Result', 'depletion of arsenate from the nutrient solution')), ('NOT_ASSOCIATED_WITH', ('Allele / Variant', 'hac1-1'), ('Trait / Phenotype', 'arsenate uptake')), ('NOT_ASSOCIATED_WITH', ('Allele / Variant', 'hac1-2'), ('Trait / Phenotype', 'arsenate uptake')), ('NOT_ASSOCIATED_WITH', ('Accession / Line', 'Kr-0'), ('Trait / Phenotype', 'arsenate uptake'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 33\n",
            "Text: On the other hand, arsenate uptake, measured as depletion of arsenate from the nutrient solution, was no different in hac1-1, hac1-2, or Kr-0 compared to wild-type Col-0 at both time points tested (Figure 8B).\n",
            "Predicted Relation Type: NOT_ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Accession / Line  //  wild-type Col-0\n",
            "Pred Arg2 (Type/Text): Trait / Phenotype  //  arsenate uptake\n",
            "All Gold Relations: [('HAS_MEASUREMENT', ('Trait / Phenotype', 'arsenate uptake'), ('Measurement / Statistical Result', 'depletion of arsenate from the nutrient solution')), ('NOT_ASSOCIATED_WITH', ('Allele / Variant', 'hac1-1'), ('Trait / Phenotype', 'arsenate uptake')), ('NOT_ASSOCIATED_WITH', ('Allele / Variant', 'hac1-2'), ('Trait / Phenotype', 'arsenate uptake')), ('NOT_ASSOCIATED_WITH', ('Accession / Line', 'Kr-0'), ('Trait / Phenotype', 'arsenate uptake'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 34\n",
            "Text: In summary, A. thaliana has evolved two main strategies to cope with high substrate Pb: (1) exclusion from the shoot by sequestration in the roots and (2) translocation to the shoots and storage in inactive organelles or intracellular spaces.\n",
            "Predicted Relation Type: ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Organism  //  A. thaliana\n",
            "Pred Arg2 (Type/Text): Chemical Substance  //  high substrate Pb\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Environmental Factor', 'high substrate Pb')), ('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Biological Process', 'exclusion from the shoot by sequestration in the roots')), ('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Biological Process', 'translocation to the shoots and storage in inactive organelles or intracellular spaces'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 35\n",
            "Text: In summary, A. thaliana has evolved two main strategies to cope with high substrate Pb: (1) exclusion from the shoot by sequestration in the roots and (2) translocation to the shoots and storage in inactive organelles or intracellular spaces.\n",
            "Predicted Relation Type: ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Biological Process  //  exclusion from the shoot\n",
            "Pred Arg2 (Type/Text): Chemical Substance  //  high substrate Pb\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Environmental Factor', 'high substrate Pb')), ('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Biological Process', 'exclusion from the shoot by sequestration in the roots')), ('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Biological Process', 'translocation to the shoots and storage in inactive organelles or intracellular spaces'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 36\n",
            "Text: In summary, A. thaliana has evolved two main strategies to cope with high substrate Pb: (1) exclusion from the shoot by sequestration in the roots and (2) translocation to the shoots and storage in inactive organelles or intracellular spaces.\n",
            "Predicted Relation Type: LOCATED_IN\n",
            "Pred Arg1 (Type/Text): Biological Process  //  exclusion from the shoot\n",
            "Pred Arg2 (Type/Text): Tissue / Plant Structure  //  roots\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Environmental Factor', 'high substrate Pb')), ('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Biological Process', 'exclusion from the shoot by sequestration in the roots')), ('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Biological Process', 'translocation to the shoots and storage in inactive organelles or intracellular spaces'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 37\n",
            "Text: In summary, A. thaliana has evolved two main strategies to cope with high substrate Pb: (1) exclusion from the shoot by sequestration in the roots and (2) translocation to the shoots and storage in inactive organelles or intracellular spaces.\n",
            "Predicted Relation Type: ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Biological Process  //  sequestration\n",
            "Pred Arg2 (Type/Text): Chemical Substance  //  high substrate Pb\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Environmental Factor', 'high substrate Pb')), ('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Biological Process', 'exclusion from the shoot by sequestration in the roots')), ('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Biological Process', 'translocation to the shoots and storage in inactive organelles or intracellular spaces'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 38\n",
            "Text: In summary, A. thaliana has evolved two main strategies to cope with high substrate Pb: (1) exclusion from the shoot by sequestration in the roots and (2) translocation to the shoots and storage in inactive organelles or intracellular spaces.\n",
            "Predicted Relation Type: LOCATED_IN\n",
            "Pred Arg1 (Type/Text): Biological Process  //  sequestration\n",
            "Pred Arg2 (Type/Text): Tissue / Plant Structure  //  roots\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Environmental Factor', 'high substrate Pb')), ('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Biological Process', 'exclusion from the shoot by sequestration in the roots')), ('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Biological Process', 'translocation to the shoots and storage in inactive organelles or intracellular spaces'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 39\n",
            "Text: In summary, A. thaliana has evolved two main strategies to cope with high substrate Pb: (1) exclusion from the shoot by sequestration in the roots and (2) translocation to the shoots and storage in inactive organelles or intracellular spaces.\n",
            "Predicted Relation Type: ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Biological Process  //  translocation to the shoots\n",
            "Pred Arg2 (Type/Text): Chemical Substance  //  high substrate Pb\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Environmental Factor', 'high substrate Pb')), ('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Biological Process', 'exclusion from the shoot by sequestration in the roots')), ('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Biological Process', 'translocation to the shoots and storage in inactive organelles or intracellular spaces'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 40\n",
            "Text: In summary, A. thaliana has evolved two main strategies to cope with high substrate Pb: (1) exclusion from the shoot by sequestration in the roots and (2) translocation to the shoots and storage in inactive organelles or intracellular spaces.\n",
            "Predicted Relation Type: LOCATED_IN\n",
            "Pred Arg1 (Type/Text): Biological Process  //  translocation to the shoots\n",
            "Pred Arg2 (Type/Text): Biological Process  //  inactive organelles\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Environmental Factor', 'high substrate Pb')), ('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Biological Process', 'exclusion from the shoot by sequestration in the roots')), ('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Biological Process', 'translocation to the shoots and storage in inactive organelles or intracellular spaces'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 41\n",
            "Text: In summary, A. thaliana has evolved two main strategies to cope with high substrate Pb: (1) exclusion from the shoot by sequestration in the roots and (2) translocation to the shoots and storage in inactive organelles or intracellular spaces.\n",
            "Predicted Relation Type: CONTRIBUTES_TO\n",
            "Pred Arg1 (Type/Text): Biological Process  //  translocation to the shoots\n",
            "Pred Arg2 (Type/Text): Biological Process  //  storage in inactive organelles or intracellular spaces\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Environmental Factor', 'high substrate Pb')), ('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Biological Process', 'exclusion from the shoot by sequestration in the roots')), ('ASSOCIATED_WITH', ('Organism', 'A. thaliana'), ('Biological Process', 'translocation to the shoots and storage in inactive organelles or intracellular spaces'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 42\n",
            "Text: This suggested the possibility that HAC1 functions as an arsenate reductase, reducing arsenate (As[V]) to arsenite (As[III]).\n",
            "Predicted Relation Type: CONTRIBUTES_TO\n",
            "Pred Arg1 (Type/Text): Protein / Polypeptide  //  arsenate reductase\n",
            "Pred Arg2 (Type/Text): Protein / Polypeptide  //  reducing arsenate (As[V]) to arsenite (As[III]\n",
            "All Gold Relations: [('ENCODES', ('Gene', 'HAC1'), ('Protein / Polypeptide', 'arsenate reductase')), ('CAUSES', ('Protein / Polypeptide', 'arsenate reductase'), ('Biological Process', 'reducing arsenate (As[V]) to arsenite (As[III])'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 43\n",
            "Text: However, foliar uptake is an important second pathway in areas with high fallout of airborne particles (Uzu et al., 2010; Luo et al., 2019).\n",
            "Predicted Relation Type: ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Biological Process  //  foliar uptake\n",
            "Pred Arg2 (Type/Text): Environmental Factor  //  high fallout of airborne particles\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Biological Process', 'foliar uptake'), ('Environmental Factor', 'areas with high fallout of airborne particles')), ('IDENTIFIED_IN', ('Biological Process', 'foliar uptake'), ('Study / Analysis', 'Uzu et al., 2010')), ('IDENTIFIED_IN', ('Biological Process', 'foliar uptake'), ('Study / Analysis', 'Luo et al., 2019'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 44\n",
            "Text: It is also interesting to note that since the majority of the A. thaliana accessions tested have the HAC1 allele associated with higher leaf arsenic, they are likely to have weak alleles of HAC1.\n",
            "Predicted Relation Type: ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Population / Group  //  the majority of the A. thaliana accessions tested\n",
            "Pred Arg2 (Type/Text): Allele / Variant  //  HAC1 allele\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Allele / Variant', 'the HAC1 allele'), ('Trait / Phenotype', 'higher leaf arsenic')), ('ASSOCIATED_WITH', ('Population / Group', 'the majority of the A. thaliana accessions tested'), ('Allele / Variant', 'the HAC1 allele')), ('ASSOCIATED_WITH', ('Population / Group', 'the majority of the A. thaliana accessions tested'), ('Allele / Variant', 'weak alleles of HAC1'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 45\n",
            "Text: It is also interesting to note that since the majority of the A. thaliana accessions tested have the HAC1 allele associated with higher leaf arsenic, they are likely to have weak alleles of HAC1.\n",
            "Predicted Relation Type: ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Allele / Variant  //  HAC1 allele\n",
            "Pred Arg2 (Type/Text): Trait / Phenotype  //  higher leaf arsenic\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Allele / Variant', 'the HAC1 allele'), ('Trait / Phenotype', 'higher leaf arsenic')), ('ASSOCIATED_WITH', ('Population / Group', 'the majority of the A. thaliana accessions tested'), ('Allele / Variant', 'the HAC1 allele')), ('ASSOCIATED_WITH', ('Population / Group', 'the majority of the A. thaliana accessions tested'), ('Allele / Variant', 'weak alleles of HAC1'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 46\n",
            "Text: It is also interesting to note that since the majority of the A. thaliana accessions tested have the HAC1 allele associated with higher leaf arsenic, they are likely to have weak alleles of HAC1.\n",
            "Predicted Relation Type: PART_OF\n",
            "Pred Arg1 (Type/Text): Allele / Variant  //  HAC1 allele\n",
            "Pred Arg2 (Type/Text): Gene  //  HAC1\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Allele / Variant', 'the HAC1 allele'), ('Trait / Phenotype', 'higher leaf arsenic')), ('ASSOCIATED_WITH', ('Population / Group', 'the majority of the A. thaliana accessions tested'), ('Allele / Variant', 'the HAC1 allele')), ('ASSOCIATED_WITH', ('Population / Group', 'the majority of the A. thaliana accessions tested'), ('Allele / Variant', 'weak alleles of HAC1'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 47\n",
            "Text: It is also interesting to note that since the majority of the A. thaliana accessions tested have the HAC1 allele associated with higher leaf arsenic, they are likely to have weak alleles of HAC1.\n",
            "Predicted Relation Type: PART_OF\n",
            "Pred Arg1 (Type/Text): Allele / Variant  //  weak alleles of HAC1\n",
            "Pred Arg2 (Type/Text): Gene  //  HAC1\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Allele / Variant', 'the HAC1 allele'), ('Trait / Phenotype', 'higher leaf arsenic')), ('ASSOCIATED_WITH', ('Population / Group', 'the majority of the A. thaliana accessions tested'), ('Allele / Variant', 'the HAC1 allele')), ('ASSOCIATED_WITH', ('Population / Group', 'the majority of the A. thaliana accessions tested'), ('Allele / Variant', 'weak alleles of HAC1'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 48\n",
            "Text: Here, we expand on this observation by establishing the strength of the AtHKT1;1 alleles in four further high Na[+] accumulating accessions (Bur-1, Duk, PHW-20 and UKNW06-386) that all contain a T at Chr4:6392276, along with a low leaf Na[+] accession (Nd-1) with a C at Chr4:6392276.\n",
            "Predicted Relation Type: ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Accession / Line  //  Bur-1\n",
            "Pred Arg2 (Type/Text): Allele / Variant  //  AtHKT1;1 alleles\n",
            "All Gold Relations: [('IS_A', ('Accession / Line', 'Bur-1'), ('Trait / Phenotype', 'high Na[+] accumulating')), ('IS_A', ('Accession / Line', 'Duk'), ('Trait / Phenotype', 'high Na[+] accumulating')), ('IS_A', ('Accession / Line', 'PHW-20'), ('Trait / Phenotype', 'high Na[+] accumulating')), ('IS_A', ('Accession / Line', 'UKNW06-386'), ('Trait / Phenotype', 'high Na[+] accumulating')), ('ASSOCIATED_WITH', ('Accession / Line', 'Bur-1'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Duk'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'PHW-20'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'UKNW06-386'), ('Allele / Variant', 'T at Chr4:6392276')), ('IS_A', ('Accession / Line', 'Nd-1'), ('Trait / Phenotype', 'low leaf Na[+]')), ('ASSOCIATED_WITH', ('Accession / Line', 'Nd-1'), ('Allele / Variant', 'C at Chr4:6392276'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 49\n",
            "Text: Here, we expand on this observation by establishing the strength of the AtHKT1;1 alleles in four further high Na[+] accumulating accessions (Bur-1, Duk, PHW-20 and UKNW06-386) that all contain a T at Chr4:6392276, along with a low leaf Na[+] accession (Nd-1) with a C at Chr4:6392276.\n",
            "Predicted Relation Type: ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Accession / Line  //  Nd-1)\n",
            "Pred Arg2 (Type/Text): Allele / Variant  //  C at Chr4:6392276\n",
            "All Gold Relations: [('IS_A', ('Accession / Line', 'Bur-1'), ('Trait / Phenotype', 'high Na[+] accumulating')), ('IS_A', ('Accession / Line', 'Duk'), ('Trait / Phenotype', 'high Na[+] accumulating')), ('IS_A', ('Accession / Line', 'PHW-20'), ('Trait / Phenotype', 'high Na[+] accumulating')), ('IS_A', ('Accession / Line', 'UKNW06-386'), ('Trait / Phenotype', 'high Na[+] accumulating')), ('ASSOCIATED_WITH', ('Accession / Line', 'Bur-1'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'Duk'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'PHW-20'), ('Allele / Variant', 'T at Chr4:6392276')), ('ASSOCIATED_WITH', ('Accession / Line', 'UKNW06-386'), ('Allele / Variant', 'T at Chr4:6392276')), ('IS_A', ('Accession / Line', 'Nd-1'), ('Trait / Phenotype', 'low leaf Na[+]')), ('ASSOCIATED_WITH', ('Accession / Line', 'Nd-1'), ('Allele / Variant', 'C at Chr4:6392276'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 50\n",
            "Text: In the hac1 A. thaliana mutant that lacks the arsenate reductase activity required for arsenite efflux, we observe an over-accumulation of arsenic within the central stele of the root. \n",
            "Predicted Relation Type: NOT_ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Accession / Line  //  hac1 A. thaliana mutant\n",
            "Pred Arg2 (Type/Text): Protein / Polypeptide  //  arsenate reductase activity\n",
            "All Gold Relations: [('NOT_ASSOCIATED_WITH', ('Accession / Line', 'hac1 A. thaliana mutant'), ('Biological Process', 'arsenate reductase activity')), ('CONTRIBUTES_TO', ('Biological Process', 'arsenate reductase activity'), ('Biological Process', 'arsenite efflux')), ('LOCATED_IN', ('Trait / Phenotype', 'over-accumulation of arsenic'), ('Tissue / Plant Structure', 'central stele')), ('PART_OF', ('Tissue / Plant Structure', 'central stele'), ('Tissue / Plant Structure', 'root'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 51\n",
            "Text: In the hac1 A. thaliana mutant that lacks the arsenate reductase activity required for arsenite efflux, we observe an over-accumulation of arsenic within the central stele of the root. \n",
            "Predicted Relation Type: CONTRIBUTES_TO\n",
            "Pred Arg1 (Type/Text): Protein / Polypeptide  //  arsenate reductase activity\n",
            "Pred Arg2 (Type/Text): Biological Process  //  arsenite efflux\n",
            "All Gold Relations: [('NOT_ASSOCIATED_WITH', ('Accession / Line', 'hac1 A. thaliana mutant'), ('Biological Process', 'arsenate reductase activity')), ('CONTRIBUTES_TO', ('Biological Process', 'arsenate reductase activity'), ('Biological Process', 'arsenite efflux')), ('LOCATED_IN', ('Trait / Phenotype', 'over-accumulation of arsenic'), ('Tissue / Plant Structure', 'central stele')), ('PART_OF', ('Tissue / Plant Structure', 'central stele'), ('Tissue / Plant Structure', 'root'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 52\n",
            "Text: In the hac1 A. thaliana mutant that lacks the arsenate reductase activity required for arsenite efflux, we observe an over-accumulation of arsenic within the central stele of the root. \n",
            "Predicted Relation Type: LOCATED_IN\n",
            "Pred Arg1 (Type/Text): Chemical Substance  //  arsenic\n",
            "Pred Arg2 (Type/Text): Tissue / Plant Structure  //  central stele of the root\n",
            "All Gold Relations: [('NOT_ASSOCIATED_WITH', ('Accession / Line', 'hac1 A. thaliana mutant'), ('Biological Process', 'arsenate reductase activity')), ('CONTRIBUTES_TO', ('Biological Process', 'arsenate reductase activity'), ('Biological Process', 'arsenite efflux')), ('LOCATED_IN', ('Trait / Phenotype', 'over-accumulation of arsenic'), ('Tissue / Plant Structure', 'central stele')), ('PART_OF', ('Tissue / Plant Structure', 'central stele'), ('Tissue / Plant Structure', 'root'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 53\n",
            "Text: Pb entering the symplast is removed by the activity of efflux pumps present in the plasma membrane.\n",
            "Predicted Relation Type: REGULATES\n",
            "Pred Arg1 (Type/Text): Biological Process  //  efflux pumps\n",
            "Pred Arg2 (Type/Text): Biological Process  //  efflux pumps\n",
            "All Gold Relations: [('LOCATED_IN', ('Protein / Polypeptide', 'efflux pumps'), ('Tissue / Plant Structure', 'plasma membrane')), ('LOCATED_IN', ('Chemical Substance', 'Pb'), ('Tissue / Plant Structure', 'symplast'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? n\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 54\n",
            "Text: In addition, Na[+]  compartmentalization facilitates vacuolar osmotic adjustment that is necessary to compensate for the osmotic effects of salinity by maintaining turgor pressure for cell expansion and growth.\n",
            "Predicted Relation Type: CONTRIBUTES_TO\n",
            "Pred Arg1 (Type/Text): Biological Process  //  vacuolar osmotic adjustment\n",
            "Pred Arg2 (Type/Text): Biological Process  //  growth\n",
            "All Gold Relations: [('CONTRIBUTES_TO', ('Biological Process', 'Na[+]  compartmentalization'), ('Biological Process', 'vacuolar osmotic adjustment')), ('CONTRIBUTES_TO', ('Biological Process', 'vacuolar osmotic adjustment'), ('Trait / Phenotype', 'turgor pressure')), ('CONTRIBUTES_TO', ('Trait / Phenotype', 'turgor pressure'), ('Biological Process', 'cell expansion and growth')), ('CONTRIBUTES_TO', ('Biological Process', 'vacuolar osmotic adjustment'), ('Environmental Factor', 'salinity'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 55\n",
            "Text: Whether the wide variety of natural phenotypic and genetic variation observed in A. thaliana [12] contributes to its local adaptation is an important unresolved question that is currently attracting a significant amount of attention [13].\n",
            "Predicted Relation Type: ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Concept / Abstract Entity  //  the wide variety of natural phenotypic and genetic variation\n",
            "Pred Arg2 (Type/Text): Organism  //  A. thaliana\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Trait / Phenotype', 'natural phenotypic and'), ('Organism', 'A. thaliana')), ('ASSOCIATED_WITH', ('Concept / Abstract Entity', 'genetic variation'), ('Organism', 'A. thaliana')), ('CONTRIBUTES_TO', ('Trait / Phenotype', 'natural phenotypic and'), ('Biological Process', 'local adaptation')), ('CONTRIBUTES_TO', ('Concept / Abstract Entity', 'genetic variation'), ('Biological Process', 'local adaptation'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 56\n",
            "Text: Whether the wide variety of natural phenotypic and genetic variation observed in A. thaliana [12] contributes to its local adaptation is an important unresolved question that is currently attracting a significant amount of attention [13].\n",
            "Predicted Relation Type: CONTRIBUTES_TO\n",
            "Pred Arg1 (Type/Text): Concept / Abstract Entity  //  the wide variety of natural phenotypic and genetic variation\n",
            "Pred Arg2 (Type/Text): Biological Process  //  local adaptation\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Trait / Phenotype', 'natural phenotypic and'), ('Organism', 'A. thaliana')), ('ASSOCIATED_WITH', ('Concept / Abstract Entity', 'genetic variation'), ('Organism', 'A. thaliana')), ('CONTRIBUTES_TO', ('Trait / Phenotype', 'natural phenotypic and'), ('Biological Process', 'local adaptation')), ('CONTRIBUTES_TO', ('Concept / Abstract Entity', 'genetic variation'), ('Biological Process', 'local adaptation'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 57\n",
            "Text: This increase in HAC1 mRNA levels reaches a maximum at 100 mM arsenate in the growth media (Figure 7A).\n",
            "Predicted Relation Type: ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Trait / Phenotype  //  increase in HAC1 mRNA levels\n",
            "Pred Arg2 (Type/Text): Trait / Phenotype  //  100 mM arsenate in the growth media\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Trait / Phenotype', 'HAC1 mRNA levels'), ('Chemical Substance', '100 mM arsenate')), ('LOCATED_IN', ('Chemical Substance', '100 mM arsenate'), ('Environmental Factor', 'growth media'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 58\n",
            "Text: To validate the existence of this leaf arsenic QTL identified by GWA analysis we created a synthetic F2 recombinant population in which alternative alleles of the dialleic SNP Chr2:9008060 were segregating.\n",
            "Predicted Relation Type: IDENTIFIED_IN\n",
            "Pred Arg1 (Type/Text): Trait / Phenotype  //  leaf arsenic QTL\n",
            "Pred Arg2 (Type/Text): Study / Analysis  //  GWA analysis\n",
            "All Gold Relations: [('IDENTIFIED_IN', ('Allele / Variant', 'leaf arsenic QTL'), ('Study / Analysis', 'GWA analysis')), ('ASSOCIATED_WITH', ('Population / Group', 'synthetic F2 recombinant population'), ('Allele / Variant', 'alternative alleles of the dialleic SNP Chr2:9008060'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 59\n",
            "Text: To validate the existence of this leaf arsenic QTL identified by GWA analysis we created a synthetic F2 recombinant population in which alternative alleles of the dialleic SNP Chr2:9008060 were segregating.\n",
            "Predicted Relation Type: PART_OF\n",
            "Pred Arg1 (Type/Text): Allele / Variant  //  alternative alleles of the dialleic SNP Chr2:9008060\n",
            "Pred Arg2 (Type/Text): Population / Group  //  a synthetic F2 recombinant population\n",
            "All Gold Relations: [('IDENTIFIED_IN', ('Allele / Variant', 'leaf arsenic QTL'), ('Study / Analysis', 'GWA analysis')), ('ASSOCIATED_WITH', ('Population / Group', 'synthetic F2 recombinant population'), ('Allele / Variant', 'alternative alleles of the dialleic SNP Chr2:9008060'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 60\n",
            "Text: To validate the existence of this leaf arsenic QTL identified by GWA analysis we created a synthetic F2 recombinant population in which alternative alleles of the dialleic SNP Chr2:9008060 were segregating.\n",
            "Predicted Relation Type: PART_OF\n",
            "Pred Arg1 (Type/Text): Allele / Variant  //  the dialleic SNP Chr2:9008060\n",
            "Pred Arg2 (Type/Text): Population / Group  //  a synthetic F2 recombinant population\n",
            "All Gold Relations: [('IDENTIFIED_IN', ('Allele / Variant', 'leaf arsenic QTL'), ('Study / Analysis', 'GWA analysis')), ('ASSOCIATED_WITH', ('Population / Group', 'synthetic F2 recombinant population'), ('Allele / Variant', 'alternative alleles of the dialleic SNP Chr2:9008060'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 61\n",
            "Text: This knowledge creates new opportunities to limit arsenic accumulation in food crops, thereby helping to reduce the cancer risk from this food-chain contaminant.\n",
            "Predicted Relation Type: CONTRIBUTES_TO\n",
            "Pred Arg1 (Type/Text): Concept / Abstract Entity  //  This knowledge\n",
            "Pred Arg2 (Type/Text): Trait / Phenotype  //  arsenic accumulation in food crops\n",
            "All Gold Relations: [('LOCATED_IN', ('Trait / Phenotype', 'arsenic accumulation'), ('Population / Group', 'food crops')), ('ASSOCIATED_WITH', ('Trait / Phenotype', 'arsenic accumulation'), ('Chemical Substance', 'food-chain contaminant')), ('CAUSES', ('Chemical Substance', 'food-chain contaminant'), ('Concept / Abstract Entity', 'cancer risk'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 62\n",
            "Text: This knowledge creates new opportunities to limit arsenic accumulation in food crops, thereby helping to reduce the cancer risk from this food-chain contaminant.\n",
            "Predicted Relation Type: CONTRIBUTES_TO\n",
            "Pred Arg1 (Type/Text): Trait / Phenotype  //  arsenic accumulation in food crops\n",
            "Pred Arg2 (Type/Text): Chemical Substance  //  this food-chain contaminant\n",
            "All Gold Relations: [('LOCATED_IN', ('Trait / Phenotype', 'arsenic accumulation'), ('Population / Group', 'food crops')), ('ASSOCIATED_WITH', ('Trait / Phenotype', 'arsenic accumulation'), ('Chemical Substance', 'food-chain contaminant')), ('CAUSES', ('Chemical Substance', 'food-chain contaminant'), ('Concept / Abstract Entity', 'cancer risk'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 63\n",
            "Text: In particular, our goals are (1) to elucidate the mechanisms developed by Pb-tolerant accessions that differ in their Pb sequestration strategies, and (2) to stablish the role of the novel candidate genes identified by GWAs in Pb tolerance.\n",
            "Predicted Relation Type: ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Accession / Line  //  Pb-tolerant accessions\n",
            "Pred Arg2 (Type/Text): Biological Process  //  Pb sequestration strategies\n",
            "All Gold Relations: [('IDENTIFIED_IN', ('Gene', 'novel candidate genes'), ('Study / Analysis', 'GWAs')), ('CONTRIBUTES_TO', ('Gene', 'novel candidate genes'), ('Trait / Phenotype', 'Pb tolerance')), ('CONTRIBUTES_TO', ('Biological Process', 'Pb sequestration strategies'), ('Trait / Phenotype', 'Pb tolerance'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 64\n",
            "Text: If polymorphism at FRI is taken into account, CRP is no longer significantly associated, but this does not necessarily mean that the association is spurious.\n",
            "Predicted Relation Type: NOT_ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Allele / Variant  //  polymorphism at FRI\n",
            "Pred Arg2 (Type/Text): Gene  //  CRP\n",
            "All Gold Relations: [('NOT_ASSOCIATED_WITH', ('Allele / Variant', 'polymorphism at FRI'), ('Protein / Polypeptide', 'CRP'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 65\n",
            "Text: However, hac1-1 and hac1-2 were both significantly more sensitive to arsenate compared to the Col-0 wild type at relatively high concentrations of arsenate when growth was measured as root length or overall plant fresh weight (Figure 10).\n",
            "Predicted Relation Type: CAUSES\n",
            "Pred Arg1 (Type/Text): Allele / Variant  //  hac1-1\n",
            "Pred Arg2 (Type/Text): Trait / Phenotype  //  significantly more sensitive\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Allele / Variant', 'hac1-1'), ('Trait / Phenotype', 'more sensitive to arsenate')), ('ASSOCIATED_WITH', ('Allele / Variant', 'hac1-2'), ('Trait / Phenotype', 'more sensitive to arsenate')), ('ASSOCIATED_WITH', ('Trait / Phenotype', 'more sensitive to arsenate'), ('Environmental Factor', 'relatively high concentrations of arsenate'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 66\n",
            "Text: However, hac1-1 and hac1-2 were both significantly more sensitive to arsenate compared to the Col-0 wild type at relatively high concentrations of arsenate when growth was measured as root length or overall plant fresh weight (Figure 10).\n",
            "Predicted Relation Type: CAUSES\n",
            "Pred Arg1 (Type/Text): Allele / Variant  //  hac1-2\n",
            "Pred Arg2 (Type/Text): Trait / Phenotype  //  significantly more sensitive\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Allele / Variant', 'hac1-1'), ('Trait / Phenotype', 'more sensitive to arsenate')), ('ASSOCIATED_WITH', ('Allele / Variant', 'hac1-2'), ('Trait / Phenotype', 'more sensitive to arsenate')), ('ASSOCIATED_WITH', ('Trait / Phenotype', 'more sensitive to arsenate'), ('Environmental Factor', 'relatively high concentrations of arsenate'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 67\n",
            "Text: However, hac1-1 and hac1-2 were both significantly more sensitive to arsenate compared to the Col-0 wild type at relatively high concentrations of arsenate when growth was measured as root length or overall plant fresh weight (Figure 10).\n",
            "Predicted Relation Type: ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Accession / Line  //  Col-0 wild type\n",
            "Pred Arg2 (Type/Text): Chemical Substance  //  arsenate\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Allele / Variant', 'hac1-1'), ('Trait / Phenotype', 'more sensitive to arsenate')), ('ASSOCIATED_WITH', ('Allele / Variant', 'hac1-2'), ('Trait / Phenotype', 'more sensitive to arsenate')), ('ASSOCIATED_WITH', ('Trait / Phenotype', 'more sensitive to arsenate'), ('Environmental Factor', 'relatively high concentrations of arsenate'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 68\n",
            "Text: On the other hand, both T-HLL and Pbsensitive (S) accessions translocated similar Pb concentrations (Figure 3F), suggesting that (S) plants are not able to maintain the Pb out of the cytosol and it becomes toxic for them.\n",
            "Predicted Relation Type: ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Accession / Line  //  T-HLL\n",
            "Pred Arg2 (Type/Text): Accession / Line  //  Pbsensitive (S) accessions\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Accession / Line', 'T-HLL'), ('Trait / Phenotype', 'Pb concentrations')), ('ASSOCIATED_WITH', ('Accession / Line', 'Pbsensitive (S) accessions'), ('Trait / Phenotype', 'Pb concentrations')), ('LOCATED_IN', ('Chemical Substance', 'Pb'), ('Tissue / Plant Structure', 'cytosol')), ('CAUSES', ('Chemical Substance', 'Pb'), ('Trait / Phenotype', 'toxic'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? c\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 69\n",
            "Text: On the other hand, both T-HLL and Pbsensitive (S) accessions translocated similar Pb concentrations (Figure 3F), suggesting that (S) plants are not able to maintain the Pb out of the cytosol and it becomes toxic for them.\n",
            "Predicted Relation Type: NOT_ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Chemical Substance  //  Pb\n",
            "Pred Arg2 (Type/Text): Tissue / Plant Structure  //  cytosol\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Accession / Line', 'T-HLL'), ('Trait / Phenotype', 'Pb concentrations')), ('ASSOCIATED_WITH', ('Accession / Line', 'Pbsensitive (S) accessions'), ('Trait / Phenotype', 'Pb concentrations')), ('LOCATED_IN', ('Chemical Substance', 'Pb'), ('Tissue / Plant Structure', 'cytosol')), ('CAUSES', ('Chemical Substance', 'Pb'), ('Trait / Phenotype', 'toxic'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? y\n",
            "\n",
            "---------------------------------------------------------\n",
            "Row index: 70\n",
            "Text: To test this hypothesis, we evaluated the oxidation state of arsenic in plants lacking a functional HAC1 gene.\n",
            "Predicted Relation Type: ASSOCIATED_WITH\n",
            "Pred Arg1 (Type/Text): Trait / Phenotype  //  oxidation state of arsenic\n",
            "Pred Arg2 (Type/Text): Chemical Substance  //  arsenic\n",
            "All Gold Relations: [('ASSOCIATED_WITH', ('Trait / Phenotype', 'the oxidation state of arsenic'), ('Population / Group', 'plants lacking a functional HAC1 gene'))]\n",
            "[y] if the prediction is valid and should have been in gold\n",
            "[n] if the prediction is truly wrong\n",
            "[c] if it is basically covered by an existing gold relation\n",
            "=> Your choice (y/n/c)? n\n",
            "\n",
            "=== Manual Review Complete ===\n",
            "TP-extra Count (valid but missing in gold) = 38\n",
            "Covered Count                              = 27\n",
            "FP Count (truly invalid)                   = 6\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 12,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "qYDCStJ-tfsU",
        "outputId": "8d366d6b-69b9-45da-af08-a54e653c4c32"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "100% 3/3 [00:30<00:00, 10.12s/it]\n",
            "[Cannot reconstruct]: linked DNA polymorphisms genome-wide ['By', 'coupling', 'these', 'events', 'with', 'linked', 'DNA', 'polymorphisms', 'genome', '-', 'wide', ',', 'the', 'phenotypic', 'effect', 'of', 'multiple', 'alleles', 'across', 'the', 'genome', 'can', 'be', 'tested', '.']\n",
            "\n",
            "[Cannot reconstruct]: [9] ['Because', 'of', 'this', 'serious', 'and', 'widespread', 'food', 'safety', 'concern', 'research', 'into', 'understanding', 'the', 'mechanisms', 'driving', 'arsenic', 'accumulation', 'in', 'plants', 'has', 'become', 'a', 'priority', '[', '9', ']', '.']\n",
            "\n",
            "[Cannot reconstruct]: M.T. et al., unpublished observations ['This', 'association', 'has', 'also', 'recently', 'been', 'experimentally', 'confirmed', '(', 'M.T.', 'et', 'al', '.', ',', 'unpublished', 'observations', ')', '.']\n",
            "\n",
            "[Cannot reconstruct]: sites with elevated soil Pb content (and acidic pH) ['We', 'observed', 'that', 'plants', 'coming', 'from', 'sites', 'with', 'elevated', 'soil', 'Pb', 'content', '(', 'and', 'acidic', 'pH', ')', 'had', 'better', 'capacity', 'to', 'tolerate', 'this', 'stress', '(', 'Figure', '2B', ')', '.']\n",
            "\n",
            "[Cannot reconstruct]: the (often much larger set of) strong associations ['This', 'is', 'an', 'upper', 'bound', 'because', 'it', 'seems', 'almost', 'certain', 'that', 'many', 'of', 'the', '(', 'often', 'much', 'larger', 'set', 'of', ')', 'strong', 'associations', 'that', 'are', 'not', 'close', 'to', 'a', '-', 'priori', 'candidates', 'will', 'also', 'turn', 'out', 'to', 'be', 'real', '.']\n",
            "\n",
            "[Cannot reconstruct]: a bi-parental cross ['However', ',', 'unlike', 'in', 'synthetic', 'RILs', 'made', 'from', 'a', 'bi', '-', 'parental', 'cross', ',', 'the', 'low', 'frequency', 'of', 'rare', 'alleles', 'in', 'a', 'natural', 'population', 'makes', 'it', 'difficult', 'to', 'detect', 'their', 'phenotypic', 'effect', 'using', 'GWA', 'mapping', '.']\n",
            "\n",
            "[Cannot reconstruct]: [28] ['This', 'natural', 'variation', 'has', 'been', 'used', 'to', 'identify', 'specific', 'genes', 'involved', 'in', 'controlling', 'variation', 'in', 'many', 'traits', '[', '28', ']', '.']\n",
            "\n",
            "[Cannot reconstruct]: ACCELERATED CELL DEATH 6 (ACD6) ['Finally', ',', 'ACCELERATED', 'CELL', 'DEATH', '6', '(', 'ACD6', ')', ',', 'which', 'has', 'been', 'experimentally', 'shown', 'to', 'be', 'directly', 'involved', 'in', 'lesioning', '[', '22', ']', ',', 'is', 'detected', 'here', 'as', 'associated', 'with', 'several', 'lesioning', 'and', 'chlorosis', 'phenotypes', '.']\n",
            "\n",
            "[Cannot reconstruct]: single, strong peaks of association ['Regardless', 'of', 'method', 'used', ',', 'six', 'phenotypes', 'yield', 'single', ',', 'strong', 'peaks', 'of', 'association', 'that', 'can', 'be', 'seen', 'by', 'inspection', '.']\n",
            "\n",
            "[Cannot reconstruct]: hac1-1 ['On', 'the', 'other', 'hand', ',', 'arsenate', 'uptake', ',', 'measured', 'as', 'depletion', 'of', 'arsenate', 'from', 'the', 'nutrient', 'solution', ',', 'was', 'no', 'different', 'in', 'hac1', '-', '1', ',', 'hac1', '-', '2', ',', 'or', 'Kr-0', 'compared', 'to', 'wild', '-', 'type', 'Col-0', 'at', 'both', 'time', 'points', 'tested', '(', 'Figure', '8B', ')', '.']\n",
            "\n",
            "[Cannot reconstruct]: reducing arsenate (As[V]) to arsenite (As[III] ['This', 'suggested', 'the', 'possibility', 'that', 'HAC1', 'functions', 'as', 'an', 'arsenate', 'reductase', ',', 'reducing', 'arsenate', '(', 'As[V', ']', ')', 'to', 'arsenite', '(', 'As[III', ']', ')', '.']\n",
            "\n",
            "[Cannot reconstruct]: Nd-1) ['Here', ',', 'we', 'expand', 'on', 'this', 'observation', 'by', 'establishing', 'the', 'strength', 'of', 'the', 'AtHKT1;1', 'alleles', 'in', 'four', 'further', 'high', 'Na[+', ']', 'accumulating', 'accessions', '(', 'Bur-1', ',', 'Duk', ',', 'PHW-20', 'and', 'UKNW06', '-', '386', ')', 'that', 'all', 'contain', 'a', 'T', 'at', 'Chr4:6392276', ',', 'along', 'with', 'a', 'low', 'leaf', 'Na[+', ']', 'accession', '(', 'Nd-1', ')', 'with', 'a', 'C', 'at', 'Chr4:6392276', '.']\n",
            "\n",
            "[Cannot reconstruct]: TRAM/LAG1/CLN8 lipid-sensing domain ['TRAM', '/', 'LAG1', '/', 'CLN8', 'lipid', '-', 'sensing', 'domain', 'has', 'been', 'well', 'described', 'in', 'humans', 'and', 'yeast', ',', 'where', 'it', 'is', 'associated', 'with', 'ER', '-', 'Golgi', 'trafficking', '(', 'Winter', 'and', 'Ponting', ',', '2002', ')', '.']\n",
            "\n",
            "[Cannot reconstruct]: this food-chain contaminant ['This', 'knowledge', 'creates', 'new', 'opportunities', 'to', 'limit', 'arsenic', 'accumulation', 'in', 'food', 'crops', ',', 'thereby', 'helping', 'to', 'reduce', 'the', 'cancer', 'risk', 'from', 'this', 'food', '-', 'chain', 'contaminant', '.']\n",
            "\n",
            "[Cannot reconstruct]: Pb-tolerant accessions ['In', 'particular', ',', 'our', 'goals', 'are', '(', '1', ')', 'to', 'elucidate', 'the', 'mechanisms', 'developed', 'by', 'Pb', '-', 'tolerant', 'accessions', 'that', 'differ', 'in', 'their', 'Pb', 'sequestration', 'strategies', ',', 'and', '(', '2', ')', 'to', 'stablish', 'the', 'role', 'of', 'the', 'novel', 'candidate', 'genes', 'identified', 'by', 'GWAs', 'in', 'Pb', 'tolerance', '.']\n",
            "\n",
            "[Cannot reconstruct]: hac1-1 ['However', ',', 'hac1', '-', '1', 'and', 'hac1', '-', '2', 'were', 'both', 'significantly', 'more', 'sensitive', 'to', 'arsenate', 'compared', 'to', 'the', 'Col-0', 'wild', 'type', 'at', 'relatively', 'high', 'concentrations', 'of', 'arsenate', 'when', 'growth', 'was', 'measured', 'as', 'root', 'length', 'or', 'overall', 'plant', 'fresh', 'weight', '(', 'Figure', '10', ')', '.']\n",
            "\n",
            "[Cannot reconstruct]: T-HLL ['On', 'the', 'other', 'hand', ',', 'both', 'T', '-', 'HLL', 'and', 'Pbsensitive', '(', 'S', ')', 'accessions', 'translocated', 'similar', 'Pb', 'concentrations', '(', 'Figure', '3F', ')', ',', 'suggesting', 'that', '(', 'S', ')', 'plants', 'are', 'not', 'able', 'to', 'maintain', 'the', 'Pb', 'out', 'of', 'the', 'cytosol', 'and', 'it', 'becomes', 'toxic', 'for', 'them', '.']\n",
            "\n",
            "100% 3/3 [00:22<00:00,  7.52s/it]\n",
            "[Cannot reconstruct]: T-DNA insertions ['Both', 'mutants', 'contained', 'T', '-', 'DNA', 'insertions', 'in', 'gene', 'At2g21045', ',', 'suggesting', 'this', 'gene', 'is', 'causal', 'for', 'HAC1', '.']\n",
            "\n",
            "[Cannot reconstruct]: Loss-of-function alleles of HAC1 ['Loss', '-', 'of', '-', 'function', 'alleles', 'of', 'HAC1', 'were', 'unable', 'to', 'complement', 'the', 'high', 'leaf', 'arsenic', 'of', 'Kr-0', 'in', 'F1', 'hybrids', '(', 'Figure', '2', 'G', ')', ',', 'whereas', 'the', 'Col-0', 'At2g21045', 'genomic', 'fragment', 'introduced', 'transgenically', 'into', 'Kr-0', 'was', 'able', 'to', 'complement', '(', 'Figure', '2H', ')', ',', 'confirming', 'that', 'At2g21045', 'is', 'the', 'causal', 'gene', 'for', 'HAC1', '.']\n",
            "\n",
            "[Cannot reconstruct]: P-value distribution ['First', ',', 'although', 'EMMA', 'appears', 'to', 'produce', 'a', 'P', '-', 'value', 'distribution', 'that', 'conforms', 'to', 'the', 'null', 'expectation', '(', 'except', 'for', 'extreme', 'values', ';', 'see', 'Supplementary', 'Figs12–118', ')', ',', 'it', 'seems', 'almost', 'certain', 'that', 'some', 'confounding', 'remains', '(', 'see', 'below', ')', '.']\n",
            "\n",
            "[Cannot reconstruct]: loss-of-function allele of HAC1 ['The', 'Kr-0', 'accession', 'contains', 'the', 'major', 'allele', 'for', 'this', 'locus', ',', 'and', 'the', 'elevated', 'leaf', 'arsenic', 'in', 'this', 'accession', 'is', 'explained', 'by', 'the', 'presence', 'of', 'a', 'loss', '-', 'of', '-', 'function', 'allele', 'of', 'HAC1', '.']\n",
            "\n",
            "[Cannot reconstruct]: Na[+] accumulation in the shoot ['Therefore', ',', 'modulation', 'of', 'its', 'function', 'would', 'allow', 'the', 'balancing', 'of', 'Na[+', ']', 'accumulation', 'in', 'the', 'shoot', 'with', 'soil', 'salinity', '.']\n",
            "\n",
            "[Cannot reconstruct]: EXTENSIN18 (EXT18) ['Genes', 'associated', 'with', 'significant', 'SNP', 'markers', 'were', 'evaluated', 'and', 'we', 'selected', 'EXTENSIN18', '(', 'EXT18', ')', 'and', 'TLC', '(', 'TRAM', '-', 'LAG1', '-', 'CLN8', ')', 'as', 'candidates', 'for', 'having', 'a', 'role', 'in', 'Pb', 'homeostasis', '.']\n",
            "\n",
            "[Cannot reconstruct]: genome wide association studies (GWAs) ['These', 'phenotypic', 'data', 'were', 'subjected', 'to', 'genome', 'wide', 'association', 'studies', '(', 'GWAs', ')', 'and', 'we', 'found', 'a', 'significant', 'association', 'on', 'chromosome', '1', 'for', 'low', 'leaf', 'Pb', 'accumulation', '.']\n",
            "\n",
            "[Cannot reconstruct]: an estimated error rate of 1.6% ['We', 'called', '216,130', 'SNPs', ',', 'at', 'an', 'estimated', 'error', 'rate', 'of', '1.6', '%', '.']\n",
            "\n",
            "[Cannot reconstruct]: leaf Pb** accumulation ['The', 'most', 'relevant', 'differences', 'were', 'found', 'in', 'leaf', 'Pb', '*', '*', 'accumulation', 'from', 'plants', 'grown', 'in', 'the', 'mine', 'soil', '(', 'RSD', '=', '1.2', ')', '.']\n",
            "\n",
            "[Cannot reconstruct]: short-range linkage disequilibrium ['Any', 'cross', 'will', 'eliminate', 'long', '-', 'range', 'linkage', 'disequilibrium', ',', 'and', 'short', '-', 'range', 'linkage', 'disequilibrium', 'can', 'be', 'overcome', 'by', 'choosing', 'appropriate', 'parental', 'strains', '.']\n",
            "\n",
            "[Cannot reconstruct]: position ['The', 'scores', ',', 'positions', ',', 'genes', 'associated', 'with', 'these', 'SNPs', ',', 'and', 'the', 'LD', 'of', 'the', 'regions', 'are', 'detailed', 'in', '*', '*', 'Supplementary', 'Table', '1', '.']\n",
            "\n",
            "[Cannot reconstruct]: genome-wide association (GWA) mapping ['Using', 'genome', '-', 'wide', 'association', '(', 'GWA', ')', 'mapping', 'of', 'loci', 'controlling', 'natural', 'variation', 'in', 'arsenic', 'accumulation', 'in', 'Arabidopsis', 'thaliana', 'allowed', 'us', 'to', 'identify', 'the', 'arsenate', 'reductase', 'required', 'for', 'this', 'reduction', ',', 'which', 'we', 'named', 'High', 'Arsenic', 'Content', '1', '(', 'HAC1', ')', '.']\n",
            "\n",
            "[Cannot reconstruct]: To´th et al., [23] ['Known', 'areas', 'of', 'saline', 'soils', 'were', 'obtained', 'from', 'To´th', 'et', 'al', '.', ',', '[', '23', ']', 'and', 'are', 'represented', 'here', 'in', 'green', '.']\n",
            "\n",
            "[Cannot reconstruct]: a single, major locus ['First', ',', 'it', 'becomes', 'clear', 'why', 'GWA', 'works', 'so', 'well', 'for', 'traits', 'that', 'are', 'monogenic', ',', 'or', 'at', 'least', 'mostly', 'due', 'to', 'a', 'single', ',', 'major', 'locus', '.']\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!python inference.py \\\n",
        "    --data data/text2spotasoc/relation/GWAS \\\n",
        "    --model output/Model_V4 \\\n",
        "    --max_source_length 512 \\\n",
        "    --max_target_length 512"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "lfxpu-7lc58B",
        "outputId": "1cae3c09-b718-462c-9e2e-e3d6c1d95907"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Inference Batches: 100% 2/2 [00:17<00:00,  8.91s/it]\n",
            "Converting seq2seq to records:   0% 0/50 [00:00<?, ?it/s][Cannot reconstruct]: Similar results ['▁Similar', '▁results', '▁were', '▁obtained', '▁for', '▁other', '▁disease', '-', 're', 's', 'i', 'stance', '▁responses', ',', '▁sodium', '▁concentration', ',', '▁le', 'sion', 'ing', '▁and', '▁', 'FR', 'IG', 'IDA', '▁(', 'FR', 'I', ')', '▁expression', '▁(', 'see', '▁Supplement', 'ary', '▁', 'Fig', 's', '▁35', 'â', '€', '“', '38', ',', '▁60', ',', '▁', '76', '▁and', '▁21,', '▁respectively', ').']\n",
            "\n",
            "[Cannot reconstruct]: arsenate ['▁Since', '▁ar', 's', 'en', 'ate', '▁is', '▁known', '▁to', '▁act', '▁as', '▁', 'a', '▁', 'phosphat', 'e', '▁an', 'a', 'logue', ',', '▁and', '▁can', '▁therefore', '▁be', '▁potentially', '▁loaded', '▁into', '▁the', '▁', 'x', 'y', 'le', 'm', '▁via', '▁the', '▁', 'phosphat', 'e', '▁transport', '▁system', ',', '▁', 'limiting', '▁ar', 's', 'en', 'ate', '▁concentration', 's', '▁in', '▁the', '▁', 's', 'tele', '▁would', '▁be', '▁an', '▁effective', '▁mechanism', '▁to', '▁minimise', '▁ar', 's', 'en', 'ic', '▁trans', 'location', '▁to', '▁the', '▁shoot', '.']\n",
            "\n",
            "[Cannot reconstruct]: the genome of A.thaliana ['▁Because', '▁the', '▁genome', '▁of', '▁A', '.', 'thal', 'iana', '▁is', '▁around', '▁120', '▁mega', 'base', 's', '▁long', '▁and', '▁the', '▁extent', '▁of', '▁link', 'age', '▁dis', 'e', 'qui', 'lib', 'rium', '▁there', 'in', '▁is', '▁comparable', '▁to', '▁that', '▁in', '▁humans', '[', '7', ',', '8', '],', '▁the', '▁', 'resulting', '▁S', 'NP', '▁density', '▁of', '▁one', '▁S', 'NP', '▁per', '▁500', '▁base', '▁pairs', '▁is', '▁considerably', '▁higher', '▁than', '▁is', '▁commonly', '▁used', '▁in', '▁human', '▁studies', '[', '6', '].']\n",
            "\n",
            "[Cannot reconstruct]: HAC1 protein ['▁The', '▁H', 'AC', '1', '▁protein', '▁accumulate', 's', '▁in', '▁the', '▁epi', 'der', 'm', 'is', ',', '▁the', '▁outer', '▁cell', '▁layer', '▁of', '▁the', '▁root', ',', '▁and', '▁also', '▁in', '▁the', '▁', 'peri', 'cycle', '▁cells', '▁surrounding', '▁the', '▁central', '▁', 'vascular', '▁tissue', '.']\n",
            "\n",
            "[Cannot reconstruct]: the same cohort ['▁In', '▁parallel', ',', '▁the', '▁same', '▁cohort', '▁was', '▁', 'cultivated', '▁in', '▁', 'a', '▁control', '▁non', '-', 'metall', 'if', 'er', 'ous', '▁soil', '▁with', '▁similar', '▁characteristics', '▁(', 'S', 'up', 'p', 'lement', 'ary', '▁Figure', '▁', '1).']\n",
            "\n",
            "[Cannot reconstruct]: HAC1 ['▁This', '▁suggests', '▁that', '▁H', 'AC', '1', '▁is', '▁involved', '▁in', '▁both', '▁constitu', 'tive', '▁and', '▁plastic', '▁responses', '▁of', '▁A', '.', '▁', 'thal', 'iana', '▁to', '▁ar', 's', 'en', 'ate', '▁in', '▁the', '▁environment', ',', '▁', 'allowing', '▁plants', '▁to', '▁maintain', '▁effective', '▁ar', 's', 'e', 'nite', '▁eff', 'lux', '▁from', '▁roots', '▁across', '▁', 'a', '▁range', '▁of', '▁ar', 's', 'en', 'ic', '▁concentration', 's', '▁in', '▁the', '▁soil', '.']\n",
            "\n",
            "[Cannot reconstruct]: Exposure to arsenate in agar solidified growth medium for 3 days ['▁Expo', 'sure', '▁to', '▁ar', 's', 'en', 'ate', '▁in', '▁', 'a', 'gar', '▁solid', 'ified', '▁growth', '▁medium', '▁for', '▁3', '▁days', '▁causes', '▁', 'a', '▁significant', ',', '▁dose', '▁dependent', ',', '▁increase', '▁in', '▁the', '▁steady', '▁state', '▁levels', '▁of', '▁H', 'AC', '1', '▁', 'm', 'RNA', '▁in', '▁roots', '▁of', '▁Col', '-', '0', '▁wild', '▁type', '▁(', 'Figure', '▁7', 'A', ').']\n",
            "\n",
            "[Cannot reconstruct]: leaf Pb accumulation ['▁Although', '▁non', '-', 'hyp', 'er', 'accumul', 'ator', '▁species', '▁usually', '▁restrict', '▁P', 'b', '▁transport', '▁to', '▁the', '▁shoot', 's', ',', '▁leaf', '▁P', 'b', '▁accumulation', '▁may', '▁considerably', '▁vary', '▁among', '▁species', '▁(', 'Be', 'ch', '▁', 'e', 't', '▁al', '.', ',', '▁2012', ';', '▁N', 'g', '▁', 'e', 't', '▁al', '.', ',', '▁2016', ';', '▁Z', 'hou', '▁', 'e', 't', '▁al', '.', ',', '▁2016', ')', '▁and', '▁even', '▁varieties', '▁(', 'L', 'i', 'u', '▁', 'e', 't', '▁al', '.', ',', '▁2013', ').']\n",
            "\n",
            "[Cannot reconstruct]: HAC1 ['▁We', '▁demonstrate', '▁that', '▁H', 'AC', '1', '▁plays', '▁', 'a', '▁critical', '▁role', '▁in', '▁controlling', '▁the', '▁amount', '▁of', '▁ar', 's', 'en', 'ic', '▁', 'accumulated', '▁in', '▁roots', '▁and', '▁shoot', 's', '▁in', '▁A', '.', '▁', 'thal', 'iana', '.']\n",
            "\n",
            "[Cannot reconstruct]: the double mutant acr2 hac1 ['▁To', '▁test', '▁this', ',', '▁we', '▁generated', '▁the', '▁double', '▁mutant', '▁', 'a', 'c', 'r', '2', '▁ha', 'c', '1', '▁homo', 'zy', 'g', 'ous', '▁for', '▁loss', '-', 'of', '-', 'function', '▁all', 'ele', 's', '▁of', '▁both', '▁A', 'CR', '2', '▁and', '▁H', 'AC', '1.']\n",
            "\n",
            "[Cannot reconstruct]: PHO1 ['▁P', 'HO', '1', '▁is', '▁the', '▁main', '▁', 'phosphat', 'e', '▁eff', 'lux', 'er', '▁involved', '▁in', '▁', 'x', 'y', 'le', 'm', '▁loading', '▁of', '▁', 'phosphat', 'e', '▁[', '71', ',', '72', ']', '▁in', '▁A', '.', '▁', 'thal', 'iana', '.']\n",
            "\n",
            "[Cannot reconstruct]: metal tolerant plants ['▁Except', '▁for', '▁hyper', 'accumul', 'ator', '▁species', ',', '▁metal', '▁', 'tolerant', '▁plants', '▁tend', '▁to', '▁restrict', '▁soil', 'â', '€', '“', 'root', '▁and', '▁root', 'â', '€', '“', 'sho', 'o', 't', '▁transfers', '.']\n",
            "\n",
            "[Cannot reconstruct]: ATQ1 ['▁We', '▁go', '▁beyond', '▁the', '▁findings', '▁of', '▁these', '▁authors', '▁by', '▁', 'revealing', '▁the', '▁functional', '▁role', '▁of', '▁AT', 'Q', '1', '▁in', '▁ar', 's', 'en', 'ate', '▁resistance', '▁and', '▁investigating', '▁the', '▁role', '▁of', '▁H', 'AC', '1/', 'AT', 'Q', '1', '▁in', '▁ar', 's', 'en', 'ic', '▁accumulation', '.']\n",
            "\n",
            "[Cannot reconstruct]: strongest associations ['▁The', '▁strongest', '▁associations', '▁do', '▁not', '▁always', '▁correspond', '▁to', '▁obvious', '▁candidates', '▁and', '▁are', '▁perhaps', '▁more', '▁interesting', '▁than', '▁associations', '▁in', '▁genes', '▁with', '▁known', '▁functions', '.']\n",
            "\n",
            "[Cannot reconstruct]: HMA3 protein ['▁The', '▁H', 'MA', '3', '▁protein', '▁is', '▁involved', '▁in', '▁P', 'b', '▁detox', 'ification', '▁by', '▁participating', '▁in', '▁its', '▁va', 'cu', 'olar', '▁se', 'quest', 'ration', '▁(', 'M', 'or', 'e', 'l', '▁', 'e', 't', '▁al', '.', ',', '▁2009', ').']\n",
            "\n",
            "[Cannot reconstruct]: Thicker cell walls ['▁Thi', 'cker', '▁cell', '▁walls', '▁are', '▁essential', '▁to', '▁accommodate', '▁P', 'b', '▁deposits', '.']\n",
            "\n",
            "[Cannot reconstruct]: excess Na[+] ['▁The', '▁primary', '▁effects', '▁of', '▁excess', '▁Na', '[', '+', ']', '▁on', '▁plants', '▁are', '▁water', '▁deficit', '▁', 'resulting', '▁from', '▁', 'a', '▁water', '▁potential', '▁gradient', '▁between', '▁the', '▁soil', '▁solution', '▁and', '▁plant', '▁cells', ',', '▁and', '▁', 'cyto', 'toxicity', '▁due', '▁of', '▁intra', 'cellular', '▁Na', '[', '+', ']', '▁accumulation', '▁[', '14', '].']\n",
            "\n",
            "[Cannot reconstruct]: most phenotypes ['▁Indeed', ',', '▁most', '▁', 'phen', 'o', 'type', 's', '▁gave', '▁rise', '▁to', '▁', 'a', '▁distribution', '▁of', '▁P', '▁values', '▁that', '▁was', '▁strongly', '▁', 's', 'k', 'ewed', '▁towards', '▁zero', '▁(', 'S', 'up', 'p', 'lement', 'ary', '▁Information', ',', '▁section', '▁2.', '1.6', ').']\n",
            "\n",
            "[Cannot reconstruct]: arsenite efflux process ['▁Critical', 'ly', ',', '▁it', '▁is', '▁this', '▁ar', 's', 'e', 'nite', '▁eff', 'lux', '▁process', '▁in', '▁roots', '▁that', '▁allows', '▁plants', '▁to', '▁maintain', '▁low', '▁ar', 's', 'en', 'ic', '▁concentration', 's', '▁in', '▁their', '▁shoot', 's', '▁when', '▁they', '▁are', '▁exposed', '▁to', '▁environmentally', '▁relevant', '▁ar', 's', 'en', 'ate', '▁concentration', 's', '▁in', '▁the', '▁soil', '▁(', 'Figure', '▁13', ').']\n",
            "\n",
            "[Cannot reconstruct]: Extensins ['▁Ex', 'ten', 's', 'in', 's', '▁are', '▁cell', '▁wall', '-', 'loc', 'ated', ',', '▁basic', ',', '▁Hy', 'p', '-', 'rich', '▁structural', '▁glyco', 'protein', 's', '▁with', '▁', 'alternating', '▁hydro', 'phil', 'ic', '▁and', '▁hydro', 'pho', 'bic', '▁', 'motifs', '▁(', 'L', 'am', 'port', '▁', 'e', 't', '▁al', '.', ',', '▁2011', ').']\n",
            "\n",
            "[Cannot reconstruct]: loss-of-function of ACR2 ['▁However', ',', '▁loss', '-', 'of', '-', 'function', '▁of', '▁A', 'CR', '2', '▁has', '▁no', '▁effect', '▁on', '▁ar', 's', 'en', 'ate', '▁reduction', ',', '▁ar', 's', 'e', 'nite', '▁eff', 'lux', ',', '▁ar', 's', 'en', 'ic', '▁accumulation', '▁or', '▁resistance', '▁to', '▁ar', 's', 'en', 'ate', '▁in', '▁', 'viv', 'o', ',', '▁as', '▁previous', '▁published', '▁[', '26', ']', '▁and', '▁repeated', '▁here', '.']\n",
            "\n",
            "[Cannot reconstruct]: Pb ['▁AC', 'BP', '1', '▁and', '▁P', 'DR', '8', '▁were', '▁enhanced', '▁by', '▁P', 'b', '▁in', '▁the', '▁roots', '▁of', '▁the', '▁T', '-', 'L', 'LL', '▁acces', 'sion', '▁but', '▁also', '▁in', '▁the', '▁roots', '▁of', '▁the', '▁(', 'S', ')', '▁plants', '▁(', 'Figure', 's', '▁4', 'C', ',', 'D', ').']\n",
            "\n",
            "[Cannot reconstruct]: loci ['▁', 'Using', '▁genome', '-', 'wide', '▁association', '▁(', 'GW', 'A', ')', '▁mapping', '▁of', '▁loc', 'i', '▁controlling', '▁natural', '▁variation', '▁in', '▁ar', 's', 'en', 'ic', '▁accumulation', '▁in', '▁Arab', 'i', 'd', 'ops', 'is', '▁', 'thal', 'iana', '▁allowed', '▁us', '▁to', '▁identify', '▁the', '▁ar', 's', 'en', 'ate', '▁reduc', 't', 'a', 's', 'e', '▁required', '▁for', '▁this', '▁reduction', ',', '▁which', '▁we', '▁named', '▁High', '▁Ar', 's', 'en', 'ic', '▁Content', '▁1', '▁(', 'HA', 'C', '1).']\n",
            "\n",
            "[Cannot reconstruct]: EXTs ['▁Furthermore', ',', '▁E', 'X', 'T', 's', '▁can', '▁interact', '▁with', '▁other', '▁cell', '▁wall', '▁components', '▁such', '▁as', '▁pe', 'c', 't', 'in', 's', '▁(', 'Q', 'i', '▁', 'e', 't', '▁al', '.', ',', '▁1995', ').']\n",
            "\n",
            "[Cannot reconstruct]: this allelic variation ['▁Further', ',', '▁we', '▁have', '▁confirmed', '▁by', '▁G', 'WA', '▁mapping', ',', '▁experimental', '▁complement', 'ation', '▁crosses', ',', '▁and', '▁gene', '▁expression', '▁studies', ',', '▁that', '▁this', '▁alle', 'lic', '▁variation', '▁directly', '▁causes', '▁changes', '▁in', '▁the', '▁', 'clin', 'ally', '▁', 'varying', '▁leaf', '▁Na', '[', '+', ']', '▁accumulation', '▁', 'phen', 'o', 'type', '▁via', '▁', 'c', 'is', '-', 're', 'gul', 'atory', '▁poly', 'morph', 'is', 'm', 's', '▁(', 'Figure', '▁2', '▁and', '▁Figure', '▁3', ').']\n",
            "\n",
            "[Cannot reconstruct]: the wide variety of natural phenotypic and genetic variation ['▁', 'Whether', '▁the', '▁wide', '▁variety', '▁of', '▁natural', '▁', 'phen', 'o', 'typ', 'ic', '▁and', '▁genetic', '▁variation', '▁observed', '▁in', '▁A', '.', '▁', 'thal', 'iana', '▁[', '12', ']', '▁contribute', 's', '▁to', '▁its', '▁local', '▁adaptation', '▁is', '▁an', '▁important', '▁un', 're', 'solv', 'e', 'd', '▁question', '▁that', '▁is', '▁currently', '▁', 'attracting', '▁', 'a', '▁significant', '▁amount', '▁of', '▁attention', '▁[', '13', '].']\n",
            "\n",
            "[Cannot reconstruct]: HAC1facilitated ['▁To', '▁understand', '▁the', '▁role', '▁of', '▁H', 'AC', '1', 'facilitated', '▁ar', 's', 'e', 'nite', '▁eff', 'lux', '▁in', '▁', 'limiting', '▁ar', 's', 'en', 'ic', '▁trans', 'location', '▁to', '▁the', '▁shoot', ',', '▁we', '▁investigated', '▁the', '▁distribution', '▁of', '▁ar', 's', 'en', 'ic', '▁in', '▁roots', '▁in', '▁the', '▁ha', 'c', '1', '-1', '▁and', '▁ha', 'c', '1-2', '▁loss', '-', 'of', '-', 'function', '▁mutant', 's', '.']\n",
            "\n",
            "[Cannot reconstruct]: appropriate parental strains ['▁Any', '▁cross', '▁will', '▁eliminate', '▁long', '-', 'range', '▁link', 'age', '▁dis', 'e', 'qui', 'lib', 'rium', ',', '▁and', '▁short', '-', 'range', '▁link', 'age', '▁dis', 'e', 'qui', 'lib', 'rium', '▁can', '▁be', '▁overcome', '▁by', '▁choosing', '▁appropriate', '▁parental', '▁strain', 's', '.']\n",
            "\n",
            "[Cannot reconstruct]: all the tested accessions ['▁We', '▁noticed', '▁that', '▁all', '▁the', '▁tested', '▁access', 'ions', '▁suffered', '▁', 'a', '▁root', '▁length', '▁reduction', '▁under', '▁the', '▁P', 'b', '▁treatment', ',', '▁but', '▁', 'ext', '18', '▁also', '▁had', '▁shorter', '▁roots', '▁under', '▁control', '▁conditions', '▁(', 'S', 'up', 'p', 'lement', 'ary', '▁Figure', '▁', '2).']\n",
            "\n",
            "[Cannot reconstruct]: excess of nominally significant association ['▁The', '▁excess', '▁of', '▁nominal', 'ly', '▁significant', '▁association', '▁for', '▁flower', 'ing', '-', 'related', '▁', 'phen', 'o', 'type', 's', '▁has', '▁been', '▁eliminated', ',', '▁as', '▁would', '▁be', '▁expected', '▁', 'if', '▁this', '▁excess', '▁were', '▁mostly', '▁due', '▁to', '▁con', 'found', 'ing', '▁by', '▁population', '▁structure', '.']\n",
            "\n",
            "[Cannot reconstruct]: regulatory networks ['▁By', '▁', 'combining', '▁results', '▁from', '▁G', 'WA', ',', '▁link', 'age', '▁mapping', '▁and', '▁perhaps', '▁also', '▁intermediate', '▁', 'phen', 'o', 'type', 's', '▁(', 'for', '▁example', '▁expression', '▁data', '),', '▁it', '▁will', '▁be', '▁possible', '▁to', '▁make', '▁progress', '▁on', '▁de', 'constructing', '▁the', '▁regulatory', '▁networks', '▁that', '▁determine', '▁natural', '▁variation', '.']\n",
            "\n",
            "[Cannot reconstruct]: Loss-of-function alleles of HAC1 ['▁Los', 's', '-', 'of', '-', 'function', '▁all', 'ele', 's', '▁of', '▁H', 'AC', '1', '▁were', '▁', 'unable', '▁to', '▁complement', '▁the', '▁high', '▁leaf', '▁ar', 's', 'en', 'ic', '▁of', '▁K', 'r', '-', '0', '▁in', '▁F', '1', '▁hybrid', 's', '▁(', 'Figure', '▁2', 'G', '),', '▁', 'whereas', '▁the', '▁Col', '-', '0', '▁At', '2', 'g', '210', '45', '▁genomic', '▁fragment', '▁introduced', '▁trans', 'gen', 'ically', '▁into', '▁K', 'r', '-', '0', '▁was', '▁', 'able', '▁to', '▁complement', '▁(', 'Figure', '▁2', 'H', '),', '▁', 'confirming', '▁that', '▁At', '2', 'g', '210', '45', '▁is', '▁the', '▁causal', '▁gene', '▁for', '▁H', 'AC', '1.']\n",
            "\n",
            "[Cannot reconstruct]: 48 T-DNA insertion alleles ['▁To', '▁determine', '▁which', '▁gene', '▁is', '▁casual', '▁for', '▁H', 'AC', '1', '▁we', '▁analyse', 'd', '▁the', '▁ar', 's', 'en', 'ic', '▁concentration', '▁of', '▁leaves', '▁from', '▁48', '▁T', '-', 'DNA', '▁', 'insertion', '▁all', 'ele', 's', '▁of', '▁genes', '▁in', '▁this', '▁region', '▁(', 'Table', '▁S', '1).']\n",
            "\n",
            "[Cannot reconstruct]: leaf arsenic ['▁Close', '▁to', '▁18', '%', '▁of', '▁the', '▁total', '▁variation', '▁in', '▁leaf', '▁ar', 's', 'en', 'ic', '▁observed', '▁in', '▁the', '▁3', '49', '▁A', '.', '▁', 'thal', 'iana', '▁access', 'ions', '▁tested', '▁is', '▁explained', '▁by', '▁S', 'NP', 's', '▁marking', '▁the', '▁link', 'age', '▁block', '▁that', '▁contains', '▁H', 'AC', '1,', '▁and', '▁plants', '▁with', '▁the', '▁major', '▁all', 'ele', '▁of', '▁this', '▁loc', 'us', '▁contain', ',', '▁on', '▁average', ',', '▁3', '7%', '▁higher', '▁leaf', '▁ar', 's', 'en', 'ic', '▁concentration', '.']\n",
            "\n",
            "[Cannot reconstruct]: extensin induction ['▁Different', '▁studies', '▁have', '▁reported', '▁extensi', 'n', '▁in', 'duction', '▁after', '▁wound', 'ing', ',', '▁however', ',', '▁to', '▁date', '▁they', '▁have', '▁not', '▁been', '▁associated', '▁with', '▁P', 'b', '▁stress', '.']\n",
            "\n",
            "[Cannot reconstruct]: hac1-1 ['▁', 'Moreover', ',', '▁in', '▁roots', '▁of', '▁ha', 'c', '1', '-1', ',', '▁ha', 'c', '1-2', ',', '▁and', '▁K', 'r', '-', '0', '▁we', '▁observe', '▁', 'a', '▁reduction', '▁in', '▁the', '▁percentage', '▁of', '▁the', '▁total', '▁', 'accumulated', '▁ar', 's', 'en', 'ic', '▁present', '▁as', '▁ar', 's', 'e', 'nite', '.']\n",
            "\n",
            "[Cannot reconstruct]: Pb stress ['▁P', 'b', '▁stress', '▁caused', '▁', 'a', '▁', 'pronounced', '▁', 'th', 'inning', '▁of', '▁the', '▁root', '▁cell', '▁walls', '▁of', '▁the', '▁sensitive', '▁access', 'ions', '▁B', 'sch', '-', '0', '[', 'S', ']', '▁and', '▁', 'ext', '18', '▁(', 'Figure', '▁5', ').']\n",
            "\n",
            "[Cannot reconstruct]: differential Pb accumulation ['▁The', '▁mechanisms', '▁behind', '▁such', '▁differential', '▁P', 'b', '▁accumulation', '▁in', '▁plants', '▁are', '▁still', '▁poorly', '▁understood', '.']\n",
            "\n",
            "[Cannot reconstruct]: peak of association ['▁This', '▁is', '▁indeed', '▁the', '▁case', ',', '▁but', '▁rather', '▁than', '▁', 'a', '▁single', ',', '▁sharp', '▁peak', '▁of', '▁association', '▁centre', 'd', '▁on', '▁F', 'RI', ',', '▁we', '▁have', '▁', 'a', '▁', 'â', '€', '▁', '̃', 'mount', 'a', 'in', '▁range', 'â', '€', 'TM', '▁covering', '▁500', '▁', 'k', 'b', '▁and', '▁on', '▁the', '▁order', '▁of', '▁', 'a', '▁hundred', '▁genes', '▁(', 'Fig', '.', '▁4', 'a', ').']\n",
            "\n",
            "[Cannot reconstruct]: Lead ['▁Lead', '▁is', '▁', 'a', '▁non', '-', 'essen', 'tial', '▁metal', '▁that', '▁does', '▁not', '▁have', '▁', 'a', '▁role', '▁in', '▁any', '▁cell', '▁metabolism', '▁process', '▁but', '▁it', '▁is', '▁easily', '▁', 'absorbed', '▁and', '▁', 'accumulated', '▁in', '▁various', '▁plant', '▁tissues', '.']\n",
            "\n",
            "[Cannot reconstruct]: the 360 A. thaliana accessions ['▁Our', '▁screen', '▁of', '▁the', '▁360', '▁A', '.', '▁', 'thal', 'iana', '▁access', 'ions', '▁in', '▁', 'a', '▁P', 'b', '-', 'enriched', '▁mine', '▁soil', '▁revealed', '▁', 'a', '▁variety', '▁of', '▁responses', '▁to', '▁P', 'b', '▁stress', ',', '▁from', '▁access', 'ions', '▁that', '▁did', '▁not', '▁germ', 'inate', ',', '▁to', '▁some', '▁that', '▁survived', '▁', 'a', '▁few', '▁days', ',', '▁to', '▁access', 'ions', '▁that', '▁', 'grew', '▁even', '▁better', '▁in', '▁high', '▁P', 'b', '▁conditions', '▁than', '▁control', '▁soil', '.']\n",
            "\n",
            "[Cannot reconstruct]: EXTENSIN18 (EXT18) ['▁Gene', 's', '▁associated', '▁with', '▁significant', '▁S', 'NP', '▁markers', '▁were', '▁evaluated', '▁and', '▁we', '▁selected', '▁E', 'X', 'TEN', 'S', 'IN', '18', '▁(', 'E', 'X', 'T', '18', ')', '▁and', '▁T', 'LC', '▁(', 'TRA', 'M', '-', 'LA', 'G', '1', '-', 'CL', 'N', '8)', '▁as', '▁candidates', '▁for', '▁having', '▁', 'a', '▁role', '▁in', '▁P', 'b', '▁home', 'ost', 'a', 's', 'is', '.']\n",
            "\n",
            "[Cannot reconstruct]: SNPs ['▁S', 'NP', 's', '▁in', '▁the', '▁F', 'RI', '▁region', '▁should', '▁thus', '▁be', '▁associated', '▁with', '▁FL', 'C', '▁expression', '.']\n",
            "\n",
            "[Cannot reconstruct]: mixed-model[12 ['▁For', '▁analysis', '▁with', '▁con', 'found', 'ing', ',', '▁', 'a', '▁mixed', '-', 'model', '[', '12', ']', '▁implemented', '▁in', '▁the', '▁program', '▁E', 'MMA', '[', '13', ']', '▁was', '▁used', '.']\n",
            "\n",
            "[Cannot reconstruct]: natural variation at the HAC1 locus ['▁We', '▁find', '▁that', '▁natural', '▁variation', '▁at', '▁the', '▁H', 'AC', '1', '▁loc', 'us', '▁accounts', '▁for', '▁', 'a', '▁significant', '▁proportion', '▁of', '▁the', '▁species', '-', 'wide', '▁diversity', '▁in', '▁leaf', '▁ar', 's', 'en', 'ic', '▁accumulation', '▁in', '▁A', '.', '▁', 'thal', 'iana', '▁when', '▁plants', '▁are', '▁grown', '▁in', '▁soil', '▁', 'containing', '▁environmentally', '▁relevant', '▁trace', '▁concentration', 's', '▁of', '▁ar', 's', 'en', 'ic', '.']\n",
            "\n",
            "[Cannot reconstruct]: genetic architecture ['▁Progress', '▁has', '▁been', '▁made', '▁in', '▁out', 'lining', '▁the', '▁genetic', '▁architecture', '▁of', '▁these', '▁adaptive', '▁traits', '▁[', '5,', '8', 'â', '€', '“', '10', '],', '▁though', '▁', 'a', '▁mo', 'le', 'cular', '▁genetic', '▁understanding', '▁is', '▁still', '▁needed', '.']\n",
            "\n",
            "Converting seq2seq to records: 100% 50/50 [00:00<00:00, 1615.75it/s]\n",
            "[INFO] Finished inference!\n",
            "[INFO] Seq2Seq predictions written to: results_file_inference/preds_seq2seq.txt\n",
            "[INFO] Record predictions written to: results_file_inference/preds_record.jsonl\n"
          ]
        }
      ],
      "source": [
        "!python transform_text.py \\\n",
        "  --input_file sentences_validation.txt \\\n",
        "  --output_dir results_file_inference \\\n",
        "  --model output/Model_V4 \\\n",
        "  --schema_file data/text2spotasoc/relation/GWAS/record.schema \\\n",
        "  --batch_size 45"
      ]
    }
  ],
  "metadata": {
    "accelerator": "GPU",
    "colab": {
      "gpuType": "A100",
      "machine_shape": "hm",
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}